{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Identifying Lipid related genes using TCGA subtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook creates a list by taking all proteins and comparing their abundance by grouping by Mesenchymal/ Normal and Mesenchymal/Proneural. The common genes between the two t-test were then searched to see if they were in Uniprot's list of genes related to Lipid metabolism "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "#sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\GBM_for_CPTAC\\\\')\n",
    "#import cis_functions as f\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_significance_col(results_df, num_comparisons):\n",
    "    \"bonferroni multiple hypothesis\"\"\"\n",
    "    alpha = .05\n",
    "    bonferroni_cutoff = alpha / num_comparisons\n",
    "    \n",
    "    pval = results_df['P_Value']\n",
    "    if float(pval[0]) <= bonferroni_cutoff:\n",
    "        results_df['Significant'] = True\n",
    "    else: \n",
    "        results_df['Significant'] = False\n",
    "    return results_df\n",
    "\n",
    "def wrap_ttest_return_all(df, label_column, comparison_columns, total_tests, alpha=.05):\n",
    "    try:\n",
    "        #Verify precondition that label column exists and has exactly 2 unique values\n",
    "        label_values = df[label_column].unique()\n",
    "        if len(label_values) != 2:\n",
    "            print(\"Incorrectly Formatted Dataframe! Label column must have exactly 2 unique values.\")\n",
    "            return None\n",
    "        \n",
    "        #Partition dataframe into two sets, one for each of the two unique values from the label column\n",
    "        partition1 = df.loc[df[label_column] == label_values[0]]\n",
    "        partition2 = df.loc[df[label_column] == label_values[1]]\n",
    "        \n",
    "        #Determine the number of real valued columns on which we will do t-tests\n",
    "        #sites = len(comparison_columns.columns)\n",
    "        number_of_comparisons = total_tests # ? phospho sites or num freq mut genes doing cis comp\n",
    "        \n",
    "        #Use a bonferroni correction to adjust for multiple testing by altering the p-value needed for acceptance\n",
    "        bonferroni_cutoff = alpha/number_of_comparisons\n",
    "        \n",
    "        #Store all comparisons with their p-values in a dictionary\n",
    "        all_comparisons = {}\n",
    "        \n",
    "        #Loop through each comparison column, perform the t-test, and determine whether it meets the significance cutoff'''\n",
    "        for column in comparison_columns:\n",
    "            stat, pval = scipy.stats.ttest_ind(partition1[column].dropna(axis=0), partition2[column].dropna(axis=0))\n",
    "            all_comparisons[column] = pval\n",
    "    \n",
    "        #Sort dictionary to list smallest p-values first\n",
    "        sorted_comparisons = sorted(all_comparisons.items(), key=lambda kv: kv[1])\n",
    "        #Format as a dataframe and return to caller\n",
    "        all_comparisons_df = pd.DataFrame.from_dict(sorted_comparisons)\n",
    "        all_comparisons_df.columns = ['Comparison', 'P_Value']\n",
    "        \n",
    "                                               \n",
    "        all_comparisons_sig_col = add_significance_col(all_comparisons_df, number_of_comparisons)\n",
    "        return all_comparisons_sig_col\n",
    "                                \n",
    "    except:\n",
    "        print(\"Incorrectly Formatted Dataframe!\")\n",
    "        return None\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "#cptac.download(dataset='ccrcc', version='0.0')\n",
    "brain= cptac.Gbm()\n",
    "desired_cutoff = 0.05\n",
    "gene = 'RB1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: S021, S048, S064, S073, S075, S103 (/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>case</th>\n",
       "      <th>Sample_Tumor_Normal</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>height</th>\n",
       "      <th>weight</th>\n",
       "      <th>bmi</th>\n",
       "      <th>country_of_origin</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSCAN31_proteomics</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00104</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>188.00</td>\n",
       "      <td>115.00</td>\n",
       "      <td>32.54</td>\n",
       "      <td>United States</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.047437</td>\n",
       "      <td>-0.105908</td>\n",
       "      <td>-0.347076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.459635</td>\n",
       "      <td>0.079452</td>\n",
       "      <td>-0.784983</td>\n",
       "      <td>-0.488441</td>\n",
       "      <td>0.167990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>C3L-00365</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.00</td>\n",
       "      <td>54.00</td>\n",
       "      <td>20.61</td>\n",
       "      <td>United States</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.161975</td>\n",
       "      <td>-0.213093</td>\n",
       "      <td>0.235571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.107421</td>\n",
       "      <td>0.048724</td>\n",
       "      <td>0.138403</td>\n",
       "      <td>-0.290141</td>\n",
       "      <td>0.405037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>C3L-00674</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>193.00</td>\n",
       "      <td>102.00</td>\n",
       "      <td>27.44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.065534</td>\n",
       "      <td>-0.306717</td>\n",
       "      <td>0.879991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.883564</td>\n",
       "      <td>-0.172222</td>\n",
       "      <td>0.011876</td>\n",
       "      <td>-0.131889</td>\n",
       "      <td>-0.503581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>164.00</td>\n",
       "      <td>52.00</td>\n",
       "      <td>19.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.062127</td>\n",
       "      <td>-0.254535</td>\n",
       "      <td>0.463653</td>\n",
       "      <td>0.580230</td>\n",
       "      <td>0.503044</td>\n",
       "      <td>-0.604986</td>\n",
       "      <td>0.178077</td>\n",
       "      <td>-0.720059</td>\n",
       "      <td>-0.150197</td>\n",
       "      <td>-0.268715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>C3L-01040</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>77.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>170.00</td>\n",
       "      <td>70.00</td>\n",
       "      <td>24.22</td>\n",
       "      <td>Russia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.092502</td>\n",
       "      <td>0.010639</td>\n",
       "      <td>-0.465079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.500083</td>\n",
       "      <td>0.112651</td>\n",
       "      <td>1.004660</td>\n",
       "      <td>-0.230304</td>\n",
       "      <td>-0.102416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S111</td>\n",
       "      <td>PT-RN5K</td>\n",
       "      <td>Normal</td>\n",
       "      <td>56.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.56</td>\n",
       "      <td>49.90</td>\n",
       "      <td>18.88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.298664</td>\n",
       "      <td>-0.403930</td>\n",
       "      <td>-0.482597</td>\n",
       "      <td>0.240966</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.609314</td>\n",
       "      <td>-1.293464</td>\n",
       "      <td>0.225910</td>\n",
       "      <td>-0.013637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S112</td>\n",
       "      <td>PT-RU72</td>\n",
       "      <td>Normal</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.56</td>\n",
       "      <td>63.05</td>\n",
       "      <td>23.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Asian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155988</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>-0.834434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.073908</td>\n",
       "      <td>0.637394</td>\n",
       "      <td>-0.795886</td>\n",
       "      <td>0.326046</td>\n",
       "      <td>-0.217014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S113</td>\n",
       "      <td>PT-UTHO</td>\n",
       "      <td>Normal</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>182.88</td>\n",
       "      <td>62.87</td>\n",
       "      <td>18.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.016857</td>\n",
       "      <td>-0.171858</td>\n",
       "      <td>-0.462233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.278683</td>\n",
       "      <td>0.616285</td>\n",
       "      <td>-0.734952</td>\n",
       "      <td>0.339283</td>\n",
       "      <td>-0.419138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S114</td>\n",
       "      <td>PT-WVLH</td>\n",
       "      <td>Normal</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>182.88</td>\n",
       "      <td>90.72</td>\n",
       "      <td>27.12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.450926</td>\n",
       "      <td>0.114686</td>\n",
       "      <td>-0.245149</td>\n",
       "      <td>-1.202774</td>\n",
       "      <td>-0.709850</td>\n",
       "      <td>-0.233022</td>\n",
       "      <td>0.732273</td>\n",
       "      <td>-1.158899</td>\n",
       "      <td>0.354608</td>\n",
       "      <td>-0.468881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S115</td>\n",
       "      <td>PT-Y8DK</td>\n",
       "      <td>Normal</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>172.72</td>\n",
       "      <td>81.65</td>\n",
       "      <td>27.37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.262036</td>\n",
       "      <td>-0.184673</td>\n",
       "      <td>-0.775883</td>\n",
       "      <td>1.778184</td>\n",
       "      <td>-0.250798</td>\n",
       "      <td>0.623839</td>\n",
       "      <td>-0.892930</td>\n",
       "      <td>0.249729</td>\n",
       "      <td>-0.376205</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>115 rows × 11169 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name            case Sample_Tumor_Normal   age  gender  height  weight    bmi  \\\n",
       "Sample_ID                                                                       \n",
       "S001       C3L-00104               Tumor  58.0    Male  188.00  115.00  32.54   \n",
       "S002       C3L-00365               Tumor  59.0  Female  162.00   54.00  20.61   \n",
       "S003       C3L-00674               Tumor  45.0    Male  193.00  102.00  27.44   \n",
       "S004       C3L-00677               Tumor  69.0  Female  164.00   52.00  19.32   \n",
       "S005       C3L-01040               Tumor  77.0  Female  170.00   70.00  24.22   \n",
       "...              ...                 ...   ...     ...     ...     ...    ...   \n",
       "S111         PT-RN5K              Normal  56.0  Female  162.56   49.90  18.88   \n",
       "S112         PT-RU72              Normal  59.0  Female  162.56   63.05  23.86   \n",
       "S113         PT-UTHO              Normal  68.0    Male  182.88   62.87  18.80   \n",
       "S114         PT-WVLH              Normal  58.0    Male  182.88   90.72  27.12   \n",
       "S115         PT-Y8DK              Normal  62.0    Male  172.72   81.65  27.37   \n",
       "\n",
       "Name      country_of_origin   race               ethnicity  ...  \\\n",
       "Sample_ID                                                   ...   \n",
       "S001          United States  White  Not-Hispanic or Latino  ...   \n",
       "S002          United States  White  Not-Hispanic or Latino  ...   \n",
       "S003                    NaN  White  Not-Hispanic or Latino  ...   \n",
       "S004                    NaN  White  Not-Hispanic or Latino  ...   \n",
       "S005                 Russia    NaN                     NaN  ...   \n",
       "...                     ...    ...                     ...  ...   \n",
       "S111                    NaN  White  Not Hispanic or Latino  ...   \n",
       "S112                    NaN  Asian                     NaN  ...   \n",
       "S113                    NaN  White                     NaN  ...   \n",
       "S114                    NaN  White                     NaN  ...   \n",
       "S115                    NaN  White  Not Hispanic or Latino  ...   \n",
       "\n",
       "Name      ZSCAN31_proteomics ZSWIM8_proteomics ZW10_proteomics  \\\n",
       "Sample_ID                                                        \n",
       "S001                     NaN         -0.047437       -0.105908   \n",
       "S002                     NaN          0.161975       -0.213093   \n",
       "S003                     NaN         -0.065534       -0.306717   \n",
       "S004               -0.062127         -0.254535        0.463653   \n",
       "S005                     NaN         -0.092502        0.010639   \n",
       "...                      ...               ...             ...   \n",
       "S111                     NaN          0.298664       -0.403930   \n",
       "S112                     NaN          0.155988        0.002551   \n",
       "S113                     NaN          0.016857       -0.171858   \n",
       "S114                0.450926          0.114686       -0.245149   \n",
       "S115                     NaN          0.262036       -0.184673   \n",
       "\n",
       "Name      ZWILCH_proteomics ZWINT_proteomics ZXDC_proteomics  \\\n",
       "Sample_ID                                                      \n",
       "S001              -0.347076              NaN        0.459635   \n",
       "S002               0.235571              NaN        0.107421   \n",
       "S003               0.879991              NaN        0.883564   \n",
       "S004               0.580230         0.503044       -0.604986   \n",
       "S005              -0.465079              NaN       -0.500083   \n",
       "...                     ...              ...             ...   \n",
       "S111              -0.482597         0.240966             NaN   \n",
       "S112              -0.834434              NaN       -0.073908   \n",
       "S113              -0.462233              NaN        1.278683   \n",
       "S114              -1.202774        -0.709850       -0.233022   \n",
       "S115              -0.775883         1.778184       -0.250798   \n",
       "\n",
       "Name      ZYG11B_proteomics  ZYX_proteomics  ZZEF1_proteomics ZZZ3_proteomics  \n",
       "Sample_ID                                                                      \n",
       "S001               0.079452       -0.784983         -0.488441        0.167990  \n",
       "S002               0.048724        0.138403         -0.290141        0.405037  \n",
       "S003              -0.172222        0.011876         -0.131889       -0.503581  \n",
       "S004               0.178077       -0.720059         -0.150197       -0.268715  \n",
       "S005               0.112651        1.004660         -0.230304       -0.102416  \n",
       "...                     ...             ...               ...             ...  \n",
       "S111               0.609314       -1.293464          0.225910       -0.013637  \n",
       "S112               0.637394       -0.795886          0.326046       -0.217014  \n",
       "S113               0.616285       -0.734952          0.339283       -0.419138  \n",
       "S114               0.732273       -1.158899          0.354608       -0.468881  \n",
       "S115               0.623839       -0.892930          0.249729       -0.376205  \n",
       "\n",
       "[115 rows x 11169 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#join clinical and proteomic data\n",
    "clin_and_prot = brain.join_metadata_to_omics(metadata_df_name=\"clinical\", omics_df_name=\"proteomics\")\n",
    "clin_and_prot = clin_and_prot.rename(columns = {\"Patient_ID\": \"case\"})\n",
    "#clin_and_prot = clin_and_prot.reset_index()\n",
    "clin_and_prot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>nmf_consensus</th>\n",
       "      <th>nmf_cluster_membership</th>\n",
       "      <th>rna_wang_cancer_cell_2017</th>\n",
       "      <th>mRNA_stemness_index</th>\n",
       "      <th>dna_methyl</th>\n",
       "      <th>is_gcimp</th>\n",
       "      <th>immune</th>\n",
       "      <th>telomere</th>\n",
       "      <th>lipid</th>\n",
       "      <th>mirna</th>\n",
       "      <th>ancestry_prediction</th>\n",
       "      <th>ancestry_prediction_afr_prob</th>\n",
       "      <th>ancestry_prediction_amr_prob</th>\n",
       "      <th>ancestry_prediction_eas_prob</th>\n",
       "      <th>ancestry_prediction_eur_prob</th>\n",
       "      <th>ancestry_prediction_sas_prob</th>\n",
       "      <th>wxs_total_mutation</th>\n",
       "      <th>wgs_total_mutation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>C3L-00104</td>\n",
       "      <td>tumor</td>\n",
       "      <td>nmf1</td>\n",
       "      <td>0.743</td>\n",
       "      <td>Proneural</td>\n",
       "      <td>0.678244</td>\n",
       "      <td>dm2</td>\n",
       "      <td>True</td>\n",
       "      <td>low</td>\n",
       "      <td>normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mi5</td>\n",
       "      <td>EUR</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>60.0</td>\n",
       "      <td>2632.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>C3L-00365</td>\n",
       "      <td>tumor</td>\n",
       "      <td>nmf3</td>\n",
       "      <td>0.614</td>\n",
       "      <td>Classical</td>\n",
       "      <td>0.681122</td>\n",
       "      <td>dm4</td>\n",
       "      <td>False</td>\n",
       "      <td>low</td>\n",
       "      <td>normal</td>\n",
       "      <td>TAG_enriched</td>\n",
       "      <td>mi1</td>\n",
       "      <td>EUR</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>57.0</td>\n",
       "      <td>7628.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>C3L-00674</td>\n",
       "      <td>tumor</td>\n",
       "      <td>nmf1</td>\n",
       "      <td>0.507</td>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>0.744635</td>\n",
       "      <td>dm5</td>\n",
       "      <td>False</td>\n",
       "      <td>high</td>\n",
       "      <td>normal</td>\n",
       "      <td>TAG_enriched</td>\n",
       "      <td>mi3</td>\n",
       "      <td>EUR</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.00</td>\n",
       "      <td>37.0</td>\n",
       "      <td>1233.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>tumor</td>\n",
       "      <td>nmf1</td>\n",
       "      <td>0.536</td>\n",
       "      <td>Proneural</td>\n",
       "      <td>0.900896</td>\n",
       "      <td>dm5</td>\n",
       "      <td>False</td>\n",
       "      <td>low</td>\n",
       "      <td>long</td>\n",
       "      <td>TAG_enriched</td>\n",
       "      <td>mi5</td>\n",
       "      <td>EUR</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.02</td>\n",
       "      <td>925.0</td>\n",
       "      <td>16955.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>C3L-01040</td>\n",
       "      <td>tumor</td>\n",
       "      <td>nmf1</td>\n",
       "      <td>0.589</td>\n",
       "      <td>Classical</td>\n",
       "      <td>0.647288</td>\n",
       "      <td>dm5</td>\n",
       "      <td>False</td>\n",
       "      <td>low</td>\n",
       "      <td>normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mi1</td>\n",
       "      <td>EUR</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>85.0</td>\n",
       "      <td>4298.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        case sample_type nmf_consensus  nmf_cluster_membership  \\\n",
       "0  C3L-00104       tumor          nmf1                   0.743   \n",
       "1  C3L-00365       tumor          nmf3                   0.614   \n",
       "2  C3L-00674       tumor          nmf1                   0.507   \n",
       "3  C3L-00677       tumor          nmf1                   0.536   \n",
       "4  C3L-01040       tumor          nmf1                   0.589   \n",
       "\n",
       "  rna_wang_cancer_cell_2017  mRNA_stemness_index dna_methyl is_gcimp immune  \\\n",
       "0                 Proneural             0.678244        dm2     True    low   \n",
       "1                 Classical             0.681122        dm4    False    low   \n",
       "2               Mesenchymal             0.744635        dm5    False   high   \n",
       "3                 Proneural             0.900896        dm5    False    low   \n",
       "4                 Classical             0.647288        dm5    False    low   \n",
       "\n",
       "  telomere         lipid mirna ancestry_prediction  \\\n",
       "0   normal           NaN   mi5                 EUR   \n",
       "1   normal  TAG_enriched   mi1                 EUR   \n",
       "2   normal  TAG_enriched   mi3                 EUR   \n",
       "3     long  TAG_enriched   mi5                 EUR   \n",
       "4   normal           NaN   mi1                 EUR   \n",
       "\n",
       "   ancestry_prediction_afr_prob  ancestry_prediction_amr_prob  \\\n",
       "0                          0.00                          0.03   \n",
       "1                          0.00                          0.02   \n",
       "2                          0.01                          0.00   \n",
       "3                          0.02                          0.11   \n",
       "4                          0.00                          0.03   \n",
       "\n",
       "   ancestry_prediction_eas_prob  ancestry_prediction_eur_prob  \\\n",
       "0                          0.00                          0.97   \n",
       "1                          0.00                          0.98   \n",
       "2                          0.01                          0.98   \n",
       "3                          0.00                          0.85   \n",
       "4                          0.00                          0.97   \n",
       "\n",
       "   ancestry_prediction_sas_prob  wxs_total_mutation  wgs_total_mutation  \n",
       "0                          0.00                60.0              2632.0  \n",
       "1                          0.00                57.0              7628.0  \n",
       "2                          0.00                37.0              1233.0  \n",
       "3                          0.02               925.0             16955.0  \n",
       "4                          0.00                85.0              4298.0  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Read in files with TCGA subtypes\n",
    "subtypes = pd.read_csv(\"/Users/Lindsey/Downloads/gbm_all_subtype_collections.2019-11-13.tsv\", sep= \"\\t\")\n",
    "\n",
    "case_subtype = subtypes[[\"case\",'rna_wang_cancer_cell_2017']] #only need subtype and case\n",
    "case_subtype = case_subtype.rename(columns = {\"rna_wang_cancer_cell_2017\": \"TCGA_subtype\"})\n",
    "\n",
    "\n",
    "subtypes.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA_subtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>C3L-00104</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00365</td>\n",
       "      <td>Classical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00674</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01040</td>\n",
       "      <td>Classical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-RN5K</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-RU72</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-UTHO</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-WVLH</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-Y8DK</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          TCGA_subtype\n",
       "case                  \n",
       "C3L-00104    Proneural\n",
       "C3L-00365    Classical\n",
       "C3L-00674  Mesenchymal\n",
       "C3L-00677    Proneural\n",
       "C3L-01040    Classical\n",
       "...                ...\n",
       "PT-RN5K         normal\n",
       "PT-RU72         normal\n",
       "PT-UTHO         normal\n",
       "PT-WVLH         normal\n",
       "PT-Y8DK         normal\n",
       "\n",
       "[109 rows x 1 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "case_subtype = case_subtype.replace(np.nan, 'normal', regex=True)\n",
    "case_subtype = case_subtype.set_index(\"case\")\n",
    "\n",
    "case_subtype\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Tumor_Normal</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>height</th>\n",
       "      <th>weight</th>\n",
       "      <th>bmi</th>\n",
       "      <th>country_of_origin</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>ethnicity_self_identify</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>TCGA_subtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>C3L-00104</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>188.00</td>\n",
       "      <td>115.00</td>\n",
       "      <td>32.54</td>\n",
       "      <td>United States</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047437</td>\n",
       "      <td>-0.105908</td>\n",
       "      <td>-0.347076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.459635</td>\n",
       "      <td>0.079452</td>\n",
       "      <td>-0.784983</td>\n",
       "      <td>-0.488441</td>\n",
       "      <td>0.167990</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00365</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.00</td>\n",
       "      <td>54.00</td>\n",
       "      <td>20.61</td>\n",
       "      <td>United States</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>0.161975</td>\n",
       "      <td>-0.213093</td>\n",
       "      <td>0.235571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.107421</td>\n",
       "      <td>0.048724</td>\n",
       "      <td>0.138403</td>\n",
       "      <td>-0.290141</td>\n",
       "      <td>0.405037</td>\n",
       "      <td>Classical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00674</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>193.00</td>\n",
       "      <td>102.00</td>\n",
       "      <td>27.44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.065534</td>\n",
       "      <td>-0.306717</td>\n",
       "      <td>0.879991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.883564</td>\n",
       "      <td>-0.172222</td>\n",
       "      <td>0.011876</td>\n",
       "      <td>-0.131889</td>\n",
       "      <td>-0.503581</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>164.00</td>\n",
       "      <td>52.00</td>\n",
       "      <td>19.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.254535</td>\n",
       "      <td>0.463653</td>\n",
       "      <td>0.580230</td>\n",
       "      <td>0.503044</td>\n",
       "      <td>-0.604986</td>\n",
       "      <td>0.178077</td>\n",
       "      <td>-0.720059</td>\n",
       "      <td>-0.150197</td>\n",
       "      <td>-0.268715</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01040</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>77.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>170.00</td>\n",
       "      <td>70.00</td>\n",
       "      <td>24.22</td>\n",
       "      <td>Russia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.092502</td>\n",
       "      <td>0.010639</td>\n",
       "      <td>-0.465079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.500083</td>\n",
       "      <td>0.112651</td>\n",
       "      <td>1.004660</td>\n",
       "      <td>-0.230304</td>\n",
       "      <td>-0.102416</td>\n",
       "      <td>Classical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-RN5K</td>\n",
       "      <td>Normal</td>\n",
       "      <td>56.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.56</td>\n",
       "      <td>49.90</td>\n",
       "      <td>18.88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.298664</td>\n",
       "      <td>-0.403930</td>\n",
       "      <td>-0.482597</td>\n",
       "      <td>0.240966</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.609314</td>\n",
       "      <td>-1.293464</td>\n",
       "      <td>0.225910</td>\n",
       "      <td>-0.013637</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-RU72</td>\n",
       "      <td>Normal</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.56</td>\n",
       "      <td>63.05</td>\n",
       "      <td>23.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Asian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.155988</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>-0.834434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.073908</td>\n",
       "      <td>0.637394</td>\n",
       "      <td>-0.795886</td>\n",
       "      <td>0.326046</td>\n",
       "      <td>-0.217014</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-UTHO</td>\n",
       "      <td>Normal</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>182.88</td>\n",
       "      <td>62.87</td>\n",
       "      <td>18.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.016857</td>\n",
       "      <td>-0.171858</td>\n",
       "      <td>-0.462233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.278683</td>\n",
       "      <td>0.616285</td>\n",
       "      <td>-0.734952</td>\n",
       "      <td>0.339283</td>\n",
       "      <td>-0.419138</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-WVLH</td>\n",
       "      <td>Normal</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>182.88</td>\n",
       "      <td>90.72</td>\n",
       "      <td>27.12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.114686</td>\n",
       "      <td>-0.245149</td>\n",
       "      <td>-1.202774</td>\n",
       "      <td>-0.709850</td>\n",
       "      <td>-0.233022</td>\n",
       "      <td>0.732273</td>\n",
       "      <td>-1.158899</td>\n",
       "      <td>0.354608</td>\n",
       "      <td>-0.468881</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-Y8DK</td>\n",
       "      <td>Normal</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>172.72</td>\n",
       "      <td>81.65</td>\n",
       "      <td>27.37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.262036</td>\n",
       "      <td>-0.184673</td>\n",
       "      <td>-0.775883</td>\n",
       "      <td>1.778184</td>\n",
       "      <td>-0.250798</td>\n",
       "      <td>0.623839</td>\n",
       "      <td>-0.892930</td>\n",
       "      <td>0.249729</td>\n",
       "      <td>-0.376205</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 11169 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Sample_Tumor_Normal   age  gender  height  weight    bmi  \\\n",
       "case                                                                 \n",
       "C3L-00104               Tumor  58.0    Male  188.00  115.00  32.54   \n",
       "C3L-00365               Tumor  59.0  Female  162.00   54.00  20.61   \n",
       "C3L-00674               Tumor  45.0    Male  193.00  102.00  27.44   \n",
       "C3L-00677               Tumor  69.0  Female  164.00   52.00  19.32   \n",
       "C3L-01040               Tumor  77.0  Female  170.00   70.00  24.22   \n",
       "...                       ...   ...     ...     ...     ...    ...   \n",
       "PT-RN5K                Normal  56.0  Female  162.56   49.90  18.88   \n",
       "PT-RU72                Normal  59.0  Female  162.56   63.05  23.86   \n",
       "PT-UTHO                Normal  68.0    Male  182.88   62.87  18.80   \n",
       "PT-WVLH                Normal  58.0    Male  182.88   90.72  27.12   \n",
       "PT-Y8DK                Normal  62.0    Male  172.72   81.65  27.37   \n",
       "\n",
       "          country_of_origin   race               ethnicity  \\\n",
       "case                                                         \n",
       "C3L-00104     United States  White  Not-Hispanic or Latino   \n",
       "C3L-00365     United States  White  Not-Hispanic or Latino   \n",
       "C3L-00674               NaN  White  Not-Hispanic or Latino   \n",
       "C3L-00677               NaN  White  Not-Hispanic or Latino   \n",
       "C3L-01040            Russia    NaN                     NaN   \n",
       "...                     ...    ...                     ...   \n",
       "PT-RN5K                 NaN  White  Not Hispanic or Latino   \n",
       "PT-RU72                 NaN  Asian                     NaN   \n",
       "PT-UTHO                 NaN  White                     NaN   \n",
       "PT-WVLH                 NaN  White                     NaN   \n",
       "PT-Y8DK                 NaN  White  Not Hispanic or Latino   \n",
       "\n",
       "          ethnicity_self_identify  ... ZSWIM8_proteomics ZW10_proteomics  \\\n",
       "case                               ...                                     \n",
       "C3L-00104               Caucasian  ...         -0.047437       -0.105908   \n",
       "C3L-00365               Caucasian  ...          0.161975       -0.213093   \n",
       "C3L-00674               Caucasian  ...         -0.065534       -0.306717   \n",
       "C3L-00677               Caucasian  ...         -0.254535        0.463653   \n",
       "C3L-01040               Caucasian  ...         -0.092502        0.010639   \n",
       "...                           ...  ...               ...             ...   \n",
       "PT-RN5K                       NaN  ...          0.298664       -0.403930   \n",
       "PT-RU72                       NaN  ...          0.155988        0.002551   \n",
       "PT-UTHO                       NaN  ...          0.016857       -0.171858   \n",
       "PT-WVLH                       NaN  ...          0.114686       -0.245149   \n",
       "PT-Y8DK                       NaN  ...          0.262036       -0.184673   \n",
       "\n",
       "          ZWILCH_proteomics ZWINT_proteomics ZXDC_proteomics  \\\n",
       "case                                                           \n",
       "C3L-00104         -0.347076              NaN        0.459635   \n",
       "C3L-00365          0.235571              NaN        0.107421   \n",
       "C3L-00674          0.879991              NaN        0.883564   \n",
       "C3L-00677          0.580230         0.503044       -0.604986   \n",
       "C3L-01040         -0.465079              NaN       -0.500083   \n",
       "...                     ...              ...             ...   \n",
       "PT-RN5K           -0.482597         0.240966             NaN   \n",
       "PT-RU72           -0.834434              NaN       -0.073908   \n",
       "PT-UTHO           -0.462233              NaN        1.278683   \n",
       "PT-WVLH           -1.202774        -0.709850       -0.233022   \n",
       "PT-Y8DK           -0.775883         1.778184       -0.250798   \n",
       "\n",
       "          ZYG11B_proteomics  ZYX_proteomics  ZZEF1_proteomics ZZZ3_proteomics  \\\n",
       "case                                                                            \n",
       "C3L-00104          0.079452       -0.784983         -0.488441        0.167990   \n",
       "C3L-00365          0.048724        0.138403         -0.290141        0.405037   \n",
       "C3L-00674         -0.172222        0.011876         -0.131889       -0.503581   \n",
       "C3L-00677          0.178077       -0.720059         -0.150197       -0.268715   \n",
       "C3L-01040          0.112651        1.004660         -0.230304       -0.102416   \n",
       "...                     ...             ...               ...             ...   \n",
       "PT-RN5K            0.609314       -1.293464          0.225910       -0.013637   \n",
       "PT-RU72            0.637394       -0.795886          0.326046       -0.217014   \n",
       "PT-UTHO            0.616285       -0.734952          0.339283       -0.419138   \n",
       "PT-WVLH            0.732273       -1.158899          0.354608       -0.468881   \n",
       "PT-Y8DK            0.623839       -0.892930          0.249729       -0.376205   \n",
       "\n",
       "          TCGA_subtype  \n",
       "case                    \n",
       "C3L-00104    Proneural  \n",
       "C3L-00365    Classical  \n",
       "C3L-00674  Mesenchymal  \n",
       "C3L-00677    Proneural  \n",
       "C3L-01040    Classical  \n",
       "...                ...  \n",
       "PT-RN5K         normal  \n",
       "PT-RU72         normal  \n",
       "PT-UTHO         normal  \n",
       "PT-WVLH         normal  \n",
       "PT-Y8DK         normal  \n",
       "\n",
       "[109 rows x 11169 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge tgca subtypes with proteomics and clincal df\n",
    "prot_subtype= clin_and_prot.merge(case_subtype, on='case')\n",
    "\n",
    "prot_subtype = prot_subtype.set_index(\"case\")\n",
    "\n",
    "prot_subtype"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mesenchymal VS Normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Tumor_Normal</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>height</th>\n",
       "      <th>weight</th>\n",
       "      <th>bmi</th>\n",
       "      <th>country_of_origin</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>ethnicity_self_identify</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>TCGA_subtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>PT-NPJ7</td>\n",
       "      <td>Normal</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>160.02</td>\n",
       "      <td>74.84</td>\n",
       "      <td>29.23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.337310</td>\n",
       "      <td>0.019538</td>\n",
       "      <td>-1.375146</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.164100</td>\n",
       "      <td>0.767921</td>\n",
       "      <td>-1.128100</td>\n",
       "      <td>0.209480</td>\n",
       "      <td>-0.205739</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-P44H</td>\n",
       "      <td>Normal</td>\n",
       "      <td>43.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>177.80</td>\n",
       "      <td>112.04</td>\n",
       "      <td>35.44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.335445</td>\n",
       "      <td>-0.199247</td>\n",
       "      <td>-0.798595</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.517515</td>\n",
       "      <td>0.379683</td>\n",
       "      <td>-1.087567</td>\n",
       "      <td>0.210812</td>\n",
       "      <td>-0.176034</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-Q2AG</td>\n",
       "      <td>Normal</td>\n",
       "      <td>42.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>177.80</td>\n",
       "      <td>102.06</td>\n",
       "      <td>32.28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.252516</td>\n",
       "      <td>-0.102542</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.791386</td>\n",
       "      <td>-0.667928</td>\n",
       "      <td>0.053734</td>\n",
       "      <td>-0.229870</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-QVJO</td>\n",
       "      <td>Normal</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>170.18</td>\n",
       "      <td>55.79</td>\n",
       "      <td>19.26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.320193</td>\n",
       "      <td>-0.098267</td>\n",
       "      <td>-1.321314</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.330917</td>\n",
       "      <td>0.667393</td>\n",
       "      <td>-1.172195</td>\n",
       "      <td>0.303740</td>\n",
       "      <td>-0.556137</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>PT-R55F</td>\n",
       "      <td>Normal</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>177.80</td>\n",
       "      <td>77.11</td>\n",
       "      <td>24.39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.326426</td>\n",
       "      <td>-0.109841</td>\n",
       "      <td>-0.676704</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.080553</td>\n",
       "      <td>0.563370</td>\n",
       "      <td>-0.973314</td>\n",
       "      <td>0.478914</td>\n",
       "      <td>-0.415546</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11169 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Sample_Tumor_Normal   age  gender  height  weight    bmi  \\\n",
       "case                                                               \n",
       "PT-NPJ7              Normal  68.0  Female  160.02   74.84  29.23   \n",
       "PT-P44H              Normal  43.0    Male  177.80  112.04  35.44   \n",
       "PT-Q2AG              Normal  42.0  Female  177.80  102.06  32.28   \n",
       "PT-QVJO              Normal  64.0  Female  170.18   55.79  19.26   \n",
       "PT-R55F              Normal  55.0    Male  177.80   77.11  24.39   \n",
       "\n",
       "        country_of_origin   race               ethnicity  \\\n",
       "case                                                       \n",
       "PT-NPJ7               NaN  White                     NaN   \n",
       "PT-P44H               NaN  White                     NaN   \n",
       "PT-Q2AG               NaN  White                     NaN   \n",
       "PT-QVJO               NaN  White  Not Hispanic or Latino   \n",
       "PT-R55F               NaN  White  Not Hispanic or Latino   \n",
       "\n",
       "        ethnicity_self_identify  ... ZSWIM8_proteomics ZW10_proteomics  \\\n",
       "case                             ...                                     \n",
       "PT-NPJ7                     NaN  ...          0.337310        0.019538   \n",
       "PT-P44H                     NaN  ...          0.335445       -0.199247   \n",
       "PT-Q2AG                     NaN  ...          0.252516       -0.102542   \n",
       "PT-QVJO                     NaN  ...          0.320193       -0.098267   \n",
       "PT-R55F                     NaN  ...          0.326426       -0.109841   \n",
       "\n",
       "        ZWILCH_proteomics ZWINT_proteomics ZXDC_proteomics ZYG11B_proteomics  \\\n",
       "case                                                                           \n",
       "PT-NPJ7         -1.375146              NaN        0.164100          0.767921   \n",
       "PT-P44H         -0.798595              NaN        0.517515          0.379683   \n",
       "PT-Q2AG               NaN              NaN             NaN          0.791386   \n",
       "PT-QVJO         -1.321314              NaN       -1.330917          0.667393   \n",
       "PT-R55F         -0.676704              NaN        0.080553          0.563370   \n",
       "\n",
       "         ZYX_proteomics  ZZEF1_proteomics ZZZ3_proteomics TCGA_subtype  \n",
       "case                                                                    \n",
       "PT-NPJ7       -1.128100          0.209480       -0.205739       normal  \n",
       "PT-P44H       -1.087567          0.210812       -0.176034       normal  \n",
       "PT-Q2AG       -0.667928          0.053734       -0.229870       normal  \n",
       "PT-QVJO       -1.172195          0.303740       -0.556137       normal  \n",
       "PT-R55F       -0.973314          0.478914       -0.415546       normal  \n",
       "\n",
       "[5 rows x 11169 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create mesenchymal and normal only df\n",
    "Mesenchymal = (prot_subtype.loc[prot_subtype['TCGA_subtype'] == 'Mesenchymal'])\n",
    "Normal = (prot_subtype.loc[prot_subtype['Sample_Tumor_Normal'] == 'Normal'])\n",
    "Normal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# create mesenchymal and normal df \n",
    "Mesench_Normal = Mesenchymal.append(Normal)\n",
    "Mesench_Normal.drop(Mesench_Normal.iloc[:, :28], axis=1, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A2M_proteomics',\n",
       " 'AAAS_proteomics',\n",
       " 'AACS_proteomics',\n",
       " 'AADAT_proteomics',\n",
       " 'AAED1_proteomics',\n",
       " 'AAGAB_proteomics',\n",
       " 'AAK1_proteomics',\n",
       " 'AAMDC_proteomics',\n",
       " 'AAMP_proteomics',\n",
       " 'AAR2_proteomics',\n",
       " 'AARS_proteomics',\n",
       " 'AARS2_proteomics',\n",
       " 'AARSD1_proteomics',\n",
       " 'AASDHPPT_proteomics',\n",
       " 'AASS_proteomics',\n",
       " 'AATF_proteomics',\n",
       " 'AATK_proteomics',\n",
       " 'ABAT_proteomics',\n",
       " 'ABCA1_proteomics',\n",
       " 'ABCA2_proteomics',\n",
       " 'ABCA3_proteomics',\n",
       " 'ABCA5_proteomics',\n",
       " 'ABCA8_proteomics',\n",
       " 'ABCB1_proteomics',\n",
       " 'ABCB10_proteomics',\n",
       " 'ABCB6_proteomics',\n",
       " 'ABCB7_proteomics',\n",
       " 'ABCB8_proteomics',\n",
       " 'ABCB9_proteomics',\n",
       " 'ABCC1_proteomics',\n",
       " 'ABCC3_proteomics',\n",
       " 'ABCC4_proteomics',\n",
       " 'ABCD1_proteomics',\n",
       " 'ABCD2_proteomics',\n",
       " 'ABCD3_proteomics',\n",
       " 'ABCD4_proteomics',\n",
       " 'ABCE1_proteomics',\n",
       " 'ABCF1_proteomics',\n",
       " 'ABCF2_proteomics',\n",
       " 'ABCF3_proteomics',\n",
       " 'ABCG1_proteomics',\n",
       " 'ABCG2_proteomics',\n",
       " 'ABHD10_proteomics',\n",
       " 'ABHD11_proteomics',\n",
       " 'ABHD12_proteomics',\n",
       " 'ABHD13_proteomics',\n",
       " 'ABHD14A-ACY1_proteomics',\n",
       " 'ABHD14B_proteomics',\n",
       " 'ABHD16A_proteomics',\n",
       " 'ABHD17A_proteomics',\n",
       " 'ABHD17B_proteomics',\n",
       " 'ABHD17C_proteomics',\n",
       " 'ABHD2_proteomics',\n",
       " 'ABHD3_proteomics',\n",
       " 'ABHD4_proteomics',\n",
       " 'ABHD5_proteomics',\n",
       " 'ABHD6_proteomics',\n",
       " 'ABI1_proteomics',\n",
       " 'ABI2_proteomics',\n",
       " 'ABI3_proteomics',\n",
       " 'ABI3BP_proteomics',\n",
       " 'ABL1_proteomics',\n",
       " 'ABL2_proteomics',\n",
       " 'ABLIM1_proteomics',\n",
       " 'ABLIM2_proteomics',\n",
       " 'ABLIM3_proteomics',\n",
       " 'ABR_proteomics',\n",
       " 'ABRACL_proteomics',\n",
       " 'ABRAXAS1_proteomics',\n",
       " 'ABRAXAS2_proteomics',\n",
       " 'ABT1_proteomics',\n",
       " 'ACAA1_proteomics',\n",
       " 'ACAA2_proteomics',\n",
       " 'ACACA_proteomics',\n",
       " 'ACACB_proteomics',\n",
       " 'ACAD10_proteomics',\n",
       " 'ACAD11_proteomics',\n",
       " 'ACAD8_proteomics',\n",
       " 'ACAD9_proteomics',\n",
       " 'ACADL_proteomics',\n",
       " 'ACADM_proteomics',\n",
       " 'ACADS_proteomics',\n",
       " 'ACADSB_proteomics',\n",
       " 'ACADVL_proteomics',\n",
       " 'ACAN_proteomics',\n",
       " 'ACAP1_proteomics',\n",
       " 'ACAP2_proteomics',\n",
       " 'ACAP3_proteomics',\n",
       " 'ACAT1_proteomics',\n",
       " 'ACAT2_proteomics',\n",
       " 'ACBD3_proteomics',\n",
       " 'ACBD4_proteomics',\n",
       " 'ACBD5_proteomics',\n",
       " 'ACBD6_proteomics',\n",
       " 'ACBD7_proteomics',\n",
       " 'ACD_proteomics',\n",
       " 'ACE_proteomics',\n",
       " 'ACHE_proteomics',\n",
       " 'ACIN1_proteomics',\n",
       " 'ACLY_proteomics',\n",
       " 'ACO1_proteomics',\n",
       " 'ACO2_proteomics',\n",
       " 'ACOT1_proteomics',\n",
       " 'ACOT11_proteomics',\n",
       " 'ACOT13_proteomics',\n",
       " 'ACOT7_proteomics',\n",
       " 'ACOT8_proteomics',\n",
       " 'ACOT9_proteomics',\n",
       " 'ACOX1_proteomics',\n",
       " 'ACOX2_proteomics',\n",
       " 'ACOX3_proteomics',\n",
       " 'ACP1_proteomics',\n",
       " 'ACP2_proteomics',\n",
       " 'ACP5_proteomics',\n",
       " 'ACP6_proteomics',\n",
       " 'ACPP_proteomics',\n",
       " 'ACSBG1_proteomics',\n",
       " 'ACSF2_proteomics',\n",
       " 'ACSF3_proteomics',\n",
       " 'ACSL1_proteomics',\n",
       " 'ACSL3_proteomics',\n",
       " 'ACSL4_proteomics',\n",
       " 'ACSL5_proteomics',\n",
       " 'ACSL6_proteomics',\n",
       " 'ACSM5_proteomics',\n",
       " 'ACSS1_proteomics',\n",
       " 'ACSS2_proteomics',\n",
       " 'ACSS3_proteomics',\n",
       " 'ACTA1_proteomics',\n",
       " 'ACTB_proteomics',\n",
       " 'ACTBL2_proteomics',\n",
       " 'ACTG1_proteomics',\n",
       " 'ACTG2_proteomics',\n",
       " 'ACTL6A_proteomics',\n",
       " 'ACTL6B_proteomics',\n",
       " 'ACTN1_proteomics',\n",
       " 'ACTN2_proteomics',\n",
       " 'ACTN3_proteomics',\n",
       " 'ACTN4_proteomics',\n",
       " 'ACTR10_proteomics',\n",
       " 'ACTR1A_proteomics',\n",
       " 'ACTR1B_proteomics',\n",
       " 'ACTR2_proteomics',\n",
       " 'ACTR3_proteomics',\n",
       " 'ACTR3B_proteomics',\n",
       " 'ACTR5_proteomics',\n",
       " 'ACTR6_proteomics',\n",
       " 'ACTR8_proteomics',\n",
       " 'ACVR1_proteomics',\n",
       " 'ACVR1B_proteomics',\n",
       " 'ACVR2A_proteomics',\n",
       " 'ACY3_proteomics',\n",
       " 'ACYP1_proteomics',\n",
       " 'ACYP2_proteomics',\n",
       " 'ADA_proteomics',\n",
       " 'ADA2_proteomics',\n",
       " 'ADAL_proteomics',\n",
       " 'ADAM10_proteomics',\n",
       " 'ADAM11_proteomics',\n",
       " 'ADAM12_proteomics',\n",
       " 'ADAM15_proteomics',\n",
       " 'ADAM17_proteomics',\n",
       " 'ADAM22_proteomics',\n",
       " 'ADAM23_proteomics',\n",
       " 'ADAM28_proteomics',\n",
       " 'ADAM9_proteomics',\n",
       " 'ADAMTS1_proteomics',\n",
       " 'ADAMTS4_proteomics',\n",
       " 'ADAMTSL2_proteomics',\n",
       " 'ADAMTSL4_proteomics',\n",
       " 'ADAP1_proteomics',\n",
       " 'ADAP2_proteomics',\n",
       " 'ADAR_proteomics',\n",
       " 'ADARB1_proteomics',\n",
       " 'ADAT1_proteomics',\n",
       " 'ADAT2_proteomics',\n",
       " 'ADAT3_proteomics',\n",
       " 'ADCK1_proteomics',\n",
       " 'ADCK5_proteomics',\n",
       " 'ADCY1_proteomics',\n",
       " 'ADCY2_proteomics',\n",
       " 'ADCY3_proteomics',\n",
       " 'ADCY5_proteomics',\n",
       " 'ADCY6_proteomics',\n",
       " 'ADCY7_proteomics',\n",
       " 'ADCY8_proteomics',\n",
       " 'ADCY9_proteomics',\n",
       " 'ADCYAP1_proteomics',\n",
       " 'ADCYAP1R1_proteomics',\n",
       " 'ADD1_proteomics',\n",
       " 'ADD2_proteomics',\n",
       " 'ADD3_proteomics',\n",
       " 'ADGRA1_proteomics',\n",
       " 'ADGRB1_proteomics',\n",
       " 'ADGRB2_proteomics',\n",
       " 'ADGRB3_proteomics',\n",
       " 'ADGRE2_proteomics',\n",
       " 'ADGRE5_proteomics',\n",
       " 'ADGRF5_proteomics',\n",
       " 'ADGRG1_proteomics',\n",
       " 'ADGRG3_proteomics',\n",
       " 'ADGRL1_proteomics',\n",
       " 'ADGRL2_proteomics',\n",
       " 'ADGRL3_proteomics',\n",
       " 'ADGRL4_proteomics',\n",
       " 'ADH1B_proteomics',\n",
       " 'ADH5_proteomics',\n",
       " 'ADHFE1_proteomics',\n",
       " 'ADI1_proteomics',\n",
       " 'ADIPOQ_proteomics',\n",
       " 'ADIPOR2_proteomics',\n",
       " 'ADIRF_proteomics',\n",
       " 'ADK_proteomics',\n",
       " 'ADM_proteomics',\n",
       " 'ADNP_proteomics',\n",
       " 'ADO_proteomics',\n",
       " 'ADORA1_proteomics',\n",
       " 'ADPGK_proteomics',\n",
       " 'ADPRH_proteomics',\n",
       " 'ADPRHL1_proteomics',\n",
       " 'ADPRHL2_proteomics',\n",
       " 'ADPRM_proteomics',\n",
       " 'ADRM1_proteomics',\n",
       " 'ADSL_proteomics',\n",
       " 'ADSS_proteomics',\n",
       " 'ADSSL1_proteomics',\n",
       " 'AEBP1_proteomics',\n",
       " 'AEBP2_proteomics',\n",
       " 'AES_proteomics',\n",
       " 'AFAP1_proteomics',\n",
       " 'AFAP1L1_proteomics',\n",
       " 'AFAP1L2_proteomics',\n",
       " 'AFDN_proteomics',\n",
       " 'AFF4_proteomics',\n",
       " 'AFG1L_proteomics',\n",
       " 'AFG3L2_proteomics',\n",
       " 'AFM_proteomics',\n",
       " 'AFMID_proteomics',\n",
       " 'AFTPH_proteomics',\n",
       " 'AGA_proteomics',\n",
       " 'AGAP1_proteomics',\n",
       " 'AGAP2_proteomics',\n",
       " 'AGAP3_proteomics',\n",
       " 'AGBL4_proteomics',\n",
       " 'AGFG1_proteomics',\n",
       " 'AGFG2_proteomics',\n",
       " 'AGGF1_proteomics',\n",
       " 'AGK_proteomics',\n",
       " 'AGL_proteomics',\n",
       " 'AGMAT_proteomics',\n",
       " 'AGO1_proteomics',\n",
       " 'AGO2_proteomics',\n",
       " 'AGO3_proteomics',\n",
       " 'AGO4_proteomics',\n",
       " 'AGPAT1_proteomics',\n",
       " 'AGPAT2_proteomics',\n",
       " 'AGPAT3_proteomics',\n",
       " 'AGPAT4_proteomics',\n",
       " 'AGPAT5_proteomics',\n",
       " 'AGPS_proteomics',\n",
       " 'AGR2_proteomics',\n",
       " 'AGR3_proteomics',\n",
       " 'AGRN_proteomics',\n",
       " 'AGT_proteomics',\n",
       " 'AGTPBP1_proteomics',\n",
       " 'AGTRAP_proteomics',\n",
       " 'AHCTF1_proteomics',\n",
       " 'AHCY_proteomics',\n",
       " 'AHCYL1_proteomics',\n",
       " 'AHCYL2_proteomics',\n",
       " 'AHDC1_proteomics',\n",
       " 'AHNAK_proteomics',\n",
       " 'AHNAK2_proteomics',\n",
       " 'AHR_proteomics',\n",
       " 'AHSA1_proteomics',\n",
       " 'AHSG_proteomics',\n",
       " 'AHSP_proteomics',\n",
       " 'AIDA_proteomics',\n",
       " 'AIF1_proteomics',\n",
       " 'AIF1L_proteomics',\n",
       " 'AIFM1_proteomics',\n",
       " 'AIFM2_proteomics',\n",
       " 'AIFM3_proteomics',\n",
       " 'AIG1_proteomics',\n",
       " 'AIMP1_proteomics',\n",
       " 'AIMP2_proteomics',\n",
       " 'AIP_proteomics',\n",
       " 'AJAP1_proteomics',\n",
       " 'AJM1_proteomics',\n",
       " 'AJUBA_proteomics',\n",
       " 'AK1_proteomics',\n",
       " 'AK2_proteomics',\n",
       " 'AK3_proteomics',\n",
       " 'AK4_proteomics',\n",
       " 'AK5_proteomics',\n",
       " 'AK6_proteomics',\n",
       " 'AKAP1_proteomics',\n",
       " 'AKAP10_proteomics',\n",
       " 'AKAP11_proteomics',\n",
       " 'AKAP12_proteomics',\n",
       " 'AKAP13_proteomics',\n",
       " 'AKAP17A_proteomics',\n",
       " 'AKAP5_proteomics',\n",
       " 'AKAP6_proteomics',\n",
       " 'AKAP7_proteomics',\n",
       " 'AKAP8_proteomics',\n",
       " 'AKAP8L_proteomics',\n",
       " 'AKAP9_proteomics',\n",
       " 'AKIP1_proteomics',\n",
       " 'AKIRIN2_proteomics',\n",
       " 'AKR1A1_proteomics',\n",
       " 'AKR1B1_proteomics',\n",
       " 'AKR1B10_proteomics',\n",
       " 'AKR1C1_proteomics',\n",
       " 'AKR1C2_proteomics',\n",
       " 'AKR1C3_proteomics',\n",
       " 'AKR1D1_proteomics',\n",
       " 'AKR7A2_proteomics',\n",
       " 'AKR7A3_proteomics',\n",
       " 'AKR7L_proteomics',\n",
       " 'AKT1_proteomics',\n",
       " 'AKT1S1_proteomics',\n",
       " 'AKT2_proteomics',\n",
       " 'AKT3_proteomics',\n",
       " 'AKTIP_proteomics',\n",
       " 'ALAD_proteomics',\n",
       " 'ALAS1_proteomics',\n",
       " 'ALB_proteomics',\n",
       " 'ALCAM_proteomics',\n",
       " 'ALDH16A1_proteomics',\n",
       " 'ALDH18A1_proteomics',\n",
       " 'ALDH1A1_proteomics',\n",
       " 'ALDH1A2_proteomics',\n",
       " 'ALDH1A3_proteomics',\n",
       " 'ALDH1B1_proteomics',\n",
       " 'ALDH1L1_proteomics',\n",
       " 'ALDH1L2_proteomics',\n",
       " 'ALDH2_proteomics',\n",
       " 'ALDH3A1_proteomics',\n",
       " 'ALDH3A2_proteomics',\n",
       " 'ALDH3B1_proteomics',\n",
       " 'ALDH4A1_proteomics',\n",
       " 'ALDH5A1_proteomics',\n",
       " 'ALDH6A1_proteomics',\n",
       " 'ALDH7A1_proteomics',\n",
       " 'ALDH9A1_proteomics',\n",
       " 'ALDOA_proteomics',\n",
       " 'ALDOB_proteomics',\n",
       " 'ALDOC_proteomics',\n",
       " 'ALG1_proteomics',\n",
       " 'ALG11_proteomics',\n",
       " 'ALG12_proteomics',\n",
       " 'ALG13_proteomics',\n",
       " 'ALG14_proteomics',\n",
       " 'ALG2_proteomics',\n",
       " 'ALG5_proteomics',\n",
       " 'ALG6_proteomics',\n",
       " 'ALG8_proteomics',\n",
       " 'ALG9_proteomics',\n",
       " 'ALKBH1_proteomics',\n",
       " 'ALKBH2_proteomics',\n",
       " 'ALKBH3_proteomics',\n",
       " 'ALKBH4_proteomics',\n",
       " 'ALKBH5_proteomics',\n",
       " 'ALKBH6_proteomics',\n",
       " 'ALKBH7_proteomics',\n",
       " 'ALKBH8_proteomics',\n",
       " 'ALMS1_proteomics',\n",
       " 'ALOX12_proteomics',\n",
       " 'ALOX15B_proteomics',\n",
       " 'ALOX5_proteomics',\n",
       " 'ALOX5AP_proteomics',\n",
       " 'ALPL_proteomics',\n",
       " 'ALS2_proteomics',\n",
       " 'ALYREF_proteomics',\n",
       " 'AMACR_proteomics',\n",
       " 'AMBP_proteomics',\n",
       " 'AMBRA1_proteomics',\n",
       " 'AMD1_proteomics',\n",
       " 'AMDHD2_proteomics',\n",
       " 'AMER2_proteomics',\n",
       " 'AMFR_proteomics',\n",
       " 'AMIGO1_proteomics',\n",
       " 'AMN1_proteomics',\n",
       " 'AMOT_proteomics',\n",
       " 'AMOTL1_proteomics',\n",
       " 'AMOTL2_proteomics',\n",
       " 'AMPD2_proteomics',\n",
       " 'AMPD3_proteomics',\n",
       " 'AMPH_proteomics',\n",
       " 'AMT_proteomics',\n",
       " 'AMZ2_proteomics',\n",
       " 'ANAPC1_proteomics',\n",
       " 'ANAPC10_proteomics',\n",
       " 'ANAPC11_proteomics',\n",
       " 'ANAPC13_proteomics',\n",
       " 'ANAPC16_proteomics',\n",
       " 'ANAPC2_proteomics',\n",
       " 'ANAPC4_proteomics',\n",
       " 'ANAPC5_proteomics',\n",
       " 'ANAPC7_proteomics',\n",
       " 'ANG_proteomics',\n",
       " 'ANGEL1_proteomics',\n",
       " 'ANGEL2_proteomics',\n",
       " 'ANGPT2_proteomics',\n",
       " 'ANGPTL1_proteomics',\n",
       " 'ANGPTL2_proteomics',\n",
       " 'ANGPTL4_proteomics',\n",
       " 'ANK1_proteomics',\n",
       " 'ANK2_proteomics',\n",
       " 'ANK3_proteomics',\n",
       " 'ANKFY1_proteomics',\n",
       " 'ANKH_proteomics',\n",
       " 'ANKHD1_proteomics',\n",
       " 'ANKHD1-EIF4EBP3_proteomics',\n",
       " 'ANKIB1_proteomics',\n",
       " 'ANKLE2_proteomics',\n",
       " 'ANKMY2_proteomics',\n",
       " 'ANKRD10_proteomics',\n",
       " 'ANKRD11_proteomics',\n",
       " 'ANKRD12_proteomics',\n",
       " 'ANKRD13A_proteomics',\n",
       " 'ANKRD13D_proteomics',\n",
       " 'ANKRD16_proteomics',\n",
       " 'ANKRD17_proteomics',\n",
       " 'ANKRD22_proteomics',\n",
       " 'ANKRD24_proteomics',\n",
       " 'ANKRD27_proteomics',\n",
       " 'ANKRD28_proteomics',\n",
       " 'ANKRD29_proteomics',\n",
       " 'ANKRD34A_proteomics',\n",
       " 'ANKRD35_proteomics',\n",
       " 'ANKRD39_proteomics',\n",
       " 'ANKRD40_proteomics',\n",
       " 'ANKRD44_proteomics',\n",
       " 'ANKRD45_proteomics',\n",
       " 'ANKRD46_proteomics',\n",
       " 'ANKRD49_proteomics',\n",
       " 'ANKRD50_proteomics',\n",
       " 'ANKRD52_proteomics',\n",
       " 'ANKRD54_proteomics',\n",
       " 'ANKRD63_proteomics',\n",
       " 'ANKS1A_proteomics',\n",
       " 'ANKS1B_proteomics',\n",
       " 'ANKS6_proteomics',\n",
       " 'ANKZF1_proteomics',\n",
       " 'ANLN_proteomics',\n",
       " 'ANO10_proteomics',\n",
       " 'ANO6_proteomics',\n",
       " 'ANO8_proteomics',\n",
       " 'ANOS1_proteomics',\n",
       " 'ANP32A_proteomics',\n",
       " 'ANP32B_proteomics',\n",
       " 'ANP32C_proteomics',\n",
       " 'ANP32E_proteomics',\n",
       " 'ANPEP_proteomics',\n",
       " 'ANTXR1_proteomics',\n",
       " 'ANTXR2_proteomics',\n",
       " 'ANXA1_proteomics',\n",
       " 'ANXA11_proteomics',\n",
       " 'ANXA13_proteomics',\n",
       " 'ANXA2_proteomics',\n",
       " 'ANXA3_proteomics',\n",
       " 'ANXA4_proteomics',\n",
       " 'ANXA5_proteomics',\n",
       " 'ANXA6_proteomics',\n",
       " 'ANXA7_proteomics',\n",
       " 'ANXA9_proteomics',\n",
       " 'AOAH_proteomics',\n",
       " 'AOC3_proteomics',\n",
       " 'AOX1_proteomics',\n",
       " 'AP1AR_proteomics',\n",
       " 'AP1B1_proteomics',\n",
       " 'AP1G1_proteomics',\n",
       " 'AP1G2_proteomics',\n",
       " 'AP1M1_proteomics',\n",
       " 'AP1M2_proteomics',\n",
       " 'AP1S1_proteomics',\n",
       " 'AP1S2_proteomics',\n",
       " 'AP1S3_proteomics',\n",
       " 'AP2A1_proteomics',\n",
       " 'AP2A2_proteomics',\n",
       " 'AP2B1_proteomics',\n",
       " 'AP2M1_proteomics',\n",
       " 'AP2S1_proteomics',\n",
       " 'AP3B1_proteomics',\n",
       " 'AP3B2_proteomics',\n",
       " 'AP3D1_proteomics',\n",
       " 'AP3M1_proteomics',\n",
       " 'AP3M2_proteomics',\n",
       " 'AP3S1_proteomics',\n",
       " 'AP3S2_proteomics',\n",
       " 'AP4B1_proteomics',\n",
       " 'AP4E1_proteomics',\n",
       " 'AP4M1_proteomics',\n",
       " 'AP4S1_proteomics',\n",
       " 'AP5B1_proteomics',\n",
       " 'AP5M1_proteomics',\n",
       " 'AP5S1_proteomics',\n",
       " 'AP5Z1_proteomics',\n",
       " 'APAF1_proteomics',\n",
       " 'APBA1_proteomics',\n",
       " 'APBA2_proteomics',\n",
       " 'APBA3_proteomics',\n",
       " 'APBB1_proteomics',\n",
       " 'APBB1IP_proteomics',\n",
       " 'APBB2_proteomics',\n",
       " 'APC_proteomics',\n",
       " 'APC2_proteomics',\n",
       " 'APCS_proteomics',\n",
       " 'APEH_proteomics',\n",
       " 'APEX1_proteomics',\n",
       " 'APH1A_proteomics',\n",
       " 'APH1B_proteomics',\n",
       " 'API5_proteomics',\n",
       " 'APIP_proteomics',\n",
       " 'APLF_proteomics',\n",
       " 'APLN_proteomics',\n",
       " 'APLNR_proteomics',\n",
       " 'APLP1_proteomics',\n",
       " 'APLP2_proteomics',\n",
       " 'APMAP_proteomics',\n",
       " 'APOA1_proteomics',\n",
       " 'APOA2_proteomics',\n",
       " 'APOA4_proteomics',\n",
       " 'APOB_proteomics',\n",
       " 'APOBEC3B_proteomics',\n",
       " 'APOBEC3C_proteomics',\n",
       " 'APOBEC3D_proteomics',\n",
       " 'APOBEC3F_proteomics',\n",
       " 'APOBEC3G_proteomics',\n",
       " 'APOBR_proteomics',\n",
       " 'APOC1_proteomics',\n",
       " 'APOC2_proteomics',\n",
       " 'APOC3_proteomics',\n",
       " 'APOC4_proteomics',\n",
       " 'APOD_proteomics',\n",
       " 'APOE_proteomics',\n",
       " 'APOF_proteomics',\n",
       " 'APOH_proteomics',\n",
       " 'APOL1_proteomics',\n",
       " 'APOL2_proteomics',\n",
       " 'APOL3_proteomics',\n",
       " 'APOM_proteomics',\n",
       " 'APOO_proteomics',\n",
       " 'APOOL_proteomics',\n",
       " 'APP_proteomics',\n",
       " 'APPBP2_proteomics',\n",
       " 'APPL1_proteomics',\n",
       " 'APPL2_proteomics',\n",
       " 'APRT_proteomics',\n",
       " 'APTX_proteomics',\n",
       " 'AQP1_proteomics',\n",
       " 'AQP4_proteomics',\n",
       " 'AQR_proteomics',\n",
       " 'ARAF_proteomics',\n",
       " 'ARAP1_proteomics',\n",
       " 'ARAP2_proteomics',\n",
       " 'ARAP3_proteomics',\n",
       " 'ARC_proteomics',\n",
       " 'ARCN1_proteomics',\n",
       " 'ARF1_proteomics',\n",
       " 'ARF3_proteomics',\n",
       " 'ARF4_proteomics',\n",
       " 'ARF5_proteomics',\n",
       " 'ARF6_proteomics',\n",
       " 'ARFGAP1_proteomics',\n",
       " 'ARFGAP2_proteomics',\n",
       " 'ARFGAP3_proteomics',\n",
       " 'ARFGEF1_proteomics',\n",
       " 'ARFGEF2_proteomics',\n",
       " 'ARFGEF3_proteomics',\n",
       " 'ARFIP1_proteomics',\n",
       " 'ARFIP2_proteomics',\n",
       " 'ARFRP1_proteomics',\n",
       " 'ARG1_proteomics',\n",
       " 'ARG2_proteomics',\n",
       " 'ARGLU1_proteomics',\n",
       " 'ARHGAP1_proteomics',\n",
       " 'ARHGAP10_proteomics',\n",
       " 'ARHGAP12_proteomics',\n",
       " 'ARHGAP15_proteomics',\n",
       " 'ARHGAP17_proteomics',\n",
       " 'ARHGAP18_proteomics',\n",
       " 'ARHGAP21_proteomics',\n",
       " 'ARHGAP22_proteomics',\n",
       " 'ARHGAP23_proteomics',\n",
       " 'ARHGAP24_proteomics',\n",
       " 'ARHGAP25_proteomics',\n",
       " 'ARHGAP26_proteomics',\n",
       " 'ARHGAP27_proteomics',\n",
       " 'ARHGAP29_proteomics',\n",
       " 'ARHGAP30_proteomics',\n",
       " 'ARHGAP31_proteomics',\n",
       " 'ARHGAP32_proteomics',\n",
       " 'ARHGAP33_proteomics',\n",
       " 'ARHGAP35_proteomics',\n",
       " 'ARHGAP39_proteomics',\n",
       " 'ARHGAP4_proteomics',\n",
       " 'ARHGAP42_proteomics',\n",
       " 'ARHGAP44_proteomics',\n",
       " 'ARHGAP45_proteomics',\n",
       " 'ARHGAP5_proteomics',\n",
       " 'ARHGAP6_proteomics',\n",
       " 'ARHGAP9_proteomics',\n",
       " 'ARHGDIA_proteomics',\n",
       " 'ARHGDIB_proteomics',\n",
       " 'ARHGDIG_proteomics',\n",
       " 'ARHGEF1_proteomics',\n",
       " 'ARHGEF10_proteomics',\n",
       " 'ARHGEF10L_proteomics',\n",
       " 'ARHGEF11_proteomics',\n",
       " 'ARHGEF12_proteomics',\n",
       " 'ARHGEF17_proteomics',\n",
       " 'ARHGEF18_proteomics',\n",
       " 'ARHGEF2_proteomics',\n",
       " 'ARHGEF25_proteomics',\n",
       " 'ARHGEF26_proteomics',\n",
       " 'ARHGEF3_proteomics',\n",
       " 'ARHGEF37_proteomics',\n",
       " 'ARHGEF4_proteomics',\n",
       " 'ARHGEF40_proteomics',\n",
       " 'ARHGEF6_proteomics',\n",
       " 'ARHGEF7_proteomics',\n",
       " 'ARHGEF9_proteomics',\n",
       " 'ARID1A_proteomics',\n",
       " 'ARID1B_proteomics',\n",
       " 'ARID2_proteomics',\n",
       " 'ARID4A_proteomics',\n",
       " 'ARID4B_proteomics',\n",
       " 'ARID5B_proteomics',\n",
       " 'ARIH1_proteomics',\n",
       " 'ARIH2_proteomics',\n",
       " 'ARL1_proteomics',\n",
       " 'ARL10_proteomics',\n",
       " 'ARL13B_proteomics',\n",
       " 'ARL14EP_proteomics',\n",
       " 'ARL15_proteomics',\n",
       " 'ARL16_proteomics',\n",
       " 'ARL2_proteomics',\n",
       " 'ARL2BP_proteomics',\n",
       " 'ARL3_proteomics',\n",
       " 'ARL4A_proteomics',\n",
       " 'ARL4C_proteomics',\n",
       " 'ARL5A_proteomics',\n",
       " 'ARL5B_proteomics',\n",
       " 'ARL6_proteomics',\n",
       " 'ARL6IP1_proteomics',\n",
       " 'ARL6IP4_proteomics',\n",
       " 'ARL6IP5_proteomics',\n",
       " 'ARL6IP6_proteomics',\n",
       " 'ARL8A_proteomics',\n",
       " 'ARL8B_proteomics',\n",
       " 'ARMC1_proteomics',\n",
       " 'ARMC10_proteomics',\n",
       " 'ARMC5_proteomics',\n",
       " 'ARMC6_proteomics',\n",
       " 'ARMC7_proteomics',\n",
       " 'ARMC8_proteomics',\n",
       " 'ARMC9_proteomics',\n",
       " 'ARMCX1_proteomics',\n",
       " 'ARMCX2_proteomics',\n",
       " 'ARMCX3_proteomics',\n",
       " 'ARMCX4_proteomics',\n",
       " 'ARMCX5_proteomics',\n",
       " 'ARMCX6_proteomics',\n",
       " 'ARMH3_proteomics',\n",
       " 'ARMT1_proteomics',\n",
       " 'ARNT_proteomics',\n",
       " 'ARNT2_proteomics',\n",
       " 'ARNTL_proteomics',\n",
       " 'ARPC1A_proteomics',\n",
       " 'ARPC1B_proteomics',\n",
       " 'ARPC2_proteomics',\n",
       " 'ARPC3_proteomics',\n",
       " 'ARPC4_proteomics',\n",
       " 'ARPC4-TTLL3_proteomics',\n",
       " 'ARPC5_proteomics',\n",
       " 'ARPC5L_proteomics',\n",
       " 'ARPIN_proteomics',\n",
       " 'ARPP19_proteomics',\n",
       " 'ARPP21_proteomics',\n",
       " 'ARR3_proteomics',\n",
       " 'ARRB1_proteomics',\n",
       " 'ARRB2_proteomics',\n",
       " 'ARRDC1_proteomics',\n",
       " 'ARSA_proteomics',\n",
       " 'ARSB_proteomics',\n",
       " 'ARSD_proteomics',\n",
       " 'ARSE_proteomics',\n",
       " 'ARSF_proteomics',\n",
       " 'ARSK_proteomics',\n",
       " 'ART3_proteomics',\n",
       " 'ART4_proteomics',\n",
       " 'ARV1_proteomics',\n",
       " 'ARVCF_proteomics',\n",
       " 'ARX_proteomics',\n",
       " 'AS3MT_proteomics',\n",
       " 'ASAH1_proteomics',\n",
       " 'ASAP1_proteomics',\n",
       " 'ASAP2_proteomics',\n",
       " 'ASAP3_proteomics',\n",
       " 'ASB1_proteomics',\n",
       " 'ASB13_proteomics',\n",
       " 'ASB3_proteomics',\n",
       " 'ASB6_proteomics',\n",
       " 'ASB8_proteomics',\n",
       " 'ASCC1_proteomics',\n",
       " 'ASCC2_proteomics',\n",
       " 'ASCC3_proteomics',\n",
       " 'ASCL1_proteomics',\n",
       " 'ASDURF_proteomics',\n",
       " 'ASF1A_proteomics',\n",
       " 'ASF1B_proteomics',\n",
       " 'ASH2L_proteomics',\n",
       " 'ASIC1_proteomics',\n",
       " 'ASIC4_proteomics',\n",
       " 'ASL_proteomics',\n",
       " 'ASMTL_proteomics',\n",
       " 'ASNA1_proteomics',\n",
       " 'ASNS_proteomics',\n",
       " 'ASNSD1_proteomics',\n",
       " 'ASPA_proteomics',\n",
       " 'ASPH_proteomics',\n",
       " 'ASPHD1_proteomics',\n",
       " 'ASPHD2_proteomics',\n",
       " 'ASPN_proteomics',\n",
       " 'ASPRV1_proteomics',\n",
       " 'ASPSCR1_proteomics',\n",
       " 'ASRGL1_proteomics',\n",
       " 'ASS1_proteomics',\n",
       " 'ASTN1_proteomics',\n",
       " 'ASTN2_proteomics',\n",
       " 'ASXL2_proteomics',\n",
       " 'ATAD1_proteomics',\n",
       " 'ATAD2_proteomics',\n",
       " 'ATAD2B_proteomics',\n",
       " 'ATAD3A_proteomics',\n",
       " 'ATAD3B_proteomics',\n",
       " 'ATAT1_proteomics',\n",
       " 'ATCAY_proteomics',\n",
       " 'ATE1_proteomics',\n",
       " 'ATF1_proteomics',\n",
       " 'ATF2_proteomics',\n",
       " 'ATF3_proteomics',\n",
       " 'ATF6_proteomics',\n",
       " 'ATF6B_proteomics',\n",
       " 'ATF7_proteomics',\n",
       " 'ATF7IP_proteomics',\n",
       " 'ATG10_proteomics',\n",
       " 'ATG101_proteomics',\n",
       " 'ATG12_proteomics',\n",
       " 'ATG13_proteomics',\n",
       " 'ATG14_proteomics',\n",
       " 'ATG16L1_proteomics',\n",
       " 'ATG2A_proteomics',\n",
       " 'ATG2B_proteomics',\n",
       " 'ATG3_proteomics',\n",
       " 'ATG4A_proteomics',\n",
       " 'ATG4B_proteomics',\n",
       " 'ATG4C_proteomics',\n",
       " 'ATG5_proteomics',\n",
       " 'ATG7_proteomics',\n",
       " 'ATG9A_proteomics',\n",
       " 'ATIC_proteomics',\n",
       " 'ATL1_proteomics',\n",
       " 'ATL2_proteomics',\n",
       " 'ATL3_proteomics',\n",
       " 'ATM_proteomics',\n",
       " 'ATN1_proteomics',\n",
       " 'ATOX1_proteomics',\n",
       " 'ATP11A_proteomics',\n",
       " 'ATP11B_proteomics',\n",
       " 'ATP11C_proteomics',\n",
       " 'ATP12A_proteomics',\n",
       " 'ATP13A1_proteomics',\n",
       " 'ATP13A2_proteomics',\n",
       " 'ATP13A3_proteomics',\n",
       " 'ATP13A4_proteomics',\n",
       " 'ATP1A1_proteomics',\n",
       " 'ATP1A2_proteomics',\n",
       " 'ATP1A3_proteomics',\n",
       " 'ATP1A4_proteomics',\n",
       " 'ATP1B1_proteomics',\n",
       " 'ATP1B2_proteomics',\n",
       " 'ATP1B3_proteomics',\n",
       " 'ATP1B4_proteomics',\n",
       " 'ATP23_proteomics',\n",
       " 'ATP2A1_proteomics',\n",
       " 'ATP2A2_proteomics',\n",
       " 'ATP2A3_proteomics',\n",
       " 'ATP2B1_proteomics',\n",
       " 'ATP2B2_proteomics',\n",
       " 'ATP2B3_proteomics',\n",
       " 'ATP2B4_proteomics',\n",
       " 'ATP2C1_proteomics',\n",
       " 'ATP4A_proteomics',\n",
       " 'ATP5F1A_proteomics',\n",
       " 'ATP5F1B_proteomics',\n",
       " 'ATP5F1C_proteomics',\n",
       " 'ATP5F1D_proteomics',\n",
       " 'ATP5F1E_proteomics',\n",
       " 'ATP5IF1_proteomics',\n",
       " 'ATP5MC1_proteomics',\n",
       " 'ATP5MD_proteomics',\n",
       " 'ATP5ME_proteomics',\n",
       " 'ATP5MF_proteomics',\n",
       " 'ATP5MF-PTCD1_proteomics',\n",
       " 'ATP5MG_proteomics',\n",
       " 'ATP5MPL_proteomics',\n",
       " 'ATP5PB_proteomics',\n",
       " 'ATP5PD_proteomics',\n",
       " 'ATP5PF_proteomics',\n",
       " 'ATP5PO_proteomics',\n",
       " 'ATP5S_proteomics',\n",
       " 'ATP6AP1_proteomics',\n",
       " 'ATP6AP2_proteomics',\n",
       " 'ATP6V0A1_proteomics',\n",
       " 'ATP6V0A2_proteomics',\n",
       " 'ATP6V0C_proteomics',\n",
       " 'ATP6V0D1_proteomics',\n",
       " 'ATP6V0D2_proteomics',\n",
       " 'ATP6V1A_proteomics',\n",
       " 'ATP6V1B2_proteomics',\n",
       " 'ATP6V1C1_proteomics',\n",
       " 'ATP6V1D_proteomics',\n",
       " 'ATP6V1E1_proteomics',\n",
       " 'ATP6V1E2_proteomics',\n",
       " 'ATP6V1F_proteomics',\n",
       " 'ATP6V1G1_proteomics',\n",
       " 'ATP6V1G2_proteomics',\n",
       " 'ATP6V1H_proteomics',\n",
       " 'ATP7A_proteomics',\n",
       " 'ATP8A1_proteomics',\n",
       " 'ATP8A2_proteomics',\n",
       " 'ATP9A_proteomics',\n",
       " 'ATP9B_proteomics',\n",
       " 'ATPAF1_proteomics',\n",
       " 'ATPAF2_proteomics',\n",
       " 'ATR_proteomics',\n",
       " 'ATRAID_proteomics',\n",
       " 'ATRIP_proteomics',\n",
       " 'ATRN_proteomics',\n",
       " 'ATRNL1_proteomics',\n",
       " 'ATRX_proteomics',\n",
       " 'ATXN1_proteomics',\n",
       " 'ATXN10_proteomics',\n",
       " 'ATXN1L_proteomics',\n",
       " 'ATXN2_proteomics',\n",
       " 'ATXN2L_proteomics',\n",
       " 'ATXN3_proteomics',\n",
       " 'ATXN7_proteomics',\n",
       " 'ATXN7L3_proteomics',\n",
       " 'ATXN7L3B_proteomics',\n",
       " 'AUH_proteomics',\n",
       " 'AUP1_proteomics',\n",
       " 'AURKA_proteomics',\n",
       " 'AURKAIP1_proteomics',\n",
       " 'AURKB_proteomics',\n",
       " 'AUTS2_proteomics',\n",
       " 'AVEN_proteomics',\n",
       " 'AVIL_proteomics',\n",
       " 'AVL9_proteomics',\n",
       " 'AXIN1_proteomics',\n",
       " 'AXL_proteomics',\n",
       " 'AZGP1_proteomics',\n",
       " 'AZI2_proteomics',\n",
       " 'AZU1_proteomics',\n",
       " 'B2M_proteomics',\n",
       " 'B3GALNT1_proteomics',\n",
       " 'B3GALT6_proteomics',\n",
       " 'B3GAT1_proteomics',\n",
       " 'B3GAT3_proteomics',\n",
       " 'B3GLCT_proteomics',\n",
       " 'B3GNT2_proteomics',\n",
       " 'B3GNT9_proteomics',\n",
       " 'B3GNTL1_proteomics',\n",
       " 'B4GALNT1_proteomics',\n",
       " 'B4GALT1_proteomics',\n",
       " 'B4GALT3_proteomics',\n",
       " 'B4GALT4_proteomics',\n",
       " 'B4GALT5_proteomics',\n",
       " 'B4GALT7_proteomics',\n",
       " 'B4GAT1_proteomics',\n",
       " 'B9D1_proteomics',\n",
       " 'B9D2_proteomics',\n",
       " 'BAALC_proteomics',\n",
       " 'BABAM1_proteomics',\n",
       " 'BABAM2_proteomics',\n",
       " 'BACE1_proteomics',\n",
       " 'BACH1_proteomics',\n",
       " 'BAD_proteomics',\n",
       " 'BAG1_proteomics',\n",
       " 'BAG2_proteomics',\n",
       " 'BAG3_proteomics',\n",
       " 'BAG4_proteomics',\n",
       " 'BAG5_proteomics',\n",
       " 'BAG6_proteomics',\n",
       " 'BAHD1_proteomics',\n",
       " 'BAIAP2_proteomics',\n",
       " 'BAIAP2L1_proteomics',\n",
       " 'BAIAP3_proteomics',\n",
       " 'BAK1_proteomics',\n",
       " 'BAMBI_proteomics',\n",
       " 'BANF1_proteomics',\n",
       " 'BANP_proteomics',\n",
       " 'BAP1_proteomics',\n",
       " 'BARD1_proteomics',\n",
       " 'BASP1_proteomics',\n",
       " 'BATF3_proteomics',\n",
       " 'BAX_proteomics',\n",
       " 'BAZ1A_proteomics',\n",
       " 'BAZ1B_proteomics',\n",
       " 'BAZ2A_proteomics',\n",
       " 'BAZ2B_proteomics',\n",
       " 'BBC3_proteomics',\n",
       " 'BBIP1_proteomics',\n",
       " 'BBOX1_proteomics',\n",
       " 'BBS1_proteomics',\n",
       " 'BBS2_proteomics',\n",
       " 'BBS4_proteomics',\n",
       " 'BBS5_proteomics',\n",
       " 'BBS7_proteomics',\n",
       " 'BBS9_proteomics',\n",
       " 'BBX_proteomics',\n",
       " 'BCAM_proteomics',\n",
       " 'BCAN_proteomics',\n",
       " 'BCAP29_proteomics',\n",
       " 'BCAP31_proteomics',\n",
       " 'BCAR1_proteomics',\n",
       " 'BCAR3_proteomics',\n",
       " 'BCAS1_proteomics',\n",
       " 'BCAS2_proteomics',\n",
       " 'BCAS3_proteomics',\n",
       " 'BCAT1_proteomics',\n",
       " 'BCAT2_proteomics',\n",
       " 'BCCIP_proteomics',\n",
       " 'BCDIN3D_proteomics',\n",
       " 'BCHE_proteomics',\n",
       " 'BCKDHA_proteomics',\n",
       " 'BCKDHB_proteomics',\n",
       " 'BCKDK_proteomics',\n",
       " 'BCL10_proteomics',\n",
       " 'BCL11A_proteomics',\n",
       " 'BCL11B_proteomics',\n",
       " 'BCL2_proteomics',\n",
       " 'BCL2L1_proteomics',\n",
       " 'BCL2L11_proteomics',\n",
       " 'BCL2L13_proteomics',\n",
       " 'BCL2L15_proteomics',\n",
       " 'BCL2L2_proteomics',\n",
       " 'BCL3_proteomics',\n",
       " 'BCL6_proteomics',\n",
       " 'BCL7A_proteomics',\n",
       " 'BCL7B_proteomics',\n",
       " 'BCL7C_proteomics',\n",
       " 'BCL9_proteomics',\n",
       " 'BCL9L_proteomics',\n",
       " 'BCLAF1_proteomics',\n",
       " 'BCLAF3_proteomics',\n",
       " 'BCO1_proteomics',\n",
       " 'BCR_proteomics',\n",
       " 'BCS1L_proteomics',\n",
       " 'BDH1_proteomics',\n",
       " 'BDH2_proteomics',\n",
       " 'BDNF_proteomics',\n",
       " 'BECN1_proteomics',\n",
       " 'BEGAIN_proteomics',\n",
       " 'BEND3_proteomics',\n",
       " 'BEND5_proteomics',\n",
       " 'BET1_proteomics',\n",
       " 'BET1L_proteomics',\n",
       " 'BEX4_proteomics',\n",
       " 'BGN_proteomics',\n",
       " 'BHLHB9_proteomics',\n",
       " 'BHLHE22_proteomics',\n",
       " 'BHLHE40_proteomics',\n",
       " 'BHLHE41_proteomics',\n",
       " 'BHMT_proteomics',\n",
       " 'BICC1_proteomics',\n",
       " 'BICD1_proteomics',\n",
       " 'BICD2_proteomics',\n",
       " 'BICRA_proteomics',\n",
       " 'BICRAL_proteomics',\n",
       " 'BID_proteomics',\n",
       " 'BIN1_proteomics',\n",
       " 'BIN2_proteomics',\n",
       " 'BIN3_proteomics',\n",
       " 'BIRC2_proteomics',\n",
       " 'BIRC3_proteomics',\n",
       " 'BIRC6_proteomics',\n",
       " 'BLM_proteomics',\n",
       " 'BLMH_proteomics',\n",
       " 'BLNK_proteomics',\n",
       " 'BLOC1S1_proteomics',\n",
       " 'BLOC1S2_proteomics',\n",
       " 'BLOC1S3_proteomics',\n",
       " 'BLOC1S4_proteomics',\n",
       " 'BLOC1S5_proteomics',\n",
       " 'BLOC1S6_proteomics',\n",
       " ...]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create gene list for t-test\n",
    "prot_col_list = list(Mesench_Normal.columns)\n",
    "prot_col_list.remove('TCGA_subtype')\n",
    "\n",
    "prot_col_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5291 significant comparisons!\n",
      "Number of comparisons: 11140\n",
      "Bonferroni cutoff =  4.4883303411131066e-06\n",
      "Logged Bonferroni cutoff =  -5.347915186501691\n"
     ]
    }
   ],
   "source": [
    "#Call wrap_ttest, pass in formatted dataframe\n",
    "\n",
    "prot_all_comparisons = u.wrap_ttest(Mesench_Normal, 'TCGA_subtype', prot_col_list)\n",
    "prot_num_comparisons = len(prot_col_list)\n",
    "print(\"Number of comparisons:\", prot_num_comparisons)\n",
    "prot_bonferroni_cutoff = .05 / prot_num_comparisons\n",
    "print(\"Bonferroni cutoff = \", prot_bonferroni_cutoff)\n",
    "print(\"Logged Bonferroni cutoff = \", np.log10(prot_bonferroni_cutoff))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant Proteomics comparisons:  5291 \n",
      "\n",
      "               Comparison       P_Value\n",
      "0        PI4KA_proteomics  1.106746e-35\n",
      "1         WDR7_proteomics  2.413637e-33\n",
      "2       ANKS1B_proteomics  4.412780e-33\n",
      "3        DMXL2_proteomics  1.011996e-32\n",
      "4         MADD_proteomics  2.561057e-32\n",
      "...                   ...           ...\n",
      "5286  C12orf73_proteomics  4.344286e-06\n",
      "5287     KDM4A_proteomics  4.350479e-06\n",
      "5288      AGO2_proteomics  4.353342e-06\n",
      "5289    ZNF618_proteomics  4.381527e-06\n",
      "5290     CYTH4_proteomics  4.432903e-06\n",
      "\n",
      "[5291 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "#print significant comparisons \n",
    "prot_all_comparisons = prot_all_comparisons.dropna(axis=0)\n",
    "prot_sig_comparisons = prot_all_comparisons.loc[prot_all_comparisons['P_Value'] <= prot_bonferroni_cutoff]\n",
    "print(\"Number of significant Proteomics comparisons: \", len(prot_sig_comparisons), '\\n')\n",
    "\n",
    "if len(prot_sig_comparisons) > 0:\n",
    "    print(prot_sig_comparisons)\n",
    "\n",
    "prot_sig_comparisons_Mesench_Normal = prot_sig_comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value_MN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>PI4KA_proteomics</td>\n",
       "      <td>1.106746e-35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>WDR7_proteomics</td>\n",
       "      <td>2.413637e-33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ANKS1B_proteomics</td>\n",
       "      <td>4.412780e-33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>DMXL2_proteomics</td>\n",
       "      <td>1.011996e-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>MADD_proteomics</td>\n",
       "      <td>2.561057e-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5286</td>\n",
       "      <td>C12orf73_proteomics</td>\n",
       "      <td>4.344286e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5287</td>\n",
       "      <td>KDM4A_proteomics</td>\n",
       "      <td>4.350479e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5288</td>\n",
       "      <td>AGO2_proteomics</td>\n",
       "      <td>4.353342e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5289</td>\n",
       "      <td>ZNF618_proteomics</td>\n",
       "      <td>4.381527e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5290</td>\n",
       "      <td>CYTH4_proteomics</td>\n",
       "      <td>4.432903e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5291 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Comparison    P_Value_MN\n",
       "0        PI4KA_proteomics  1.106746e-35\n",
       "1         WDR7_proteomics  2.413637e-33\n",
       "2       ANKS1B_proteomics  4.412780e-33\n",
       "3        DMXL2_proteomics  1.011996e-32\n",
       "4         MADD_proteomics  2.561057e-32\n",
       "...                   ...           ...\n",
       "5286  C12orf73_proteomics  4.344286e-06\n",
       "5287     KDM4A_proteomics  4.350479e-06\n",
       "5288      AGO2_proteomics  4.353342e-06\n",
       "5289    ZNF618_proteomics  4.381527e-06\n",
       "5290     CYTH4_proteomics  4.432903e-06\n",
       "\n",
       "[5291 rows x 2 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_sig_comparisons_Mesench_Normal.set_index(\"Comparison\")\n",
    "prot_sig_comparisons_Mesench_Normal = prot_sig_comparisons_Mesench_Normal.rename(columns = {\"P_Value\": \"P_Value_MN\"})\n",
    "prot_sig_comparisons_Mesench_Normal"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mesenchymal and Proneural"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Tumor_Normal</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>height</th>\n",
       "      <th>weight</th>\n",
       "      <th>bmi</th>\n",
       "      <th>country_of_origin</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>ethnicity_self_identify</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>TCGA_subtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>C3L-00104</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>188.0</td>\n",
       "      <td>115.0</td>\n",
       "      <td>32.54</td>\n",
       "      <td>United States</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047437</td>\n",
       "      <td>-0.105908</td>\n",
       "      <td>-0.347076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.459635</td>\n",
       "      <td>0.079452</td>\n",
       "      <td>-0.784983</td>\n",
       "      <td>-0.488441</td>\n",
       "      <td>0.167990</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>164.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>19.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.254535</td>\n",
       "      <td>0.463653</td>\n",
       "      <td>0.580230</td>\n",
       "      <td>0.503044</td>\n",
       "      <td>-0.604986</td>\n",
       "      <td>0.178077</td>\n",
       "      <td>-0.720059</td>\n",
       "      <td>-0.150197</td>\n",
       "      <td>-0.268715</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01146</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>175.0</td>\n",
       "      <td>96.0</td>\n",
       "      <td>31.35</td>\n",
       "      <td>Russia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.176744</td>\n",
       "      <td>0.134078</td>\n",
       "      <td>-0.143252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.425889</td>\n",
       "      <td>-0.008761</td>\n",
       "      <td>0.277088</td>\n",
       "      <td>-0.296844</td>\n",
       "      <td>-0.054669</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01149</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>173.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>30.41</td>\n",
       "      <td>Russia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.191377</td>\n",
       "      <td>0.402318</td>\n",
       "      <td>2.018452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.504515</td>\n",
       "      <td>0.732463</td>\n",
       "      <td>-0.613257</td>\n",
       "      <td>-0.115444</td>\n",
       "      <td>0.566719</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01157</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>66.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>28.58</td>\n",
       "      <td>Russia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.059525</td>\n",
       "      <td>0.087678</td>\n",
       "      <td>-0.080362</td>\n",
       "      <td>0.013261</td>\n",
       "      <td>0.472166</td>\n",
       "      <td>0.335460</td>\n",
       "      <td>0.150904</td>\n",
       "      <td>-0.010514</td>\n",
       "      <td>-0.230476</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 11169 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Sample_Tumor_Normal   age  gender  height  weight    bmi  \\\n",
       "case                                                                 \n",
       "C3L-00104               Tumor  58.0    Male   188.0   115.0  32.54   \n",
       "C3L-00677               Tumor  69.0  Female   164.0    52.0  19.32   \n",
       "C3L-01146               Tumor  59.0    Male   175.0    96.0  31.35   \n",
       "C3L-01149               Tumor  48.0    Male   173.0    91.0  30.41   \n",
       "C3L-01157               Tumor  66.0  Female   162.0    75.0  28.58   \n",
       "\n",
       "          country_of_origin   race               ethnicity  \\\n",
       "case                                                         \n",
       "C3L-00104     United States  White  Not-Hispanic or Latino   \n",
       "C3L-00677               NaN  White  Not-Hispanic or Latino   \n",
       "C3L-01146            Russia    NaN                     NaN   \n",
       "C3L-01149            Russia    NaN                     NaN   \n",
       "C3L-01157            Russia    NaN                     NaN   \n",
       "\n",
       "          ethnicity_self_identify  ... ZSWIM8_proteomics ZW10_proteomics  \\\n",
       "case                               ...                                     \n",
       "C3L-00104               Caucasian  ...         -0.047437       -0.105908   \n",
       "C3L-00677               Caucasian  ...         -0.254535        0.463653   \n",
       "C3L-01146               Caucasian  ...         -0.176744        0.134078   \n",
       "C3L-01149               Caucasian  ...         -0.191377        0.402318   \n",
       "C3L-01157               Caucasian  ...         -0.059525        0.087678   \n",
       "\n",
       "          ZWILCH_proteomics ZWINT_proteomics ZXDC_proteomics  \\\n",
       "case                                                           \n",
       "C3L-00104         -0.347076              NaN        0.459635   \n",
       "C3L-00677          0.580230         0.503044       -0.604986   \n",
       "C3L-01146         -0.143252              NaN       -0.425889   \n",
       "C3L-01149          2.018452              NaN        0.504515   \n",
       "C3L-01157         -0.080362         0.013261        0.472166   \n",
       "\n",
       "          ZYG11B_proteomics  ZYX_proteomics  ZZEF1_proteomics ZZZ3_proteomics  \\\n",
       "case                                                                            \n",
       "C3L-00104          0.079452       -0.784983         -0.488441        0.167990   \n",
       "C3L-00677          0.178077       -0.720059         -0.150197       -0.268715   \n",
       "C3L-01146         -0.008761        0.277088         -0.296844       -0.054669   \n",
       "C3L-01149          0.732463       -0.613257         -0.115444        0.566719   \n",
       "C3L-01157          0.335460        0.150904         -0.010514       -0.230476   \n",
       "\n",
       "          TCGA_subtype  \n",
       "case                    \n",
       "C3L-00104    Proneural  \n",
       "C3L-00677    Proneural  \n",
       "C3L-01146    Proneural  \n",
       "C3L-01149    Proneural  \n",
       "C3L-01157    Proneural  \n",
       "\n",
       "[5 rows x 11169 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create df that are only Mesenchymal and Proneural \n",
    "Mesenchymal = (prot_subtype.loc[prot_subtype['TCGA_subtype'] == 'Mesenchymal'])\n",
    "Proneural = (prot_subtype.loc[prot_subtype['TCGA_subtype'] == 'Proneural'])\n",
    "Proneural.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A2M_proteomics</th>\n",
       "      <th>AAAS_proteomics</th>\n",
       "      <th>AACS_proteomics</th>\n",
       "      <th>AADAT_proteomics</th>\n",
       "      <th>AAED1_proteomics</th>\n",
       "      <th>AAGAB_proteomics</th>\n",
       "      <th>AAK1_proteomics</th>\n",
       "      <th>AAMDC_proteomics</th>\n",
       "      <th>AAMP_proteomics</th>\n",
       "      <th>AAR2_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "      <th>TCGA_subtype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>C3L-00674</td>\n",
       "      <td>1.096470</td>\n",
       "      <td>-0.094421</td>\n",
       "      <td>-0.106304</td>\n",
       "      <td>0.084578</td>\n",
       "      <td>0.176402</td>\n",
       "      <td>-0.248151</td>\n",
       "      <td>0.014061</td>\n",
       "      <td>-0.699773</td>\n",
       "      <td>-0.638462</td>\n",
       "      <td>0.592874</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.065534</td>\n",
       "      <td>-0.306717</td>\n",
       "      <td>0.879991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.883564</td>\n",
       "      <td>-0.172222</td>\n",
       "      <td>0.011876</td>\n",
       "      <td>-0.131889</td>\n",
       "      <td>-0.503581</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01043</td>\n",
       "      <td>0.769231</td>\n",
       "      <td>-0.011039</td>\n",
       "      <td>-0.152467</td>\n",
       "      <td>-0.457411</td>\n",
       "      <td>0.679201</td>\n",
       "      <td>-0.106294</td>\n",
       "      <td>-0.428006</td>\n",
       "      <td>-0.781859</td>\n",
       "      <td>0.039258</td>\n",
       "      <td>-0.080868</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.026241</td>\n",
       "      <td>0.118676</td>\n",
       "      <td>0.444517</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.835224</td>\n",
       "      <td>-0.271005</td>\n",
       "      <td>0.371339</td>\n",
       "      <td>-0.168640</td>\n",
       "      <td>0.144939</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01045</td>\n",
       "      <td>-0.735991</td>\n",
       "      <td>0.125775</td>\n",
       "      <td>-0.285444</td>\n",
       "      <td>-0.620125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.166882</td>\n",
       "      <td>-0.110721</td>\n",
       "      <td>-0.749020</td>\n",
       "      <td>-0.041834</td>\n",
       "      <td>0.601213</td>\n",
       "      <td>...</td>\n",
       "      <td>0.033878</td>\n",
       "      <td>0.051231</td>\n",
       "      <td>0.211483</td>\n",
       "      <td>-0.287100</td>\n",
       "      <td>-0.230124</td>\n",
       "      <td>-0.047399</td>\n",
       "      <td>0.761362</td>\n",
       "      <td>-0.376203</td>\n",
       "      <td>0.476902</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01046</td>\n",
       "      <td>-0.037553</td>\n",
       "      <td>0.239725</td>\n",
       "      <td>0.152238</td>\n",
       "      <td>-0.505107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.133864</td>\n",
       "      <td>0.191524</td>\n",
       "      <td>0.030741</td>\n",
       "      <td>-0.036606</td>\n",
       "      <td>0.115695</td>\n",
       "      <td>...</td>\n",
       "      <td>0.176411</td>\n",
       "      <td>0.003886</td>\n",
       "      <td>0.720046</td>\n",
       "      <td>-0.295464</td>\n",
       "      <td>-0.543543</td>\n",
       "      <td>0.118416</td>\n",
       "      <td>0.963697</td>\n",
       "      <td>0.243561</td>\n",
       "      <td>0.013539</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3L-01049</td>\n",
       "      <td>-0.310086</td>\n",
       "      <td>0.087263</td>\n",
       "      <td>-0.163304</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.009380</td>\n",
       "      <td>0.091515</td>\n",
       "      <td>-0.393903</td>\n",
       "      <td>-0.217689</td>\n",
       "      <td>-0.195168</td>\n",
       "      <td>0.302267</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.053609</td>\n",
       "      <td>0.060704</td>\n",
       "      <td>0.635137</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.199433</td>\n",
       "      <td>-0.340400</td>\n",
       "      <td>0.271082</td>\n",
       "      <td>-0.122117</td>\n",
       "      <td>0.177207</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-02785</td>\n",
       "      <td>0.691145</td>\n",
       "      <td>0.369580</td>\n",
       "      <td>0.196114</td>\n",
       "      <td>0.028247</td>\n",
       "      <td>0.878197</td>\n",
       "      <td>-0.152089</td>\n",
       "      <td>0.457016</td>\n",
       "      <td>0.017247</td>\n",
       "      <td>-0.838311</td>\n",
       "      <td>0.022492</td>\n",
       "      <td>...</td>\n",
       "      <td>0.144039</td>\n",
       "      <td>0.046498</td>\n",
       "      <td>0.667878</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.068733</td>\n",
       "      <td>-0.198584</td>\n",
       "      <td>-0.307852</td>\n",
       "      <td>-0.092755</td>\n",
       "      <td>0.376837</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-02788</td>\n",
       "      <td>-0.340301</td>\n",
       "      <td>-0.381604</td>\n",
       "      <td>0.176850</td>\n",
       "      <td>0.056150</td>\n",
       "      <td>-0.381644</td>\n",
       "      <td>-0.249617</td>\n",
       "      <td>0.737844</td>\n",
       "      <td>0.114329</td>\n",
       "      <td>-0.200811</td>\n",
       "      <td>-0.477411</td>\n",
       "      <td>...</td>\n",
       "      <td>0.196207</td>\n",
       "      <td>-0.076559</td>\n",
       "      <td>-0.723503</td>\n",
       "      <td>-0.915281</td>\n",
       "      <td>0.171156</td>\n",
       "      <td>-0.126355</td>\n",
       "      <td>-0.806935</td>\n",
       "      <td>0.017285</td>\n",
       "      <td>0.148011</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-03088</td>\n",
       "      <td>0.836600</td>\n",
       "      <td>-0.519413</td>\n",
       "      <td>0.408439</td>\n",
       "      <td>0.150180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.201109</td>\n",
       "      <td>1.047781</td>\n",
       "      <td>0.235729</td>\n",
       "      <td>-0.264215</td>\n",
       "      <td>-0.348418</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008570</td>\n",
       "      <td>-0.247399</td>\n",
       "      <td>-0.764748</td>\n",
       "      <td>-0.347858</td>\n",
       "      <td>-0.133599</td>\n",
       "      <td>-0.148703</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>0.253013</td>\n",
       "      <td>0.284701</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-03186</td>\n",
       "      <td>0.767029</td>\n",
       "      <td>-0.231897</td>\n",
       "      <td>0.050841</td>\n",
       "      <td>0.304359</td>\n",
       "      <td>-0.384608</td>\n",
       "      <td>-0.205284</td>\n",
       "      <td>0.513575</td>\n",
       "      <td>0.151769</td>\n",
       "      <td>-0.307037</td>\n",
       "      <td>-0.297650</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.110846</td>\n",
       "      <td>-0.069605</td>\n",
       "      <td>-0.183921</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.681170</td>\n",
       "      <td>0.095601</td>\n",
       "      <td>-0.096716</td>\n",
       "      <td>0.049811</td>\n",
       "      <td>-0.409252</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C3N-03473</td>\n",
       "      <td>-0.333312</td>\n",
       "      <td>0.404318</td>\n",
       "      <td>-0.570473</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.590039</td>\n",
       "      <td>0.423253</td>\n",
       "      <td>0.114616</td>\n",
       "      <td>0.366197</td>\n",
       "      <td>0.193688</td>\n",
       "      <td>0.316815</td>\n",
       "      <td>...</td>\n",
       "      <td>0.241629</td>\n",
       "      <td>0.070969</td>\n",
       "      <td>0.768267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.060488</td>\n",
       "      <td>-0.340455</td>\n",
       "      <td>-0.124841</td>\n",
       "      <td>-0.270490</td>\n",
       "      <td>0.145806</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 11141 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           A2M_proteomics  AAAS_proteomics  AACS_proteomics  AADAT_proteomics  \\\n",
       "case                                                                            \n",
       "C3L-00674        1.096470        -0.094421        -0.106304          0.084578   \n",
       "C3L-01043        0.769231        -0.011039        -0.152467         -0.457411   \n",
       "C3L-01045       -0.735991         0.125775        -0.285444         -0.620125   \n",
       "C3L-01046       -0.037553         0.239725         0.152238         -0.505107   \n",
       "C3L-01049       -0.310086         0.087263        -0.163304               NaN   \n",
       "...                   ...              ...              ...               ...   \n",
       "C3N-02785        0.691145         0.369580         0.196114          0.028247   \n",
       "C3N-02788       -0.340301        -0.381604         0.176850          0.056150   \n",
       "C3N-03088        0.836600        -0.519413         0.408439          0.150180   \n",
       "C3N-03186        0.767029        -0.231897         0.050841          0.304359   \n",
       "C3N-03473       -0.333312         0.404318        -0.570473               NaN   \n",
       "\n",
       "           AAED1_proteomics  AAGAB_proteomics  AAK1_proteomics  \\\n",
       "case                                                             \n",
       "C3L-00674          0.176402         -0.248151         0.014061   \n",
       "C3L-01043          0.679201         -0.106294        -0.428006   \n",
       "C3L-01045               NaN          0.166882        -0.110721   \n",
       "C3L-01046               NaN         -0.133864         0.191524   \n",
       "C3L-01049         -0.009380          0.091515        -0.393903   \n",
       "...                     ...               ...              ...   \n",
       "C3N-02785          0.878197         -0.152089         0.457016   \n",
       "C3N-02788         -0.381644         -0.249617         0.737844   \n",
       "C3N-03088               NaN         -0.201109         1.047781   \n",
       "C3N-03186         -0.384608         -0.205284         0.513575   \n",
       "C3N-03473         -0.590039          0.423253         0.114616   \n",
       "\n",
       "           AAMDC_proteomics  AAMP_proteomics  AAR2_proteomics  ...  \\\n",
       "case                                                           ...   \n",
       "C3L-00674         -0.699773        -0.638462         0.592874  ...   \n",
       "C3L-01043         -0.781859         0.039258        -0.080868  ...   \n",
       "C3L-01045         -0.749020        -0.041834         0.601213  ...   \n",
       "C3L-01046          0.030741        -0.036606         0.115695  ...   \n",
       "C3L-01049         -0.217689        -0.195168         0.302267  ...   \n",
       "...                     ...              ...              ...  ...   \n",
       "C3N-02785          0.017247        -0.838311         0.022492  ...   \n",
       "C3N-02788          0.114329        -0.200811        -0.477411  ...   \n",
       "C3N-03088          0.235729        -0.264215        -0.348418  ...   \n",
       "C3N-03186          0.151769        -0.307037        -0.297650  ...   \n",
       "C3N-03473          0.366197         0.193688         0.316815  ...   \n",
       "\n",
       "           ZSWIM8_proteomics  ZW10_proteomics  ZWILCH_proteomics  \\\n",
       "case                                                               \n",
       "C3L-00674          -0.065534        -0.306717           0.879991   \n",
       "C3L-01043          -0.026241         0.118676           0.444517   \n",
       "C3L-01045           0.033878         0.051231           0.211483   \n",
       "C3L-01046           0.176411         0.003886           0.720046   \n",
       "C3L-01049          -0.053609         0.060704           0.635137   \n",
       "...                      ...              ...                ...   \n",
       "C3N-02785           0.144039         0.046498           0.667878   \n",
       "C3N-02788           0.196207        -0.076559          -0.723503   \n",
       "C3N-03088           0.008570        -0.247399          -0.764748   \n",
       "C3N-03186          -0.110846        -0.069605          -0.183921   \n",
       "C3N-03473           0.241629         0.070969           0.768267   \n",
       "\n",
       "           ZWINT_proteomics  ZXDC_proteomics  ZYG11B_proteomics  \\\n",
       "case                                                              \n",
       "C3L-00674               NaN         0.883564          -0.172222   \n",
       "C3L-01043               NaN         0.835224          -0.271005   \n",
       "C3L-01045         -0.287100        -0.230124          -0.047399   \n",
       "C3L-01046         -0.295464        -0.543543           0.118416   \n",
       "C3L-01049               NaN         0.199433          -0.340400   \n",
       "...                     ...              ...                ...   \n",
       "C3N-02785               NaN        -0.068733          -0.198584   \n",
       "C3N-02788         -0.915281         0.171156          -0.126355   \n",
       "C3N-03088         -0.347858        -0.133599          -0.148703   \n",
       "C3N-03186               NaN        -0.681170           0.095601   \n",
       "C3N-03473               NaN        -0.060488          -0.340455   \n",
       "\n",
       "           ZYX_proteomics  ZZEF1_proteomics  ZZZ3_proteomics  TCGA_subtype  \n",
       "case                                                                        \n",
       "C3L-00674        0.011876         -0.131889        -0.503581   Mesenchymal  \n",
       "C3L-01043        0.371339         -0.168640         0.144939   Mesenchymal  \n",
       "C3L-01045        0.761362         -0.376203         0.476902   Mesenchymal  \n",
       "C3L-01046        0.963697          0.243561         0.013539   Mesenchymal  \n",
       "C3L-01049        0.271082         -0.122117         0.177207   Mesenchymal  \n",
       "...                   ...               ...              ...           ...  \n",
       "C3N-02785       -0.307852         -0.092755         0.376837     Proneural  \n",
       "C3N-02788       -0.806935          0.017285         0.148011     Proneural  \n",
       "C3N-03088        0.000129          0.253013         0.284701     Proneural  \n",
       "C3N-03186       -0.096716          0.049811        -0.409252     Proneural  \n",
       "C3N-03473       -0.124841         -0.270490         0.145806     Proneural  \n",
       "\n",
       "[68 rows x 11141 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create Mesenchymal and Proneural df \n",
    "Mesench_Proneural = Mesenchymal.append(Proneural)\n",
    "Mesench_Proneural.drop(Mesench_Proneural.iloc[:, :28], axis=1, inplace=True)\n",
    "\n",
    "Mesench_Proneural"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['A2M_proteomics',\n",
       " 'AAAS_proteomics',\n",
       " 'AACS_proteomics',\n",
       " 'AADAT_proteomics',\n",
       " 'AAED1_proteomics',\n",
       " 'AAGAB_proteomics',\n",
       " 'AAK1_proteomics',\n",
       " 'AAMDC_proteomics',\n",
       " 'AAMP_proteomics',\n",
       " 'AAR2_proteomics',\n",
       " 'AARS_proteomics',\n",
       " 'AARS2_proteomics',\n",
       " 'AARSD1_proteomics',\n",
       " 'AASDHPPT_proteomics',\n",
       " 'AASS_proteomics',\n",
       " 'AATF_proteomics',\n",
       " 'AATK_proteomics',\n",
       " 'ABAT_proteomics',\n",
       " 'ABCA1_proteomics',\n",
       " 'ABCA2_proteomics',\n",
       " 'ABCA3_proteomics',\n",
       " 'ABCA5_proteomics',\n",
       " 'ABCA8_proteomics',\n",
       " 'ABCB1_proteomics',\n",
       " 'ABCB10_proteomics',\n",
       " 'ABCB6_proteomics',\n",
       " 'ABCB7_proteomics',\n",
       " 'ABCB8_proteomics',\n",
       " 'ABCB9_proteomics',\n",
       " 'ABCC1_proteomics',\n",
       " 'ABCC3_proteomics',\n",
       " 'ABCC4_proteomics',\n",
       " 'ABCD1_proteomics',\n",
       " 'ABCD2_proteomics',\n",
       " 'ABCD3_proteomics',\n",
       " 'ABCD4_proteomics',\n",
       " 'ABCE1_proteomics',\n",
       " 'ABCF1_proteomics',\n",
       " 'ABCF2_proteomics',\n",
       " 'ABCF3_proteomics',\n",
       " 'ABCG1_proteomics',\n",
       " 'ABCG2_proteomics',\n",
       " 'ABHD10_proteomics',\n",
       " 'ABHD11_proteomics',\n",
       " 'ABHD12_proteomics',\n",
       " 'ABHD13_proteomics',\n",
       " 'ABHD14A-ACY1_proteomics',\n",
       " 'ABHD14B_proteomics',\n",
       " 'ABHD16A_proteomics',\n",
       " 'ABHD17A_proteomics',\n",
       " 'ABHD17B_proteomics',\n",
       " 'ABHD17C_proteomics',\n",
       " 'ABHD2_proteomics',\n",
       " 'ABHD3_proteomics',\n",
       " 'ABHD4_proteomics',\n",
       " 'ABHD5_proteomics',\n",
       " 'ABHD6_proteomics',\n",
       " 'ABI1_proteomics',\n",
       " 'ABI2_proteomics',\n",
       " 'ABI3_proteomics',\n",
       " 'ABI3BP_proteomics',\n",
       " 'ABL1_proteomics',\n",
       " 'ABL2_proteomics',\n",
       " 'ABLIM1_proteomics',\n",
       " 'ABLIM2_proteomics',\n",
       " 'ABLIM3_proteomics',\n",
       " 'ABR_proteomics',\n",
       " 'ABRACL_proteomics',\n",
       " 'ABRAXAS1_proteomics',\n",
       " 'ABRAXAS2_proteomics',\n",
       " 'ABT1_proteomics',\n",
       " 'ACAA1_proteomics',\n",
       " 'ACAA2_proteomics',\n",
       " 'ACACA_proteomics',\n",
       " 'ACACB_proteomics',\n",
       " 'ACAD10_proteomics',\n",
       " 'ACAD11_proteomics',\n",
       " 'ACAD8_proteomics',\n",
       " 'ACAD9_proteomics',\n",
       " 'ACADL_proteomics',\n",
       " 'ACADM_proteomics',\n",
       " 'ACADS_proteomics',\n",
       " 'ACADSB_proteomics',\n",
       " 'ACADVL_proteomics',\n",
       " 'ACAN_proteomics',\n",
       " 'ACAP1_proteomics',\n",
       " 'ACAP2_proteomics',\n",
       " 'ACAP3_proteomics',\n",
       " 'ACAT1_proteomics',\n",
       " 'ACAT2_proteomics',\n",
       " 'ACBD3_proteomics',\n",
       " 'ACBD4_proteomics',\n",
       " 'ACBD5_proteomics',\n",
       " 'ACBD6_proteomics',\n",
       " 'ACBD7_proteomics',\n",
       " 'ACD_proteomics',\n",
       " 'ACE_proteomics',\n",
       " 'ACHE_proteomics',\n",
       " 'ACIN1_proteomics',\n",
       " 'ACLY_proteomics',\n",
       " 'ACO1_proteomics',\n",
       " 'ACO2_proteomics',\n",
       " 'ACOT1_proteomics',\n",
       " 'ACOT11_proteomics',\n",
       " 'ACOT13_proteomics',\n",
       " 'ACOT7_proteomics',\n",
       " 'ACOT8_proteomics',\n",
       " 'ACOT9_proteomics',\n",
       " 'ACOX1_proteomics',\n",
       " 'ACOX2_proteomics',\n",
       " 'ACOX3_proteomics',\n",
       " 'ACP1_proteomics',\n",
       " 'ACP2_proteomics',\n",
       " 'ACP5_proteomics',\n",
       " 'ACP6_proteomics',\n",
       " 'ACPP_proteomics',\n",
       " 'ACSBG1_proteomics',\n",
       " 'ACSF2_proteomics',\n",
       " 'ACSF3_proteomics',\n",
       " 'ACSL1_proteomics',\n",
       " 'ACSL3_proteomics',\n",
       " 'ACSL4_proteomics',\n",
       " 'ACSL5_proteomics',\n",
       " 'ACSL6_proteomics',\n",
       " 'ACSM5_proteomics',\n",
       " 'ACSS1_proteomics',\n",
       " 'ACSS2_proteomics',\n",
       " 'ACSS3_proteomics',\n",
       " 'ACTA1_proteomics',\n",
       " 'ACTB_proteomics',\n",
       " 'ACTBL2_proteomics',\n",
       " 'ACTG1_proteomics',\n",
       " 'ACTG2_proteomics',\n",
       " 'ACTL6A_proteomics',\n",
       " 'ACTL6B_proteomics',\n",
       " 'ACTN1_proteomics',\n",
       " 'ACTN2_proteomics',\n",
       " 'ACTN3_proteomics',\n",
       " 'ACTN4_proteomics',\n",
       " 'ACTR10_proteomics',\n",
       " 'ACTR1A_proteomics',\n",
       " 'ACTR1B_proteomics',\n",
       " 'ACTR2_proteomics',\n",
       " 'ACTR3_proteomics',\n",
       " 'ACTR3B_proteomics',\n",
       " 'ACTR5_proteomics',\n",
       " 'ACTR6_proteomics',\n",
       " 'ACTR8_proteomics',\n",
       " 'ACVR1_proteomics',\n",
       " 'ACVR1B_proteomics',\n",
       " 'ACVR2A_proteomics',\n",
       " 'ACY3_proteomics',\n",
       " 'ACYP1_proteomics',\n",
       " 'ACYP2_proteomics',\n",
       " 'ADA_proteomics',\n",
       " 'ADA2_proteomics',\n",
       " 'ADAL_proteomics',\n",
       " 'ADAM10_proteomics',\n",
       " 'ADAM11_proteomics',\n",
       " 'ADAM12_proteomics',\n",
       " 'ADAM15_proteomics',\n",
       " 'ADAM17_proteomics',\n",
       " 'ADAM22_proteomics',\n",
       " 'ADAM23_proteomics',\n",
       " 'ADAM28_proteomics',\n",
       " 'ADAM9_proteomics',\n",
       " 'ADAMTS1_proteomics',\n",
       " 'ADAMTS4_proteomics',\n",
       " 'ADAMTSL2_proteomics',\n",
       " 'ADAMTSL4_proteomics',\n",
       " 'ADAP1_proteomics',\n",
       " 'ADAP2_proteomics',\n",
       " 'ADAR_proteomics',\n",
       " 'ADARB1_proteomics',\n",
       " 'ADAT1_proteomics',\n",
       " 'ADAT2_proteomics',\n",
       " 'ADAT3_proteomics',\n",
       " 'ADCK1_proteomics',\n",
       " 'ADCK5_proteomics',\n",
       " 'ADCY1_proteomics',\n",
       " 'ADCY2_proteomics',\n",
       " 'ADCY3_proteomics',\n",
       " 'ADCY5_proteomics',\n",
       " 'ADCY6_proteomics',\n",
       " 'ADCY7_proteomics',\n",
       " 'ADCY8_proteomics',\n",
       " 'ADCY9_proteomics',\n",
       " 'ADCYAP1_proteomics',\n",
       " 'ADCYAP1R1_proteomics',\n",
       " 'ADD1_proteomics',\n",
       " 'ADD2_proteomics',\n",
       " 'ADD3_proteomics',\n",
       " 'ADGRA1_proteomics',\n",
       " 'ADGRB1_proteomics',\n",
       " 'ADGRB2_proteomics',\n",
       " 'ADGRB3_proteomics',\n",
       " 'ADGRE2_proteomics',\n",
       " 'ADGRE5_proteomics',\n",
       " 'ADGRF5_proteomics',\n",
       " 'ADGRG1_proteomics',\n",
       " 'ADGRG3_proteomics',\n",
       " 'ADGRL1_proteomics',\n",
       " 'ADGRL2_proteomics',\n",
       " 'ADGRL3_proteomics',\n",
       " 'ADGRL4_proteomics',\n",
       " 'ADH1B_proteomics',\n",
       " 'ADH5_proteomics',\n",
       " 'ADHFE1_proteomics',\n",
       " 'ADI1_proteomics',\n",
       " 'ADIPOQ_proteomics',\n",
       " 'ADIPOR2_proteomics',\n",
       " 'ADIRF_proteomics',\n",
       " 'ADK_proteomics',\n",
       " 'ADM_proteomics',\n",
       " 'ADNP_proteomics',\n",
       " 'ADO_proteomics',\n",
       " 'ADORA1_proteomics',\n",
       " 'ADPGK_proteomics',\n",
       " 'ADPRH_proteomics',\n",
       " 'ADPRHL1_proteomics',\n",
       " 'ADPRHL2_proteomics',\n",
       " 'ADPRM_proteomics',\n",
       " 'ADRM1_proteomics',\n",
       " 'ADSL_proteomics',\n",
       " 'ADSS_proteomics',\n",
       " 'ADSSL1_proteomics',\n",
       " 'AEBP1_proteomics',\n",
       " 'AEBP2_proteomics',\n",
       " 'AES_proteomics',\n",
       " 'AFAP1_proteomics',\n",
       " 'AFAP1L1_proteomics',\n",
       " 'AFAP1L2_proteomics',\n",
       " 'AFDN_proteomics',\n",
       " 'AFF4_proteomics',\n",
       " 'AFG1L_proteomics',\n",
       " 'AFG3L2_proteomics',\n",
       " 'AFM_proteomics',\n",
       " 'AFMID_proteomics',\n",
       " 'AFTPH_proteomics',\n",
       " 'AGA_proteomics',\n",
       " 'AGAP1_proteomics',\n",
       " 'AGAP2_proteomics',\n",
       " 'AGAP3_proteomics',\n",
       " 'AGBL4_proteomics',\n",
       " 'AGFG1_proteomics',\n",
       " 'AGFG2_proteomics',\n",
       " 'AGGF1_proteomics',\n",
       " 'AGK_proteomics',\n",
       " 'AGL_proteomics',\n",
       " 'AGMAT_proteomics',\n",
       " 'AGO1_proteomics',\n",
       " 'AGO2_proteomics',\n",
       " 'AGO3_proteomics',\n",
       " 'AGO4_proteomics',\n",
       " 'AGPAT1_proteomics',\n",
       " 'AGPAT2_proteomics',\n",
       " 'AGPAT3_proteomics',\n",
       " 'AGPAT4_proteomics',\n",
       " 'AGPAT5_proteomics',\n",
       " 'AGPS_proteomics',\n",
       " 'AGR2_proteomics',\n",
       " 'AGR3_proteomics',\n",
       " 'AGRN_proteomics',\n",
       " 'AGT_proteomics',\n",
       " 'AGTPBP1_proteomics',\n",
       " 'AGTRAP_proteomics',\n",
       " 'AHCTF1_proteomics',\n",
       " 'AHCY_proteomics',\n",
       " 'AHCYL1_proteomics',\n",
       " 'AHCYL2_proteomics',\n",
       " 'AHDC1_proteomics',\n",
       " 'AHNAK_proteomics',\n",
       " 'AHNAK2_proteomics',\n",
       " 'AHR_proteomics',\n",
       " 'AHSA1_proteomics',\n",
       " 'AHSG_proteomics',\n",
       " 'AHSP_proteomics',\n",
       " 'AIDA_proteomics',\n",
       " 'AIF1_proteomics',\n",
       " 'AIF1L_proteomics',\n",
       " 'AIFM1_proteomics',\n",
       " 'AIFM2_proteomics',\n",
       " 'AIFM3_proteomics',\n",
       " 'AIG1_proteomics',\n",
       " 'AIMP1_proteomics',\n",
       " 'AIMP2_proteomics',\n",
       " 'AIP_proteomics',\n",
       " 'AJAP1_proteomics',\n",
       " 'AJM1_proteomics',\n",
       " 'AJUBA_proteomics',\n",
       " 'AK1_proteomics',\n",
       " 'AK2_proteomics',\n",
       " 'AK3_proteomics',\n",
       " 'AK4_proteomics',\n",
       " 'AK5_proteomics',\n",
       " 'AK6_proteomics',\n",
       " 'AKAP1_proteomics',\n",
       " 'AKAP10_proteomics',\n",
       " 'AKAP11_proteomics',\n",
       " 'AKAP12_proteomics',\n",
       " 'AKAP13_proteomics',\n",
       " 'AKAP17A_proteomics',\n",
       " 'AKAP5_proteomics',\n",
       " 'AKAP6_proteomics',\n",
       " 'AKAP7_proteomics',\n",
       " 'AKAP8_proteomics',\n",
       " 'AKAP8L_proteomics',\n",
       " 'AKAP9_proteomics',\n",
       " 'AKIP1_proteomics',\n",
       " 'AKIRIN2_proteomics',\n",
       " 'AKR1A1_proteomics',\n",
       " 'AKR1B1_proteomics',\n",
       " 'AKR1B10_proteomics',\n",
       " 'AKR1C1_proteomics',\n",
       " 'AKR1C2_proteomics',\n",
       " 'AKR1C3_proteomics',\n",
       " 'AKR1D1_proteomics',\n",
       " 'AKR7A2_proteomics',\n",
       " 'AKR7A3_proteomics',\n",
       " 'AKR7L_proteomics',\n",
       " 'AKT1_proteomics',\n",
       " 'AKT1S1_proteomics',\n",
       " 'AKT2_proteomics',\n",
       " 'AKT3_proteomics',\n",
       " 'AKTIP_proteomics',\n",
       " 'ALAD_proteomics',\n",
       " 'ALAS1_proteomics',\n",
       " 'ALB_proteomics',\n",
       " 'ALCAM_proteomics',\n",
       " 'ALDH16A1_proteomics',\n",
       " 'ALDH18A1_proteomics',\n",
       " 'ALDH1A1_proteomics',\n",
       " 'ALDH1A2_proteomics',\n",
       " 'ALDH1A3_proteomics',\n",
       " 'ALDH1B1_proteomics',\n",
       " 'ALDH1L1_proteomics',\n",
       " 'ALDH1L2_proteomics',\n",
       " 'ALDH2_proteomics',\n",
       " 'ALDH3A1_proteomics',\n",
       " 'ALDH3A2_proteomics',\n",
       " 'ALDH3B1_proteomics',\n",
       " 'ALDH4A1_proteomics',\n",
       " 'ALDH5A1_proteomics',\n",
       " 'ALDH6A1_proteomics',\n",
       " 'ALDH7A1_proteomics',\n",
       " 'ALDH9A1_proteomics',\n",
       " 'ALDOA_proteomics',\n",
       " 'ALDOB_proteomics',\n",
       " 'ALDOC_proteomics',\n",
       " 'ALG1_proteomics',\n",
       " 'ALG11_proteomics',\n",
       " 'ALG12_proteomics',\n",
       " 'ALG13_proteomics',\n",
       " 'ALG14_proteomics',\n",
       " 'ALG2_proteomics',\n",
       " 'ALG5_proteomics',\n",
       " 'ALG6_proteomics',\n",
       " 'ALG8_proteomics',\n",
       " 'ALG9_proteomics',\n",
       " 'ALKBH1_proteomics',\n",
       " 'ALKBH2_proteomics',\n",
       " 'ALKBH3_proteomics',\n",
       " 'ALKBH4_proteomics',\n",
       " 'ALKBH5_proteomics',\n",
       " 'ALKBH6_proteomics',\n",
       " 'ALKBH7_proteomics',\n",
       " 'ALKBH8_proteomics',\n",
       " 'ALMS1_proteomics',\n",
       " 'ALOX12_proteomics',\n",
       " 'ALOX15B_proteomics',\n",
       " 'ALOX5_proteomics',\n",
       " 'ALOX5AP_proteomics',\n",
       " 'ALPL_proteomics',\n",
       " 'ALS2_proteomics',\n",
       " 'ALYREF_proteomics',\n",
       " 'AMACR_proteomics',\n",
       " 'AMBP_proteomics',\n",
       " 'AMBRA1_proteomics',\n",
       " 'AMD1_proteomics',\n",
       " 'AMDHD2_proteomics',\n",
       " 'AMER2_proteomics',\n",
       " 'AMFR_proteomics',\n",
       " 'AMIGO1_proteomics',\n",
       " 'AMN1_proteomics',\n",
       " 'AMOT_proteomics',\n",
       " 'AMOTL1_proteomics',\n",
       " 'AMOTL2_proteomics',\n",
       " 'AMPD2_proteomics',\n",
       " 'AMPD3_proteomics',\n",
       " 'AMPH_proteomics',\n",
       " 'AMT_proteomics',\n",
       " 'AMZ2_proteomics',\n",
       " 'ANAPC1_proteomics',\n",
       " 'ANAPC10_proteomics',\n",
       " 'ANAPC11_proteomics',\n",
       " 'ANAPC13_proteomics',\n",
       " 'ANAPC16_proteomics',\n",
       " 'ANAPC2_proteomics',\n",
       " 'ANAPC4_proteomics',\n",
       " 'ANAPC5_proteomics',\n",
       " 'ANAPC7_proteomics',\n",
       " 'ANG_proteomics',\n",
       " 'ANGEL1_proteomics',\n",
       " 'ANGEL2_proteomics',\n",
       " 'ANGPT2_proteomics',\n",
       " 'ANGPTL1_proteomics',\n",
       " 'ANGPTL2_proteomics',\n",
       " 'ANGPTL4_proteomics',\n",
       " 'ANK1_proteomics',\n",
       " 'ANK2_proteomics',\n",
       " 'ANK3_proteomics',\n",
       " 'ANKFY1_proteomics',\n",
       " 'ANKH_proteomics',\n",
       " 'ANKHD1_proteomics',\n",
       " 'ANKHD1-EIF4EBP3_proteomics',\n",
       " 'ANKIB1_proteomics',\n",
       " 'ANKLE2_proteomics',\n",
       " 'ANKMY2_proteomics',\n",
       " 'ANKRD10_proteomics',\n",
       " 'ANKRD11_proteomics',\n",
       " 'ANKRD12_proteomics',\n",
       " 'ANKRD13A_proteomics',\n",
       " 'ANKRD13D_proteomics',\n",
       " 'ANKRD16_proteomics',\n",
       " 'ANKRD17_proteomics',\n",
       " 'ANKRD22_proteomics',\n",
       " 'ANKRD24_proteomics',\n",
       " 'ANKRD27_proteomics',\n",
       " 'ANKRD28_proteomics',\n",
       " 'ANKRD29_proteomics',\n",
       " 'ANKRD34A_proteomics',\n",
       " 'ANKRD35_proteomics',\n",
       " 'ANKRD39_proteomics',\n",
       " 'ANKRD40_proteomics',\n",
       " 'ANKRD44_proteomics',\n",
       " 'ANKRD45_proteomics',\n",
       " 'ANKRD46_proteomics',\n",
       " 'ANKRD49_proteomics',\n",
       " 'ANKRD50_proteomics',\n",
       " 'ANKRD52_proteomics',\n",
       " 'ANKRD54_proteomics',\n",
       " 'ANKRD63_proteomics',\n",
       " 'ANKS1A_proteomics',\n",
       " 'ANKS1B_proteomics',\n",
       " 'ANKS6_proteomics',\n",
       " 'ANKZF1_proteomics',\n",
       " 'ANLN_proteomics',\n",
       " 'ANO10_proteomics',\n",
       " 'ANO6_proteomics',\n",
       " 'ANO8_proteomics',\n",
       " 'ANOS1_proteomics',\n",
       " 'ANP32A_proteomics',\n",
       " 'ANP32B_proteomics',\n",
       " 'ANP32C_proteomics',\n",
       " 'ANP32E_proteomics',\n",
       " 'ANPEP_proteomics',\n",
       " 'ANTXR1_proteomics',\n",
       " 'ANTXR2_proteomics',\n",
       " 'ANXA1_proteomics',\n",
       " 'ANXA11_proteomics',\n",
       " 'ANXA13_proteomics',\n",
       " 'ANXA2_proteomics',\n",
       " 'ANXA3_proteomics',\n",
       " 'ANXA4_proteomics',\n",
       " 'ANXA5_proteomics',\n",
       " 'ANXA6_proteomics',\n",
       " 'ANXA7_proteomics',\n",
       " 'ANXA9_proteomics',\n",
       " 'AOAH_proteomics',\n",
       " 'AOC3_proteomics',\n",
       " 'AOX1_proteomics',\n",
       " 'AP1AR_proteomics',\n",
       " 'AP1B1_proteomics',\n",
       " 'AP1G1_proteomics',\n",
       " 'AP1G2_proteomics',\n",
       " 'AP1M1_proteomics',\n",
       " 'AP1M2_proteomics',\n",
       " 'AP1S1_proteomics',\n",
       " 'AP1S2_proteomics',\n",
       " 'AP1S3_proteomics',\n",
       " 'AP2A1_proteomics',\n",
       " 'AP2A2_proteomics',\n",
       " 'AP2B1_proteomics',\n",
       " 'AP2M1_proteomics',\n",
       " 'AP2S1_proteomics',\n",
       " 'AP3B1_proteomics',\n",
       " 'AP3B2_proteomics',\n",
       " 'AP3D1_proteomics',\n",
       " 'AP3M1_proteomics',\n",
       " 'AP3M2_proteomics',\n",
       " 'AP3S1_proteomics',\n",
       " 'AP3S2_proteomics',\n",
       " 'AP4B1_proteomics',\n",
       " 'AP4E1_proteomics',\n",
       " 'AP4M1_proteomics',\n",
       " 'AP4S1_proteomics',\n",
       " 'AP5B1_proteomics',\n",
       " 'AP5M1_proteomics',\n",
       " 'AP5S1_proteomics',\n",
       " 'AP5Z1_proteomics',\n",
       " 'APAF1_proteomics',\n",
       " 'APBA1_proteomics',\n",
       " 'APBA2_proteomics',\n",
       " 'APBA3_proteomics',\n",
       " 'APBB1_proteomics',\n",
       " 'APBB1IP_proteomics',\n",
       " 'APBB2_proteomics',\n",
       " 'APC_proteomics',\n",
       " 'APC2_proteomics',\n",
       " 'APCS_proteomics',\n",
       " 'APEH_proteomics',\n",
       " 'APEX1_proteomics',\n",
       " 'APH1A_proteomics',\n",
       " 'APH1B_proteomics',\n",
       " 'API5_proteomics',\n",
       " 'APIP_proteomics',\n",
       " 'APLF_proteomics',\n",
       " 'APLN_proteomics',\n",
       " 'APLNR_proteomics',\n",
       " 'APLP1_proteomics',\n",
       " 'APLP2_proteomics',\n",
       " 'APMAP_proteomics',\n",
       " 'APOA1_proteomics',\n",
       " 'APOA2_proteomics',\n",
       " 'APOA4_proteomics',\n",
       " 'APOB_proteomics',\n",
       " 'APOBEC3B_proteomics',\n",
       " 'APOBEC3C_proteomics',\n",
       " 'APOBEC3D_proteomics',\n",
       " 'APOBEC3F_proteomics',\n",
       " 'APOBEC3G_proteomics',\n",
       " 'APOBR_proteomics',\n",
       " 'APOC1_proteomics',\n",
       " 'APOC2_proteomics',\n",
       " 'APOC3_proteomics',\n",
       " 'APOC4_proteomics',\n",
       " 'APOD_proteomics',\n",
       " 'APOE_proteomics',\n",
       " 'APOF_proteomics',\n",
       " 'APOH_proteomics',\n",
       " 'APOL1_proteomics',\n",
       " 'APOL2_proteomics',\n",
       " 'APOL3_proteomics',\n",
       " 'APOM_proteomics',\n",
       " 'APOO_proteomics',\n",
       " 'APOOL_proteomics',\n",
       " 'APP_proteomics',\n",
       " 'APPBP2_proteomics',\n",
       " 'APPL1_proteomics',\n",
       " 'APPL2_proteomics',\n",
       " 'APRT_proteomics',\n",
       " 'APTX_proteomics',\n",
       " 'AQP1_proteomics',\n",
       " 'AQP4_proteomics',\n",
       " 'AQR_proteomics',\n",
       " 'ARAF_proteomics',\n",
       " 'ARAP1_proteomics',\n",
       " 'ARAP2_proteomics',\n",
       " 'ARAP3_proteomics',\n",
       " 'ARC_proteomics',\n",
       " 'ARCN1_proteomics',\n",
       " 'ARF1_proteomics',\n",
       " 'ARF3_proteomics',\n",
       " 'ARF4_proteomics',\n",
       " 'ARF5_proteomics',\n",
       " 'ARF6_proteomics',\n",
       " 'ARFGAP1_proteomics',\n",
       " 'ARFGAP2_proteomics',\n",
       " 'ARFGAP3_proteomics',\n",
       " 'ARFGEF1_proteomics',\n",
       " 'ARFGEF2_proteomics',\n",
       " 'ARFGEF3_proteomics',\n",
       " 'ARFIP1_proteomics',\n",
       " 'ARFIP2_proteomics',\n",
       " 'ARFRP1_proteomics',\n",
       " 'ARG1_proteomics',\n",
       " 'ARG2_proteomics',\n",
       " 'ARGLU1_proteomics',\n",
       " 'ARHGAP1_proteomics',\n",
       " 'ARHGAP10_proteomics',\n",
       " 'ARHGAP12_proteomics',\n",
       " 'ARHGAP15_proteomics',\n",
       " 'ARHGAP17_proteomics',\n",
       " 'ARHGAP18_proteomics',\n",
       " 'ARHGAP21_proteomics',\n",
       " 'ARHGAP22_proteomics',\n",
       " 'ARHGAP23_proteomics',\n",
       " 'ARHGAP24_proteomics',\n",
       " 'ARHGAP25_proteomics',\n",
       " 'ARHGAP26_proteomics',\n",
       " 'ARHGAP27_proteomics',\n",
       " 'ARHGAP29_proteomics',\n",
       " 'ARHGAP30_proteomics',\n",
       " 'ARHGAP31_proteomics',\n",
       " 'ARHGAP32_proteomics',\n",
       " 'ARHGAP33_proteomics',\n",
       " 'ARHGAP35_proteomics',\n",
       " 'ARHGAP39_proteomics',\n",
       " 'ARHGAP4_proteomics',\n",
       " 'ARHGAP42_proteomics',\n",
       " 'ARHGAP44_proteomics',\n",
       " 'ARHGAP45_proteomics',\n",
       " 'ARHGAP5_proteomics',\n",
       " 'ARHGAP6_proteomics',\n",
       " 'ARHGAP9_proteomics',\n",
       " 'ARHGDIA_proteomics',\n",
       " 'ARHGDIB_proteomics',\n",
       " 'ARHGDIG_proteomics',\n",
       " 'ARHGEF1_proteomics',\n",
       " 'ARHGEF10_proteomics',\n",
       " 'ARHGEF10L_proteomics',\n",
       " 'ARHGEF11_proteomics',\n",
       " 'ARHGEF12_proteomics',\n",
       " 'ARHGEF17_proteomics',\n",
       " 'ARHGEF18_proteomics',\n",
       " 'ARHGEF2_proteomics',\n",
       " 'ARHGEF25_proteomics',\n",
       " 'ARHGEF26_proteomics',\n",
       " 'ARHGEF3_proteomics',\n",
       " 'ARHGEF37_proteomics',\n",
       " 'ARHGEF4_proteomics',\n",
       " 'ARHGEF40_proteomics',\n",
       " 'ARHGEF6_proteomics',\n",
       " 'ARHGEF7_proteomics',\n",
       " 'ARHGEF9_proteomics',\n",
       " 'ARID1A_proteomics',\n",
       " 'ARID1B_proteomics',\n",
       " 'ARID2_proteomics',\n",
       " 'ARID4A_proteomics',\n",
       " 'ARID4B_proteomics',\n",
       " 'ARID5B_proteomics',\n",
       " 'ARIH1_proteomics',\n",
       " 'ARIH2_proteomics',\n",
       " 'ARL1_proteomics',\n",
       " 'ARL10_proteomics',\n",
       " 'ARL13B_proteomics',\n",
       " 'ARL14EP_proteomics',\n",
       " 'ARL15_proteomics',\n",
       " 'ARL16_proteomics',\n",
       " 'ARL2_proteomics',\n",
       " 'ARL2BP_proteomics',\n",
       " 'ARL3_proteomics',\n",
       " 'ARL4A_proteomics',\n",
       " 'ARL4C_proteomics',\n",
       " 'ARL5A_proteomics',\n",
       " 'ARL5B_proteomics',\n",
       " 'ARL6_proteomics',\n",
       " 'ARL6IP1_proteomics',\n",
       " 'ARL6IP4_proteomics',\n",
       " 'ARL6IP5_proteomics',\n",
       " 'ARL6IP6_proteomics',\n",
       " 'ARL8A_proteomics',\n",
       " 'ARL8B_proteomics',\n",
       " 'ARMC1_proteomics',\n",
       " 'ARMC10_proteomics',\n",
       " 'ARMC5_proteomics',\n",
       " 'ARMC6_proteomics',\n",
       " 'ARMC7_proteomics',\n",
       " 'ARMC8_proteomics',\n",
       " 'ARMC9_proteomics',\n",
       " 'ARMCX1_proteomics',\n",
       " 'ARMCX2_proteomics',\n",
       " 'ARMCX3_proteomics',\n",
       " 'ARMCX4_proteomics',\n",
       " 'ARMCX5_proteomics',\n",
       " 'ARMCX6_proteomics',\n",
       " 'ARMH3_proteomics',\n",
       " 'ARMT1_proteomics',\n",
       " 'ARNT_proteomics',\n",
       " 'ARNT2_proteomics',\n",
       " 'ARNTL_proteomics',\n",
       " 'ARPC1A_proteomics',\n",
       " 'ARPC1B_proteomics',\n",
       " 'ARPC2_proteomics',\n",
       " 'ARPC3_proteomics',\n",
       " 'ARPC4_proteomics',\n",
       " 'ARPC4-TTLL3_proteomics',\n",
       " 'ARPC5_proteomics',\n",
       " 'ARPC5L_proteomics',\n",
       " 'ARPIN_proteomics',\n",
       " 'ARPP19_proteomics',\n",
       " 'ARPP21_proteomics',\n",
       " 'ARR3_proteomics',\n",
       " 'ARRB1_proteomics',\n",
       " 'ARRB2_proteomics',\n",
       " 'ARRDC1_proteomics',\n",
       " 'ARSA_proteomics',\n",
       " 'ARSB_proteomics',\n",
       " 'ARSD_proteomics',\n",
       " 'ARSE_proteomics',\n",
       " 'ARSF_proteomics',\n",
       " 'ARSK_proteomics',\n",
       " 'ART3_proteomics',\n",
       " 'ART4_proteomics',\n",
       " 'ARV1_proteomics',\n",
       " 'ARVCF_proteomics',\n",
       " 'ARX_proteomics',\n",
       " 'AS3MT_proteomics',\n",
       " 'ASAH1_proteomics',\n",
       " 'ASAP1_proteomics',\n",
       " 'ASAP2_proteomics',\n",
       " 'ASAP3_proteomics',\n",
       " 'ASB1_proteomics',\n",
       " 'ASB13_proteomics',\n",
       " 'ASB3_proteomics',\n",
       " 'ASB6_proteomics',\n",
       " 'ASB8_proteomics',\n",
       " 'ASCC1_proteomics',\n",
       " 'ASCC2_proteomics',\n",
       " 'ASCC3_proteomics',\n",
       " 'ASCL1_proteomics',\n",
       " 'ASDURF_proteomics',\n",
       " 'ASF1A_proteomics',\n",
       " 'ASF1B_proteomics',\n",
       " 'ASH2L_proteomics',\n",
       " 'ASIC1_proteomics',\n",
       " 'ASIC4_proteomics',\n",
       " 'ASL_proteomics',\n",
       " 'ASMTL_proteomics',\n",
       " 'ASNA1_proteomics',\n",
       " 'ASNS_proteomics',\n",
       " 'ASNSD1_proteomics',\n",
       " 'ASPA_proteomics',\n",
       " 'ASPH_proteomics',\n",
       " 'ASPHD1_proteomics',\n",
       " 'ASPHD2_proteomics',\n",
       " 'ASPN_proteomics',\n",
       " 'ASPRV1_proteomics',\n",
       " 'ASPSCR1_proteomics',\n",
       " 'ASRGL1_proteomics',\n",
       " 'ASS1_proteomics',\n",
       " 'ASTN1_proteomics',\n",
       " 'ASTN2_proteomics',\n",
       " 'ASXL2_proteomics',\n",
       " 'ATAD1_proteomics',\n",
       " 'ATAD2_proteomics',\n",
       " 'ATAD2B_proteomics',\n",
       " 'ATAD3A_proteomics',\n",
       " 'ATAD3B_proteomics',\n",
       " 'ATAT1_proteomics',\n",
       " 'ATCAY_proteomics',\n",
       " 'ATE1_proteomics',\n",
       " 'ATF1_proteomics',\n",
       " 'ATF2_proteomics',\n",
       " 'ATF3_proteomics',\n",
       " 'ATF6_proteomics',\n",
       " 'ATF6B_proteomics',\n",
       " 'ATF7_proteomics',\n",
       " 'ATF7IP_proteomics',\n",
       " 'ATG10_proteomics',\n",
       " 'ATG101_proteomics',\n",
       " 'ATG12_proteomics',\n",
       " 'ATG13_proteomics',\n",
       " 'ATG14_proteomics',\n",
       " 'ATG16L1_proteomics',\n",
       " 'ATG2A_proteomics',\n",
       " 'ATG2B_proteomics',\n",
       " 'ATG3_proteomics',\n",
       " 'ATG4A_proteomics',\n",
       " 'ATG4B_proteomics',\n",
       " 'ATG4C_proteomics',\n",
       " 'ATG5_proteomics',\n",
       " 'ATG7_proteomics',\n",
       " 'ATG9A_proteomics',\n",
       " 'ATIC_proteomics',\n",
       " 'ATL1_proteomics',\n",
       " 'ATL2_proteomics',\n",
       " 'ATL3_proteomics',\n",
       " 'ATM_proteomics',\n",
       " 'ATN1_proteomics',\n",
       " 'ATOX1_proteomics',\n",
       " 'ATP11A_proteomics',\n",
       " 'ATP11B_proteomics',\n",
       " 'ATP11C_proteomics',\n",
       " 'ATP12A_proteomics',\n",
       " 'ATP13A1_proteomics',\n",
       " 'ATP13A2_proteomics',\n",
       " 'ATP13A3_proteomics',\n",
       " 'ATP13A4_proteomics',\n",
       " 'ATP1A1_proteomics',\n",
       " 'ATP1A2_proteomics',\n",
       " 'ATP1A3_proteomics',\n",
       " 'ATP1A4_proteomics',\n",
       " 'ATP1B1_proteomics',\n",
       " 'ATP1B2_proteomics',\n",
       " 'ATP1B3_proteomics',\n",
       " 'ATP1B4_proteomics',\n",
       " 'ATP23_proteomics',\n",
       " 'ATP2A1_proteomics',\n",
       " 'ATP2A2_proteomics',\n",
       " 'ATP2A3_proteomics',\n",
       " 'ATP2B1_proteomics',\n",
       " 'ATP2B2_proteomics',\n",
       " 'ATP2B3_proteomics',\n",
       " 'ATP2B4_proteomics',\n",
       " 'ATP2C1_proteomics',\n",
       " 'ATP4A_proteomics',\n",
       " 'ATP5F1A_proteomics',\n",
       " 'ATP5F1B_proteomics',\n",
       " 'ATP5F1C_proteomics',\n",
       " 'ATP5F1D_proteomics',\n",
       " 'ATP5F1E_proteomics',\n",
       " 'ATP5IF1_proteomics',\n",
       " 'ATP5MC1_proteomics',\n",
       " 'ATP5MD_proteomics',\n",
       " 'ATP5ME_proteomics',\n",
       " 'ATP5MF_proteomics',\n",
       " 'ATP5MF-PTCD1_proteomics',\n",
       " 'ATP5MG_proteomics',\n",
       " 'ATP5MPL_proteomics',\n",
       " 'ATP5PB_proteomics',\n",
       " 'ATP5PD_proteomics',\n",
       " 'ATP5PF_proteomics',\n",
       " 'ATP5PO_proteomics',\n",
       " 'ATP5S_proteomics',\n",
       " 'ATP6AP1_proteomics',\n",
       " 'ATP6AP2_proteomics',\n",
       " 'ATP6V0A1_proteomics',\n",
       " 'ATP6V0A2_proteomics',\n",
       " 'ATP6V0C_proteomics',\n",
       " 'ATP6V0D1_proteomics',\n",
       " 'ATP6V0D2_proteomics',\n",
       " 'ATP6V1A_proteomics',\n",
       " 'ATP6V1B2_proteomics',\n",
       " 'ATP6V1C1_proteomics',\n",
       " 'ATP6V1D_proteomics',\n",
       " 'ATP6V1E1_proteomics',\n",
       " 'ATP6V1E2_proteomics',\n",
       " 'ATP6V1F_proteomics',\n",
       " 'ATP6V1G1_proteomics',\n",
       " 'ATP6V1G2_proteomics',\n",
       " 'ATP6V1H_proteomics',\n",
       " 'ATP7A_proteomics',\n",
       " 'ATP8A1_proteomics',\n",
       " 'ATP8A2_proteomics',\n",
       " 'ATP9A_proteomics',\n",
       " 'ATP9B_proteomics',\n",
       " 'ATPAF1_proteomics',\n",
       " 'ATPAF2_proteomics',\n",
       " 'ATR_proteomics',\n",
       " 'ATRAID_proteomics',\n",
       " 'ATRIP_proteomics',\n",
       " 'ATRN_proteomics',\n",
       " 'ATRNL1_proteomics',\n",
       " 'ATRX_proteomics',\n",
       " 'ATXN1_proteomics',\n",
       " 'ATXN10_proteomics',\n",
       " 'ATXN1L_proteomics',\n",
       " 'ATXN2_proteomics',\n",
       " 'ATXN2L_proteomics',\n",
       " 'ATXN3_proteomics',\n",
       " 'ATXN7_proteomics',\n",
       " 'ATXN7L3_proteomics',\n",
       " 'ATXN7L3B_proteomics',\n",
       " 'AUH_proteomics',\n",
       " 'AUP1_proteomics',\n",
       " 'AURKA_proteomics',\n",
       " 'AURKAIP1_proteomics',\n",
       " 'AURKB_proteomics',\n",
       " 'AUTS2_proteomics',\n",
       " 'AVEN_proteomics',\n",
       " 'AVIL_proteomics',\n",
       " 'AVL9_proteomics',\n",
       " 'AXIN1_proteomics',\n",
       " 'AXL_proteomics',\n",
       " 'AZGP1_proteomics',\n",
       " 'AZI2_proteomics',\n",
       " 'AZU1_proteomics',\n",
       " 'B2M_proteomics',\n",
       " 'B3GALNT1_proteomics',\n",
       " 'B3GALT6_proteomics',\n",
       " 'B3GAT1_proteomics',\n",
       " 'B3GAT3_proteomics',\n",
       " 'B3GLCT_proteomics',\n",
       " 'B3GNT2_proteomics',\n",
       " 'B3GNT9_proteomics',\n",
       " 'B3GNTL1_proteomics',\n",
       " 'B4GALNT1_proteomics',\n",
       " 'B4GALT1_proteomics',\n",
       " 'B4GALT3_proteomics',\n",
       " 'B4GALT4_proteomics',\n",
       " 'B4GALT5_proteomics',\n",
       " 'B4GALT7_proteomics',\n",
       " 'B4GAT1_proteomics',\n",
       " 'B9D1_proteomics',\n",
       " 'B9D2_proteomics',\n",
       " 'BAALC_proteomics',\n",
       " 'BABAM1_proteomics',\n",
       " 'BABAM2_proteomics',\n",
       " 'BACE1_proteomics',\n",
       " 'BACH1_proteomics',\n",
       " 'BAD_proteomics',\n",
       " 'BAG1_proteomics',\n",
       " 'BAG2_proteomics',\n",
       " 'BAG3_proteomics',\n",
       " 'BAG4_proteomics',\n",
       " 'BAG5_proteomics',\n",
       " 'BAG6_proteomics',\n",
       " 'BAHD1_proteomics',\n",
       " 'BAIAP2_proteomics',\n",
       " 'BAIAP2L1_proteomics',\n",
       " 'BAIAP3_proteomics',\n",
       " 'BAK1_proteomics',\n",
       " 'BAMBI_proteomics',\n",
       " 'BANF1_proteomics',\n",
       " 'BANP_proteomics',\n",
       " 'BAP1_proteomics',\n",
       " 'BARD1_proteomics',\n",
       " 'BASP1_proteomics',\n",
       " 'BATF3_proteomics',\n",
       " 'BAX_proteomics',\n",
       " 'BAZ1A_proteomics',\n",
       " 'BAZ1B_proteomics',\n",
       " 'BAZ2A_proteomics',\n",
       " 'BAZ2B_proteomics',\n",
       " 'BBC3_proteomics',\n",
       " 'BBIP1_proteomics',\n",
       " 'BBOX1_proteomics',\n",
       " 'BBS1_proteomics',\n",
       " 'BBS2_proteomics',\n",
       " 'BBS4_proteomics',\n",
       " 'BBS5_proteomics',\n",
       " 'BBS7_proteomics',\n",
       " 'BBS9_proteomics',\n",
       " 'BBX_proteomics',\n",
       " 'BCAM_proteomics',\n",
       " 'BCAN_proteomics',\n",
       " 'BCAP29_proteomics',\n",
       " 'BCAP31_proteomics',\n",
       " 'BCAR1_proteomics',\n",
       " 'BCAR3_proteomics',\n",
       " 'BCAS1_proteomics',\n",
       " 'BCAS2_proteomics',\n",
       " 'BCAS3_proteomics',\n",
       " 'BCAT1_proteomics',\n",
       " 'BCAT2_proteomics',\n",
       " 'BCCIP_proteomics',\n",
       " 'BCDIN3D_proteomics',\n",
       " 'BCHE_proteomics',\n",
       " 'BCKDHA_proteomics',\n",
       " 'BCKDHB_proteomics',\n",
       " 'BCKDK_proteomics',\n",
       " 'BCL10_proteomics',\n",
       " 'BCL11A_proteomics',\n",
       " 'BCL11B_proteomics',\n",
       " 'BCL2_proteomics',\n",
       " 'BCL2L1_proteomics',\n",
       " 'BCL2L11_proteomics',\n",
       " 'BCL2L13_proteomics',\n",
       " 'BCL2L15_proteomics',\n",
       " 'BCL2L2_proteomics',\n",
       " 'BCL3_proteomics',\n",
       " 'BCL6_proteomics',\n",
       " 'BCL7A_proteomics',\n",
       " 'BCL7B_proteomics',\n",
       " 'BCL7C_proteomics',\n",
       " 'BCL9_proteomics',\n",
       " 'BCL9L_proteomics',\n",
       " 'BCLAF1_proteomics',\n",
       " 'BCLAF3_proteomics',\n",
       " 'BCO1_proteomics',\n",
       " 'BCR_proteomics',\n",
       " 'BCS1L_proteomics',\n",
       " 'BDH1_proteomics',\n",
       " 'BDH2_proteomics',\n",
       " 'BDNF_proteomics',\n",
       " 'BECN1_proteomics',\n",
       " 'BEGAIN_proteomics',\n",
       " 'BEND3_proteomics',\n",
       " 'BEND5_proteomics',\n",
       " 'BET1_proteomics',\n",
       " 'BET1L_proteomics',\n",
       " 'BEX4_proteomics',\n",
       " 'BGN_proteomics',\n",
       " 'BHLHB9_proteomics',\n",
       " 'BHLHE22_proteomics',\n",
       " 'BHLHE40_proteomics',\n",
       " 'BHLHE41_proteomics',\n",
       " 'BHMT_proteomics',\n",
       " 'BICC1_proteomics',\n",
       " 'BICD1_proteomics',\n",
       " 'BICD2_proteomics',\n",
       " 'BICRA_proteomics',\n",
       " 'BICRAL_proteomics',\n",
       " 'BID_proteomics',\n",
       " 'BIN1_proteomics',\n",
       " 'BIN2_proteomics',\n",
       " 'BIN3_proteomics',\n",
       " 'BIRC2_proteomics',\n",
       " 'BIRC3_proteomics',\n",
       " 'BIRC6_proteomics',\n",
       " 'BLM_proteomics',\n",
       " 'BLMH_proteomics',\n",
       " 'BLNK_proteomics',\n",
       " 'BLOC1S1_proteomics',\n",
       " 'BLOC1S2_proteomics',\n",
       " 'BLOC1S3_proteomics',\n",
       " 'BLOC1S4_proteomics',\n",
       " 'BLOC1S5_proteomics',\n",
       " 'BLOC1S6_proteomics',\n",
       " ...]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create list of genes to pass into t-test\n",
    "prot_col_list = list(Mesench_Proneural.columns)\n",
    "prot_col_list.remove('TCGA_subtype')\n",
    "\n",
    "prot_col_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "961 significant comparisons!\n",
      "Number of comparisons: 11140\n",
      "Bonferroni cutoff =  4.4883303411131066e-06\n",
      "Logged Bonferroni cutoff =  -5.347915186501691\n"
     ]
    }
   ],
   "source": [
    "#Call wrap_ttest, pass in formatted dataframe\n",
    "\n",
    "prot_all_comparisons = u.wrap_ttest(Mesench_Proneural, 'TCGA_subtype', prot_col_list)\n",
    "prot_num_comparisons = len(prot_col_list)\n",
    "print(\"Number of comparisons:\", prot_num_comparisons)\n",
    "prot_bonferroni_cutoff = .05 / prot_num_comparisons\n",
    "print(\"Bonferroni cutoff = \", prot_bonferroni_cutoff)\n",
    "print(\"Logged Bonferroni cutoff = \", np.log10(prot_bonferroni_cutoff))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant Proteomics comparisons:  961 \n",
      "\n",
      "             Comparison       P_Value\n",
      "0     PODXL2_proteomics  2.047362e-14\n",
      "1      BASP1_proteomics  2.654212e-14\n",
      "2     GPRIN1_proteomics  7.094843e-14\n",
      "3      PHF24_proteomics  7.994311e-14\n",
      "4       SCAI_proteomics  1.492919e-13\n",
      "..                  ...           ...\n",
      "956  CNTNAP1_proteomics  4.392254e-06\n",
      "957    HSPB1_proteomics  4.421559e-06\n",
      "958    NOVA1_proteomics  4.426351e-06\n",
      "959   THSD7A_proteomics  4.466200e-06\n",
      "960   PARP14_proteomics  4.483369e-06\n",
      "\n",
      "[961 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "#Print significant comparisons \n",
    "prot_all_comparisons = prot_all_comparisons.dropna(axis=0)\n",
    "prot_sig_comparisons = prot_all_comparisons.loc[prot_all_comparisons['P_Value'] <= prot_bonferroni_cutoff]\n",
    "print(\"Number of significant Proteomics comparisons: \", len(prot_sig_comparisons), '\\n')\n",
    "\n",
    "if len(prot_sig_comparisons) > 0:\n",
    "    print(prot_sig_comparisons)\n",
    "    \n",
    "prot_sig_comparisons_Mesench_Proneural = prot_sig_comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value_MP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>PODXL2_proteomics</td>\n",
       "      <td>2.047362e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>BASP1_proteomics</td>\n",
       "      <td>2.654212e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>GPRIN1_proteomics</td>\n",
       "      <td>7.094843e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>PHF24_proteomics</td>\n",
       "      <td>7.994311e-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>SCAI_proteomics</td>\n",
       "      <td>1.492919e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>956</td>\n",
       "      <td>CNTNAP1_proteomics</td>\n",
       "      <td>4.392254e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>957</td>\n",
       "      <td>HSPB1_proteomics</td>\n",
       "      <td>4.421559e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>958</td>\n",
       "      <td>NOVA1_proteomics</td>\n",
       "      <td>4.426351e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>959</td>\n",
       "      <td>THSD7A_proteomics</td>\n",
       "      <td>4.466200e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>960</td>\n",
       "      <td>PARP14_proteomics</td>\n",
       "      <td>4.483369e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>961 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Comparison    P_Value_MP\n",
       "0     PODXL2_proteomics  2.047362e-14\n",
       "1      BASP1_proteomics  2.654212e-14\n",
       "2     GPRIN1_proteomics  7.094843e-14\n",
       "3      PHF24_proteomics  7.994311e-14\n",
       "4       SCAI_proteomics  1.492919e-13\n",
       "..                  ...           ...\n",
       "956  CNTNAP1_proteomics  4.392254e-06\n",
       "957    HSPB1_proteomics  4.421559e-06\n",
       "958    NOVA1_proteomics  4.426351e-06\n",
       "959   THSD7A_proteomics  4.466200e-06\n",
       "960   PARP14_proteomics  4.483369e-06\n",
       "\n",
       "[961 rows x 2 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_sig_comparisons_Mesench_Proneural.set_index(\"Comparison\")\n",
    "prot_sig_comparisons_Mesench_Proneural = prot_sig_comparisons_Mesench_Proneural.rename(columns = {\"P_Value\": \"P_Value_MP\"})\n",
    "prot_sig_comparisons_Mesench_Proneural"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       PODXL2\n",
       "1        BASP1\n",
       "2       GPRIN1\n",
       "3        PHF24\n",
       "4         SCAI\n",
       "        ...   \n",
       "909       CD47\n",
       "910    CNTNAP1\n",
       "911      HSPB1\n",
       "912     THSD7A\n",
       "913     PARP14\n",
       "Name: Comparison, Length: 914, dtype: object"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#create list of common significant genes\n",
    "common_sig = prot_sig_comparisons_Mesench_Proneural.merge(prot_sig_comparisons_Mesench_Normal, on='Comparison')\n",
    "common_sig.replace(to_replace = '_proteomics', value = '', \n",
    "                        inplace = True, regex = True)# shorten column names\n",
    "\n",
    "\n",
    "sig_list  = common_sig['Comparison'] \n",
    "sig_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-0.20449052751586466"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Calculate mean\n",
    "Mesenchymal[\"MTMR1_proteomics\"].mean()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9241884660140893"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check to see if mean is up in normal\n",
    "Normal[\"MTMR1_proteomics\"].mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Q14032</td>\n",
       "      <td>BAAT_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Bile acid-CoA:amino acid N-acyltransferase (BA...</td>\n",
       "      <td>BAAT</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>P55084</td>\n",
       "      <td>ECHB_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Trifunctional enzyme subunit beta, mitochondri...</td>\n",
       "      <td>HADHB MSTP029</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Q3SYC2</td>\n",
       "      <td>MOGT2_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>2-acylglycerol O-acyltransferase 2 (EC 2.3.1.2...</td>\n",
       "      <td>MOGAT2 DC5 DGAT2L5</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Q15800</td>\n",
       "      <td>MSMO1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Methylsterol monooxygenase 1 (EC 1.14.18.9) (C...</td>\n",
       "      <td>MSMO1 DESP4 ERG25 SC4MOL</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>O60494</td>\n",
       "      <td>CUBN_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Cubilin (460 kDa receptor) (Intestinal intrins...</td>\n",
       "      <td>CUBN IFCR</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>3623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>480</td>\n",
       "      <td>Q49AA0</td>\n",
       "      <td>ZFP69_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Zinc finger protein 69 homolog (Zinc finger pr...</td>\n",
       "      <td>ZFP69 ZNF642</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>481</td>\n",
       "      <td>O95477</td>\n",
       "      <td>ABCA1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Phospholipid-transporting ATPase ABCA1 (EC 7.6...</td>\n",
       "      <td>ABCA1 ABC1 CERP</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>2261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>482</td>\n",
       "      <td>O94905</td>\n",
       "      <td>ERLN2_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Erlin-2 (Endoplasmic reticulum lipid raft-asso...</td>\n",
       "      <td>ERLIN2 C8orf2 SPFH2 UNQ2441/PRO5003/PRO9924</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>483</td>\n",
       "      <td>A1L3X0</td>\n",
       "      <td>ELOV7_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Elongation of very long chain fatty acids prot...</td>\n",
       "      <td>ELOVL7</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>484</td>\n",
       "      <td>Q8N9I5</td>\n",
       "      <td>FADS6_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Fatty acid desaturase 6 (EC 1.14.19.-)</td>\n",
       "      <td>FADS6 FP18279</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>356</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>485 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Entry   Entry name    Status  \\\n",
       "0    Q14032   BAAT_HUMAN  reviewed   \n",
       "1    P55084   ECHB_HUMAN  reviewed   \n",
       "2    Q3SYC2  MOGT2_HUMAN  reviewed   \n",
       "3    Q15800  MSMO1_HUMAN  reviewed   \n",
       "4    O60494   CUBN_HUMAN  reviewed   \n",
       "..      ...          ...       ...   \n",
       "480  Q49AA0  ZFP69_HUMAN  reviewed   \n",
       "481  O95477  ABCA1_HUMAN  reviewed   \n",
       "482  O94905  ERLN2_HUMAN  reviewed   \n",
       "483  A1L3X0  ELOV7_HUMAN  reviewed   \n",
       "484  Q8N9I5  FADS6_HUMAN  reviewed   \n",
       "\n",
       "                                         Protein names  \\\n",
       "0    Bile acid-CoA:amino acid N-acyltransferase (BA...   \n",
       "1    Trifunctional enzyme subunit beta, mitochondri...   \n",
       "2    2-acylglycerol O-acyltransferase 2 (EC 2.3.1.2...   \n",
       "3    Methylsterol monooxygenase 1 (EC 1.14.18.9) (C...   \n",
       "4    Cubilin (460 kDa receptor) (Intestinal intrins...   \n",
       "..                                                 ...   \n",
       "480  Zinc finger protein 69 homolog (Zinc finger pr...   \n",
       "481  Phospholipid-transporting ATPase ABCA1 (EC 7.6...   \n",
       "482  Erlin-2 (Endoplasmic reticulum lipid raft-asso...   \n",
       "483  Elongation of very long chain fatty acids prot...   \n",
       "484             Fatty acid desaturase 6 (EC 1.14.19.-)   \n",
       "\n",
       "                                      Gene names              Organism  Length  \n",
       "0                                           BAAT  Homo sapiens (Human)     418  \n",
       "1                                  HADHB MSTP029  Homo sapiens (Human)     474  \n",
       "2                             MOGAT2 DC5 DGAT2L5  Homo sapiens (Human)     334  \n",
       "3                       MSMO1 DESP4 ERG25 SC4MOL  Homo sapiens (Human)     293  \n",
       "4                                      CUBN IFCR  Homo sapiens (Human)    3623  \n",
       "..                                           ...                   ...     ...  \n",
       "480                                 ZFP69 ZNF642  Homo sapiens (Human)     526  \n",
       "481                              ABCA1 ABC1 CERP  Homo sapiens (Human)    2261  \n",
       "482  ERLIN2 C8orf2 SPFH2 UNQ2441/PRO5003/PRO9924  Homo sapiens (Human)     339  \n",
       "483                                       ELOVL7  Homo sapiens (Human)     281  \n",
       "484                                FADS6 FP18279  Homo sapiens (Human)     356  \n",
       "\n",
       "[485 rows x 7 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read in uniProt file for lipid metabolism \n",
    "lipid_genes = pd.read_csv(\"/Users/Lindsey/Downloads/uniprot_keyword_Lipid.tab\", sep= \"\\t\")\n",
    "lipid_genes\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['PLCB1',\n",
       " 'B4GALT5',\n",
       " 'MTMR1',\n",
       " 'SULT4A1',\n",
       " 'SLC27A4',\n",
       " 'ACOT7',\n",
       " 'PLBD1',\n",
       " 'PTPRN2',\n",
       " 'BSCL2',\n",
       " 'PCYT1A',\n",
       " 'SMPD3',\n",
       " 'DAGLA',\n",
       " 'PLCG2',\n",
       " 'HDLBP']"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create list of significant genes (common in both t-tests) that are also in Uniprot df \n",
    "genes_dirty = list(lipid_genes['Gene names'])\n",
    "genes_clean = []\n",
    "for e in genes_dirty:\n",
    "    bits = e.split(\" \")\n",
    "    genes_clean.append(bits[0])\n",
    "    \n",
    " \n",
    "def intersection(lst1, lst2): \n",
    "    return list(set(lst1) & set(lst2)) \n",
    "\n",
    "winners = intersection(sig_list, genes_clean)\n",
    "#print (winners)\n",
    "#list from proteomics correlation\n",
    "winners"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
