{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "#sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\GBM_for_CPTAC\\\\')\n",
    "#import cis_functions as f\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_significance_col(results_df, num_comparisons):\n",
    "    \"bonferroni multiple hypothesis\"\"\"\n",
    "    alpha = .05\n",
    "    bonferroni_cutoff = alpha / num_comparisons\n",
    "    \n",
    "    pval = results_df['P_Value']\n",
    "    if float(pval[0]) <= bonferroni_cutoff:\n",
    "        results_df['Significant'] = True\n",
    "    else: \n",
    "        results_df['Significant'] = False\n",
    "    return results_df\n",
    "\n",
    "def wrap_ttest_return_all(df, label_column, comparison_columns, total_tests, alpha=.05):\n",
    "    try:\n",
    "        #Verify precondition that label column exists and has exactly 2 unique values\n",
    "        label_values = df[label_column].unique()\n",
    "        if len(label_values) != 2:\n",
    "            print(\"Incorrectly Formatted Dataframe! Label column must have exactly 2 unique values.\")\n",
    "            return None\n",
    "        \n",
    "        #Partition dataframe into two sets, one for each of the two unique values from the label column\n",
    "        partition1 = df.loc[df[label_column] == label_values[0]]\n",
    "        partition2 = df.loc[df[label_column] == label_values[1]]\n",
    "        \n",
    "        #Determine the number of real valued columns on which we will do t-tests\n",
    "        #sites = len(comparison_columns.columns)\n",
    "        number_of_comparisons = total_tests # ? phospho sites or num freq mut genes doing cis comp\n",
    "        \n",
    "        #Use a bonferroni correction to adjust for multiple testing by altering the p-value needed for acceptance\n",
    "        bonferroni_cutoff = alpha/number_of_comparisons\n",
    "        \n",
    "        #Store all comparisons with their p-values in a dictionary\n",
    "        all_comparisons = {}\n",
    "        \n",
    "        #Loop through each comparison column, perform the t-test, and determine whether it meets the significance cutoff'''\n",
    "        for column in comparison_columns:\n",
    "            stat, pval = scipy.stats.ttest_ind(partition1[column].dropna(axis=0), partition2[column].dropna(axis=0))\n",
    "            all_comparisons[column] = pval\n",
    "    \n",
    "        #Sort dictionary to list smallest p-values first\n",
    "        sorted_comparisons = sorted(all_comparisons.items(), key=lambda kv: kv[1])\n",
    "        #Format as a dataframe and return to caller\n",
    "        all_comparisons_df = pd.DataFrame.from_dict(sorted_comparisons)\n",
    "        all_comparisons_df.columns = ['Comparison', 'P_Value']\n",
    "        \n",
    "                                               \n",
    "        all_comparisons_sig_col = add_significance_col(all_comparisons_df, number_of_comparisons)\n",
    "        return all_comparisons_sig_col\n",
    "                                \n",
    "    except:\n",
    "        print(\"Incorrectly Formatted Dataframe!\")\n",
    "        return None\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "#cptac.download(dataset='ccrcc', version='0.0')\n",
    "brain= cptac.Gbm()\n",
    "desired_cutoff = 0.05\n",
    "gene = 'RB1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Below are the dataframes contained in this dataset:\n",
      "\tacetylproteomics\n",
      "\t\tDimensions: (109, 18767)\n",
      "\tcircular_RNA\n",
      "\t\tDimensions: (99, 3670)\n",
      "\tclinical\n",
      "\t\tDimensions: (115, 28)\n",
      "\tCNV\n",
      "\t\tDimensions: (98, 19907)\n",
      "\texperimental_design\n",
      "\t\tDimensions: (115, 8)\n",
      "\tgene_fusion\n",
      "\t\tDimensions: (2090, 8)\n",
      "\tlipidomics\n",
      "\t\tDimensions: (88, 582)\n",
      "\tmetabolomics\n",
      "\t\tDimensions: (87, 134)\n",
      "\tmiRNA\n",
      "\t\tDimensions: (87, 2883)\n",
      "\tphosphoproteomics\n",
      "\t\tDimensions: (109, 101266)\n",
      "\tproteomics\n",
      "\t\tDimensions: (109, 11141)\n",
      "\tsomatic_mutation\n",
      "\t\tDimensions: (5774, 3)\n",
      "\ttranscriptomics\n",
      "\t\tDimensions: (108, 60483)\n"
     ]
    }
   ],
   "source": [
    "brain.list_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: proteomics data was not found for the following samples, so proteomics data columns were filled with NaN for these samples: S021, S048, S064, S073, S075, S103 (/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Sample_Tumor_Normal</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>height</th>\n",
       "      <th>weight</th>\n",
       "      <th>bmi</th>\n",
       "      <th>country_of_origin</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSCAN31_proteomics</th>\n",
       "      <th>ZSWIM8_proteomics</th>\n",
       "      <th>ZW10_proteomics</th>\n",
       "      <th>ZWILCH_proteomics</th>\n",
       "      <th>ZWINT_proteomics</th>\n",
       "      <th>ZXDC_proteomics</th>\n",
       "      <th>ZYG11B_proteomics</th>\n",
       "      <th>ZYX_proteomics</th>\n",
       "      <th>ZZEF1_proteomics</th>\n",
       "      <th>ZZZ3_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00104</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>188.00</td>\n",
       "      <td>115.00</td>\n",
       "      <td>32.54</td>\n",
       "      <td>United States</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.047437</td>\n",
       "      <td>-0.105908</td>\n",
       "      <td>-0.347076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.459635</td>\n",
       "      <td>0.079452</td>\n",
       "      <td>-0.784983</td>\n",
       "      <td>-0.488441</td>\n",
       "      <td>0.167990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>C3L-00365</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.00</td>\n",
       "      <td>54.00</td>\n",
       "      <td>20.61</td>\n",
       "      <td>United States</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.161975</td>\n",
       "      <td>-0.213093</td>\n",
       "      <td>0.235571</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.107421</td>\n",
       "      <td>0.048724</td>\n",
       "      <td>0.138403</td>\n",
       "      <td>-0.290141</td>\n",
       "      <td>0.405037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>C3L-00674</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>45.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>193.00</td>\n",
       "      <td>102.00</td>\n",
       "      <td>27.44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.065534</td>\n",
       "      <td>-0.306717</td>\n",
       "      <td>0.879991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.883564</td>\n",
       "      <td>-0.172222</td>\n",
       "      <td>0.011876</td>\n",
       "      <td>-0.131889</td>\n",
       "      <td>-0.503581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>164.00</td>\n",
       "      <td>52.00</td>\n",
       "      <td>19.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.062127</td>\n",
       "      <td>-0.254535</td>\n",
       "      <td>0.463653</td>\n",
       "      <td>0.580230</td>\n",
       "      <td>0.503044</td>\n",
       "      <td>-0.604986</td>\n",
       "      <td>0.178077</td>\n",
       "      <td>-0.720059</td>\n",
       "      <td>-0.150197</td>\n",
       "      <td>-0.268715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>C3L-01040</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>77.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>170.00</td>\n",
       "      <td>70.00</td>\n",
       "      <td>24.22</td>\n",
       "      <td>Russia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.092502</td>\n",
       "      <td>0.010639</td>\n",
       "      <td>-0.465079</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.500083</td>\n",
       "      <td>0.112651</td>\n",
       "      <td>1.004660</td>\n",
       "      <td>-0.230304</td>\n",
       "      <td>-0.102416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S111</td>\n",
       "      <td>PT-RN5K</td>\n",
       "      <td>Normal</td>\n",
       "      <td>56.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.56</td>\n",
       "      <td>49.90</td>\n",
       "      <td>18.88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.298664</td>\n",
       "      <td>-0.403930</td>\n",
       "      <td>-0.482597</td>\n",
       "      <td>0.240966</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.609314</td>\n",
       "      <td>-1.293464</td>\n",
       "      <td>0.225910</td>\n",
       "      <td>-0.013637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S112</td>\n",
       "      <td>PT-RU72</td>\n",
       "      <td>Normal</td>\n",
       "      <td>59.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>162.56</td>\n",
       "      <td>63.05</td>\n",
       "      <td>23.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Asian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.155988</td>\n",
       "      <td>0.002551</td>\n",
       "      <td>-0.834434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.073908</td>\n",
       "      <td>0.637394</td>\n",
       "      <td>-0.795886</td>\n",
       "      <td>0.326046</td>\n",
       "      <td>-0.217014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S113</td>\n",
       "      <td>PT-UTHO</td>\n",
       "      <td>Normal</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>182.88</td>\n",
       "      <td>62.87</td>\n",
       "      <td>18.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.016857</td>\n",
       "      <td>-0.171858</td>\n",
       "      <td>-0.462233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.278683</td>\n",
       "      <td>0.616285</td>\n",
       "      <td>-0.734952</td>\n",
       "      <td>0.339283</td>\n",
       "      <td>-0.419138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S114</td>\n",
       "      <td>PT-WVLH</td>\n",
       "      <td>Normal</td>\n",
       "      <td>58.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>182.88</td>\n",
       "      <td>90.72</td>\n",
       "      <td>27.12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.450926</td>\n",
       "      <td>0.114686</td>\n",
       "      <td>-0.245149</td>\n",
       "      <td>-1.202774</td>\n",
       "      <td>-0.709850</td>\n",
       "      <td>-0.233022</td>\n",
       "      <td>0.732273</td>\n",
       "      <td>-1.158899</td>\n",
       "      <td>0.354608</td>\n",
       "      <td>-0.468881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S115</td>\n",
       "      <td>PT-Y8DK</td>\n",
       "      <td>Normal</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>172.72</td>\n",
       "      <td>81.65</td>\n",
       "      <td>27.37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>White</td>\n",
       "      <td>Not Hispanic or Latino</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.262036</td>\n",
       "      <td>-0.184673</td>\n",
       "      <td>-0.775883</td>\n",
       "      <td>1.778184</td>\n",
       "      <td>-0.250798</td>\n",
       "      <td>0.623839</td>\n",
       "      <td>-0.892930</td>\n",
       "      <td>0.249729</td>\n",
       "      <td>-0.376205</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>115 rows Ã— 11169 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name      Patient_ID Sample_Tumor_Normal   age  gender  height  weight    bmi  \\\n",
       "Sample_ID                                                                       \n",
       "S001       C3L-00104               Tumor  58.0    Male  188.00  115.00  32.54   \n",
       "S002       C3L-00365               Tumor  59.0  Female  162.00   54.00  20.61   \n",
       "S003       C3L-00674               Tumor  45.0    Male  193.00  102.00  27.44   \n",
       "S004       C3L-00677               Tumor  69.0  Female  164.00   52.00  19.32   \n",
       "S005       C3L-01040               Tumor  77.0  Female  170.00   70.00  24.22   \n",
       "...              ...                 ...   ...     ...     ...     ...    ...   \n",
       "S111         PT-RN5K              Normal  56.0  Female  162.56   49.90  18.88   \n",
       "S112         PT-RU72              Normal  59.0  Female  162.56   63.05  23.86   \n",
       "S113         PT-UTHO              Normal  68.0    Male  182.88   62.87  18.80   \n",
       "S114         PT-WVLH              Normal  58.0    Male  182.88   90.72  27.12   \n",
       "S115         PT-Y8DK              Normal  62.0    Male  172.72   81.65  27.37   \n",
       "\n",
       "Name      country_of_origin   race               ethnicity  ...  \\\n",
       "Sample_ID                                                   ...   \n",
       "S001          United States  White  Not-Hispanic or Latino  ...   \n",
       "S002          United States  White  Not-Hispanic or Latino  ...   \n",
       "S003                    NaN  White  Not-Hispanic or Latino  ...   \n",
       "S004                    NaN  White  Not-Hispanic or Latino  ...   \n",
       "S005                 Russia    NaN                     NaN  ...   \n",
       "...                     ...    ...                     ...  ...   \n",
       "S111                    NaN  White  Not Hispanic or Latino  ...   \n",
       "S112                    NaN  Asian                     NaN  ...   \n",
       "S113                    NaN  White                     NaN  ...   \n",
       "S114                    NaN  White                     NaN  ...   \n",
       "S115                    NaN  White  Not Hispanic or Latino  ...   \n",
       "\n",
       "Name      ZSCAN31_proteomics ZSWIM8_proteomics ZW10_proteomics  \\\n",
       "Sample_ID                                                        \n",
       "S001                     NaN         -0.047437       -0.105908   \n",
       "S002                     NaN          0.161975       -0.213093   \n",
       "S003                     NaN         -0.065534       -0.306717   \n",
       "S004               -0.062127         -0.254535        0.463653   \n",
       "S005                     NaN         -0.092502        0.010639   \n",
       "...                      ...               ...             ...   \n",
       "S111                     NaN          0.298664       -0.403930   \n",
       "S112                     NaN          0.155988        0.002551   \n",
       "S113                     NaN          0.016857       -0.171858   \n",
       "S114                0.450926          0.114686       -0.245149   \n",
       "S115                     NaN          0.262036       -0.184673   \n",
       "\n",
       "Name      ZWILCH_proteomics ZWINT_proteomics ZXDC_proteomics  \\\n",
       "Sample_ID                                                      \n",
       "S001              -0.347076              NaN        0.459635   \n",
       "S002               0.235571              NaN        0.107421   \n",
       "S003               0.879991              NaN        0.883564   \n",
       "S004               0.580230         0.503044       -0.604986   \n",
       "S005              -0.465079              NaN       -0.500083   \n",
       "...                     ...              ...             ...   \n",
       "S111              -0.482597         0.240966             NaN   \n",
       "S112              -0.834434              NaN       -0.073908   \n",
       "S113              -0.462233              NaN        1.278683   \n",
       "S114              -1.202774        -0.709850       -0.233022   \n",
       "S115              -0.775883         1.778184       -0.250798   \n",
       "\n",
       "Name      ZYG11B_proteomics  ZYX_proteomics  ZZEF1_proteomics ZZZ3_proteomics  \n",
       "Sample_ID                                                                      \n",
       "S001               0.079452       -0.784983         -0.488441        0.167990  \n",
       "S002               0.048724        0.138403         -0.290141        0.405037  \n",
       "S003              -0.172222        0.011876         -0.131889       -0.503581  \n",
       "S004               0.178077       -0.720059         -0.150197       -0.268715  \n",
       "S005               0.112651        1.004660         -0.230304       -0.102416  \n",
       "...                     ...             ...               ...             ...  \n",
       "S111               0.609314       -1.293464          0.225910       -0.013637  \n",
       "S112               0.637394       -0.795886          0.326046       -0.217014  \n",
       "S113               0.616285       -0.734952          0.339283       -0.419138  \n",
       "S114               0.732273       -1.158899          0.354608       -0.468881  \n",
       "S115               0.623839       -0.892930          0.249729       -0.376205  \n",
       "\n",
       "[115 rows x 11169 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clin_and_prot = brain.join_metadata_to_omics(metadata_df_name=\"clinical\", omics_df_name=\"proteomics\")\n",
    "clin_and_prot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>TCGA_subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>C3L-00104</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>C3L-00365</td>\n",
       "      <td>Classical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>C3L-00674</td>\n",
       "      <td>Mesenchymal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C3L-00677</td>\n",
       "      <td>Proneural</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>C3L-01040</td>\n",
       "      <td>Classical</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Patient_ID TCGA_subtype\n",
       "0  C3L-00104    Proneural\n",
       "1  C3L-00365    Classical\n",
       "2  C3L-00674  Mesenchymal\n",
       "3  C3L-00677    Proneural\n",
       "4  C3L-01040    Classical"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Read in files with TCGA subtypes\n",
    "subtypes = pd.read_csv(\"/Users/Lindsey/Downloads/gbm_all_subtype_collections.2019-11-13.tsv\", sep= \"\\t\")\n",
    "case_subtype = subtypes[['case','rna_wang_cancer_cell_2017']] #only need subtype and case\n",
    "case_subtype = case_subtype.rename(columns = {\"rna_wang_cancer_cell_2017\": \"TCGA_subtype\", \"case\":\"Patient_ID\"})\n",
    "case_subtype.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-26-287568225538>, line 3)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-26-287568225538>\"\u001b[0;36m, line \u001b[0;32m3\u001b[0m\n\u001b[0;31m    prot_subtype = prot_subtype.drop([,3:8])\u001b[0m\n\u001b[0m                                      ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "# merge tgca subtypes with proteomics and clincal df\n",
    "prot_subtype= clin_and_prot.merge(case_subtype, on='Patient_ID')\n",
    "prot_subtype = prot_subtype.drop([,3:8])\n",
    "prot_subtype.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
