{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ATRX Mutation Trans Effect on Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Library Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as al\n",
    "\n",
    "brain = cptac.Gbm()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = \"ATRX\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Investigate Proteomics, Phosphoproteomics, Acetylproteomics, or Transcriptomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#omics = \"proteomics\"\n",
    "omics = \"transcriptomics\"\n",
    "#omics = \"phosphoproteomics\"\n",
    "#omics = \"acetylproteomics\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Track all significant comparisons in Dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_significant_comparisons = pd.DataFrame(columns=['Cancer_Type', 'Gene', 'Comparison','Interacting_Protein','P_Value'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_to_all_significant_comparisons(df, cancer, interacting, all_sig_comp):\n",
    "    expanded = df\n",
    "    expanded['Gene'] = gene\n",
    "    expanded['Cancer_Type'] = cancer\n",
    "    expanded['Interacting_Protein'] = interacting\n",
    "    \n",
    "    updated_all_comparisons = pd.concat([all_sig_comp, expanded], sort=False)\n",
    "    \n",
    "    return updated_all_comparisons"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Interacting Proteins: Transcriptomics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate interacting protein list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a call to the cptac.algorithms get interacting proteins method, which interacts with the uniprot and string databases to generate a list of known interacting partners with the given gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Interacting Proteins:\n",
      "NKTR\n",
      "PNN\n",
      "TP53\n",
      "STAG2\n",
      "H3F3B\n",
      "SETDB1\n",
      "NIPBL\n",
      "PHIP\n",
      "BPTF\n",
      "RSF1\n",
      "PRKDC\n",
      "MUS81\n",
      "ARID4A\n",
      "RAD52\n",
      "BDP1\n",
      "XRCC6\n",
      "SMC5\n",
      "H3F3A\n",
      "TPR\n",
      "SCAF11\n",
      "ATRX\n",
      "DAXX\n",
      "RAD51\n",
      "MECP2\n",
      "XRCC3\n",
      "HIST2H3PS2\n",
      "EIF4A2\n",
      "EZH2\n",
      "Mecp2\n",
      "CBX5\n"
     ]
    }
   ],
   "source": [
    "'''Use get interacting proteins method to generate list of interacting proteins'''\n",
    "interacting_proteins = al.get_interacting_proteins(gene)\n",
    "\n",
    "'''Show interacting protein list'''\n",
    "print(\"Interacting Proteins:\")\n",
    "for interacting_protein in interacting_proteins:\n",
    "    print(interacting_protein)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test for significant comparisons in any of interacting proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: The following columns were not found in the transcriptomics dataframe, so they were inserted into joined table, but filled with NaN: Mecp2 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n",
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the ATRX gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 2)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "                                 Comparison   P_Value\n",
      "0  MECP2_transcriptomics_ENSG00000169057.18  0.000467\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "'''Create dataframe in order to do comparisons with wrap_ttest'''\n",
    "protdf = brain.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics, omics_genes=interacting_proteins)\n",
    "protdf = brain.reduce_multiindex(protdf, flatten=True)\n",
    "protdf = protdf[protdf.Sample_Status == 'Tumor'] # drop Normal samples\n",
    "\n",
    "'''Create the binary valued column needed to do the comparison'''\n",
    "for ind, row in protdf.iterrows():\n",
    "    if row[gene+\"_Mutation_Status\"] != 'Wildtype_Tumor':\n",
    "        protdf.at[ind,'Label'] = 'Mutated'\n",
    "    else:\n",
    "        protdf.at[ind,'Label'] = 'Wildtype'\n",
    "\n",
    "'''Format the dataframe correctly'''\n",
    "protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "\n",
    "'''Make list of columns to be compared using t-tests'''\n",
    "col_list = list(protdf.columns)\n",
    "col_list.remove('Label')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "'''Call wrap_ttest, pass in formatted dataframe'''\n",
    "wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "'''Print results, if anything significant was found'''\n",
    "if wrap_results is not None:\n",
    "        print(wrap_results)\n",
    "        print(\"\\n\\n\")\n",
    "        \n",
    "        all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Gbm\", True, all_significant_comparisons)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# All Proteins: Transcriptomics "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene:  ATRX\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 100 samples for the ATRX gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "                                           Comparison       P_Value\n",
      "0           TSPYL2_transcriptomics_ENSG00000184205.13  5.743268e-10\n",
      "1     RP11-296E23.2_transcriptomics_ENSG00000265948.1  6.907699e-10\n",
      "2           CCDC68_transcriptomics_ENSG00000166510.12  4.049959e-09\n",
      "3      RP11-624L4.1_transcriptomics_ENSG00000259345.4  1.148094e-08\n",
      "4           GREB1L_transcriptomics_ENSG00000141449.13  1.270396e-08\n",
      "5      RP11-80F22.7_transcriptomics_ENSG00000260153.1  2.534988e-08\n",
      "6           CDR1-AS_transcriptomics_ENSG00000281508.1  3.881078e-08\n",
      "7     RP11-541N10.3_transcriptomics_ENSG00000260461.1  4.690104e-08\n",
      "8        AC005237.4_transcriptomics_ENSG00000225521.1  4.760715e-08\n",
      "9      RP11-279F6.1_transcriptomics_ENSG00000245750.6  8.828716e-08\n",
      "10          EXOSC1_transcriptomics_ENSG00000171311.11  1.059106e-07\n",
      "11       AC093585.6_transcriptomics_ENSG00000229996.1  1.130568e-07\n",
      "12           PCAT29_transcriptomics_ENSG00000259641.4  1.398273e-07\n",
      "13             DRG2_transcriptomics_ENSG00000108591.8  1.594127e-07\n",
      "14          SEC31B_transcriptomics_ENSG00000075826.15  1.644265e-07\n",
      "15    RP11-323P17.2_transcriptomics_ENSG00000274461.1  1.707400e-07\n",
      "16             DAW1_transcriptomics_ENSG00000123977.8  1.973377e-07\n",
      "17  LL0XNC01-116E7.5_transcriptomics_ENSG000002762...  2.468243e-07\n",
      "18    RP11-370P15.2_transcriptomics_ENSG00000261014.1  3.184356e-07\n",
      "19     CTB-174O21.2_transcriptomics_ENSG00000269487.1  3.383054e-07\n",
      "20     RP4-587D13.1_transcriptomics_ENSG00000281008.1  3.497493e-07\n",
      "21      CTC-370J7.1_transcriptomics_ENSG00000253630.1  3.533912e-07\n",
      "22     CTC-543D15.1_transcriptomics_ENSG00000260112.1  3.893415e-07\n",
      "23           AGAP4_transcriptomics_ENSG00000188234.12  4.749609e-07\n",
      "24     LINC00328-2P_transcriptomics_ENSG00000225016.1  5.141748e-07\n",
      "25      RP11-99A1.2_transcriptomics_ENSG00000267311.1  5.702053e-07\n",
      "26        RN7SL284P_transcriptomics_ENSG00000243446.3  5.768131e-07\n",
      "27       AC010095.7_transcriptomics_ENSG00000235644.1  6.443425e-07\n",
      "28        KCNQ5-IT1_transcriptomics_ENSG00000233844.1  6.546887e-07\n",
      "29            PHF8_transcriptomics_ENSG00000172943.17  6.997059e-07\n",
      "30         RPS10P28_transcriptomics_ENSG00000243167.1  7.035771e-07\n",
      "31            GPX3_transcriptomics_ENSG00000211445.10  7.529478e-07\n",
      "32           CNNM2_transcriptomics_ENSG00000148842.16  8.219161e-07\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    print(\"\\nGene: \", gene)\n",
    "\n",
    "    # Use all proteins\n",
    "\n",
    "    # Create dataframe in order to do comparisons with wrap_ttest\n",
    "    protdf = brain.join_omics_to_mutations(mutations_genes=[gene], omics_df_name=omics)\n",
    "    protdf = brain.reduce_multiindex(protdf, flatten=True) # flatten for specific column names\n",
    "    protdf = protdf.loc[protdf['Sample_Status'] == 'Tumor']\n",
    "    \n",
    "    # Create binary column \n",
    "    protdf['Label'] = np.where(\n",
    "                protdf[gene+'_Mutation_Status'] == 'Wildtype_Tumor', 'Wildtype', 'Mutated')\n",
    "\n",
    "    # Format the dataframe correctly'''\n",
    "    protdf = protdf.drop(gene+\"_Mutation\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Location\",axis=1)\n",
    "    protdf = protdf.drop(gene+\"_Mutation_Status\", axis=1)\n",
    "    protdf = protdf.drop(\"Sample_Status\",axis=1)\n",
    "\n",
    "    # Make list of columns to be compared using t-tests\n",
    "    col_list = list(protdf.columns)\n",
    "    col_list.remove('Label')\n",
    "\n",
    "    print(\"Doing t-test comparisons\\n\")\n",
    "    \n",
    "    # Call wrap_ttest, pass in formatted dataframe\n",
    "    wrap_results = al.wrap_ttest(protdf, 'Label', col_list)\n",
    "\n",
    "    # Print results, if anything significant was found\n",
    "    if wrap_results is not None:\n",
    "            print(wrap_results)\n",
    "            print(\"\\n\\n\")\n",
    "            \n",
    "            all_significant_comparisons = add_to_all_significant_comparisons(wrap_results, \"Gbm\", False, all_significant_comparisons)\n",
    "\n",
    "\n",
    "except Exception as e:\n",
    "    print(\"Error in Comparison\")\n",
    "    print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Print all significant comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Comparison</th>\n",
       "      <th>Interacting_Protein</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>MECP2_transcriptomics_ENSG00000169057.18</td>\n",
       "      <td>True</td>\n",
       "      <td>4.671346e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>TSPYL2_transcriptomics_ENSG00000184205.13</td>\n",
       "      <td>False</td>\n",
       "      <td>5.743268e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-296E23.2_transcriptomics_ENSG00000265948.1</td>\n",
       "      <td>False</td>\n",
       "      <td>6.907699e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>CCDC68_transcriptomics_ENSG00000166510.12</td>\n",
       "      <td>False</td>\n",
       "      <td>4.049959e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-624L4.1_transcriptomics_ENSG00000259345.4</td>\n",
       "      <td>False</td>\n",
       "      <td>1.148094e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>GREB1L_transcriptomics_ENSG00000141449.13</td>\n",
       "      <td>False</td>\n",
       "      <td>1.270396e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-80F22.7_transcriptomics_ENSG00000260153.1</td>\n",
       "      <td>False</td>\n",
       "      <td>2.534988e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>CDR1-AS_transcriptomics_ENSG00000281508.1</td>\n",
       "      <td>False</td>\n",
       "      <td>3.881078e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-541N10.3_transcriptomics_ENSG00000260461.1</td>\n",
       "      <td>False</td>\n",
       "      <td>4.690104e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>AC005237.4_transcriptomics_ENSG00000225521.1</td>\n",
       "      <td>False</td>\n",
       "      <td>4.760715e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-279F6.1_transcriptomics_ENSG00000245750.6</td>\n",
       "      <td>False</td>\n",
       "      <td>8.828716e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>EXOSC1_transcriptomics_ENSG00000171311.11</td>\n",
       "      <td>False</td>\n",
       "      <td>1.059106e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>AC093585.6_transcriptomics_ENSG00000229996.1</td>\n",
       "      <td>False</td>\n",
       "      <td>1.130568e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>PCAT29_transcriptomics_ENSG00000259641.4</td>\n",
       "      <td>False</td>\n",
       "      <td>1.398273e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>DRG2_transcriptomics_ENSG00000108591.8</td>\n",
       "      <td>False</td>\n",
       "      <td>1.594127e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>SEC31B_transcriptomics_ENSG00000075826.15</td>\n",
       "      <td>False</td>\n",
       "      <td>1.644265e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-323P17.2_transcriptomics_ENSG00000274461.1</td>\n",
       "      <td>False</td>\n",
       "      <td>1.707400e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>DAW1_transcriptomics_ENSG00000123977.8</td>\n",
       "      <td>False</td>\n",
       "      <td>1.973377e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>LL0XNC01-116E7.5_transcriptomics_ENSG000002762...</td>\n",
       "      <td>False</td>\n",
       "      <td>2.468243e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-370P15.2_transcriptomics_ENSG00000261014.1</td>\n",
       "      <td>False</td>\n",
       "      <td>3.184356e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>CTB-174O21.2_transcriptomics_ENSG00000269487.1</td>\n",
       "      <td>False</td>\n",
       "      <td>3.383054e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP4-587D13.1_transcriptomics_ENSG00000281008.1</td>\n",
       "      <td>False</td>\n",
       "      <td>3.497493e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>CTC-370J7.1_transcriptomics_ENSG00000253630.1</td>\n",
       "      <td>False</td>\n",
       "      <td>3.533912e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>CTC-543D15.1_transcriptomics_ENSG00000260112.1</td>\n",
       "      <td>False</td>\n",
       "      <td>3.893415e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>AGAP4_transcriptomics_ENSG00000188234.12</td>\n",
       "      <td>False</td>\n",
       "      <td>4.749609e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>LINC00328-2P_transcriptomics_ENSG00000225016.1</td>\n",
       "      <td>False</td>\n",
       "      <td>5.141748e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RP11-99A1.2_transcriptomics_ENSG00000267311.1</td>\n",
       "      <td>False</td>\n",
       "      <td>5.702053e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RN7SL284P_transcriptomics_ENSG00000243446.3</td>\n",
       "      <td>False</td>\n",
       "      <td>5.768131e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>AC010095.7_transcriptomics_ENSG00000235644.1</td>\n",
       "      <td>False</td>\n",
       "      <td>6.443425e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>KCNQ5-IT1_transcriptomics_ENSG00000233844.1</td>\n",
       "      <td>False</td>\n",
       "      <td>6.546887e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>PHF8_transcriptomics_ENSG00000172943.17</td>\n",
       "      <td>False</td>\n",
       "      <td>6.997059e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>RPS10P28_transcriptomics_ENSG00000243167.1</td>\n",
       "      <td>False</td>\n",
       "      <td>7.035771e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>GPX3_transcriptomics_ENSG00000211445.10</td>\n",
       "      <td>False</td>\n",
       "      <td>7.529478e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>Gbm</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>CNNM2_transcriptomics_ENSG00000148842.16</td>\n",
       "      <td>False</td>\n",
       "      <td>8.219161e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Cancer_Type  Gene                                         Comparison  \\\n",
       "0          Gbm  ATRX           MECP2_transcriptomics_ENSG00000169057.18   \n",
       "0          Gbm  ATRX          TSPYL2_transcriptomics_ENSG00000184205.13   \n",
       "1          Gbm  ATRX    RP11-296E23.2_transcriptomics_ENSG00000265948.1   \n",
       "2          Gbm  ATRX          CCDC68_transcriptomics_ENSG00000166510.12   \n",
       "3          Gbm  ATRX     RP11-624L4.1_transcriptomics_ENSG00000259345.4   \n",
       "4          Gbm  ATRX          GREB1L_transcriptomics_ENSG00000141449.13   \n",
       "5          Gbm  ATRX     RP11-80F22.7_transcriptomics_ENSG00000260153.1   \n",
       "6          Gbm  ATRX          CDR1-AS_transcriptomics_ENSG00000281508.1   \n",
       "7          Gbm  ATRX    RP11-541N10.3_transcriptomics_ENSG00000260461.1   \n",
       "8          Gbm  ATRX       AC005237.4_transcriptomics_ENSG00000225521.1   \n",
       "9          Gbm  ATRX     RP11-279F6.1_transcriptomics_ENSG00000245750.6   \n",
       "10         Gbm  ATRX          EXOSC1_transcriptomics_ENSG00000171311.11   \n",
       "11         Gbm  ATRX       AC093585.6_transcriptomics_ENSG00000229996.1   \n",
       "12         Gbm  ATRX           PCAT29_transcriptomics_ENSG00000259641.4   \n",
       "13         Gbm  ATRX             DRG2_transcriptomics_ENSG00000108591.8   \n",
       "14         Gbm  ATRX          SEC31B_transcriptomics_ENSG00000075826.15   \n",
       "15         Gbm  ATRX    RP11-323P17.2_transcriptomics_ENSG00000274461.1   \n",
       "16         Gbm  ATRX             DAW1_transcriptomics_ENSG00000123977.8   \n",
       "17         Gbm  ATRX  LL0XNC01-116E7.5_transcriptomics_ENSG000002762...   \n",
       "18         Gbm  ATRX    RP11-370P15.2_transcriptomics_ENSG00000261014.1   \n",
       "19         Gbm  ATRX     CTB-174O21.2_transcriptomics_ENSG00000269487.1   \n",
       "20         Gbm  ATRX     RP4-587D13.1_transcriptomics_ENSG00000281008.1   \n",
       "21         Gbm  ATRX      CTC-370J7.1_transcriptomics_ENSG00000253630.1   \n",
       "22         Gbm  ATRX     CTC-543D15.1_transcriptomics_ENSG00000260112.1   \n",
       "23         Gbm  ATRX           AGAP4_transcriptomics_ENSG00000188234.12   \n",
       "24         Gbm  ATRX     LINC00328-2P_transcriptomics_ENSG00000225016.1   \n",
       "25         Gbm  ATRX      RP11-99A1.2_transcriptomics_ENSG00000267311.1   \n",
       "26         Gbm  ATRX        RN7SL284P_transcriptomics_ENSG00000243446.3   \n",
       "27         Gbm  ATRX       AC010095.7_transcriptomics_ENSG00000235644.1   \n",
       "28         Gbm  ATRX        KCNQ5-IT1_transcriptomics_ENSG00000233844.1   \n",
       "29         Gbm  ATRX            PHF8_transcriptomics_ENSG00000172943.17   \n",
       "30         Gbm  ATRX         RPS10P28_transcriptomics_ENSG00000243167.1   \n",
       "31         Gbm  ATRX            GPX3_transcriptomics_ENSG00000211445.10   \n",
       "32         Gbm  ATRX           CNNM2_transcriptomics_ENSG00000148842.16   \n",
       "\n",
       "   Interacting_Protein       P_Value  \n",
       "0                 True  4.671346e-04  \n",
       "0                False  5.743268e-10  \n",
       "1                False  6.907699e-10  \n",
       "2                False  4.049959e-09  \n",
       "3                False  1.148094e-08  \n",
       "4                False  1.270396e-08  \n",
       "5                False  2.534988e-08  \n",
       "6                False  3.881078e-08  \n",
       "7                False  4.690104e-08  \n",
       "8                False  4.760715e-08  \n",
       "9                False  8.828716e-08  \n",
       "10               False  1.059106e-07  \n",
       "11               False  1.130568e-07  \n",
       "12               False  1.398273e-07  \n",
       "13               False  1.594127e-07  \n",
       "14               False  1.644265e-07  \n",
       "15               False  1.707400e-07  \n",
       "16               False  1.973377e-07  \n",
       "17               False  2.468243e-07  \n",
       "18               False  3.184356e-07  \n",
       "19               False  3.383054e-07  \n",
       "20               False  3.497493e-07  \n",
       "21               False  3.533912e-07  \n",
       "22               False  3.893415e-07  \n",
       "23               False  4.749609e-07  \n",
       "24               False  5.141748e-07  \n",
       "25               False  5.702053e-07  \n",
       "26               False  5.768131e-07  \n",
       "27               False  6.443425e-07  \n",
       "28               False  6.546887e-07  \n",
       "29               False  6.997059e-07  \n",
       "30               False  7.035771e-07  \n",
       "31               False  7.529478e-07  \n",
       "32               False  8.219161e-07  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "if len(all_significant_comparisons) > 0:\n",
    "    display(all_significant_comparisons)\n",
    "    \n",
    "else:\n",
    "    print('No Significant Comparisons!')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Write Significant Comparisons (if any) to Shared CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\nexisting_results = pd.read_csv(gene+'_Trans_Results.csv')\\n\\nupdated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\\n\\nupdated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)\""
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "existing_results = pd.read_csv(gene+'_Trans_Results.csv')\n",
    "\n",
    "updated_results = pd.concat([existing_results, all_significant_comparisons], sort=False)\n",
    "\n",
    "updated_results.to_csv(path_or_buf = gene + '_Trans_Results.csv', index=False)'''"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
