{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RB1 cis comparison\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "cis comparisons for RB1 for transcriptomics, proteomics, and phosphoproteomics  \n",
    "Also includes combined phosphoproteomic graph with RB1 and CCNE1 phosphoproteoimcs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Library Imports"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run this cell to import the necessary libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "#sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\GBM_for_CPTAC\\\\')\n",
    "#import cis_functions as f\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_significance_col(results_df, num_comparisons):\n",
    "    \"bonferroni multiple hypothesis\"\"\"\n",
    "    alpha = .05\n",
    "    bonferroni_cutoff = alpha / num_comparisons\n",
    "    \n",
    "    pval = results_df['P_Value']\n",
    "    if float(pval[0]) <= bonferroni_cutoff:\n",
    "        results_df['Significant'] = True\n",
    "    else: \n",
    "        results_df['Significant'] = False\n",
    "    return results_df\n",
    "\n",
    "def wrap_ttest_return_all(df, label_column, comparison_columns, total_tests, alpha=.05):\n",
    "    try:\n",
    "        #Verify precondition that label column exists and has exactly 2 unique values\n",
    "        label_values = df[label_column].unique()\n",
    "        if len(label_values) != 2:\n",
    "            print(\"Incorrectly Formatted Dataframe! Label column must have exactly 2 unique values.\")\n",
    "            return None\n",
    "        \n",
    "        #Partition dataframe into two sets, one for each of the two unique values from the label column\n",
    "        partition1 = df.loc[df[label_column] == label_values[0]]\n",
    "        partition2 = df.loc[df[label_column] == label_values[1]]\n",
    "        \n",
    "        #Determine the number of real valued columns on which we will do t-tests\n",
    "        #sites = len(comparison_columns.columns)\n",
    "        number_of_comparisons = total_tests # ? phospho sites or num freq mut genes doing cis comp\n",
    "        \n",
    "        #Use a bonferroni correction to adjust for multiple testing by altering the p-value needed for acceptance\n",
    "        bonferroni_cutoff = alpha/number_of_comparisons\n",
    "        \n",
    "        #Store all comparisons with their p-values in a dictionary\n",
    "        all_comparisons = {}\n",
    "        \n",
    "        #Loop through each comparison column, perform the t-test, and determine whether it meets the significance cutoff'''\n",
    "        for column in comparison_columns:\n",
    "            stat, pval = scipy.stats.ttest_ind(partition1[column].dropna(axis=0), partition2[column].dropna(axis=0))\n",
    "            all_comparisons[column] = pval\n",
    "    \n",
    "        #Sort dictionary to list smallest p-values first\n",
    "        sorted_comparisons = sorted(all_comparisons.items(), key=lambda kv: kv[1])\n",
    "        #Format as a dataframe and return to caller\n",
    "        all_comparisons_df = pd.DataFrame.from_dict(sorted_comparisons)\n",
    "        all_comparisons_df.columns = ['Comparison', 'P_Value']\n",
    "        \n",
    "                                               \n",
    "        all_comparisons_sig_col = add_significance_col(all_comparisons_df, number_of_comparisons)\n",
    "        return all_comparisons_sig_col\n",
    "                                \n",
    "    except:\n",
    "        print(\"Incorrectly Formatted Dataframe!\")\n",
    "        return None\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# note: pass reduced omics # CHECK would include non coding, but none in fm genes\n",
    "\n",
    "def format_mutated_wt_comparison(reduced_omics_df):\n",
    "    # Step 2 - Create binary column \n",
    "    reduced_omics_df['binary_mutations'] = np.where(\n",
    "                reduced_omics_df[gene+'_Mutation_Status'] == 'Wildtype_Tumor', 'Wildtype', 'Mutated')\n",
    "\n",
    "    # Step 3 - Format the dataframe correctly for the T-test(just omics and binary columns for tumors)\n",
    "    tumors = reduced_omics_df[reduced_omics_df.Sample_Status == 'Tumor'] #drop Normal samples\n",
    "    columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "    mut_status = tumors.drop(columns_to_drop, axis = 1)\n",
    "\n",
    "    return mut_status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_missense_truncation_comparison(cancer_object, reduced_omics_df):\n",
    "    #get omics data and tumors\n",
    "    tumors = reduced_omics_df.loc[reduced_omics_df['Sample_Status'] == 'Tumor'] #drop Normal samples\n",
    "\n",
    "\n",
    "    somatic_mutations = cancer_object.get_somatic_mutation().reset_index()\n",
    "\n",
    "    if cancer_object.get_cancer_type() == 'colon':\n",
    "        missence_truncation_groups = {'frameshift substitution': 'Truncation', \n",
    "            'frameshift deletion': 'Truncation', 'frameshift insertion': 'Truncation', \n",
    "            'stopgain': 'Truncation', 'stoploss': 'Truncation', 'nonsynonymous SNV': 'Missense',\n",
    "            'nonframeshift insertion': 'Missense','nonframeshift deletion': 'Missense', \n",
    "            'nonframeshift substitution': 'Missense'}\n",
    "    else: \n",
    "        missence_truncation_groups = {'In_Frame_Del': 'Missense', 'In_Frame_Ins': 'Missense',\n",
    "            'Missense_Mutation': 'Missense', 'Frame_Shift_Del': 'Truncation','Nonsense_Mutation': 'Truncation', \n",
    "            'Splice_Site': 'Truncation', 'Frame_Shift_Ins': 'Truncation','Nonstop_Mutation':'Truncation'}\n",
    "\n",
    "    mutations_replaced_M_T = somatic_mutations.replace(missence_truncation_groups)\n",
    "    mutations_replaced_M_T = mutations_replaced_M_T.loc[mutations_replaced_M_T['Gene'] == gene]\n",
    "\n",
    "    # group mutation categories\n",
    "    miss = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'Missense']\n",
    "    trunc = mutations_replaced_M_T.loc[mutations_replaced_M_T['Mutation'] == 'Truncation']\n",
    "\n",
    "    #get lists of unique samples for missence and trucation categories\n",
    "    miss_unique_samples = list(miss['Sample_ID'].unique())\n",
    "    trunc_unique_samples = list(trunc['Sample_ID'].unique())\n",
    "    \n",
    "    #check if there is only one type of mutation for the specific gene\n",
    "    if miss_unique_samples == []:\n",
    "        print('Only truncation type mutations found for', gene+'.', \n",
    "             'Not possible to compare missense with wildtype.')\n",
    "        truncation_omics = tumors.loc[tumors.index.isin(trunc_unique_samples)]\n",
    "        truncation_omics = truncation_omics.assign(binary_mutations = 'Truncation')\n",
    "        columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "        binary_mut_omics = truncation_omics.drop(columns_to_drop, axis = 1)\n",
    "        return binary_mut_omics\n",
    "    elif trunc_unique_samples == []:\n",
    "        print('Only missence type mutations found for', gene+'.', \n",
    "             'Not possible to compare truncation with wildtype.')\n",
    "        missence_omics = tumors.loc[tumors.index.isin(miss_unique_samples)]\n",
    "        missence_omics = missence_omics.assign(binary_mutations = 'Missense')\n",
    "        columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "        binary_mut_omics = missence_omics.drop(columns_to_drop, axis = 1)\n",
    "        return binary_mut_omics\n",
    "\n",
    "    ### Step 2 - Create the binary column needed to do the comparison\n",
    "    # Get mutation catagories with omics data\n",
    "    missence_omics = tumors.loc[tumors.index.isin(miss_unique_samples)]\n",
    "    missence_omics = missence_omics.assign(binary_mutations = 'Missense')\n",
    "    truncation_omics = tumors.loc[tumors.index.isin(trunc_unique_samples)]\n",
    "    truncation_omics = truncation_omics.assign(binary_mutations = 'Truncation')\n",
    "    binary_mut_omics = missence_omics.append(truncation_omics)\n",
    "\n",
    "    # Step 3 - Format the dataframe correctly for the T-test(just omics and binary columns for tumors)\n",
    "    columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "    binary_mut_omics = binary_mut_omics.drop(columns_to_drop, axis = 1)\n",
    "\n",
    "    return binary_mut_omics\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_pval_annotation(pval_symbol, x1, x2, line_start = .05, line_height=.05):\n",
    "    # for manual adjustment to pval annotations\n",
    "    \n",
    "    y, h = line_start, line_height\n",
    "    plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "             [y, y+h, y+h, y], #vertical line\n",
    "             lw=1.5, color= '.3')\n",
    "    plt.text((x1+x2)*.5, # half between x coord\n",
    "             y+h, pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "\n",
    "# used in pval_annotation\n",
    "def get_pval_symbol(pval_df):\n",
    "    if pval_df.iloc[0:,2].bool() == True:\n",
    "        pval_symbol = '*'\n",
    "    else:\n",
    "        pval_symbol = 'ns'\n",
    "    return pval_symbol\n",
    "\n",
    "def pval_annotation(pval_df, plotted_df):\n",
    "    # pval symbol\n",
    "    num_pvals = len(pval_df)\n",
    "    omics_col = plotted_df.columns[0]\n",
    "    \n",
    "\n",
    "    # annotation for Mutated and Wiltype\n",
    "    if num_pvals == 1:\n",
    "        mut_pval = pval_df.loc[pval_df.index == \"Mutated/Wildtype\"]\n",
    "        mut_pval_symbol = get_pval_symbol(pval_df)\n",
    "        \n",
    "        x1, x2 = 0, 1   # columns (first column: 0, see plt.xticks())\n",
    "        y, h = plotted_df[omics_col].max() + .05, .05     \n",
    "        plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "                 [y, y+h, y+h, y], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x1+x2)*.5, # half between x coord\n",
    "                 y+h, mut_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "    \n",
    "    # annotation for Missense, Truncation and Wildtype\n",
    "    elif num_pvals == 2:\n",
    "        # Missense\n",
    "        miss_pval = pval_df.loc[pval_df.index == \"Missense/Wildtype\"]\n",
    "        miss_pval_symbol = get_pval_symbol(miss_pval)\n",
    "    \n",
    "        x1, x2 = 0, 1   # columns (first column: 0, see plt.xticks())\n",
    "        y, h = plotted_df[omics_col].max() + .05, .05     \n",
    "        plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "                 [y, y+h, y+h, y], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x1+x2)*.5, # half between x coord\n",
    "                 y+h, miss_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "        \n",
    "        # Truncation \n",
    "        trunc_pval = pval_df.loc[pval_df.index == \"Truncation/Wildtype\"]\n",
    "        trunc_pval_symbol = get_pval_symbol(trunc_pval)\n",
    "        \n",
    "        x3, x4 = 0, 2   # columns (first column: 0, see plt.xticks())\n",
    "        y2, h2 = plotted_df[omics_col].max() + .15, .05     \n",
    "        plt.plot([x3, x3, x4, x4], #draw horizontal line\n",
    "                 [y2, y2+h2, y2+h2, y2], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x3+x4)*.5, # half between x coord\n",
    "                 y2+h2, trunc_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Find the mutation frequency of the gene of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "#cptac.download(dataset=\"endometrial\", version=\"latest\")\n",
    "en = cptac.Endometrial()\n",
    "desired_cutoff = 0.05\n",
    "gene = 'RB1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#FIX\n",
    "#freq_mut = u.get_frequently_mutated(brain, cutoff = desired_cutoff)\n",
    "#freq_mut.loc[freq_mut['Gene'] == gene]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 3: cis comparisons for omics "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Determine if the DNA mutation has an effect on the omics measurement. In order to do this, we have a few steps in code. These steps are found in the format_cis_comparison_data function.\n",
    "1. get a table with both the omics and mutation data for tumors\n",
    "2. get a binary column from the mutation data to separate our samples\n",
    "3. format data frame to be used in the T-test\n",
    "4. send data to the T-test.\n",
    "\n",
    "The format_cis_comparison_data does the first 3 steps (for dataframes with multi-indexes)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Transcriptomics cis comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Transcriptomics: Mutated and Wildtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 104 samples for the RB1 gene (/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Wildtype    90\n",
       "Mutated      5\n",
       "Name: binary_mutations, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#num_freq_mut_genes = len(freq_mut)\n",
    "num_freq_mut_genes = 20\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest - drop \n",
    "transrcpt_mutations = en.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'transcriptomics', omics_genes = gene)\n",
    "#reduced_transcript_mutations = en.reduce_multiindex(transrcpt_mutations, levels_to_drop=1) #single col labels\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "transcript_mut_wt = format_mutated_wt_comparison(transrcpt_mutations)\n",
    "transcript_mut_wt['binary_mutations'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-9-a3aa8ce27ef7>, line 6)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-9-a3aa8ce27ef7>\"\u001b[0;36m, line \u001b[0;32m6\u001b[0m\n\u001b[0;31m    transcript_results = wrap_ttest_return_all(transcript_mut_wt 'binary_mutations', trans_col_list, num_freq_mut_genes)\u001b[0m\n\u001b[0m                                                                                  ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "# Step 4 - T-test\n",
    "trans_col_list = [gene+'_transcriptomics']\n",
    "trans_col = gene+'_transcriptomics'\n",
    "\n",
    "print(\"Doing t-test comparison for mutation status\")\n",
    "transcript_results = wrap_ttest_return_all(transcript_mut_wt 'binary_mutations', trans_col_list, num_freq_mut_genes)\n",
    "transcript_results = transcript_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "transcript_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Multiple hypothesis testing is taken into account. Other notebooks test the rest of the frequently mutated genes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#f.cis_plot(trans_mut_status, gene, 'transcriptomics', transcrpt_results)\n",
    "plt.rcParams['figure.figsize']=(8,5)\n",
    "sns.set(font_scale = 1.3)\n",
    "\n",
    "cis_boxplot = sns.boxplot(data = transcript_mutations, x = 'binary_mutations',\n",
    "                          y = trans_col, order = ['Wildtype', 'Mutated'], showfliers = False)  \n",
    "cis_boxplot.set_title(gene + \" Effect on \" + gene +\" Transcriptomics in Glioblastoma Tumors\\n\")\n",
    "cis_boxplot = sns.stripplot(data= transcript_mutations, x = 'binary_mutations',\n",
    "                            y = trans_col,jitter = True, color = \".3\", order = ['Wildtype', 'Mutated'])\n",
    "cis_boxplot.set(xlabel = \"\\n\"+gene + \" Mutation Status in Tumors\", ylabel = 'Transcriptomics')\n",
    "cis_boxplot.set_xticklabels(cis_boxplot.get_xticklabels())\n",
    "\n",
    "mut_pval_symbol = get_pval_symbol(transcript_results)\n",
    "format_pval_annotation(mut_pval_symbol, x1 = 0, x2 = 1, line_start = 12, line_height = 1)\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Transcriptomics: Missense and Truncation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# select all wildtype from earlier -add to mut_type binary column\n",
    "trans_wildtype = transcript_mut_wt.loc[transcript_mut_wt['binary_mutations'] == 'Wildtype'] \n",
    "trans_miss_trunc = get_missense_truncation_comparison(en, transrcpt_mutations)\n",
    "trans_miss_trunc_and_wt = trans_miss_trunc.append(trans_wildtype, sort=False) # use for graph\n",
    "trans_miss_trunc_and_wt.head() # 2 overlap : 102 total\n",
    "trans_miss_trunc_and_wt['binary_mutations'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# compare missense and wildtype\n",
    "miss_wt = trans_miss_trunc_and_wt.loc[trans_miss_trunc_and_wt['binary_mutations'] != 'Truncation']\n",
    "print(\"Doing t-test comparison for missense and wildtype\")\n",
    "trans_results_miss = wrap_ttest_return_all(miss_wt, 'binary_mutations', trans_col_list, num_freq_mut_genes)\n",
    "trans_results_miss = trans_results_miss.rename(index = {0: \"Missense/Wildtype\"})\n",
    "\n",
    "# compare truncation and wildtype\n",
    "trunc_wt = trans_miss_trunc_and_wt.loc[trans_miss_trunc_and_wt['binary_mutations'] != 'Missense']\n",
    "print(\"Doing t-test comparison for truncation and wildtype\")\n",
    "trans_results_trunc = wrap_ttest_return_all(trunc_wt, 'binary_mutations', trans_col_list, num_freq_mut_genes)\n",
    "trans_results_trunc = trans_results_trunc.rename(index = {0: \"Truncation/Wildtype\"})\n",
    "\n",
    "trans_results_miss_trunc = trans_results_miss.append(trans_results_trunc)\n",
    "trans_results_miss_trunc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#f.cis_plot(trans_mut_and_wildtype, gene, 'transcriptomics', trans_results_miss_trunc)\n",
    "plt.rcParams['figure.figsize']=(8,5)\n",
    "sns.set(font_scale = 1.3)\n",
    "cis_boxplot = sns.boxplot(data = trans_miss_trunc_and_wt, x = 'binary_mutations',\n",
    "                          y = trans_col, order = ['Wildtype',\"Missense\" ,'Truncation'], showfliers = False)  \n",
    "cis_boxplot.set_title(gene + \" Effect on \" + gene +\" Transcriptomics in Glioblastoma Tumors\\n\")\n",
    "cis_boxplot = sns.stripplot(data= trans_miss_trunc_and_wt, x = 'binary_mutations',\n",
    "                            y = trans_col,jitter = True, color = \".3\", order = ['Wildtype',\"Missense\", 'Truncation'])\n",
    "cis_boxplot.set(xlabel = \"\\n\"+gene + \" Mutation Status in Tumors\", ylabel = 'Transcriptomics')\n",
    "cis_boxplot.set_xticklabels(cis_boxplot.get_xticklabels())\n",
    "\n",
    "# Missense\n",
    "miss_pval = trans_results_miss_trunc.loc[trans_results_miss_trunc.index == \"Missense/Wildtype\"]\n",
    "miss_pval_symbol = get_pval_symbol(miss_pval)\n",
    "format_pval_annotation(miss_pval_symbol, x1 = 0, x2 = 1, line_start = 14, line_height = .7)\n",
    "\n",
    "# Truncation \n",
    "trunc_pval = trans_results_miss_trunc.loc[trans_results_miss_trunc.index == \"Truncation/Wildtype\"]\n",
    "trunc_pval_symbol = get_pval_symbol(trunc_pval)\n",
    "format_pval_annotation(trunc_pval_symbol, x1 = 0, x2 = 2, line_start = 13, line_height = .5)\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Proteomics cis comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Proteomics: Mutated and Wildtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest  \n",
    "prot_and_mutations = en.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'proteomics', omics_genes = gene) # single col level\n",
    "\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "prot_mut_wt = format_mutated_wt_comparison(prot_and_mutations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut) \n",
    "num_freq_mut_genes = 20\n",
    "prot_col_list = [gene+'_proteomics']\n",
    "prot_col = gene+\"_proteomics\"\n",
    "\n",
    "# Step 4 - T-test comparing means of mutated vs wildtype effect on cis omics\n",
    "print(\"Doing t-test comparison for mutated and wildtype\")\n",
    "prot_results = wrap_ttest_return_all(prot_mut_wt, 'binary_mutations', prot_col_list, num_freq_mut_genes)\n",
    "prot_results = prot_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "prot_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#f.cis_plot(prot_mut_status, gene, \"proteomics\", prot_results)\n",
    "\n",
    "plt.rcParams['figure.figsize']=(8,5)\n",
    "sns.set(font_scale = 1.3)\n",
    "cis_boxplot = sns.boxplot(data = prot_mut_wt, x = 'binary_mutations',\n",
    "                          y = prot_col, order = ['Wildtype', 'Mutated'], showfliers = False)  \n",
    "cis_boxplot.set_title(gene + \" Effect on \" + gene +\" Proteomics in Endometrial Tumors\\n\")\n",
    "cis_boxplot = sns.stripplot(data= prot_mut_wt, x = 'binary_mutations',\n",
    "                            y = prot_col,jitter = True, color = \".3\", order = ['Wildtype', 'Mutated'])\n",
    "cis_boxplot.set(xlabel = \"\\n\"+gene + \" Mutation Status in Tumors\", ylabel = 'Proteomics')\n",
    "cis_boxplot.set_xticklabels(cis_boxplot.get_xticklabels())\n",
    "\n",
    "mut_pval_symbol = get_pval_symbol(prot_results)\n",
    "format_pval_annotation(mut_pval_symbol, x1 = 0, x2 = 1, line_start = 1.65, line_height = .15)\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Proteomics: Missense and Truncation compared to wildtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Repeat Step 2 & 3 - binary column \n",
    "prot_miss_trunc = get_missense_truncation_comparison(en, prot_and_mutations)\n",
    "prot_wildtype = prot_mut_wt.loc[prot_mut_wt['binary_mutations'] == 'Wildtype'] # select all wildtype \n",
    "prot_miss_trunc_wildtype = prot_miss_trunc.append(prot_wildtype, sort=False) # use for graph\n",
    "\n",
    "# compare missense and wildtype\n",
    "prot_miss_wt = prot_miss_trunc_wildtype.loc[prot_miss_trunc_wildtype['binary_mutations'] != 'Truncation']\n",
    "print(\"Doing t-test comparison for missense and wildtype\")\n",
    "prot_results_miss = wrap_ttest_return_all(prot_miss_wt, 'binary_mutations', prot_col_list, num_freq_mut_genes)\n",
    "prot_results_miss = prot_results_miss.rename(index = {0: \"Missense/Wildtype\"})\n",
    "\n",
    "# compare truncation and wildtype\n",
    "prot_trunc_wt = prot_miss_trunc_wildtype.loc[prot_miss_trunc_wildtype['binary_mutations'] != 'Missense']\n",
    "print(\"Doing t-test comparison for truncation and wildtype\")\n",
    "prot_results_trunc = wrap_ttest_return_all(prot_trunc_wt, 'binary_mutations', prot_col_list, num_freq_mut_genes)\n",
    "prot_results_trunc = prot_results_trunc.rename(index = {0: \"Truncation/Wildtype\"})\n",
    "\n",
    "prot_results_miss_trunc = prot_results_miss.append(prot_results_trunc)\n",
    "prot_results_miss_trunc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(8,5)\n",
    "sns.set(font_scale = 1.3)\n",
    "cis_boxplot = sns.boxplot(data = prot_miss_trunc_wildtype, x = 'binary_mutations',\n",
    "                          y = prot_col, order = ['Wildtype', 'Missense', 'Truncation'], showfliers = False)  \n",
    "cis_boxplot.set_title(gene + \" Effect on \" + gene +\" Proteomics in Endometrial Tumors\\n\")\n",
    "cis_boxplot = sns.stripplot(data= prot_miss_trunc_wildtype, x = 'binary_mutations',\n",
    "                            y = prot_col,jitter = True, color = \".3\", order = ['Wildtype', 'Missense', 'Truncation'])\n",
    "cis_boxplot.set(xlabel = \"\\n\"+gene + \" Mutation Status in Tumors\", ylabel = 'Proteomics')\n",
    "cis_boxplot.set_xticklabels(cis_boxplot.get_xticklabels())\n",
    "\n",
    "# Missense\n",
    "miss_pval = prot_results_miss_trunc.loc[prot_results_miss_trunc.index == \"Missense/Wildtype\"]\n",
    "miss_pval_symbol = get_pval_symbol(miss_pval)\n",
    "format_pval_annotation(miss_pval_symbol, x1 = 0, x2 = 1, line_start = 1.65, line_height = 0.15)\n",
    "\n",
    "# Truncation \n",
    "trunc_pval = prot_results_miss_trunc.loc[prot_results_miss_trunc.index == \"Truncation/Wildtype\"]\n",
    "trunc_pval_symbol = get_pval_symbol(trunc_pval)\n",
    "format_pval_annotation(trunc_pval_symbol, x1 = 0, x2 = 2, line_start = 1.95, line_height = 0.15)\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phosphoproteomics cis comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Phosphoproteomics: Mutated and Wildtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest\n",
    "phospho_and_mut = en.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'phosphoproteomics', omics_genes = gene)\n",
    "flatten_phospho_and_mut = en.reduce_multiindex(phospho_and_mut, flatten=True)#reduce multi columns\n",
    "flatten_phospho_and_mut\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Step 2 - Create the binary column needed to do the comparison\n",
    "flatten_phospho_and_mut['binary_mutations'] = flatten_phospho_and_mut[gene+'_Mutation_Status'].apply(\n",
    "    lambda x: 'Wildtype' if x == 'Wildtype_Tumor' else 'Mutated')\n",
    "\n",
    "# Step 3 - Format\n",
    "phospho_tumors = flatten_phospho_and_mut.query('Sample_Status != \"Normal\"') # drop Normal samples (warning with .loc)\n",
    "columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "binary_phospho = phospho_tumors.drop(columns_to_drop, axis = 1)\n",
    "only_phospho = binary_phospho.drop('binary_mutations', axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use al.wrap_ttest for many columns of phospho data (takes phospho sites into account with bonferroni cutoff)\n",
    "phos_col_list = list(only_phospho.columns)\n",
    "\n",
    "# Step 4 - T-test comparing means of mutated vs wildtype effect on cis omics\n",
    "print(\"Doing t-test comparison for mutation status\")\n",
    "phos_pval_results = u.wrap_ttest(binary_phospho, 'binary_mutations', phos_col_list)\n",
    "#phos_pval_results = phos_pval_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "phos_pval_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Graph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create data frame with all significant phospho sites in order to graph\n",
    "#phospho_list = list(phos_pval_results['Comparison']) # comment when no sig phospho sites\n",
    "phospho_list = list(only_phospho.columns) # uncomment when no sig phospho sites\n",
    "phospho_df = pd.melt(binary_phospho, id_vars = 'binary_mutations', \n",
    "            value_vars = phospho_list, var_name = 'specific_phospho', value_name = 'phosphoproteomics')\n",
    "phospho_df.replace(to_replace = '_phosphoproteomics', value = '', \n",
    "                        inplace = True, regex = True) # shorten column names\n",
    "\n",
    "\n",
    "\n",
    "phospho_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(25,10) #size of plot\n",
    "sns.set(font_scale = 2)\n",
    "\n",
    "cis_boxplot = sns.boxplot(data = phospho_df, x = 'specific_phospho', y = 'phosphoproteomics', hue = 'binary_mutations',\n",
    "                             hue_order = [\"Wildtype\",\"Mutated\"], showfliers = False) \n",
    "cis_boxplot.set_title(gene + \" Effect on \" + gene + \" Phosphoproteomics in Endometrial  Tumors\")\n",
    "cis_boxplot = sns.stripplot(data = phospho_df, x = 'specific_phospho', y = 'phosphoproteomics', hue = 'binary_mutations',\n",
    "                               hue_order = [\"Wildtype\",\"Mutated\"], jitter = True, color = \".3\", dodge = True)\n",
    "cis_boxplot.set(xlabel = 'Phosphorylation Sites', ylabel = \"Phosphorylation Levels\")\n",
    "cis_boxplot.set_xticklabels(cis_boxplot.get_xticklabels(),rotation=80)\n",
    "\n",
    "# format legend\n",
    "handles, labels = cis_boxplot.get_legend_handles_labels()\n",
    "plt.legend(handles[0:2], labels[0:2])\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Phosphoproteomics: Missense and Truncation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# select all wildtype from earlier -add to mut_type binary column\n",
    "phos_wildtype = binary_phospho.loc[binary_phospho['binary_mutations'] == 'Wildtype'] \n",
    "\n",
    "phos_mut_type = get_missense_truncation_comparison(en, flatten_phospho_and_mut)\n",
    "phos_mut_and_wildtype = phos_mut_type.append(phos_wildtype, sort=True) \n",
    "phos_mut_and_wildtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "phos_col_list = list(only_phospho.columns)\n",
    "# compare missense and wildtype\n",
    "miss_wt = phos_mut_and_wildtype.loc[phos_mut_and_wildtype['binary_mutations'] != 'Truncation']\n",
    "print(\"Doing t-test comparison for missense and wildtype\\n\")\n",
    "phos_results_miss = u.wrap_ttest(miss_wt, 'binary_mutations', phos_col_list)\n",
    "print(phos_results_miss)\n",
    "\n",
    "# compare truncation and wildtype \n",
    "trunc_wt = phos_mut_and_wildtype.loc[phos_mut_and_wildtype['binary_mutations'] != 'Missense']\n",
    "print(\"\\n\\nDoing t-test comparison for truncation and wildtype\\n\")\n",
    "phos_results_trunc = u.wrap_ttest(trunc_wt, 'binary_mutations', phos_col_list)\n",
    "phos_results_trunc\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Graph Missense and Truncation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create data frame with all significant phospho sites in order to graph\n",
    "mut_type_phospho_list = list(phos_results_trunc['Comparison']) # comment when no sig phospho sites\n",
    "#mut_type_phospho_list = list(only_phospho.columns) # uncomment when no sig phospho sites\n",
    "phos_mut_type_df = pd.melt(phos_mut_and_wildtype, id_vars = 'binary_mutations', \n",
    "            value_vars = mut_type_phospho_list, var_name = 'specific_phospho', value_name = 'phosphoproteomics')\n",
    "phos_mut_type_df.replace(to_replace = '_phosphoproteomics', value = '', \n",
    "                        inplace = True, regex = True) # shorten column names\n",
    "phos_mut_type_df.replace(to_replace = '.TNILQYASTRPPTLSPIPHIPRS\\WPYK', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "phos_mut_type_df.replace(to_replace = '.\\DREGPTDHLESACPLNLPLQNNHTAADMYLSPVRS\\WPK', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "phos_mut_type_df.replace(to_replace = '.TAVIPINGS\\WPR', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "phos_mut_type_df.replace(to_replace = '.IPGGNIYIS\\WPLKS\\WPYK', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "phos_mut_type_df.replace(to_replace = '.TLQTDSIDSFETQRT\\WPRK', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "\n",
    "\n",
    "phos_mut_type_df.replace(to_replace = '.SNLDEEVNVIPPHT\\WPVR', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "\n",
    "phos_mut_type_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(20,10) #size of plot\n",
    "sns.set(font_scale = 2)\n",
    "phos_cis_boxplot = sns.boxplot(data = phos_mut_type_df, x = 'specific_phospho', y = 'phosphoproteomics', hue = 'binary_mutations',\n",
    "                             hue_order = [\"Wildtype\",\"Missense\",\"Truncation\"], showfliers = False)  \n",
    "phos_cis_boxplot.set_title(gene + \" Effect on \" + gene + \" Phosphoproteomics in Endrometrial Tumors\")\n",
    "phos_cis_boxplot = sns.stripplot(data = phos_mut_type_df, x = 'specific_phospho', y = 'phosphoproteomics', hue = 'binary_mutations',\n",
    "                               hue_order = [\"Wildtype\",\"Missense\",\"Truncation\"], jitter = True, color = \".3\", dodge = True)\n",
    "phos_cis_boxplot.set(xlabel = 'Phosphorylation Sites', ylabel = \"Phosphorylation Levels\")\n",
    "phos_cis_boxplot.set_xticklabels(phos_cis_boxplot.get_xticklabels(),rotation = 80)\n",
    "\n",
    "# format legend\n",
    "handles, labels = phos_cis_boxplot.get_legend_handles_labels()\n",
    "plt.legend(handles[0:3], labels[0:3])\n",
    "\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phosphorylation Normalization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Data already is log transformed. To normalize Phosphoproteomics - Proteomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 139 samples for the RB1 gene (/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>RB1_phosphoproteomics_S249</th>\n",
       "      <th>RB1_phosphoproteomics_S37</th>\n",
       "      <th>RB1_phosphoproteomics_S608</th>\n",
       "      <th>RB1_phosphoproteomics_S612</th>\n",
       "      <th>RB1_phosphoproteomics_S624</th>\n",
       "      <th>RB1_phosphoproteomics_S773</th>\n",
       "      <th>RB1_phosphoproteomics_S780</th>\n",
       "      <th>RB1_phosphoproteomics_S788</th>\n",
       "      <th>RB1_phosphoproteomics_S794</th>\n",
       "      <th>RB1_phosphoproteomics_S795</th>\n",
       "      <th>...</th>\n",
       "      <th>RB1_phosphoproteomics_T821</th>\n",
       "      <th>RB1_phosphoproteomics_T823</th>\n",
       "      <th>RB1_phosphoproteomics_T826</th>\n",
       "      <th>RB1_phosphoproteomics_Y771</th>\n",
       "      <th>RB1_phosphoproteomics_Y805</th>\n",
       "      <th>RB1_phosphoproteomics_Y813</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>0.6680</td>\n",
       "      <td>0.6750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.508000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.8550</td>\n",
       "      <td>...</td>\n",
       "      <td>1.490</td>\n",
       "      <td>1.320</td>\n",
       "      <td>1.320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.090</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>0.6040</td>\n",
       "      <td>0.4340</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.2740</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.040</td>\n",
       "      <td>1.040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>-0.2920</td>\n",
       "      <td>0.0170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.801</td>\n",
       "      <td>0.613</td>\n",
       "      <td>0.613</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-0.3065</td>\n",
       "      <td>0.5570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.717000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.107</td>\n",
       "      <td>-0.107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S006</td>\n",
       "      <td>-0.2205</td>\n",
       "      <td>-0.0247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9925</td>\n",
       "      <td>...</td>\n",
       "      <td>1.690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.805</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S149</td>\n",
       "      <td>-0.7220</td>\n",
       "      <td>-0.1670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.720</td>\n",
       "      <td>-0.354000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2710</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.648</td>\n",
       "      <td>-0.648</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S150</td>\n",
       "      <td>0.7640</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.994</td>\n",
       "      <td>0.489000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0550</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.030</td>\n",
       "      <td>1.030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S151</td>\n",
       "      <td>-0.5670</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0.203500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1435</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.253</td>\n",
       "      <td>-0.253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S152</td>\n",
       "      <td>-0.9760</td>\n",
       "      <td>-0.0941</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.238</td>\n",
       "      <td>-0.268000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.5400</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.050</td>\n",
       "      <td>-1.050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S153</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>-0.1720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.249</td>\n",
       "      <td>0.220357</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4420</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.977</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.684</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       RB1_phosphoproteomics_S249  RB1_phosphoproteomics_S37  \\\n",
       "Sample_ID                                                          \n",
       "S001                           0.6680                     0.6750   \n",
       "S002                           0.6040                     0.4340   \n",
       "S003                          -0.2920                     0.0170   \n",
       "S005                          -0.3065                     0.5570   \n",
       "S006                          -0.2205                    -0.0247   \n",
       "...                               ...                        ...   \n",
       "S149                          -0.7220                    -0.1670   \n",
       "S150                           0.7640                     0.0143   \n",
       "S151                          -0.5670                     0.3250   \n",
       "S152                          -0.9760                    -0.0941   \n",
       "S153                           0.6775                    -0.1720   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S608  RB1_phosphoproteomics_S612  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                         NaN   \n",
       "S150                              NaN                         NaN   \n",
       "S151                              NaN                         NaN   \n",
       "S152                              NaN                         NaN   \n",
       "S153                              NaN                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S624  RB1_phosphoproteomics_S773  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                             1.07                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                      -0.720   \n",
       "S150                              NaN                       0.994   \n",
       "S151                              NaN                       0.238   \n",
       "S152                              NaN                      -0.238   \n",
       "S153                              NaN                       0.249   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S780  RB1_phosphoproteomics_S788  \\\n",
       "Sample_ID                                                           \n",
       "S001                         0.508000                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                        -0.717000                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                        -0.354000                         NaN   \n",
       "S150                         0.489000                         NaN   \n",
       "S151                         0.203500                         NaN   \n",
       "S152                        -0.268000                         NaN   \n",
       "S153                         0.220357                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S794  RB1_phosphoproteomics_S795  ...  \\\n",
       "Sample_ID                                                          ...   \n",
       "S001                              NaN                     -0.8550  ...   \n",
       "S002                              NaN                      1.2740  ...   \n",
       "S003                              NaN                         NaN  ...   \n",
       "S005                              NaN                         NaN  ...   \n",
       "S006                              NaN                      0.9925  ...   \n",
       "...                               ...                         ...  ...   \n",
       "S149                              NaN                     -0.2710  ...   \n",
       "S150                              NaN                      1.0550  ...   \n",
       "S151                              NaN                     -0.1435  ...   \n",
       "S152                              NaN                     -1.5400  ...   \n",
       "S153                              NaN                      0.4420  ...   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T821  RB1_phosphoproteomics_T823  \\\n",
       "Sample_ID                                                           \n",
       "S001                            1.490                       1.320   \n",
       "S002                              NaN                       1.040   \n",
       "S003                            0.801                       0.613   \n",
       "S005                              NaN                      -0.107   \n",
       "S006                            1.690                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                      -0.648   \n",
       "S150                              NaN                       1.030   \n",
       "S151                              NaN                      -0.253   \n",
       "S152                              NaN                      -1.050   \n",
       "S153                              NaN                       0.977   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T826  RB1_phosphoproteomics_Y771  \\\n",
       "Sample_ID                                                           \n",
       "S001                            1.320                         NaN   \n",
       "S002                            1.040                         NaN   \n",
       "S003                            0.613                         NaN   \n",
       "S005                           -0.107                         NaN   \n",
       "S006                            1.690                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                           -0.648                         NaN   \n",
       "S150                            1.030                         NaN   \n",
       "S151                           -0.253                         NaN   \n",
       "S152                           -1.050                         NaN   \n",
       "S153                            0.977                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y805  RB1_phosphoproteomics_Y813  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                       1.090   \n",
       "S002                            1.760                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                       0.805   \n",
       "...                               ...                         ...   \n",
       "S149                           -1.170                         NaN   \n",
       "S150                            1.640                         NaN   \n",
       "S151                           -0.187                         NaN   \n",
       "S152                           -1.090                         NaN   \n",
       "S153                            0.684                         NaN   \n",
       "\n",
       "Name            RB1_Mutation   RB1_Location  RB1_Mutation_Status  \\\n",
       "Sample_ID                                                          \n",
       "S001        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S002        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S003        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S005        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S006        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "...                      ...            ...                  ...   \n",
       "S149       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S150       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S151       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S152       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S153       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "\n",
       "Name       Sample_Status  \n",
       "Sample_ID                 \n",
       "S001               Tumor  \n",
       "S002               Tumor  \n",
       "S003               Tumor  \n",
       "S005               Tumor  \n",
       "S006               Tumor  \n",
       "...                  ...  \n",
       "S149              Normal  \n",
       "S150              Normal  \n",
       "S151              Normal  \n",
       "S152              Normal  \n",
       "S153              Normal  \n",
       "\n",
       "[144 rows x 30 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create dataframe with phosphoproteomic data\n",
    "phospho_and_mut = en.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'phosphoproteomics', omics_genes = gene)\n",
    "flatten_phospho_and_mut = en.reduce_multiindex(phospho_and_mut,flatten=True)#reduce multi columns\n",
    "flatten_phospho_and_mut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 139 samples for the RB1 gene (/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>RB1_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>0.77500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>0.52700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>-0.00121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>0.13200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S006</td>\n",
       "      <td>0.17900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S149</td>\n",
       "      <td>-0.08510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S150</td>\n",
       "      <td>0.45800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S151</td>\n",
       "      <td>-0.11200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S152</td>\n",
       "      <td>-0.17400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S153</td>\n",
       "      <td>0.36100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows Ã— 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       RB1_proteomics\n",
       "Sample_ID                \n",
       "S001              0.77500\n",
       "S002              0.52700\n",
       "S003             -0.00121\n",
       "S005              0.13200\n",
       "S006              0.17900\n",
       "...                   ...\n",
       "S149             -0.08510\n",
       "S150              0.45800\n",
       "S151             -0.11200\n",
       "S152             -0.17400\n",
       "S153              0.36100\n",
       "\n",
       "[144 rows x 1 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1 - Create dataframe with proteomic data \n",
    "proteomic_and_mut = en.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'proteomics', omics_genes = gene)\n",
    "proteomic_and_mut_drop = proteomic_and_mut.drop([\"RB1_Mutation\", \"RB1_Location\", \"RB1_Mutation_Status\", \"Sample_Status\"], axis = 1)\n",
    "proteomic_and_mut_drop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>RB1_phosphoproteomics_S249</th>\n",
       "      <th>RB1_phosphoproteomics_S37</th>\n",
       "      <th>RB1_phosphoproteomics_S608</th>\n",
       "      <th>RB1_phosphoproteomics_S612</th>\n",
       "      <th>RB1_phosphoproteomics_S624</th>\n",
       "      <th>RB1_phosphoproteomics_S773</th>\n",
       "      <th>RB1_phosphoproteomics_S780</th>\n",
       "      <th>RB1_phosphoproteomics_S788</th>\n",
       "      <th>RB1_phosphoproteomics_S794</th>\n",
       "      <th>RB1_phosphoproteomics_S795</th>\n",
       "      <th>...</th>\n",
       "      <th>RB1_phosphoproteomics_T625</th>\n",
       "      <th>RB1_phosphoproteomics_T774</th>\n",
       "      <th>RB1_phosphoproteomics_T778</th>\n",
       "      <th>RB1_phosphoproteomics_T821</th>\n",
       "      <th>RB1_phosphoproteomics_T823</th>\n",
       "      <th>RB1_phosphoproteomics_T826</th>\n",
       "      <th>RB1_phosphoproteomics_Y771</th>\n",
       "      <th>RB1_phosphoproteomics_Y805</th>\n",
       "      <th>RB1_phosphoproteomics_Y813</th>\n",
       "      <th>RB1_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>0.6680</td>\n",
       "      <td>0.6750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.508000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.8550</td>\n",
       "      <td>...</td>\n",
       "      <td>0.288</td>\n",
       "      <td>-0.680000</td>\n",
       "      <td>0.5080</td>\n",
       "      <td>1.490</td>\n",
       "      <td>1.320</td>\n",
       "      <td>1.320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.090</td>\n",
       "      <td>0.77500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>0.6040</td>\n",
       "      <td>0.4340</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.2740</td>\n",
       "      <td>...</td>\n",
       "      <td>1.050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0825</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.040</td>\n",
       "      <td>1.040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.52700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>-0.2920</td>\n",
       "      <td>0.0170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.801</td>\n",
       "      <td>0.613</td>\n",
       "      <td>0.613</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.00121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-0.3065</td>\n",
       "      <td>0.5570</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.717000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.107</td>\n",
       "      <td>-0.107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.13200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S006</td>\n",
       "      <td>-0.2205</td>\n",
       "      <td>-0.0247</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9925</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.805</td>\n",
       "      <td>0.17900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S149</td>\n",
       "      <td>-0.7220</td>\n",
       "      <td>-0.1670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.720</td>\n",
       "      <td>-0.354000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2710</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039</td>\n",
       "      <td>-0.569500</td>\n",
       "      <td>-0.4040</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.648</td>\n",
       "      <td>-0.648</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.08510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S150</td>\n",
       "      <td>0.7640</td>\n",
       "      <td>0.0143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.994</td>\n",
       "      <td>0.489000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0550</td>\n",
       "      <td>...</td>\n",
       "      <td>0.652</td>\n",
       "      <td>0.560500</td>\n",
       "      <td>0.9250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.030</td>\n",
       "      <td>1.030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.45800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S151</td>\n",
       "      <td>-0.5670</td>\n",
       "      <td>0.3250</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.238</td>\n",
       "      <td>0.203500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1435</td>\n",
       "      <td>...</td>\n",
       "      <td>0.838</td>\n",
       "      <td>0.115245</td>\n",
       "      <td>0.2900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.253</td>\n",
       "      <td>-0.253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.187</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.11200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S152</td>\n",
       "      <td>-0.9760</td>\n",
       "      <td>-0.0941</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.238</td>\n",
       "      <td>-0.268000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.5400</td>\n",
       "      <td>...</td>\n",
       "      <td>1.100</td>\n",
       "      <td>-0.117740</td>\n",
       "      <td>0.1010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.050</td>\n",
       "      <td>-1.050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.090</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.17400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S153</td>\n",
       "      <td>0.6775</td>\n",
       "      <td>-0.1720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.249</td>\n",
       "      <td>0.220357</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4420</td>\n",
       "      <td>...</td>\n",
       "      <td>0.294</td>\n",
       "      <td>0.114650</td>\n",
       "      <td>0.4410</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.977</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.684</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.36100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       RB1_phosphoproteomics_S249  RB1_phosphoproteomics_S37  \\\n",
       "Sample_ID                                                          \n",
       "S001                           0.6680                     0.6750   \n",
       "S002                           0.6040                     0.4340   \n",
       "S003                          -0.2920                     0.0170   \n",
       "S005                          -0.3065                     0.5570   \n",
       "S006                          -0.2205                    -0.0247   \n",
       "...                               ...                        ...   \n",
       "S149                          -0.7220                    -0.1670   \n",
       "S150                           0.7640                     0.0143   \n",
       "S151                          -0.5670                     0.3250   \n",
       "S152                          -0.9760                    -0.0941   \n",
       "S153                           0.6775                    -0.1720   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S608  RB1_phosphoproteomics_S612  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                         NaN   \n",
       "S150                              NaN                         NaN   \n",
       "S151                              NaN                         NaN   \n",
       "S152                              NaN                         NaN   \n",
       "S153                              NaN                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S624  RB1_phosphoproteomics_S773  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                             1.07                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                      -0.720   \n",
       "S150                              NaN                       0.994   \n",
       "S151                              NaN                       0.238   \n",
       "S152                              NaN                      -0.238   \n",
       "S153                              NaN                       0.249   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S780  RB1_phosphoproteomics_S788  \\\n",
       "Sample_ID                                                           \n",
       "S001                         0.508000                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                        -0.717000                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                        -0.354000                         NaN   \n",
       "S150                         0.489000                         NaN   \n",
       "S151                         0.203500                         NaN   \n",
       "S152                        -0.268000                         NaN   \n",
       "S153                         0.220357                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S794  RB1_phosphoproteomics_S795  ...  \\\n",
       "Sample_ID                                                          ...   \n",
       "S001                              NaN                     -0.8550  ...   \n",
       "S002                              NaN                      1.2740  ...   \n",
       "S003                              NaN                         NaN  ...   \n",
       "S005                              NaN                         NaN  ...   \n",
       "S006                              NaN                      0.9925  ...   \n",
       "...                               ...                         ...  ...   \n",
       "S149                              NaN                     -0.2710  ...   \n",
       "S150                              NaN                      1.0550  ...   \n",
       "S151                              NaN                     -0.1435  ...   \n",
       "S152                              NaN                     -1.5400  ...   \n",
       "S153                              NaN                      0.4420  ...   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T625  RB1_phosphoproteomics_T774  \\\n",
       "Sample_ID                                                           \n",
       "S001                            0.288                   -0.680000   \n",
       "S002                            1.050                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                            0.117                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                           -0.039                   -0.569500   \n",
       "S150                            0.652                    0.560500   \n",
       "S151                            0.838                    0.115245   \n",
       "S152                            1.100                   -0.117740   \n",
       "S153                            0.294                    0.114650   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T778  RB1_phosphoproteomics_T821  \\\n",
       "Sample_ID                                                           \n",
       "S001                           0.5080                       1.490   \n",
       "S002                          -0.0825                         NaN   \n",
       "S003                              NaN                       0.801   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                       1.690   \n",
       "...                               ...                         ...   \n",
       "S149                          -0.4040                         NaN   \n",
       "S150                           0.9250                         NaN   \n",
       "S151                           0.2900                         NaN   \n",
       "S152                           0.1010                         NaN   \n",
       "S153                           0.4410                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T823  RB1_phosphoproteomics_T826  \\\n",
       "Sample_ID                                                           \n",
       "S001                            1.320                       1.320   \n",
       "S002                            1.040                       1.040   \n",
       "S003                            0.613                       0.613   \n",
       "S005                           -0.107                      -0.107   \n",
       "S006                              NaN                       1.690   \n",
       "...                               ...                         ...   \n",
       "S149                           -0.648                      -0.648   \n",
       "S150                            1.030                       1.030   \n",
       "S151                           -0.253                      -0.253   \n",
       "S152                           -1.050                      -1.050   \n",
       "S153                            0.977                       0.977   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y771  RB1_phosphoproteomics_Y805  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                       1.760   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                      -1.170   \n",
       "S150                              NaN                       1.640   \n",
       "S151                              NaN                      -0.187   \n",
       "S152                              NaN                      -1.090   \n",
       "S153                              NaN                       0.684   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y813  RB1_proteomics  \n",
       "Sample_ID                                              \n",
       "S001                            1.090         0.77500  \n",
       "S002                              NaN         0.52700  \n",
       "S003                              NaN        -0.00121  \n",
       "S005                              NaN         0.13200  \n",
       "S006                            0.805         0.17900  \n",
       "...                               ...             ...  \n",
       "S149                              NaN        -0.08510  \n",
       "S150                              NaN         0.45800  \n",
       "S151                              NaN        -0.11200  \n",
       "S152                              NaN        -0.17400  \n",
       "S153                              NaN         0.36100  \n",
       "\n",
       "[144 rows x 27 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phosphoprot_proteomic = flatten_phospho_and_mut.merge(proteomic_and_mut_drop, on='Sample_ID')\n",
    "phosphoprot_proteomic = phosphoprot_proteomic.drop([\"RB1_Mutation\", \"RB1_Location\", \"RB1_Mutation_Status\", \"Sample_Status\"], axis = 1)\n",
    "phosphoprot_proteomic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>RB1_phosphoproteomics_S249</th>\n",
       "      <th>RB1_phosphoproteomics_S37</th>\n",
       "      <th>RB1_phosphoproteomics_S608</th>\n",
       "      <th>RB1_phosphoproteomics_S612</th>\n",
       "      <th>RB1_phosphoproteomics_S624</th>\n",
       "      <th>RB1_phosphoproteomics_S773</th>\n",
       "      <th>RB1_phosphoproteomics_S780</th>\n",
       "      <th>RB1_phosphoproteomics_S788</th>\n",
       "      <th>RB1_phosphoproteomics_S794</th>\n",
       "      <th>RB1_phosphoproteomics_S795</th>\n",
       "      <th>...</th>\n",
       "      <th>RB1_phosphoproteomics_T625</th>\n",
       "      <th>RB1_phosphoproteomics_T774</th>\n",
       "      <th>RB1_phosphoproteomics_T778</th>\n",
       "      <th>RB1_phosphoproteomics_T821</th>\n",
       "      <th>RB1_phosphoproteomics_T823</th>\n",
       "      <th>RB1_phosphoproteomics_T826</th>\n",
       "      <th>RB1_phosphoproteomics_Y771</th>\n",
       "      <th>RB1_phosphoproteomics_Y805</th>\n",
       "      <th>RB1_phosphoproteomics_Y813</th>\n",
       "      <th>RB1_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>-0.10700</td>\n",
       "      <td>-0.10000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.267000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.6300</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.4870</td>\n",
       "      <td>-1.455000</td>\n",
       "      <td>-0.2670</td>\n",
       "      <td>0.71500</td>\n",
       "      <td>0.54500</td>\n",
       "      <td>0.54500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.315</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>0.07700</td>\n",
       "      <td>-0.09300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7470</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6095</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.51300</td>\n",
       "      <td>0.51300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.2330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>-0.29079</td>\n",
       "      <td>0.01821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.80221</td>\n",
       "      <td>0.61421</td>\n",
       "      <td>0.61421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-0.43850</td>\n",
       "      <td>0.42500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.849000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.23900</td>\n",
       "      <td>-0.23900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S006</td>\n",
       "      <td>-0.39950</td>\n",
       "      <td>-0.20370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8135</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.51100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.51100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.626</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S149</td>\n",
       "      <td>-0.63690</td>\n",
       "      <td>-0.08190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6349</td>\n",
       "      <td>-0.268900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1859</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0461</td>\n",
       "      <td>-0.484400</td>\n",
       "      <td>-0.3189</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.56290</td>\n",
       "      <td>-0.56290</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S150</td>\n",
       "      <td>0.30600</td>\n",
       "      <td>-0.44370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5360</td>\n",
       "      <td>0.031000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5970</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1940</td>\n",
       "      <td>0.102500</td>\n",
       "      <td>0.4670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.57200</td>\n",
       "      <td>0.57200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S151</td>\n",
       "      <td>-0.45500</td>\n",
       "      <td>0.43700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3500</td>\n",
       "      <td>0.315500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0315</td>\n",
       "      <td>...</td>\n",
       "      <td>0.9500</td>\n",
       "      <td>0.227245</td>\n",
       "      <td>0.4020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.14100</td>\n",
       "      <td>-0.14100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S152</td>\n",
       "      <td>-0.80200</td>\n",
       "      <td>0.07990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.094000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.3660</td>\n",
       "      <td>...</td>\n",
       "      <td>1.2740</td>\n",
       "      <td>0.056260</td>\n",
       "      <td>0.2750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.87600</td>\n",
       "      <td>-0.87600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.9160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S153</td>\n",
       "      <td>0.31650</td>\n",
       "      <td>-0.53300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1120</td>\n",
       "      <td>-0.140643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0810</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0670</td>\n",
       "      <td>-0.246350</td>\n",
       "      <td>0.0800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.61600</td>\n",
       "      <td>0.61600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       RB1_phosphoproteomics_S249  RB1_phosphoproteomics_S37  \\\n",
       "Sample_ID                                                          \n",
       "S001                         -0.10700                   -0.10000   \n",
       "S002                          0.07700                   -0.09300   \n",
       "S003                         -0.29079                    0.01821   \n",
       "S005                         -0.43850                    0.42500   \n",
       "S006                         -0.39950                   -0.20370   \n",
       "...                               ...                        ...   \n",
       "S149                         -0.63690                   -0.08190   \n",
       "S150                          0.30600                   -0.44370   \n",
       "S151                         -0.45500                    0.43700   \n",
       "S152                         -0.80200                    0.07990   \n",
       "S153                          0.31650                   -0.53300   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S608  RB1_phosphoproteomics_S612  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                         NaN   \n",
       "S150                              NaN                         NaN   \n",
       "S151                              NaN                         NaN   \n",
       "S152                              NaN                         NaN   \n",
       "S153                              NaN                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S624  RB1_phosphoproteomics_S773  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                            0.891                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                     -0.6349   \n",
       "S150                              NaN                      0.5360   \n",
       "S151                              NaN                      0.3500   \n",
       "S152                              NaN                     -0.0640   \n",
       "S153                              NaN                     -0.1120   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S780  RB1_phosphoproteomics_S788  \\\n",
       "Sample_ID                                                           \n",
       "S001                        -0.267000                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                        -0.849000                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                        -0.268900                         NaN   \n",
       "S150                         0.031000                         NaN   \n",
       "S151                         0.315500                         NaN   \n",
       "S152                        -0.094000                         NaN   \n",
       "S153                        -0.140643                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S794  RB1_phosphoproteomics_S795  ...  \\\n",
       "Sample_ID                                                          ...   \n",
       "S001                              NaN                     -1.6300  ...   \n",
       "S002                              NaN                      0.7470  ...   \n",
       "S003                              NaN                         NaN  ...   \n",
       "S005                              NaN                         NaN  ...   \n",
       "S006                              NaN                      0.8135  ...   \n",
       "...                               ...                         ...  ...   \n",
       "S149                              NaN                     -0.1859  ...   \n",
       "S150                              NaN                      0.5970  ...   \n",
       "S151                              NaN                     -0.0315  ...   \n",
       "S152                              NaN                     -1.3660  ...   \n",
       "S153                              NaN                      0.0810  ...   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T625  RB1_phosphoproteomics_T774  \\\n",
       "Sample_ID                                                           \n",
       "S001                          -0.4870                   -1.455000   \n",
       "S002                           0.5230                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                          -0.0150                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                           0.0461                   -0.484400   \n",
       "S150                           0.1940                    0.102500   \n",
       "S151                           0.9500                    0.227245   \n",
       "S152                           1.2740                    0.056260   \n",
       "S153                          -0.0670                   -0.246350   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T778  RB1_phosphoproteomics_T821  \\\n",
       "Sample_ID                                                           \n",
       "S001                          -0.2670                     0.71500   \n",
       "S002                          -0.6095                         NaN   \n",
       "S003                              NaN                     0.80221   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                     1.51100   \n",
       "...                               ...                         ...   \n",
       "S149                          -0.3189                         NaN   \n",
       "S150                           0.4670                         NaN   \n",
       "S151                           0.4020                         NaN   \n",
       "S152                           0.2750                         NaN   \n",
       "S153                           0.0800                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T823  RB1_phosphoproteomics_T826  \\\n",
       "Sample_ID                                                           \n",
       "S001                          0.54500                     0.54500   \n",
       "S002                          0.51300                     0.51300   \n",
       "S003                          0.61421                     0.61421   \n",
       "S005                         -0.23900                    -0.23900   \n",
       "S006                              NaN                     1.51100   \n",
       "...                               ...                         ...   \n",
       "S149                         -0.56290                    -0.56290   \n",
       "S150                          0.57200                     0.57200   \n",
       "S151                         -0.14100                    -0.14100   \n",
       "S152                         -0.87600                    -0.87600   \n",
       "S153                          0.61600                     0.61600   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y771  RB1_phosphoproteomics_Y805  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                      1.2330   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                     -1.0849   \n",
       "S150                              NaN                      1.1820   \n",
       "S151                              NaN                     -0.0750   \n",
       "S152                              NaN                     -0.9160   \n",
       "S153                              NaN                      0.3230   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y813  RB1_proteomics  \n",
       "Sample_ID                                              \n",
       "S001                            0.315             0.0  \n",
       "S002                              NaN             0.0  \n",
       "S003                              NaN             0.0  \n",
       "S005                              NaN             0.0  \n",
       "S006                            0.626             0.0  \n",
       "...                               ...             ...  \n",
       "S149                              NaN             0.0  \n",
       "S150                              NaN             0.0  \n",
       "S151                              NaN             0.0  \n",
       "S152                              NaN             0.0  \n",
       "S153                              NaN             0.0  \n",
       "\n",
       "[144 rows x 27 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Subtract RB1_phosphoproteomics from RB1 proteomic data \n",
    "\n",
    "column_Names = list(phosphoprot_proteomic.columns)\n",
    "column_Names\n",
    "\n",
    "for i in column_Names:\n",
    "    phosphoprot_proteomic[i] = phosphoprot_proteomic[i] - phosphoprot_proteomic[\"RB1_proteomics\"]\n",
    "    \n",
    "phosphoprot_proteomic\n",
    "\n",
    "    \n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>RB1_phosphoproteomics_S249</th>\n",
       "      <th>RB1_phosphoproteomics_S37</th>\n",
       "      <th>RB1_phosphoproteomics_S608</th>\n",
       "      <th>RB1_phosphoproteomics_S612</th>\n",
       "      <th>RB1_phosphoproteomics_S624</th>\n",
       "      <th>RB1_phosphoproteomics_S773</th>\n",
       "      <th>RB1_phosphoproteomics_S780</th>\n",
       "      <th>RB1_phosphoproteomics_S788</th>\n",
       "      <th>RB1_phosphoproteomics_S794</th>\n",
       "      <th>RB1_phosphoproteomics_S795</th>\n",
       "      <th>...</th>\n",
       "      <th>RB1_phosphoproteomics_T821</th>\n",
       "      <th>RB1_phosphoproteomics_T823</th>\n",
       "      <th>RB1_phosphoproteomics_T826</th>\n",
       "      <th>RB1_phosphoproteomics_Y771</th>\n",
       "      <th>RB1_phosphoproteomics_Y805</th>\n",
       "      <th>RB1_phosphoproteomics_Y813</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>-0.10700</td>\n",
       "      <td>-0.10000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.267000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.6300</td>\n",
       "      <td>...</td>\n",
       "      <td>0.71500</td>\n",
       "      <td>0.54500</td>\n",
       "      <td>0.54500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.315</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>0.07700</td>\n",
       "      <td>-0.09300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7470</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.51300</td>\n",
       "      <td>0.51300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.2330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>-0.29079</td>\n",
       "      <td>0.01821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.80221</td>\n",
       "      <td>0.61421</td>\n",
       "      <td>0.61421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-0.43850</td>\n",
       "      <td>0.42500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.849000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.23900</td>\n",
       "      <td>-0.23900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S006</td>\n",
       "      <td>-0.39950</td>\n",
       "      <td>-0.20370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.891</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8135</td>\n",
       "      <td>...</td>\n",
       "      <td>1.51100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.51100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.626</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S149</td>\n",
       "      <td>-0.63690</td>\n",
       "      <td>-0.08190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6349</td>\n",
       "      <td>-0.268900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1859</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.56290</td>\n",
       "      <td>-0.56290</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S150</td>\n",
       "      <td>0.30600</td>\n",
       "      <td>-0.44370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5360</td>\n",
       "      <td>0.031000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5970</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.57200</td>\n",
       "      <td>0.57200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S151</td>\n",
       "      <td>-0.45500</td>\n",
       "      <td>0.43700</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3500</td>\n",
       "      <td>0.315500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0315</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.14100</td>\n",
       "      <td>-0.14100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S152</td>\n",
       "      <td>-0.80200</td>\n",
       "      <td>0.07990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0640</td>\n",
       "      <td>-0.094000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.3660</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.87600</td>\n",
       "      <td>-0.87600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.9160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S153</td>\n",
       "      <td>0.31650</td>\n",
       "      <td>-0.53300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1120</td>\n",
       "      <td>-0.140643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0810</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.61600</td>\n",
       "      <td>0.61600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>144 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       RB1_phosphoproteomics_S249  RB1_phosphoproteomics_S37  \\\n",
       "Sample_ID                                                          \n",
       "S001                         -0.10700                   -0.10000   \n",
       "S002                          0.07700                   -0.09300   \n",
       "S003                         -0.29079                    0.01821   \n",
       "S005                         -0.43850                    0.42500   \n",
       "S006                         -0.39950                   -0.20370   \n",
       "...                               ...                        ...   \n",
       "S149                         -0.63690                   -0.08190   \n",
       "S150                          0.30600                   -0.44370   \n",
       "S151                         -0.45500                    0.43700   \n",
       "S152                         -0.80200                    0.07990   \n",
       "S153                          0.31650                   -0.53300   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S608  RB1_phosphoproteomics_S612  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                         NaN   \n",
       "S150                              NaN                         NaN   \n",
       "S151                              NaN                         NaN   \n",
       "S152                              NaN                         NaN   \n",
       "S153                              NaN                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S624  RB1_phosphoproteomics_S773  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                            0.891                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                     -0.6349   \n",
       "S150                              NaN                      0.5360   \n",
       "S151                              NaN                      0.3500   \n",
       "S152                              NaN                     -0.0640   \n",
       "S153                              NaN                     -0.1120   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S780  RB1_phosphoproteomics_S788  \\\n",
       "Sample_ID                                                           \n",
       "S001                        -0.267000                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                        -0.849000                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                        -0.268900                         NaN   \n",
       "S150                         0.031000                         NaN   \n",
       "S151                         0.315500                         NaN   \n",
       "S152                        -0.094000                         NaN   \n",
       "S153                        -0.140643                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S794  RB1_phosphoproteomics_S795  ...  \\\n",
       "Sample_ID                                                          ...   \n",
       "S001                              NaN                     -1.6300  ...   \n",
       "S002                              NaN                      0.7470  ...   \n",
       "S003                              NaN                         NaN  ...   \n",
       "S005                              NaN                         NaN  ...   \n",
       "S006                              NaN                      0.8135  ...   \n",
       "...                               ...                         ...  ...   \n",
       "S149                              NaN                     -0.1859  ...   \n",
       "S150                              NaN                      0.5970  ...   \n",
       "S151                              NaN                     -0.0315  ...   \n",
       "S152                              NaN                     -1.3660  ...   \n",
       "S153                              NaN                      0.0810  ...   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T821  RB1_phosphoproteomics_T823  \\\n",
       "Sample_ID                                                           \n",
       "S001                          0.71500                     0.54500   \n",
       "S002                              NaN                     0.51300   \n",
       "S003                          0.80221                     0.61421   \n",
       "S005                              NaN                    -0.23900   \n",
       "S006                          1.51100                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                              NaN                    -0.56290   \n",
       "S150                              NaN                     0.57200   \n",
       "S151                              NaN                    -0.14100   \n",
       "S152                              NaN                    -0.87600   \n",
       "S153                              NaN                     0.61600   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T826  RB1_phosphoproteomics_Y771  \\\n",
       "Sample_ID                                                           \n",
       "S001                          0.54500                         NaN   \n",
       "S002                          0.51300                         NaN   \n",
       "S003                          0.61421                         NaN   \n",
       "S005                         -0.23900                         NaN   \n",
       "S006                          1.51100                         NaN   \n",
       "...                               ...                         ...   \n",
       "S149                         -0.56290                         NaN   \n",
       "S150                          0.57200                         NaN   \n",
       "S151                         -0.14100                         NaN   \n",
       "S152                         -0.87600                         NaN   \n",
       "S153                          0.61600                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y805  RB1_phosphoproteomics_Y813  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                       0.315   \n",
       "S002                           1.2330                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                       0.626   \n",
       "...                               ...                         ...   \n",
       "S149                          -1.0849                         NaN   \n",
       "S150                           1.1820                         NaN   \n",
       "S151                          -0.0750                         NaN   \n",
       "S152                          -0.9160                         NaN   \n",
       "S153                           0.3230                         NaN   \n",
       "\n",
       "Name            RB1_Mutation   RB1_Location  RB1_Mutation_Status  \\\n",
       "Sample_ID                                                          \n",
       "S001        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S002        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S003        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S005        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "S006        [Wildtype_Tumor]  [No_mutation]       Wildtype_Tumor   \n",
       "...                      ...            ...                  ...   \n",
       "S149       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S150       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S151       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S152       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "S153       [Wildtype_Normal]  [No_mutation]      Wildtype_Normal   \n",
       "\n",
       "Name       Sample_Status  \n",
       "Sample_ID                 \n",
       "S001               Tumor  \n",
       "S002               Tumor  \n",
       "S003               Tumor  \n",
       "S005               Tumor  \n",
       "S006               Tumor  \n",
       "...                  ...  \n",
       "S149              Normal  \n",
       "S150              Normal  \n",
       "S151              Normal  \n",
       "S152              Normal  \n",
       "S153              Normal  \n",
       "\n",
       "[144 rows x 30 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Add back in RB1 mutation, location, status, and sample status columns\n",
    "phosphoprot_proteomic = phosphoprot_proteomic.merge(proteomic_and_mut, on='Sample_ID')\n",
    "phosphoprot_proteomic = phosphoprot_proteomic.drop([\"RB1_proteomics_x\"], axis = 1)\n",
    "phosphoprot_proteomic = phosphoprot_proteomic.drop([\"RB1_proteomics_y\"], axis = 1)\n",
    "phosphoprot_proteomic\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>RB1_phosphoproteomics_S249</th>\n",
       "      <th>RB1_phosphoproteomics_S37</th>\n",
       "      <th>RB1_phosphoproteomics_S608</th>\n",
       "      <th>RB1_phosphoproteomics_S612</th>\n",
       "      <th>RB1_phosphoproteomics_S624</th>\n",
       "      <th>RB1_phosphoproteomics_S773</th>\n",
       "      <th>RB1_phosphoproteomics_S780</th>\n",
       "      <th>RB1_phosphoproteomics_S788</th>\n",
       "      <th>RB1_phosphoproteomics_S794</th>\n",
       "      <th>RB1_phosphoproteomics_S795</th>\n",
       "      <th>...</th>\n",
       "      <th>RB1_phosphoproteomics_T625</th>\n",
       "      <th>RB1_phosphoproteomics_T774</th>\n",
       "      <th>RB1_phosphoproteomics_T778</th>\n",
       "      <th>RB1_phosphoproteomics_T821</th>\n",
       "      <th>RB1_phosphoproteomics_T823</th>\n",
       "      <th>RB1_phosphoproteomics_T826</th>\n",
       "      <th>RB1_phosphoproteomics_Y771</th>\n",
       "      <th>RB1_phosphoproteomics_Y805</th>\n",
       "      <th>RB1_phosphoproteomics_Y813</th>\n",
       "      <th>binary_mutations</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>-0.10700</td>\n",
       "      <td>-0.10000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2670</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.63000</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.4870</td>\n",
       "      <td>-1.455</td>\n",
       "      <td>-0.2670</td>\n",
       "      <td>0.71500</td>\n",
       "      <td>0.54500</td>\n",
       "      <td>0.54500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3150</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>0.07700</td>\n",
       "      <td>-0.09300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.74700</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5230</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.6095</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.51300</td>\n",
       "      <td>0.51300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>-0.29079</td>\n",
       "      <td>0.01821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.80221</td>\n",
       "      <td>0.61421</td>\n",
       "      <td>0.61421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-0.43850</td>\n",
       "      <td>0.42500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.8490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.23900</td>\n",
       "      <td>-0.23900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S006</td>\n",
       "      <td>-0.39950</td>\n",
       "      <td>-0.20370</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8910</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.81350</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.51100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.51100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6260</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S099</td>\n",
       "      <td>0.25555</td>\n",
       "      <td>0.96900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1992</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.69800</td>\n",
       "      <td>0.69800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S100</td>\n",
       "      <td>0.00450</td>\n",
       "      <td>0.33200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.27245</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.64400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.64400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.1489</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S101</td>\n",
       "      <td>0.48300</td>\n",
       "      <td>0.22900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.07300</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.63800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.63800</td>\n",
       "      <td>-0.395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S102</td>\n",
       "      <td>-1.48400</td>\n",
       "      <td>-0.42600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.10400</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.2800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.42800</td>\n",
       "      <td>-0.42800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S103</td>\n",
       "      <td>0.23320</td>\n",
       "      <td>0.43270</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3727</td>\n",
       "      <td>0.3727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.1243</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.03790</td>\n",
       "      <td>0.10610</td>\n",
       "      <td>0.10610</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows Ã— 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       RB1_phosphoproteomics_S249  RB1_phosphoproteomics_S37  \\\n",
       "Sample_ID                                                          \n",
       "S001                         -0.10700                   -0.10000   \n",
       "S002                          0.07700                   -0.09300   \n",
       "S003                         -0.29079                    0.01821   \n",
       "S005                         -0.43850                    0.42500   \n",
       "S006                         -0.39950                   -0.20370   \n",
       "...                               ...                        ...   \n",
       "S099                          0.25555                    0.96900   \n",
       "S100                          0.00450                    0.33200   \n",
       "S101                          0.48300                    0.22900   \n",
       "S102                         -1.48400                   -0.42600   \n",
       "S103                          0.23320                    0.43270   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S608  RB1_phosphoproteomics_S612  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S099                              NaN                         NaN   \n",
       "S100                              NaN                         NaN   \n",
       "S101                              NaN                         NaN   \n",
       "S102                              NaN                         NaN   \n",
       "S103                              NaN                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S624  RB1_phosphoproteomics_S773  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                           0.8910                         NaN   \n",
       "...                               ...                         ...   \n",
       "S099                              NaN                         NaN   \n",
       "S100                           0.3530                         NaN   \n",
       "S101                          -0.3065                         NaN   \n",
       "S102                              NaN                         NaN   \n",
       "S103                              NaN                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S780  RB1_phosphoproteomics_S788  \\\n",
       "Sample_ID                                                           \n",
       "S001                          -0.2670                         NaN   \n",
       "S002                              NaN                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                          -0.8490                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S099                          -0.1992                         NaN   \n",
       "S100                              NaN                         NaN   \n",
       "S101                              NaN                         NaN   \n",
       "S102                              NaN                         NaN   \n",
       "S103                           0.3727                      0.3727   \n",
       "\n",
       "Name       RB1_phosphoproteomics_S794  RB1_phosphoproteomics_S795  ...  \\\n",
       "Sample_ID                                                          ...   \n",
       "S001                              NaN                    -1.63000  ...   \n",
       "S002                              NaN                     0.74700  ...   \n",
       "S003                              NaN                         NaN  ...   \n",
       "S005                              NaN                         NaN  ...   \n",
       "S006                              NaN                     0.81350  ...   \n",
       "...                               ...                         ...  ...   \n",
       "S099                              NaN                         NaN  ...   \n",
       "S100                              NaN                     0.27245  ...   \n",
       "S101                              NaN                    -0.07300  ...   \n",
       "S102                              NaN                    -0.10400  ...   \n",
       "S103                              NaN                         NaN  ...   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T625  RB1_phosphoproteomics_T774  \\\n",
       "Sample_ID                                                           \n",
       "S001                          -0.4870                      -1.455   \n",
       "S002                           0.5230                         NaN   \n",
       "S003                              NaN                         NaN   \n",
       "S005                          -0.0150                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S099                           0.1180                         NaN   \n",
       "S100                              NaN                         NaN   \n",
       "S101                              NaN                         NaN   \n",
       "S102                           0.4120                         NaN   \n",
       "S103                          -0.1243                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T778  RB1_phosphoproteomics_T821  \\\n",
       "Sample_ID                                                           \n",
       "S001                          -0.2670                     0.71500   \n",
       "S002                          -0.6095                         NaN   \n",
       "S003                              NaN                     0.80221   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                     1.51100   \n",
       "...                               ...                         ...   \n",
       "S099                              NaN                         NaN   \n",
       "S100                              NaN                     0.64400   \n",
       "S101                              NaN                    -0.63800   \n",
       "S102                          -0.2800                         NaN   \n",
       "S103                              NaN                    -0.03790   \n",
       "\n",
       "Name       RB1_phosphoproteomics_T823  RB1_phosphoproteomics_T826  \\\n",
       "Sample_ID                                                           \n",
       "S001                          0.54500                     0.54500   \n",
       "S002                          0.51300                     0.51300   \n",
       "S003                          0.61421                     0.61421   \n",
       "S005                         -0.23900                    -0.23900   \n",
       "S006                              NaN                     1.51100   \n",
       "...                               ...                         ...   \n",
       "S099                          0.69800                     0.69800   \n",
       "S100                              NaN                     0.64400   \n",
       "S101                              NaN                    -0.63800   \n",
       "S102                         -0.42800                    -0.42800   \n",
       "S103                          0.10610                     0.10610   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y771  RB1_phosphoproteomics_Y805  \\\n",
       "Sample_ID                                                           \n",
       "S001                              NaN                         NaN   \n",
       "S002                              NaN                       1.233   \n",
       "S003                              NaN                         NaN   \n",
       "S005                              NaN                         NaN   \n",
       "S006                              NaN                         NaN   \n",
       "...                               ...                         ...   \n",
       "S099                              NaN                         NaN   \n",
       "S100                              NaN                         NaN   \n",
       "S101                           -0.395                         NaN   \n",
       "S102                              NaN                      -0.452   \n",
       "S103                              NaN                         NaN   \n",
       "\n",
       "Name       RB1_phosphoproteomics_Y813  binary_mutations  \n",
       "Sample_ID                                                \n",
       "S001                           0.3150          Wildtype  \n",
       "S002                              NaN          Wildtype  \n",
       "S003                              NaN          Wildtype  \n",
       "S005                              NaN          Wildtype  \n",
       "S006                           0.6260          Wildtype  \n",
       "...                               ...               ...  \n",
       "S099                              NaN          Wildtype  \n",
       "S100                           0.1489          Wildtype  \n",
       "S101                              NaN          Wildtype  \n",
       "S102                              NaN          Wildtype  \n",
       "S103                              NaN          Wildtype  \n",
       "\n",
       "[95 rows x 27 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 2 - Create the binary column needed to do the comparison\n",
    "phosphoprot_proteomic['binary_mutations'] = phosphoprot_proteomic[gene+'_Mutation_Status'].apply(\n",
    "    lambda x: 'Wildtype' if x == 'Wildtype_Tumor' else 'Mutated')\n",
    "\n",
    "# Step 3 - Format\n",
    "phospho_tumors = phosphoprot_proteomic.query('Sample_Status != \"Normal\"') # drop Normal samples (warning with .loc)\n",
    "columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "binary_phospho = phospho_tumors.drop(columns_to_drop, axis = 1)\n",
    "only_phospho = binary_phospho.drop('binary_mutations', axis = 1)\n",
    "\n",
    "binary_phospho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "binary_phospho.to_csv(\"RB1_phospho_Endo\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use al.wrap_ttest for many columns of phospho data (takes phospho sites into account with bonferroni cutoff)\n",
    "phos_col_list = list(only_phospho.columns)\n",
    "\n",
    "# Step 4 - T-test comparing means of mutated vs wildtype effect on cis omics\n",
    "print(\"Doing t-test comparison for mutation status\")\n",
    "phos_pval_results = u.wrap_ttest(binary_phospho, 'binary_mutations', phos_col_list)\n",
    "#phos_pval_results = phos_pval_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "phos_pval_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create data frame with all significant phospho sites in order to graph\n",
    "phospho_list = list(phos_pval_results['Comparison']) # comment when no sig phospho sites\n",
    "#phospho_list = list(only_phospho.columns) # uncomment when no sig phospho sites\n",
    "phospho_df = pd.melt(binary_phospho, id_vars = 'binary_mutations', \n",
    "            value_vars = phospho_list, var_name = 'specific_phospho', value_name = 'phosphoproteomics')\n",
    "phospho_df.replace(to_replace = '_phosphoproteomics', value = '', \n",
    "                       inplace = True, regex = True) # shorten column names\n",
    "phospho_df.replace(to_replace = '.TNILQYASTRPPTLS\\WPIPHIPR', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "phospho_df.replace(to_replace = '.TAATAAAAAAEPPAPPPPPPPEEDPEQDS\\WGPEDLPLVR', value = '', \n",
    "                        inplace = True, regex = True)\n",
    "\n",
    "\n",
    "phospho_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.rcParams['figure.figsize']=(25,10) #size of plot\n",
    "sns.set(font_scale = 2)\n",
    "\n",
    "cis_boxplot = sns.boxplot(data = phospho_df, x = 'specific_phospho', y = 'phosphoproteomics', hue = 'binary_mutations',\n",
    "                             hue_order = [\"Wildtype\",\"Mutated\"], showfliers = False) \n",
    "cis_boxplot.set_title(gene + \" Effect on \" + gene + \" Normalized Phosphoproteomics in Endometrial  Tumors\")\n",
    "cis_boxplot = sns.stripplot(data = phospho_df, x = 'specific_phospho', y = 'phosphoproteomics', hue = 'binary_mutations',\n",
    "                               hue_order = [\"Wildtype\",\"Mutated\"], jitter = True, color = \".3\", dodge = True)\n",
    "cis_boxplot.set(xlabel = 'Phosphorylation Sites', ylabel = \"Phosphorylation Levels\")\n",
    "cis_boxplot.set_xticklabels(cis_boxplot.get_xticklabels(),rotation=80)\n",
    "\n",
    "# format legend\n",
    "handles, labels = cis_boxplot.get_legend_handles_labels()\n",
    "plt.legend(handles[0:2], labels[0:2])\n",
    "\n",
    "#Uncomment for when any phospho site comparison is significant\n",
    "#pval annotations for \n",
    "format_pval_annotation('*', x1 = -.2, x2 = .25, line_start = 2, line_height = .03)\n",
    "#pval annotations for \n",
    "format_pval_annotation('*', x1 = .8, x2 = 1.2, line_start = 2, line_height = .03)\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Acetylproteomics not included in the Gbm dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## mutation types on RB1\n",
    "somatic_df = brain.get_somatic_mutation()\n",
    "df1 = brain.join_omics_to_mutations(\n",
    "    omics_df_name=\"proteomics\",\n",
    "    mutations_genes=\"RB1\", \n",
    "    omics_genes=[\"RB1\"])\n",
    "df1.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
