{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combined Proteomic Comparison \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Library Imports"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run this cell to import the necessary libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_pval_annotation(pval_symbol, x1, x2, line_start = .05, line_height=.05):\n",
    "    # for manual adjustment to pval annotations\n",
    "    \n",
    "    y, h = line_start, line_height\n",
    "    plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "             [y, y+h, y+h, y], #vertical line\n",
    "             lw=1.5, color= '.3')\n",
    "    plt.text((x1+x2)*.5, # half between x coord\n",
    "             y+h, pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "\n",
    "# used in pval_annotation\n",
    "def get_pval_symbol(pval_df):\n",
    "    if pval_df.iloc[0:,2].bool() == True:\n",
    "        pval_symbol = '*'\n",
    "    else:\n",
    "        pval_symbol = 'ns'\n",
    "    return pval_symbol\n",
    "\n",
    "def pval_annotation(pval_df, plotted_df):\n",
    "    # pval symbol\n",
    "    num_pvals = len(pval_df)\n",
    "    omics_col = plotted_df.columns[0]\n",
    "    \n",
    "\n",
    "    # annotation for Mutated and Wiltype\n",
    "    if num_pvals == 1:\n",
    "        mut_pval = pval_df.loc[pval_df.index == \"Mutated/Wildtype\"]\n",
    "        mut_pval_symbol = get_pval_symbol(pval_df)\n",
    "        \n",
    "        x1, x2 = 0, 1   # columns (first column: 0, see plt.xticks())\n",
    "        y, h = plotted_df[omics_col].max() + .05, .05     \n",
    "        plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "                 [y, y+h, y+h, y], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x1+x2)*.5, # half between x coord\n",
    "                 y+h, mut_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "    \n",
    "    # annotation for Missense, Truncation and Wildtype\n",
    "    elif num_pvals == 2:\n",
    "        # Missense\n",
    "        miss_pval = pval_df.loc[pval_df.index == \"Missense/Wildtype\"]\n",
    "        miss_pval_symbol = get_pval_symbol(miss_pval)\n",
    "    \n",
    "        x1, x2 = 0, 1   # columns (first column: 0, see plt.xticks())\n",
    "        y, h = plotted_df[omics_col].max() + .05, .05     \n",
    "        plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "                 [y, y+h, y+h, y], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x1+x2)*.5, # half between x coord\n",
    "                 y+h, miss_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "        \n",
    "        # Truncation \n",
    "        trunc_pval = pval_df.loc[pval_df.index == \"Truncation/Wildtype\"]\n",
    "        trunc_pval_symbol = get_pval_symbol(trunc_pval)\n",
    "        \n",
    "        x3, x4 = 0, 2   # columns (first column: 0, see plt.xticks())\n",
    "        y2, h2 = plotted_df[omics_col].max() + .15, .05     \n",
    "        plt.plot([x3, x3, x4, x4], #draw horizontal line\n",
    "                 [y2, y2+h2, y2+h2, y2], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x3+x4)*.5, # half between x coord\n",
    "                 y2+h2, trunc_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Find the mutation frequency of the gene of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "brain = cptac.Gbm()\n",
    "desired_cutoff = 0.05\n",
    "gene = 'RB1'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Proteomics RB1 cis comparison"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Proteomics: Mutated and Wildtype rb1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1041"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interacting_list = ['RNASEL', 'RPS6', 'COPS7A', 'NFKBIA', 'MAPK1', 'LPIN1', 'HRAS', 'AMBRA1', 'IGFBP3', 'HDAC1', 'DPF3', 'CASP9', 'APMAP', 'SP1', 'FHIT', 'WNT7B', 'CEBPD', 'ATG5', 'HBEGF', 'ZFP36', 'COL4A5', 'MAG', 'TLK2', 'IRS1', 'GABARAPL1', 'ATG12', 'IL3', 'PARP1', 'DDB2', 'ANAPC13', 'OMG', 'UBE2D2', 'BRCA2', 'TGFBR2', 'FASLG', 'LAMC3', 'UPK3A', 'TYMS', 'APEX1', 'PRKCD', 'GNAS', 'SMARCD3', 'PSMD12', 'FZD2', 'ING2', 'RBBP7', 'RASSF5', 'ITGB1', 'NOS2', 'ICAM1', 'PRKCZ', 'IFT122', 'CHK1', 'BAD', 'CREB3L4', 'STAT3', 'ORC3L', 'DGKE', 'RIPK2', 'PRMT2', 'STAT1', 'PSMD7', 'AXIN2', 'KCNIP3', 'IFT81', 'PIK3C2G', 'WEE1', 'LAMA1', 'E2F2', 'IGF1R', 'ARHGDIA', 'RARB', 'NOX4', 'MAP3K7', 'CSPG4', 'UBE2H', 'COX6C', 'CDH23', 'SMC3', 'CCNB1', 'MNAT1', 'WDR60', 'CREBBP', 'GLA', 'SMARCD1', 'SOCS3', 'EGF', 'GRIN1', 'GATA3', 'ANXA1', 'BARD1', 'TSC2', 'FADD', 'NEFL', 'GDNF', 'SH3GLB1', 'ANAPC4', 'IRS2', 'GSK3B', 'IQCB1', 'ARID1B', 'PSMD13', 'PTEN', 'NCOA1', 'LRP5', 'COL4A1', 'MAP1LC3C', 'LEP', 'CDK2', 'RXRA', 'CDC25B', 'UCP1', 'ANAPC5', 'FGF20', 'POLA1', 'MKS1', 'KLF7', 'PIK3CA', 'DNMT1', 'CXCL2', 'CBS', 'PLD1', 'MAPK3', 'FAF1', 'SMC2', 'PIDD', 'BCAN', 'LIF', 'PKMYT1', 'PIK3CG', 'GABARAP', 'CDT1', 'FUZ', 'ADFP', 'WNT4', 'FGF4', 'HIF1A', 'IGFBP7', 'VTN', 'RPA3', 'CEP170', 'TCF7L2', 'IFNB1', 'ANAPC2', 'BBS7', 'GRB2', 'DOCK5', 'CKS1B', 'RHOB', 'CNOT10', 'IL1A', 'PDK1', 'TFAP2A', 'NINL', 'IFT74', 'GNAI2', 'RXRG', 'XPNPEP3', 'CDC25C', 'CCNA1', 'MCM10', 'GNAI3', 'COL4A3', 'PBEF1', 'MAD2L2', 'UQCC1', 'SHC2', 'VEGFA', 'ASIP', 'LEF1', 'NR3C1', 'TRAF1', 'MET', 'ALK', 'TLR4', 'PIK3C3', 'LILRB3', 'ATR', 'MCM2', 'WNT10B', 'MEF2D', 'FGF2', 'WNT11', 'BIRC8', 'LIPE', 'ZBTB17', 'SQSTM1', 'SNRPB2', 'PLCB1', 'MBNL1', 'CDK5', 'WNT6', 'SRC', 'MAPK9', 'FGF10', 'PRKCI', 'HSPB1', 'PMAIP1', 'ANAPC7', 'SOCS1', 'FRZB', 'ROCK2', 'TTC26', 'CYP26A1', 'LPIN2', 'EIF5B', 'PSMC4', 'SLC39A2', 'MEF2A', 'SMARCA2', 'SKIP', 'TGFB2', 'CHST11', 'ORC2', 'NDN', 'CNOT6L', 'YPEL5', 'CLUAP1', 'MAP1LC3B', 'MAX', 'BUB1B-PAK6', 'CXCR2', 'SKP1', 'CCND3', 'CREBL1', 'IFI16', 'E2F1', 'NCOA2', 'PCNA', 'ACSL3', 'SCRIB', 'SPRY2', 'TFAP2B', 'TCTEX1D2', 'RPS6KB1', 'CSNK2B', 'AQP4', 'IL6', 'ORC1', 'TFAP2D', 'IFNG', 'RBL1', 'DLL1', 'ADIPOQ', 'MYT1', 'ARMC8', 'DLL3', 'LPIN3', 'CDKN1A', 'PPARA', 'ARAF', 'RRM1', 'RABIF', 'MAPK13', 'GLI4', 'ACAN', 'CDC45L', 'MRE11A', 'MLL', 'ADPN', 'SEMA6A', 'CREB1', 'CHUK', 'IRS4', 'FGF21', 'FRAT2', 'PGR', 'APC2', 'NOTCH3', 'EED', 'KLK8', 'RB1', 'H3F3B', 'YAP1', 'BBC3', 'CCNA2', 'CDK6', 'WNT16', 'RPGR', 'CREG1', 'AGT', 'DCAF7', 'BRAF', 'NFKB1', 'PSMD8', 'CALM1', 'PCK2', 'NR4A1', 'LRP6', 'CASP10', 'AGPAT2', 'MAPRE2', 'ING1', 'LAMB3', 'MAP2K4', 'ARID1A', 'FGF8', 'IL2', 'DAPK2', 'COPS6', 'TNFSF13B', 'IL1B', 'MRE11', 'FGFR1', 'ABL1', 'TTC30B', 'PEBP1', 'PTPRZ1', 'BTG2', 'PTK2', 'FZD5', 'YWHAZ', 'NTN1', 'IFT46', 'RPS6KB2', 'IGF1', 'EZH2', 'RHOC', 'PTCH1', 'DVL2', 'WEE2', 'GADD45G', 'MCM8', 'IL1R1', 'GNAQ', 'CXCL1', 'ANLN', 'CBX5', 'SNAP29', 'NUDC', 'STAG2', 'COL1A1', 'GID4', 'PDGFRA', 'PRIM1', 'TTK', 'SMO', 'CD47', 'RRM2', 'RMND5B', 'NOTCH4', 'PIK3CB', 'TRAF4', 'RGMA', 'GH1', 'FKBP8', 'ITGA2B', 'DNPEP', 'TRAF6', 'CCDC40', 'RELA', 'MCM3', 'BUB3', 'MAP3K1', 'COL4A4', 'CRABP1', 'FGF16', 'SCD', 'FOXO1', 'NPAT', 'IQGAP1', 'E2F4', 'CDC25A', 'IFT88', 'NEK8', 'ORC2L', 'MTA1', 'WNT5A', 'PTGS2', 'POLE2', 'PRLR', 'CBX3', 'TYMP', 'TRAF3', 'ITGA2', 'YWHAG', 'RAC2', 'RAD52', 'MAPK8', 'DYNLT1', 'IFT140', 'SMARCC1', 'CREB3L3', 'CHEK1', 'ATG16L1', 'SFN', 'EXOSC4', 'EGR2', 'TFAP2C', 'CDK4', 'SMAD2', 'FGF3', 'E2F3', 'AKT1', 'CDR2', 'MFAP1', 'WNT5B', 'ULK1', 'MAP1LC3A', 'CREB3', 'PPARD', 'PPARGC1A', 'STMN1', 'PIK3R2', 'PAK3', 'HEY2', 'TSC1', 'FKBP1A', 'STOM', 'GABARAPL2', 'FZD3', 'SMARCB1', 'SESN1', 'ARG1', 'IL6R', 'DCK', 'MAP2K2', 'CEBPB', 'POLE', 'WHRN', 'NDUFA5', 'DVL3', 'CDKN2A', 'RAP1B', 'RAB21', 'ITGA6', 'C1QB', 'ROS', 'HEY1', 'KLF15', 'MMP12', 'FOXO1A', 'MEF2C', 'COPS3', 'MED1', 'GADD45A', 'E2F5', 'TSHR', 'TNF', 'SPTAN1', 'FGF1', 'GAB1', 'PIK3C2A', 'CHK2', 'BCL2L1', 'ACTG1', 'EFHC2', 'TRAF5', 'NBN', 'GNAI1', 'LMNB1', 'ATRIP', 'STAG1', 'HMGA1', 'LAMB4', 'SREBF1', 'EXOC3', 'TP53AIP1', 'FLT4', 'CSNK1A1', 'SLC39A4', 'MYL6', 'IFT52', 'PLK1', 'RPA1', 'LAMA2', 'IFT172', 'RALA', 'TCF7', 'DPF1', 'MSK1', 'ARHGDIB', 'KIF4A', 'TSHB', 'PTTG2', 'C5', 'CASP3', 'AKT3', 'WNT3', 'UVRAG', 'SUV39H1', 'HTRA2', 'ARHGEF6', 'RNGTT', 'DBF4A', 'GNA12', 'SLIT2', 'CFLAR', 'JAG1', 'RHOA', 'TFDP2', 'LSM4', 'CUL1', 'PIK3R1', 'PIK3C2B', 'BIRC7', 'COPS4', 'NFKB2', 'BCL2', 'MCM5', 'CCDC6', 'TTC8', 'PRIM2A', 'ACTA1', 'GADD45B', 'DHFR', 'SMARCD2', 'MTOR', 'CCNB3', 'KDM1A', 'TP53', 'ERBB3', 'GLB1', 'NCOR1', 'HSPB11', 'YWHAB', 'SUZ12', 'LPL', 'GATA4', 'DCAF11', 'RALGDS', 'PRKCQ', 'ACTB', 'FGFR3', 'TGFA', 'SSNA1', 'CDKN2D', 'SOX9', 'LAMB2', 'MKLN1', 'IL8', 'SMC4', 'MSH2', 'POM121', 'NR2F1', 'EPAS1', 'MYC', 'RMND5A', 'RPS6KA1', 'RBP1', 'LAMC1', 'BBS1', 'AAR2', 'BMP2', 'CXCL10', 'ZYG11B', 'TRAF3IP1', 'SHC3', 'NUP88', 'NFKBIB', 'DYNLRB2', 'CSNK2A2', 'PAK4', 'KIAA0652', 'EXOC6', 'PLCG2', 'COPS8', 'HES5', 'MBD3', 'CD2BP2', 'LAMP2', 'RHBDD2', 'FZD6', 'RBX1', 'INS', 'NUP133', 'TGFB1', 'CCNG2', 'PRKCG', 'RALB', 'ANAPC1', 'ESPL1', 'JAK1', 'AKT1S1', 'PSMC6', 'RAB14', 'EXOC7', 'IRF5', 'IL4', 'RAB3IL1', 'CSNK1A1L', 'CDK8', 'CDK7', 'TGFBR1', 'PAK2', 'MDM4', 'TIAM1', 'CASP6', 'ACTL6B', 'BBS4', 'LZTFL1', 'TNKS1BP1', 'CCND1', 'FGFR2', 'YWHAH', 'RASSF1', 'RAD50', 'CDKN2B', 'DUSP6', 'DYNC1LI1', 'GDF10', 'IKBKG', 'DYNLT3', 'FGF23', 'HES1', 'MAP2K7', 'CCNE2', 'MAP2K5', 'NME8', 'FRAT1', 'FZD7', 'CCND2', 'MYL12B', 'INHBA', 'CEBPA', 'XYLT1', 'RFC4', 'APC', 'RB1CC1', 'GLI1', 'EHBP1', 'RANBP10', 'WDR26', 'H2AFZ', 'TNFSF13', 'KLF6', 'TNFRSF10B', 'RALBP1', 'SPOCK', 'CCNH', 'ANKS3', 'FGF19', 'EBF', 'C10orf70', 'H3F3A', 'AHR', 'FOXC2', 'FZR1', 'YWHAE', 'RTN4', 'SOS1', 'TGFB3', 'EXOC1', 'FZD4', 'INTU', 'ADCY2', 'COL3A1', 'AIMP1', 'PLA2G6', 'FGF5', 'WNT7A', 'BECN1', 'AQP1', 'FGF18', 'WNT1', 'CDK1', 'ACTL6A', 'RFC5', 'CASP8', 'EGFR', 'RARA', 'PLA2G2A', 'DLL4', 'ESR2', 'JAK3', 'GLI2', 'DPF2', 'TTC30A', 'RAD9A', 'EFTUD2', 'YWHAQ', 'BUB1', 'MMP14', 'IGFBP5', 'GTF3A', 'JUN', 'APAF1', 'GJA1', 'PRB1', 'COL4A6', 'BMP3', 'MAP2K6', 'THBS1', 'PAK5', 'ATG7', 'NEK7', 'FOXO3', 'NOTCH2', 'H2AFX', 'FGF9', 'RAD1', 'CCR2', 'SELP', 'FGF6', 'SPATA7', 'CSNK2A1', 'SERPINE1', 'BIRC2', 'TWIST1', 'COPS7B', 'FGF22', 'SAP30', 'SPARC', 'LAMA5', 'MYL6B', 'FANCD2', 'AXIN1', 'IFT22', 'NGFR', 'TOP2A', 'CDC16', 'WNT3A', 'MSH6', 'WNT10A', 'PLAT', 'NR1H3', 'PLIN', 'GNG2', 'RRM2B', 'TLK1', 'RAB2A', 'WNK1', 'MIXL1', 'PRKCA', 'MMP1', 'ARL8B', 'LCA5', 'SOS2', 'GNB1', 'STAT5A', 'ID3', 'LAMP1', 'PPP3CA', 'GNAO1', 'ESR1', 'PTTG1', 'CSNK2A3', 'RAP1A', 'BDNF', 'CDC27', 'PIK3CD', 'IKBKB', 'CCNB2', 'DAPK1', 'SLIT3', 'HMGB1', 'PLK4', 'MAPK10', 'LAMA4', 'HUS1', 'FOS', 'FZD8', 'IL6ST', 'DF', 'EP300', 'GNA13', 'PDPK1', 'PTPRA', 'CDC6', 'AIF1', 'RIPK4', 'RANBP9', 'LIFR', 'SMAD4', 'TFAP2E', 'GDI1', 'STAT2', 'CDC23', 'EHD3', 'PLAU', 'MTA2', 'CCL2', 'CYP26B1', 'VPS4A', 'FANCG', 'ECHS1', 'RETN', 'RAC1', 'TMED1', 'LAMB1', 'BAK1', 'ATG10', 'IRF1', 'IQGAP2', 'MAEA', 'BBIP1', 'EXOC5', 'TCEB2', 'HMGB2', 'RSL1D1', 'PDE4D', 'RBBP4', 'EXOSC9', 'TNFSF11', 'CDKN2C', 'CBL', 'VAPB', 'GSN', 'NRIP1', 'MCM7', 'CDC7', 'GFAP', 'TCF7L1', 'PCK1', 'BRCA1', 'RAF1', 'LCN2', 'EPHA4', 'CCNG1', 'LTB', 'PLXNA2', 'MTA3', 'PAK1', 'DYNLRB1', 'MIF', 'EXOSC7', 'DVL1', 'SMARCE1', 'CXCL8', 'RAD21', 'LGALS3', 'CBX1', 'FGF17', 'ORC3', 'DAXX', 'CNTFR', 'EXOC6B', 'ITGAV', 'FN1', 'CRABP2', 'ATG14', 'DLK1', 'TIPRL', 'COL10A1', 'CDKN1C', 'DYNLL2', 'CTBP2', 'BSCL2', 'CKS2', 'MMP9', 'SLC39A1', 'NOS1', 'SLC2A4', 'MDM2', 'ERRFI1', 'CTNNB1', 'CDC42', 'PLA2G5', 'ORC5L', 'PTGIS', 'RBL2', 'STK4', 'ORC1L', 'BMP1', 'IFT20', 'BBS2', 'DFFB', 'ADCY3', 'FZD9', 'SKP2', 'ARFGAP3', 'NOTCH1', 'OSM', 'CTSA', 'P53', 'LRPPRC', 'RBM14', 'CDC20', 'SERPINB2', 'IFT57', 'BIRC3', 'TULP3', 'CNOT1', 'SMC1A', 'TBC1D4', 'COPS2', 'PLCG1', 'EXOC4', 'NRAS', 'BBS9', 'FCGR2A', 'PGRMC2', 'COPS5', 'VIM', 'MAP2K1', 'CREB3L1', 'SMAD3', 'CUGBP1', 'RAC3', 'CDKN1B', 'CDH1', 'CGA', 'EXOSC2', 'WDR34', 'FOXO4', 'RAP1GAP', 'ORC4', 'LMNA', 'EGR1', 'CCL3', 'NCOR2', 'CSRP2', 'MAP2K3', 'LAMC2', 'TACR1', 'FZD10', 'PDGFRB', 'NDUFA9', 'BAX', 'GLI3', 'PIK3R3', 'POLK', 'KLF5', 'WWTR1', 'COL2A1', 'IFT43', 'ITGA3', 'CXCL14', 'DDIT3', 'GATA2', 'CCNE1', 'FGF7', 'ERF', 'HLTF', 'PRKDC', 'SOX4', 'COPS1', 'CHEK2', 'NF1', 'HEYL', 'TCF1', 'ANKS6', 'ANAPC11', 'CETN3', 'PRKCH', 'BID', 'TSSC1', 'RTN4R', 'MAD1L1', 'CHD4', 'SHC1', 'DAPK3', 'COL4A2', 'JAG2', 'EXOC8', 'FAS', 'DFFA', 'DYNC1H1', 'ERBB2', 'EML4', 'JAK2', 'WNT2B', 'CHD3', 'BMI1', 'ZMPSTE24', 'MCM6', 'NCOA3', 'MCM4', 'CDC45', 'MCM9', 'SHC4', 'MAPKAPK3', 'LAMA3', 'SLIT1', 'ATG3', 'STAT5B', 'CDC2', 'ORC6', 'TRAF2', 'TRIB3', 'SFRP4', 'FZD1', 'RORA', 'KIT', 'SMARCA4', 'RAD17', 'IQGAP3', 'CASP7', 'IL24', 'DDX5', 'DYNLL1', 'ORC4L', 'FRAP1', 'CAMK2A', 'MAPKAPK2', 'BMP4', 'GAP43', 'GID8', 'PML', 'PAK6', 'WNT2', 'MBP', 'IFT27', 'CYCS', 'CD44', 'PPARG', 'TNFSF15', 'BBS5', 'TFDP1', 'RAB8A', 'USH1C', 'PDYN', 'KRAS', 'POLD3', 'AFG3L2', 'AKT2', 'RFC3', 'ORC5', 'PRKCB', 'RABEP2', 'IFT80', 'ANAPC10', 'POLA2', 'VCAN', 'LMNB2', 'IRF7', 'ORC6L', 'NEFM', 'RAD51', 'ATM', 'DYNC1I2', 'MEF2B', 'NCAN', 'RXRB', 'CEP97', 'EXOC2', 'CEP290', 'LTB4R', 'SMC1B', 'MAPK14', 'MBD2', 'ZMYND19', 'SIN3A', 'RPA2', 'HDAC2', 'MMP2', 'PAFAH1B1', 'SLC39A3', 'SMARCC2', 'STAT6', 'GBL']\n",
    "\n",
    "len(interacting_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: The following columns were not found in the proteomics dataframe, so they were inserted into joined table, but filled with NaN: ADFP, ADPN, ALK, ANKS3, ASIP, AXIN2, BIRC7, BIRC8, BMI1, BMP2, BMP3, BMP4, BRCA1, BRCA2, BTG2, BUB1, C10orf70, CALM1, CCDC40, CCL3, CCNA1, CCNB3, CCND3, CCNE1, CCNE2, CCNG1, CCNG2, CCR2, CDC2, CDC25A, CDC25B, CDC25C, CDC45, CDC45L, CDC6, CDC7, CDH23, CDT1, CGA, CHK1, CHK2, COL10A1, COL2A1, COL4A3, COL4A4, COL4A5, COL4A6, COPS1, CREB3, CREB3L1, CREB3L3, CREB3L4, CREBL1, CSNK1A1L, CSNK2A3, CUGBP1, CXCL2, CXCR2, CYP26A1, CYP26B1, DBF4A, DDIT3, DF, DLK1, DLL1, DLL4, DYNLRB2, E2F1, E2F2, EBF, EFHC2, EGF, EPAS1, ESPL1, ESR1, ESR2, FANCG, FASLG, FGF10, FGF16, FGF17, FGF18, FGF19, FGF20, FGF21, FGF22, FGF23, FGF3, FGF4, FGF5, FGF6, FGF7, FGF8, FGF9, FLT4, FOXC2, FOXO1A, FRAP1, FRAT1, FRAT2, FUZ, FZD10, FZD4, FZD5, FZD8, FZD9, GADD45A, GADD45B, GADD45G, GATA2, GATA3, GATA4, GBL, GDF10, GDNF, GH1, GLI2, GLI3, GLI4, H2AFZ, H3F3B, HES5, HEY2, HEYL, HIF1A, IFNB1, IFNG, IL1A, IL1R1, IL2, IL24, IL3, IL4, IL6, IL6R, IL8, INHBA, INS, INTU, IRF1, IRF7, IRS4, JAG2, KIAA0652, KLF15, KLF5, KLF6, KLF7, KLK8, LAMA3, LAMB3, LAMB4, LAMC2, LCA5, LEP, LIF, LRP5, LTB, MAP1LC3C, MCM10, MCM8, MCM9, MEF2B, MIXL1, MKS1, MLL, MRE11A, MSK1, MYC, NINL, NME8, NOTCH4, NOX4, NR1H3, NR4A1, NRIP1, ORC1L, ORC2L, ORC3L, ORC4L, ORC5L, ORC6L, OSM, P53, PAK6, PBEF1, PGR, PIDD, PIK3C2G, PIK3R3, PKMYT1, PLA2G5, PLIN, PLK4, POLE2, PPARA, PPARD, PPARG, PPARGC1A, PRB1, PRIM2A, PRLR, PTCH1, PTTG2, RAD51, RAD52, RARB, RIPK4, ROS, RXRG, SKIP, SKP2, SLC2A4, SLC39A2, SLC39A3, SLC39A4, SLIT3, SMC1B, SMO, SOCS1, SOCS3, SPATA7, SPOCK, SUV39H1, TACR1, TCEB2, TCF1, TCF7L1, TFAP2A, TFAP2B, TFAP2C, TFAP2D, TFAP2E, TGFA, TLR4, TNF, TNFSF11, TNFSF13, TNFSF15, TP53AIP1, TRIB3, TSHB, TSHR, TSSC1, TTC30B, TWIST1, UCP1, UPK3A, USH1C, WEE2, WHRN, WNT1, WNT10A, WNT10B, WNT11, WNT16, WNT2, WNT2B, WNT3, WNT3A, WNT4, WNT5B, WNT6, WNT7A, WNT7B, ZFP36, ZMYND19 (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n",
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the RB1 gene (C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py, line 3)\n"
     ]
    }
   ],
   "source": [
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest  \n",
    "prot_and_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = ['RB1'], omics_df_name = 'proteomics', omics_genes = interacting_list) # single col level \n",
    "\n",
    "# Step 2 - Create binary column \n",
    "prot_and_mutations['RB1_binary_mutations'] = np.where(\n",
    "            prot_and_mutations['RB1_Mutation_Status'] == 'Wildtype_Tumor', 'Wildtype', 'Mutated')\n",
    "\n",
    "# Step 3 - Format the dataframe correctly for the T-test(just omics and binary columns for tumors)\n",
    "tumors = prot_and_mutations[prot_and_mutations.Sample_Status == 'Tumor'] #drop Normal samples\n",
    "columns_to_drop = [\"RB1_Mutation\", \"RB1_Location\", \"RB1_Mutation_Status\", \"Sample_Status\"]\n",
    "RB1_mut_status = tumors.drop(columns_to_drop, axis = 1)\n",
    "RB1_mut_status = RB1_mut_status.dropna(axis=1,how='all')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>RNASEL_proteomics</th>\n",
       "      <th>RPS6_proteomics</th>\n",
       "      <th>COPS7A_proteomics</th>\n",
       "      <th>NFKBIA_proteomics</th>\n",
       "      <th>MAPK1_proteomics</th>\n",
       "      <th>LPIN1_proteomics</th>\n",
       "      <th>HRAS_proteomics</th>\n",
       "      <th>AMBRA1_proteomics</th>\n",
       "      <th>IGFBP3_proteomics</th>\n",
       "      <th>HDAC1_proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>MAPK14_proteomics</th>\n",
       "      <th>MBD2_proteomics</th>\n",
       "      <th>SIN3A_proteomics</th>\n",
       "      <th>RPA2_proteomics</th>\n",
       "      <th>HDAC2_proteomics</th>\n",
       "      <th>MMP2_proteomics</th>\n",
       "      <th>PAFAH1B1_proteomics</th>\n",
       "      <th>SMARCC2_proteomics</th>\n",
       "      <th>STAT6_proteomics</th>\n",
       "      <th>RB1_binary_mutations</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>-0.074682</td>\n",
       "      <td>-0.298868</td>\n",
       "      <td>-0.375332</td>\n",
       "      <td>0.430818</td>\n",
       "      <td>0.364629</td>\n",
       "      <td>0.593744</td>\n",
       "      <td>-0.005871</td>\n",
       "      <td>0.013390</td>\n",
       "      <td>1.187890</td>\n",
       "      <td>0.156024</td>\n",
       "      <td>...</td>\n",
       "      <td>0.059122</td>\n",
       "      <td>0.086119</td>\n",
       "      <td>0.036935</td>\n",
       "      <td>0.155935</td>\n",
       "      <td>0.119510</td>\n",
       "      <td>-0.154579</td>\n",
       "      <td>-0.111372</td>\n",
       "      <td>0.281121</td>\n",
       "      <td>-0.580490</td>\n",
       "      <td>Mutated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>0.354429</td>\n",
       "      <td>-0.149719</td>\n",
       "      <td>-0.214671</td>\n",
       "      <td>0.202222</td>\n",
       "      <td>0.419605</td>\n",
       "      <td>-0.900685</td>\n",
       "      <td>-0.242413</td>\n",
       "      <td>0.114986</td>\n",
       "      <td>1.089068</td>\n",
       "      <td>0.208120</td>\n",
       "      <td>...</td>\n",
       "      <td>0.248486</td>\n",
       "      <td>0.177578</td>\n",
       "      <td>0.153798</td>\n",
       "      <td>0.327944</td>\n",
       "      <td>-0.002000</td>\n",
       "      <td>0.236139</td>\n",
       "      <td>0.026597</td>\n",
       "      <td>-0.090254</td>\n",
       "      <td>-0.532108</td>\n",
       "      <td>Mutated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>-0.006295</td>\n",
       "      <td>0.458307</td>\n",
       "      <td>-0.092492</td>\n",
       "      <td>0.006309</td>\n",
       "      <td>-0.369258</td>\n",
       "      <td>0.197488</td>\n",
       "      <td>-0.100306</td>\n",
       "      <td>0.608246</td>\n",
       "      <td>0.288977</td>\n",
       "      <td>-0.493526</td>\n",
       "      <td>...</td>\n",
       "      <td>0.131765</td>\n",
       "      <td>0.217612</td>\n",
       "      <td>-0.117634</td>\n",
       "      <td>-0.178608</td>\n",
       "      <td>0.104724</td>\n",
       "      <td>-0.190389</td>\n",
       "      <td>-0.124933</td>\n",
       "      <td>-0.254752</td>\n",
       "      <td>0.283337</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>0.200944</td>\n",
       "      <td>-0.320724</td>\n",
       "      <td>0.406828</td>\n",
       "      <td>-0.514395</td>\n",
       "      <td>-0.090202</td>\n",
       "      <td>0.560842</td>\n",
       "      <td>0.450479</td>\n",
       "      <td>0.468160</td>\n",
       "      <td>-0.842326</td>\n",
       "      <td>-0.302168</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.158649</td>\n",
       "      <td>-0.003504</td>\n",
       "      <td>0.142193</td>\n",
       "      <td>0.332077</td>\n",
       "      <td>-0.007612</td>\n",
       "      <td>-0.537830</td>\n",
       "      <td>0.138830</td>\n",
       "      <td>-0.410945</td>\n",
       "      <td>-0.194730</td>\n",
       "      <td>Mutated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-0.026010</td>\n",
       "      <td>0.239727</td>\n",
       "      <td>-0.172056</td>\n",
       "      <td>-0.373507</td>\n",
       "      <td>-0.919902</td>\n",
       "      <td>-0.195571</td>\n",
       "      <td>0.135963</td>\n",
       "      <td>0.175601</td>\n",
       "      <td>-0.480078</td>\n",
       "      <td>0.120492</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.345273</td>\n",
       "      <td>-0.096198</td>\n",
       "      <td>-0.154285</td>\n",
       "      <td>-0.144604</td>\n",
       "      <td>0.097894</td>\n",
       "      <td>1.254198</td>\n",
       "      <td>-0.184890</td>\n",
       "      <td>0.447387</td>\n",
       "      <td>-0.038862</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S100</td>\n",
       "      <td>0.991301</td>\n",
       "      <td>-0.098582</td>\n",
       "      <td>0.125135</td>\n",
       "      <td>0.097418</td>\n",
       "      <td>0.382110</td>\n",
       "      <td>0.046771</td>\n",
       "      <td>-0.906531</td>\n",
       "      <td>0.086460</td>\n",
       "      <td>0.790289</td>\n",
       "      <td>1.068121</td>\n",
       "      <td>...</td>\n",
       "      <td>0.348674</td>\n",
       "      <td>0.602169</td>\n",
       "      <td>0.571138</td>\n",
       "      <td>0.463434</td>\n",
       "      <td>0.272694</td>\n",
       "      <td>1.095257</td>\n",
       "      <td>-0.208197</td>\n",
       "      <td>0.435165</td>\n",
       "      <td>0.363866</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S101</td>\n",
       "      <td>-0.236228</td>\n",
       "      <td>-0.133548</td>\n",
       "      <td>-0.343473</td>\n",
       "      <td>-0.280943</td>\n",
       "      <td>-0.337446</td>\n",
       "      <td>-0.406649</td>\n",
       "      <td>-0.686651</td>\n",
       "      <td>0.528189</td>\n",
       "      <td>-0.339737</td>\n",
       "      <td>0.544661</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.025603</td>\n",
       "      <td>0.171727</td>\n",
       "      <td>0.587882</td>\n",
       "      <td>0.468716</td>\n",
       "      <td>0.914054</td>\n",
       "      <td>0.119959</td>\n",
       "      <td>-0.269277</td>\n",
       "      <td>0.773308</td>\n",
       "      <td>-0.620584</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S102</td>\n",
       "      <td>-0.308096</td>\n",
       "      <td>0.432225</td>\n",
       "      <td>-0.010452</td>\n",
       "      <td>0.042622</td>\n",
       "      <td>0.088424</td>\n",
       "      <td>-0.071187</td>\n",
       "      <td>0.396887</td>\n",
       "      <td>-0.047656</td>\n",
       "      <td>0.576165</td>\n",
       "      <td>-0.384968</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.164853</td>\n",
       "      <td>-0.286700</td>\n",
       "      <td>-0.275250</td>\n",
       "      <td>-0.580135</td>\n",
       "      <td>-0.171912</td>\n",
       "      <td>0.768317</td>\n",
       "      <td>-0.006391</td>\n",
       "      <td>-0.471101</td>\n",
       "      <td>0.049946</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S104</td>\n",
       "      <td>0.088947</td>\n",
       "      <td>-0.378121</td>\n",
       "      <td>-0.102066</td>\n",
       "      <td>-0.170723</td>\n",
       "      <td>0.070822</td>\n",
       "      <td>-0.070323</td>\n",
       "      <td>0.576265</td>\n",
       "      <td>-0.074904</td>\n",
       "      <td>0.519953</td>\n",
       "      <td>-0.142008</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.412712</td>\n",
       "      <td>-0.094317</td>\n",
       "      <td>-0.204850</td>\n",
       "      <td>-0.389038</td>\n",
       "      <td>-0.296534</td>\n",
       "      <td>0.560643</td>\n",
       "      <td>-0.134905</td>\n",
       "      <td>-0.223818</td>\n",
       "      <td>0.317589</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S105</td>\n",
       "      <td>0.076472</td>\n",
       "      <td>-0.615069</td>\n",
       "      <td>-0.395202</td>\n",
       "      <td>-0.414660</td>\n",
       "      <td>-0.166356</td>\n",
       "      <td>-0.090786</td>\n",
       "      <td>-0.508041</td>\n",
       "      <td>-0.203143</td>\n",
       "      <td>-0.540350</td>\n",
       "      <td>0.631873</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.500179</td>\n",
       "      <td>0.595049</td>\n",
       "      <td>0.970071</td>\n",
       "      <td>0.486166</td>\n",
       "      <td>0.904629</td>\n",
       "      <td>0.072905</td>\n",
       "      <td>0.039987</td>\n",
       "      <td>0.706528</td>\n",
       "      <td>-0.197082</td>\n",
       "      <td>Wildtype</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 775 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       RNASEL_proteomics  RPS6_proteomics  COPS7A_proteomics  \\\n",
       "Sample_ID                                                          \n",
       "S001               -0.074682        -0.298868          -0.375332   \n",
       "S002                0.354429        -0.149719          -0.214671   \n",
       "S003               -0.006295         0.458307          -0.092492   \n",
       "S004                0.200944        -0.320724           0.406828   \n",
       "S005               -0.026010         0.239727          -0.172056   \n",
       "...                      ...              ...                ...   \n",
       "S100                0.991301        -0.098582           0.125135   \n",
       "S101               -0.236228        -0.133548          -0.343473   \n",
       "S102               -0.308096         0.432225          -0.010452   \n",
       "S104                0.088947        -0.378121          -0.102066   \n",
       "S105                0.076472        -0.615069          -0.395202   \n",
       "\n",
       "Name       NFKBIA_proteomics  MAPK1_proteomics  LPIN1_proteomics  \\\n",
       "Sample_ID                                                          \n",
       "S001                0.430818          0.364629          0.593744   \n",
       "S002                0.202222          0.419605         -0.900685   \n",
       "S003                0.006309         -0.369258          0.197488   \n",
       "S004               -0.514395         -0.090202          0.560842   \n",
       "S005               -0.373507         -0.919902         -0.195571   \n",
       "...                      ...               ...               ...   \n",
       "S100                0.097418          0.382110          0.046771   \n",
       "S101               -0.280943         -0.337446         -0.406649   \n",
       "S102                0.042622          0.088424         -0.071187   \n",
       "S104               -0.170723          0.070822         -0.070323   \n",
       "S105               -0.414660         -0.166356         -0.090786   \n",
       "\n",
       "Name       HRAS_proteomics  AMBRA1_proteomics  IGFBP3_proteomics  \\\n",
       "Sample_ID                                                          \n",
       "S001             -0.005871           0.013390           1.187890   \n",
       "S002             -0.242413           0.114986           1.089068   \n",
       "S003             -0.100306           0.608246           0.288977   \n",
       "S004              0.450479           0.468160          -0.842326   \n",
       "S005              0.135963           0.175601          -0.480078   \n",
       "...                    ...                ...                ...   \n",
       "S100             -0.906531           0.086460           0.790289   \n",
       "S101             -0.686651           0.528189          -0.339737   \n",
       "S102              0.396887          -0.047656           0.576165   \n",
       "S104              0.576265          -0.074904           0.519953   \n",
       "S105             -0.508041          -0.203143          -0.540350   \n",
       "\n",
       "Name       HDAC1_proteomics  ...  MAPK14_proteomics  MBD2_proteomics  \\\n",
       "Sample_ID                    ...                                       \n",
       "S001               0.156024  ...           0.059122         0.086119   \n",
       "S002               0.208120  ...           0.248486         0.177578   \n",
       "S003              -0.493526  ...           0.131765         0.217612   \n",
       "S004              -0.302168  ...          -0.158649        -0.003504   \n",
       "S005               0.120492  ...          -0.345273        -0.096198   \n",
       "...                     ...  ...                ...              ...   \n",
       "S100               1.068121  ...           0.348674         0.602169   \n",
       "S101               0.544661  ...          -0.025603         0.171727   \n",
       "S102              -0.384968  ...          -0.164853        -0.286700   \n",
       "S104              -0.142008  ...          -0.412712        -0.094317   \n",
       "S105               0.631873  ...          -0.500179         0.595049   \n",
       "\n",
       "Name       SIN3A_proteomics  RPA2_proteomics  HDAC2_proteomics  \\\n",
       "Sample_ID                                                        \n",
       "S001               0.036935         0.155935          0.119510   \n",
       "S002               0.153798         0.327944         -0.002000   \n",
       "S003              -0.117634        -0.178608          0.104724   \n",
       "S004               0.142193         0.332077         -0.007612   \n",
       "S005              -0.154285        -0.144604          0.097894   \n",
       "...                     ...              ...               ...   \n",
       "S100               0.571138         0.463434          0.272694   \n",
       "S101               0.587882         0.468716          0.914054   \n",
       "S102              -0.275250        -0.580135         -0.171912   \n",
       "S104              -0.204850        -0.389038         -0.296534   \n",
       "S105               0.970071         0.486166          0.904629   \n",
       "\n",
       "Name       MMP2_proteomics  PAFAH1B1_proteomics  SMARCC2_proteomics  \\\n",
       "Sample_ID                                                             \n",
       "S001             -0.154579            -0.111372            0.281121   \n",
       "S002              0.236139             0.026597           -0.090254   \n",
       "S003             -0.190389            -0.124933           -0.254752   \n",
       "S004             -0.537830             0.138830           -0.410945   \n",
       "S005              1.254198            -0.184890            0.447387   \n",
       "...                    ...                  ...                 ...   \n",
       "S100              1.095257            -0.208197            0.435165   \n",
       "S101              0.119959            -0.269277            0.773308   \n",
       "S102              0.768317            -0.006391           -0.471101   \n",
       "S104              0.560643            -0.134905           -0.223818   \n",
       "S105              0.072905             0.039987            0.706528   \n",
       "\n",
       "Name       STAT6_proteomics  RB1_binary_mutations  \n",
       "Sample_ID                                          \n",
       "S001              -0.580490               Mutated  \n",
       "S002              -0.532108               Mutated  \n",
       "S003               0.283337              Wildtype  \n",
       "S004              -0.194730               Mutated  \n",
       "S005              -0.038862              Wildtype  \n",
       "...                     ...                   ...  \n",
       "S100               0.363866              Wildtype  \n",
       "S101              -0.620584              Wildtype  \n",
       "S102               0.049946              Wildtype  \n",
       "S104               0.317589              Wildtype  \n",
       "S105              -0.197082              Wildtype  \n",
       "\n",
       "[99 rows x 775 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RB1_mut_status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparisons\n",
      "\n",
      "6 significant comparisons!\n"
     ]
    }
   ],
   "source": [
    "col_list = list(RB1_mut_status.columns)\n",
    "col_list.remove('RB1_binary_mutations')\n",
    "\n",
    "print(\"Doing t-test comparisons\\n\")\n",
    "\n",
    "# Call wrap_ttest, pass in formatted dataframe\n",
    "wrap_results = u.wrap_ttest(RB1_mut_status, 'RB1_binary_mutations', col_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>RB1_proteomics</td>\n",
       "      <td>1.139144e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>CDK2_proteomics</td>\n",
       "      <td>5.174663e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>PTK2_proteomics</td>\n",
       "      <td>7.166146e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>CDK6_proteomics</td>\n",
       "      <td>1.256368e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>HSPB11_proteomics</td>\n",
       "      <td>1.865781e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>NOTCH1_proteomics</td>\n",
       "      <td>1.960593e-05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Comparison       P_Value\n",
       "0     RB1_proteomics  1.139144e-08\n",
       "1    CDK2_proteomics  5.174663e-06\n",
       "2    PTK2_proteomics  7.166146e-06\n",
       "3    CDK6_proteomics  1.256368e-05\n",
       "4  HSPB11_proteomics  1.865781e-05\n",
       "5  NOTCH1_proteomics  1.960593e-05"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wrap_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Visualize "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\brittany henderson\\Anaconda3\\envs\\Research\\lib\\site-packages\\ipykernel_launcher.py:1: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation_Exclusivity</th>\n",
       "      <th>variable</th>\n",
       "      <th>omics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCND2_proteomics</td>\n",
       "      <td>0.0207138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCND2_proteomics</td>\n",
       "      <td>0.0415693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCND2_proteomics</td>\n",
       "      <td>0.42506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCND2_proteomics</td>\n",
       "      <td>0.0578722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCND2_proteomics</td>\n",
       "      <td>1.15014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>787</td>\n",
       "      <td>Neither</td>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>-0.343235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>788</td>\n",
       "      <td>Neither</td>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>789</td>\n",
       "      <td>Neither</td>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>790</td>\n",
       "      <td>Neither</td>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>-1.06291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>791</td>\n",
       "      <td>TP53_Only</td>\n",
       "      <td>TP53_proteomics</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>792 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Mutation_Exclusivity          variable      omics\n",
       "0                    NaN  CCND2_proteomics  0.0207138\n",
       "1                    NaN  CCND2_proteomics  0.0415693\n",
       "2                    NaN  CCND2_proteomics    0.42506\n",
       "3                    NaN  CCND2_proteomics  0.0578722\n",
       "4                    NaN  CCND2_proteomics    1.15014\n",
       "..                   ...               ...        ...\n",
       "787              Neither   TP53_proteomics  -0.343235\n",
       "788              Neither   TP53_proteomics        NaN\n",
       "789              Neither   TP53_proteomics        NaN\n",
       "790              Neither   TP53_proteomics   -1.06291\n",
       "791            TP53_Only   TP53_proteomics        NaN\n",
       "\n",
       "[792 rows x 3 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RB1_all = pd.concat([CCND2_mut_status, TP53_mut_status])\n",
    "#([RB1_mut_status, CDK2_mut_status, CDK6_mut_status, CDKN2A_mut_status, TFDP1_mut_status, MORF4L2_mut_status,TP53_mut_status])\n",
    "#RB1_all = pd.merge(RB1_all, TP53_prot_miss_trunc_wildtype, on = \"Sample_ID\")\n",
    "RB1_alldf = pd.melt(RB1_all, id_vars = ['Mutation_Exclusivity'], value_name = \"omics\")\n",
    "RB1_alldf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "merge TP53 mutation type to label graph "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Neither the `x` nor `y` variable appears to be numeric.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-15-cbb8700c7ec5>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mplt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrcParams\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'figure.figsize'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m14\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m8\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;31m#size of plot\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m boxplot = sns.boxplot(x='Mutation_Exclusivity', y=\"omics\", \n\u001b[1;32m----> 4\u001b[1;33m                         data = RB1_alldf, showfliers = False)    \n\u001b[0m\u001b[0;32m      5\u001b[0m \u001b[1;31m#boxplot.set_title(\"GBM RB1 Mutation Type effect on Proteomics Abundance\")\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      6\u001b[0m \u001b[1;31m#boxplot = sns.stripplot(x='variable', y=\"omics\", hue = 'Mutation_Exclusivity', data = RB1_alldf, jitter = True,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\seaborn\\categorical.py\u001b[0m in \u001b[0;36mboxplot\u001b[1;34m(x, y, hue, data, order, hue_order, orient, color, palette, saturation, width, dodge, fliersize, linewidth, whis, notch, ax, **kwargs)\u001b[0m\n\u001b[0;32m   2229\u001b[0m     plotter = _BoxPlotter(x, y, hue, data, order, hue_order,\n\u001b[0;32m   2230\u001b[0m                           \u001b[0morient\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcolor\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpalette\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msaturation\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2231\u001b[1;33m                           width, dodge, fliersize, linewidth)\n\u001b[0m\u001b[0;32m   2232\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2233\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0max\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\seaborn\\categorical.py\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, x, y, hue, data, order, hue_order, orient, color, palette, saturation, width, dodge, fliersize, linewidth)\u001b[0m\n\u001b[0;32m    444\u001b[0m                  width, dodge, fliersize, linewidth):\n\u001b[0;32m    445\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 446\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mestablish_variables\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morient\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morder\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhue_order\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    447\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mestablish_colors\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcolor\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpalette\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msaturation\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    448\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\seaborn\\categorical.py\u001b[0m in \u001b[0;36mestablish_variables\u001b[1;34m(self, x, y, hue, data, orient, order, hue_order, units)\u001b[0m\n\u001b[0;32m    156\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    157\u001b[0m             \u001b[1;31m# Figure out the plotting orientation\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 158\u001b[1;33m             \u001b[0morient\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0minfer_orient\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morient\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    159\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    160\u001b[0m             \u001b[1;31m# Option 2a:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Anaconda3\\envs\\Research\\lib\\site-packages\\seaborn\\categorical.py\u001b[0m in \u001b[0;36minfer_orient\u001b[1;34m(self, x, y, orient)\u001b[0m\n\u001b[0;32m    359\u001b[0m         \u001b[1;32melif\u001b[0m \u001b[0mis_not_numeric\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0my\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    360\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mis_not_numeric\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mx\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 361\u001b[1;33m                 \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mno_numeric\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    362\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    363\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[1;34m\"h\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: Neither the `x` nor `y` variable appears to be numeric."
     ]
    }
   ],
   "source": [
    "# Visualize\n",
    "plt.rcParams['figure.figsize']=(14,8) #size of plot\n",
    "boxplot = sns.boxplot(x='Mutation_Exclusivity', y=\"omics\", \n",
    "                        data = RB1_alldf, showfliers = False)    \n",
    "#boxplot.set_title(\"GBM RB1 Mutation Type effect on Proteomics Abundance\")\n",
    "#boxplot = sns.stripplot(x='variable', y=\"omics\", hue = 'Mutation_Exclusivity', data = RB1_alldf, jitter = True, \n",
    " #                          color = \".3\", hue_order = ['Neither', 'RB1_Only', 'TP53_Only' 'Both'], dodge = True)\n",
    "\n",
    "boxplot.set(xlabel = \"interacting genes\", ylabel = 'omics')\n",
    "\n",
    "#boxplot.set(xlabel = \"interacting genes\", ylabel = 'value')\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Visualize\n",
    "plt.rcParams['figure.figsize']=(14,8) #size of plot\n",
    "boxplot = sns.boxplot(x='Gene', y=\"value\", hue = 'binary_mutations', \n",
    "                         hue_order = ['Wildtype', 'Truncation'], data = RB1_alldf, showfliers = False)    \n",
    "boxplot.set_title(\"GBM RB1 Mutation Type effect on Proteomics Abundance\")\n",
    "boxplot = sns.stripplot(x='Gene', y=\"value\", hue = 'binary_mutations', data = RB1_alldf, jitter = True, \n",
    "                           color = \".3\", hue_order = ['Wildtype', 'Truncation'], dodge = True)\n",
    "boxplot.set(xlabel = \"interacting genes\", ylabel = 'value')\n",
    "\n",
    "plt.show()\n",
    "\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create Dataframe \n",
    "\n",
    "RB1_cis_mut = RB1_prot_miss_trunc_wildtype.assign(Gene= \"RB1\")\n",
    "CDK2_trans_mut = CDK2_prot_miss_trunc_wildtype.assign(Gene = \"CDK2\")\n",
    "CDK6_trans_mut = CDK6_prot_miss_trunc_wildtype.assign(Gene = \"CDK6\")\n",
    "CDKN2A_trans_mut = CDKN2A_prot_miss_trunc_wildtype.assign(Gene = \"CDKN2A\")\n",
    "TFDP1_trans_mut = TFDP1_prot_miss_trunc_wildtype.assign(Gene = \"TFDP1\")\n",
    "CCND2_trans_mut = CCND2_prot_miss_trunc_wildtype.assign(Gene = \"CCND2\")\n",
    "TP53_trans_mut = TP53_prot_miss_trunc_wildtype.assign(Gene = \"TP53\")\n",
    "\n",
    "RB1_all = pd.concat([RB1_cis_mut, CDK2_trans_mut, CDK6_trans_mut, CDKN2A_trans_mut,CCND2_trans_mut, TFDP1_trans_mut])\n",
    "RB1_alldf = pd.melt(RB1_all, id_vars = [\"Gene\", 'binary_mutations'])\n",
    "RB1_alldf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Visualize\n",
    "plt.rcParams['figure.figsize']=(14,8) #size of plot\n",
    "boxplot = sns.boxplot(x='Gene', y=\"value\", hue = 'binary_mutations', \n",
    "                         hue_order = ['Wildtype', 'Truncation'], data = RB1_alldf, showfliers = False)    \n",
    "boxplot.set_title(\"GBM RB1 Mutation Type effect on Proteomics Abundance\")\n",
    "boxplot = sns.stripplot(x='Gene', y=\"value\", hue = 'binary_mutations', data = RB1_alldf, jitter = True, \n",
    "                           color = \".3\", hue_order = ['Wildtype', 'Truncation'], dodge = True)\n",
    "boxplot.set(xlabel = \"interacting genes\", ylabel = 'value')\n",
    "\n",
    "plt.show()\n",
    "\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extra checks "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "looking for E2F1 in transcriptomics and proteomics "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut)\n",
    "num_freq_mut_genes = 20\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest - drop \n",
    "transrcpt_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'transcriptomics', omics_genes = \"E2F1\")\n",
    "reduced_transcript_mutations = brain.reduce_multiindex(transrcpt_mutations, levels_to_drop=1) #single col labels\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "transcript_mut_wt = format_mutated_wt_comparison(reduced_transcript_mutations)\n",
    "transcript_mut_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 4 - T-test\n",
    "trans_col_list = ['E2F1_transcriptomics']\n",
    "trans_col = 'E2F1_transcriptomics'\n",
    "\n",
    "print(\"Doing t-test comparison for mutation status\")\n",
    "transcript_results = wrap_ttest_return_all(transcript_mut_wt, 'binary_mutations', trans_col_list, num_freq_mut_genes)\n",
    "transcript_results = transcript_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "transcript_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest  \n",
    "prot_and_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'proteomics', omics_genes = \"E2F1\") # single col level\n",
    "\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "prot_mut_wt = format_mutated_wt_comparison(prot_and_mutations)\n",
    "prot_mut_wt.loc[prot_mut_wt [\"binary_mutations\"] == \"Mutated\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut) \n",
    "num_freq_mut_genes = 20\n",
    "prot_col_list = [\"E2F1\"+'_proteomics']\n",
    "prot_col = \"E2F1\"+\"_proteomics\"\n",
    "\n",
    "# Step 4 - T-test comparing means of mutated vs wildtype effect on cis omics\n",
    "print(\"Doing t-test comparison for mutated and wildtype\")\n",
    "prot_results = wrap_ttest_return_all(prot_mut_wt, 'binary_mutations', prot_col_list, num_freq_mut_genes)\n",
    "prot_results = prot_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "prot_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = brain.get_proteomics()\n",
    "E2_cols = [col for col in df.columns if \"E2\" in col]\n",
    "print (E2_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"E2F3\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "looking for CCNE1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut)\n",
    "num_freq_mut_genes = 20\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest - drop \n",
    "transrcpt_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'transcriptomics', omics_genes = \"CCNE1\")\n",
    "reduced_transcript_mutations = brain.reduce_multiindex(transrcpt_mutations, levels_to_drop=1) #single col labels\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "transcript_mut_wt = format_mutated_wt_comparison(reduced_transcript_mutations)\n",
    "transcript_mut_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 4 - T-test\n",
    "trans_col_list = ['CCNE1_transcriptomics']\n",
    "trans_col = 'CCNE1_transcriptomics'\n",
    "\n",
    "print(\"Doing t-test comparison for mutation status\")\n",
    "transcript_results = wrap_ttest_return_all(transcript_mut_wt, 'binary_mutations', trans_col_list, num_freq_mut_genes)\n",
    "transcript_results = transcript_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "transcript_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest  \n",
    "prot_and_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'proteomics', omics_genes = \"CCNE1\") # single col level\n",
    "\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "prot_mut_wt = format_mutated_wt_comparison(prot_and_mutations)\n",
    "prot_mut_wt.loc[prot_mut_wt [\"binary_mutations\"] == \"Mutated\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut) \n",
    "num_freq_mut_genes = 20\n",
    "prot_col_list = [\"CCNE1\"+'_proteomics']\n",
    "prot_col = \"CCNE1\"+\"_proteomics\"\n",
    "\n",
    "# Step 4 - T-test comparing means of mutated vs wildtype effect on cis omics\n",
    "print(\"Doing t-test comparison for mutated and wildtype\")\n",
    "prot_results = wrap_ttest_return_all(prot_mut_wt, 'binary_mutations', prot_col_list, num_freq_mut_genes)\n",
    "prot_results = prot_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "prot_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = brain.get_proteomics()\n",
    "CCN_cols = [col for col in df.columns if \"CCN\" in col]\n",
    "print (CCN_cols)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.columns.get_loc(\"CCND1\")\n",
    "df.iloc[0:4,1465:1480]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "could not find E2F1 or CCNE1 in proteomics, but were significant in transcriptomics. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut)\n",
    "num_freq_mut_genes = 20\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest - drop \n",
    "transrcpt_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'transcriptomics', omics_genes = \"CDK2\")\n",
    "reduced_transcript_mutations = brain.reduce_multiindex(transrcpt_mutations, levels_to_drop=1) #single col labels\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "transcript_mut_wt = format_mutated_wt_comparison(reduced_transcript_mutations)\n",
    "transcript_mut_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut) \n",
    "num_freq_mut_genes = 20\n",
    "prot_col_list = [\"CDK4\"+'_proteomics']\n",
    "prot_col = \"CDK4\"+\"_proteomics\"\n",
    "\n",
    "# Step 4 - T-test comparing means of mutated vs wildtype effect on cis omics\n",
    "print(\"Doing t-test comparison for mutated and wildtype\")\n",
    "prot_results = wrap_ttest_return_all(CDK2_prot_mut_wt, 'binary_mutations', prot_col_list, num_freq_mut_genes)\n",
    "prot_results = prot_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "prot_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BIRC5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut)\n",
    "num_freq_mut_genes = 20\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest - drop \n",
    "transrcpt_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'transcriptomics', omics_genes = \"BIRC5\")\n",
    "reduced_transcript_mutations = brain.reduce_multiindex(transrcpt_mutations, levels_to_drop=1) #single col labels\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "transcript_mut_wt = format_mutated_wt_comparison(reduced_transcript_mutations)\n",
    "transcript_mut_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Step 4 - T-test\n",
    "trans_col_list = ['BIRC5_transcriptomics']\n",
    "trans_col = 'BIRC5_transcriptomics'\n",
    "\n",
    "print(\"Doing t-test comparison for mutation status\")\n",
    "transcript_results = wrap_ttest_return_all(transcript_mut_wt, 'binary_mutations', trans_col_list, num_freq_mut_genes)\n",
    "transcript_results = transcript_results.rename(index = {0: \"Mutated/Wildtype\"})\n",
    "transcript_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest  \n",
    "prot_and_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'proteomics', omics_genes = \"BIRC5\") # single col level\n",
    "\n",
    "\n",
    "# Step 2 & 3 - Get binary column and format\n",
    "prot_mut_wt = format_mutated_wt_comparison(prot_and_mutations)\n",
    "prot_mut_wt.loc[prot_mut_wt [\"binary_mutations\"] == \"Mutated\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
