{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy.stats\n",
    "import re\n",
    "import sys \n",
    "#sys.path.append('C:\\\\Users\\\\brittany henderson\\\\GitHub\\\\GBM_for_CPTAC\\\\')\n",
    "#import cis_functions as f\n",
    "\n",
    "import cptac\n",
    "import cptac.utils as u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# note: pass reduced omics # CHECK would include non coding, but none in fm genes\n",
    "\n",
    "def format_mutated_wt_comparison(reduced_omics_df):\n",
    "    # Step 2 - Create binary column \n",
    "    reduced_omics_df['binary_mutations'] = np.where(\n",
    "                reduced_omics_df[gene+'_Mutation_Status'] == 'Wildtype_Tumor', 'Wildtype', 'Mutated')\n",
    "\n",
    "    # Step 3 - Format the dataframe correctly for the T-test(just omics and binary columns for tumors)\n",
    "    tumors = reduced_omics_df[reduced_omics_df.Sample_Status == 'Tumor'] #drop Normal samples\n",
    "    columns_to_drop = [gene+\"_Mutation\", gene+\"_Location\", gene+\"_Mutation_Status\", \"Sample_Status\"]\n",
    "    mut_status = tumors.drop(columns_to_drop, axis = 1)\n",
    "\n",
    "    return mut_status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_significance_col(results_df, num_comparisons):\n",
    "    \"bonferroni multiple hypothesis\"\"\"\n",
    "    alpha = .05\n",
    "    bonferroni_cutoff = alpha / num_comparisons\n",
    "    \n",
    "    pval = results_df['P_Value']\n",
    "    if float(pval[0]) <= bonferroni_cutoff:\n",
    "        results_df['Significant'] = True\n",
    "    else: \n",
    "        results_df['Significant'] = False\n",
    "    return results_df\n",
    "\n",
    "def wrap_ttest_return_all(df, label_column, comparison_columns, total_tests, alpha=.05):\n",
    "    try:\n",
    "        #Verify precondition that label column exists and has exactly 2 unique values\n",
    "        label_values = df[label_column].unique()\n",
    "        if len(label_values) != 2:\n",
    "            print(\"Incorrectly Formatted Dataframe! Label column must have exactly 2 unique values.\")\n",
    "            return None\n",
    "        \n",
    "        #Partition dataframe into two sets, one for each of the two unique values from the label column\n",
    "        partition1 = df.loc[df[label_column] == label_values[0]]\n",
    "        partition2 = df.loc[df[label_column] == label_values[1]]\n",
    "        \n",
    "        #Determine the number of real valued columns on which we will do t-tests\n",
    "        #sites = len(comparison_columns.columns)\n",
    "        number_of_comparisons = total_tests # ? phospho sites or num freq mut genes doing cis comp\n",
    "        \n",
    "        #Use a bonferroni correction to adjust for multiple testing by altering the p-value needed for acceptance\n",
    "        bonferroni_cutoff = alpha/number_of_comparisons\n",
    "        \n",
    "        #Store all comparisons with their p-values in a dictionary\n",
    "        all_comparisons = {}\n",
    "        \n",
    "        #Loop through each comparison column, perform the t-test, and determine whether it meets the significance cutoff'''\n",
    "        for column in comparison_columns:\n",
    "            stat, pval = scipy.stats.ttest_ind(partition1[column].dropna(axis=0), partition2[column].dropna(axis=0))\n",
    "            all_comparisons[column] = pval\n",
    "    \n",
    "        #Sort dictionary to list smallest p-values first\n",
    "        sorted_comparisons = sorted(all_comparisons.items(), key=lambda kv: kv[1])\n",
    "        #Format as a dataframe and return to caller\n",
    "        all_comparisons_df = pd.DataFrame.from_dict(sorted_comparisons)\n",
    "        all_comparisons_df.columns = ['Comparison', 'P_Value']\n",
    "        \n",
    "                                               \n",
    "        all_comparisons_sig_col = add_significance_col(all_comparisons_df, number_of_comparisons)\n",
    "        return all_comparisons_sig_col\n",
    "                                \n",
    "    except:\n",
    "        print(\"Incorrectly Formatted Dataframe!\")\n",
    "        return None\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_pval_annotation(pval_symbol, x1, x2, line_start = .05, line_height=.05):\n",
    "    # for manual adjustment to pval annotations\n",
    "    \n",
    "    y, h = line_start, line_height\n",
    "    plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "             [y, y+h, y+h, y], #vertical line\n",
    "             lw=1.5, color= '.3')\n",
    "    plt.text((x1+x2)*.5, # half between x coord\n",
    "             y+h, pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "\n",
    "# used in pval_annotation\n",
    "def get_pval_symbol(pval_df):\n",
    "    if pval_df.iloc[0:,2].bool() == True:\n",
    "        pval_symbol = '*'\n",
    "    else:\n",
    "        pval_symbol = 'ns'\n",
    "    return pval_symbol\n",
    "\n",
    "def pval_annotation(pval_df, plotted_df):\n",
    "    # pval symbol\n",
    "    num_pvals = len(pval_df)\n",
    "    omics_col = plotted_df.columns[0]\n",
    "    \n",
    "\n",
    "    # annotation for Mutated and Wiltype\n",
    "    if num_pvals == 1:\n",
    "        mut_pval = pval_df.loc[pval_df.index == \"Mutated/Wildtype\"]\n",
    "        mut_pval_symbol = get_pval_symbol(pval_df)\n",
    "        \n",
    "        x1, x2 = 0, 1   # columns (first column: 0, see plt.xticks())\n",
    "        y, h = plotted_df[omics_col].max() + .05, .05     \n",
    "        plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "                 [y, y+h, y+h, y], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x1+x2)*.5, # half between x coord\n",
    "                 y+h, mut_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "    \n",
    "    # annotation for Missense, Truncation and Wildtype\n",
    "    elif num_pvals == 2:\n",
    "        # Missense\n",
    "        miss_pval = pval_df.loc[pval_df.index == \"Missense/Wildtype\"]\n",
    "        miss_pval_symbol = get_pval_symbol(miss_pval)\n",
    "    \n",
    "        x1, x2 = 0, 1   # columns (first column: 0, see plt.xticks())\n",
    "        y, h = plotted_df[omics_col].max() + .05, .05     \n",
    "        plt.plot([x1, x1, x2, x2], #draw horizontal line\n",
    "                 [y, y+h, y+h, y], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x1+x2)*.5, # half between x coord\n",
    "                 y+h, miss_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n",
    "        \n",
    "        # Truncation \n",
    "        trunc_pval = pval_df.loc[pval_df.index == \"Truncation/Wildtype\"]\n",
    "        trunc_pval_symbol = get_pval_symbol(trunc_pval)\n",
    "        \n",
    "        x3, x4 = 0, 2   # columns (first column: 0, see plt.xticks())\n",
    "        y2, h2 = plotted_df[omics_col].max() + .15, .05     \n",
    "        plt.plot([x3, x3, x4, x4], #draw horizontal line\n",
    "                 [y2, y2+h2, y2+h2, y2], #vertical line\n",
    "                 lw=1.5, color= '.3')\n",
    "        plt.text((x3+x4)*.5, # half between x coord\n",
    "                 y2+h2, trunc_pval_symbol, horizontalalignment='center', verticalalignment='bottom', color = \"black\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cptac.download(dataset=\"GBM\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                    \r"
     ]
    }
   ],
   "source": [
    "brain = cptac.Gbm()\n",
    "gene = [\"RB1\",\"TP53\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 98 samples for the RB1 gene, 76 samples for the TP53 gene (/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>APAF1_transcriptomics</th>\n",
       "      <th>ARF1_transcriptomics</th>\n",
       "      <th>BID_transcriptomics</th>\n",
       "      <th>CASP3_transcriptomics</th>\n",
       "      <th>CASP7_transcriptomics</th>\n",
       "      <th>CASP9_transcriptomics</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>TP53_Mutation</th>\n",
       "      <th>TP53_Location</th>\n",
       "      <th>TP53_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>121584.924414</td>\n",
       "      <td>8.755904e+05</td>\n",
       "      <td>172824.575576</td>\n",
       "      <td>282155.799107</td>\n",
       "      <td>94757.856468</td>\n",
       "      <td>96998.516011</td>\n",
       "      <td>[Splice_Site]</td>\n",
       "      <td>[p.X654_splice]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G266R]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>127543.335077</td>\n",
       "      <td>1.143540e+06</td>\n",
       "      <td>129224.679935</td>\n",
       "      <td>338958.989116</td>\n",
       "      <td>152333.359473</td>\n",
       "      <td>74437.051565</td>\n",
       "      <td>[Frame_Shift_Del]</td>\n",
       "      <td>[p.F117Lfs*8]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.K132R]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>159717.892717</td>\n",
       "      <td>1.105102e+06</td>\n",
       "      <td>201906.077059</td>\n",
       "      <td>397178.747372</td>\n",
       "      <td>122607.182408</td>\n",
       "      <td>63949.687681</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>108777.078394</td>\n",
       "      <td>9.163683e+05</td>\n",
       "      <td>49391.078560</td>\n",
       "      <td>155861.917352</td>\n",
       "      <td>57094.934837</td>\n",
       "      <td>62775.668485</td>\n",
       "      <td>[Splice_Region, Splice_Site]</td>\n",
       "      <td>[p.D349=, p.X605_splice]</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>[Missense_Mutation, Missense_Mutation]</td>\n",
       "      <td>[p.R248Q, p.R158H]</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>132169.668704</td>\n",
       "      <td>1.052542e+06</td>\n",
       "      <td>81962.838430</td>\n",
       "      <td>272108.246951</td>\n",
       "      <td>84073.177294</td>\n",
       "      <td>57767.798647</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S111</td>\n",
       "      <td>76799.044768</td>\n",
       "      <td>1.139062e+06</td>\n",
       "      <td>86646.834455</td>\n",
       "      <td>108005.583119</td>\n",
       "      <td>204269.839972</td>\n",
       "      <td>93971.250012</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S112</td>\n",
       "      <td>62509.908726</td>\n",
       "      <td>1.169101e+06</td>\n",
       "      <td>86477.195668</td>\n",
       "      <td>123151.060839</td>\n",
       "      <td>76319.084848</td>\n",
       "      <td>75349.198211</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S113</td>\n",
       "      <td>71335.626885</td>\n",
       "      <td>1.113513e+06</td>\n",
       "      <td>76232.860484</td>\n",
       "      <td>105063.867772</td>\n",
       "      <td>45654.394073</td>\n",
       "      <td>72438.677038</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S114</td>\n",
       "      <td>55162.287761</td>\n",
       "      <td>1.795846e+06</td>\n",
       "      <td>104819.979966</td>\n",
       "      <td>113218.183627</td>\n",
       "      <td>27233.943751</td>\n",
       "      <td>35006.290734</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S115</td>\n",
       "      <td>56450.009051</td>\n",
       "      <td>1.663497e+06</td>\n",
       "      <td>108939.060438</td>\n",
       "      <td>115969.270287</td>\n",
       "      <td>29941.390886</td>\n",
       "      <td>47978.793249</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>108 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       APAF1_transcriptomics  ARF1_transcriptomics  BID_transcriptomics  \\\n",
       "Sample_ID                                                                     \n",
       "S001               121584.924414          8.755904e+05        172824.575576   \n",
       "S002               127543.335077          1.143540e+06        129224.679935   \n",
       "S003               159717.892717          1.105102e+06        201906.077059   \n",
       "S004               108777.078394          9.163683e+05         49391.078560   \n",
       "S005               132169.668704          1.052542e+06         81962.838430   \n",
       "...                          ...                   ...                  ...   \n",
       "S111                76799.044768          1.139062e+06         86646.834455   \n",
       "S112                62509.908726          1.169101e+06         86477.195668   \n",
       "S113                71335.626885          1.113513e+06         76232.860484   \n",
       "S114                55162.287761          1.795846e+06        104819.979966   \n",
       "S115                56450.009051          1.663497e+06        108939.060438   \n",
       "\n",
       "Name       CASP3_transcriptomics  CASP7_transcriptomics  \\\n",
       "Sample_ID                                                 \n",
       "S001               282155.799107           94757.856468   \n",
       "S002               338958.989116          152333.359473   \n",
       "S003               397178.747372          122607.182408   \n",
       "S004               155861.917352           57094.934837   \n",
       "S005               272108.246951           84073.177294   \n",
       "...                          ...                    ...   \n",
       "S111               108005.583119          204269.839972   \n",
       "S112               123151.060839           76319.084848   \n",
       "S113               105063.867772           45654.394073   \n",
       "S114               113218.183627           27233.943751   \n",
       "S115               115969.270287           29941.390886   \n",
       "\n",
       "Name       CASP9_transcriptomics                  RB1_Mutation  \\\n",
       "Sample_ID                                                        \n",
       "S001                96998.516011                 [Splice_Site]   \n",
       "S002                74437.051565             [Frame_Shift_Del]   \n",
       "S003                63949.687681              [Wildtype_Tumor]   \n",
       "S004                62775.668485  [Splice_Region, Splice_Site]   \n",
       "S005                57767.798647              [Wildtype_Tumor]   \n",
       "...                          ...                           ...   \n",
       "S111                93971.250012             [Wildtype_Normal]   \n",
       "S112                75349.198211             [Wildtype_Normal]   \n",
       "S113                72438.677038             [Wildtype_Normal]   \n",
       "S114                35006.290734             [Wildtype_Normal]   \n",
       "S115                47978.793249             [Wildtype_Normal]   \n",
       "\n",
       "Name                   RB1_Location RB1_Mutation_Status  \\\n",
       "Sample_ID                                                 \n",
       "S001                [p.X654_splice]     Single_mutation   \n",
       "S002                  [p.F117Lfs*8]     Single_mutation   \n",
       "S003                  [No_mutation]      Wildtype_Tumor   \n",
       "S004       [p.D349=, p.X605_splice]   Multiple_mutation   \n",
       "S005                  [No_mutation]      Wildtype_Tumor   \n",
       "...                             ...                 ...   \n",
       "S111                  [No_mutation]     Wildtype_Normal   \n",
       "S112                  [No_mutation]     Wildtype_Normal   \n",
       "S113                  [No_mutation]     Wildtype_Normal   \n",
       "S114                  [No_mutation]     Wildtype_Normal   \n",
       "S115                  [No_mutation]     Wildtype_Normal   \n",
       "\n",
       "Name                                TP53_Mutation       TP53_Location  \\\n",
       "Sample_ID                                                               \n",
       "S001                          [Missense_Mutation]           [p.G266R]   \n",
       "S002                          [Missense_Mutation]           [p.K132R]   \n",
       "S003                             [Wildtype_Tumor]       [No_mutation]   \n",
       "S004       [Missense_Mutation, Missense_Mutation]  [p.R248Q, p.R158H]   \n",
       "S005                             [Wildtype_Tumor]       [No_mutation]   \n",
       "...                                           ...                 ...   \n",
       "S111                            [Wildtype_Normal]       [No_mutation]   \n",
       "S112                            [Wildtype_Normal]       [No_mutation]   \n",
       "S113                            [Wildtype_Normal]       [No_mutation]   \n",
       "S114                            [Wildtype_Normal]       [No_mutation]   \n",
       "S115                            [Wildtype_Normal]       [No_mutation]   \n",
       "\n",
       "Name      TP53_Mutation_Status Sample_Status  \n",
       "Sample_ID                                     \n",
       "S001           Single_mutation         Tumor  \n",
       "S002           Single_mutation         Tumor  \n",
       "S003            Wildtype_Tumor         Tumor  \n",
       "S004         Multiple_mutation         Tumor  \n",
       "S005            Wildtype_Tumor         Tumor  \n",
       "...                        ...           ...  \n",
       "S111           Wildtype_Normal        Normal  \n",
       "S112           Wildtype_Normal        Normal  \n",
       "S113           Wildtype_Normal        Normal  \n",
       "S114           Wildtype_Normal        Normal  \n",
       "S115           Wildtype_Normal        Normal  \n",
       "\n",
       "[108 rows x 13 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#num_freq_mut_genes = len(freq_mut)\n",
    "num_freq_mut_genes = 20\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest - drop \n",
    "transrcpt_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'transcriptomics', omics_genes = [\"ARF1\",\"CASP3\",\"CASP7\",\"CASP7\",\"CASP9\",\"BID\",\"APAF1\"])\n",
    "reduced_transcript_mutations = brain.reduce_multiindex(transrcpt_mutations, levels_to_drop=1) #single col labels\n",
    "reduced_transcript_mutations\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>APAF1_transcriptomics</th>\n",
       "      <th>ARF1_transcriptomics</th>\n",
       "      <th>BID_transcriptomics</th>\n",
       "      <th>CASP3_transcriptomics</th>\n",
       "      <th>CASP7_transcriptomics</th>\n",
       "      <th>CASP9_transcriptomics</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>TP53_Mutation</th>\n",
       "      <th>TP53_Location</th>\n",
       "      <th>TP53_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>121584.924414</td>\n",
       "      <td>8.755904e+05</td>\n",
       "      <td>172824.575576</td>\n",
       "      <td>282155.799107</td>\n",
       "      <td>94757.856468</td>\n",
       "      <td>96998.516011</td>\n",
       "      <td>[Splice_Site]</td>\n",
       "      <td>[p.X654_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G266R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>127543.335077</td>\n",
       "      <td>1.143540e+06</td>\n",
       "      <td>129224.679935</td>\n",
       "      <td>338958.989116</td>\n",
       "      <td>152333.359473</td>\n",
       "      <td>74437.051565</td>\n",
       "      <td>[Frame_Shift_Del]</td>\n",
       "      <td>[p.F117Lfs*8]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.K132R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>159717.892717</td>\n",
       "      <td>1.105102e+06</td>\n",
       "      <td>201906.077059</td>\n",
       "      <td>397178.747372</td>\n",
       "      <td>122607.182408</td>\n",
       "      <td>63949.687681</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>108777.078394</td>\n",
       "      <td>9.163683e+05</td>\n",
       "      <td>49391.078560</td>\n",
       "      <td>155861.917352</td>\n",
       "      <td>57094.934837</td>\n",
       "      <td>62775.668485</td>\n",
       "      <td>[Splice_Region, Splice_Site]</td>\n",
       "      <td>[p.D349=, p.X605_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation, Missense_Mutation]</td>\n",
       "      <td>[p.R248Q, p.R158H]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>132169.668704</td>\n",
       "      <td>1.052542e+06</td>\n",
       "      <td>81962.838430</td>\n",
       "      <td>272108.246951</td>\n",
       "      <td>84073.177294</td>\n",
       "      <td>57767.798647</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S111</td>\n",
       "      <td>76799.044768</td>\n",
       "      <td>1.139062e+06</td>\n",
       "      <td>86646.834455</td>\n",
       "      <td>108005.583119</td>\n",
       "      <td>204269.839972</td>\n",
       "      <td>93971.250012</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S112</td>\n",
       "      <td>62509.908726</td>\n",
       "      <td>1.169101e+06</td>\n",
       "      <td>86477.195668</td>\n",
       "      <td>123151.060839</td>\n",
       "      <td>76319.084848</td>\n",
       "      <td>75349.198211</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S113</td>\n",
       "      <td>71335.626885</td>\n",
       "      <td>1.113513e+06</td>\n",
       "      <td>76232.860484</td>\n",
       "      <td>105063.867772</td>\n",
       "      <td>45654.394073</td>\n",
       "      <td>72438.677038</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S114</td>\n",
       "      <td>55162.287761</td>\n",
       "      <td>1.795846e+06</td>\n",
       "      <td>104819.979966</td>\n",
       "      <td>113218.183627</td>\n",
       "      <td>27233.943751</td>\n",
       "      <td>35006.290734</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S115</td>\n",
       "      <td>56450.009051</td>\n",
       "      <td>1.663497e+06</td>\n",
       "      <td>108939.060438</td>\n",
       "      <td>115969.270287</td>\n",
       "      <td>29941.390886</td>\n",
       "      <td>47978.793249</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>108 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       APAF1_transcriptomics  ARF1_transcriptomics  BID_transcriptomics  \\\n",
       "Sample_ID                                                                     \n",
       "S001               121584.924414          8.755904e+05        172824.575576   \n",
       "S002               127543.335077          1.143540e+06        129224.679935   \n",
       "S003               159717.892717          1.105102e+06        201906.077059   \n",
       "S004               108777.078394          9.163683e+05         49391.078560   \n",
       "S005               132169.668704          1.052542e+06         81962.838430   \n",
       "...                          ...                   ...                  ...   \n",
       "S111                76799.044768          1.139062e+06         86646.834455   \n",
       "S112                62509.908726          1.169101e+06         86477.195668   \n",
       "S113                71335.626885          1.113513e+06         76232.860484   \n",
       "S114                55162.287761          1.795846e+06        104819.979966   \n",
       "S115                56450.009051          1.663497e+06        108939.060438   \n",
       "\n",
       "Name       CASP3_transcriptomics  CASP7_transcriptomics  \\\n",
       "Sample_ID                                                 \n",
       "S001               282155.799107           94757.856468   \n",
       "S002               338958.989116          152333.359473   \n",
       "S003               397178.747372          122607.182408   \n",
       "S004               155861.917352           57094.934837   \n",
       "S005               272108.246951           84073.177294   \n",
       "...                          ...                    ...   \n",
       "S111               108005.583119          204269.839972   \n",
       "S112               123151.060839           76319.084848   \n",
       "S113               105063.867772           45654.394073   \n",
       "S114               113218.183627           27233.943751   \n",
       "S115               115969.270287           29941.390886   \n",
       "\n",
       "Name       CASP9_transcriptomics                  RB1_Mutation  \\\n",
       "Sample_ID                                                        \n",
       "S001                96998.516011                 [Splice_Site]   \n",
       "S002                74437.051565             [Frame_Shift_Del]   \n",
       "S003                63949.687681              [Wildtype_Tumor]   \n",
       "S004                62775.668485  [Splice_Region, Splice_Site]   \n",
       "S005                57767.798647              [Wildtype_Tumor]   \n",
       "...                          ...                           ...   \n",
       "S111                93971.250012             [Wildtype_Normal]   \n",
       "S112                75349.198211             [Wildtype_Normal]   \n",
       "S113                72438.677038             [Wildtype_Normal]   \n",
       "S114                35006.290734             [Wildtype_Normal]   \n",
       "S115                47978.793249             [Wildtype_Normal]   \n",
       "\n",
       "Name                   RB1_Location RB1_Mutation_Status  \\\n",
       "Sample_ID                                                 \n",
       "S001                [p.X654_splice]            Mutation   \n",
       "S002                  [p.F117Lfs*8]            Mutation   \n",
       "S003                  [No_mutation]            Wildtype   \n",
       "S004       [p.D349=, p.X605_splice]            Mutation   \n",
       "S005                  [No_mutation]            Wildtype   \n",
       "...                             ...                 ...   \n",
       "S111                  [No_mutation]     Wildtype_Normal   \n",
       "S112                  [No_mutation]     Wildtype_Normal   \n",
       "S113                  [No_mutation]     Wildtype_Normal   \n",
       "S114                  [No_mutation]     Wildtype_Normal   \n",
       "S115                  [No_mutation]     Wildtype_Normal   \n",
       "\n",
       "Name                                TP53_Mutation       TP53_Location  \\\n",
       "Sample_ID                                                               \n",
       "S001                          [Missense_Mutation]           [p.G266R]   \n",
       "S002                          [Missense_Mutation]           [p.K132R]   \n",
       "S003                             [Wildtype_Tumor]       [No_mutation]   \n",
       "S004       [Missense_Mutation, Missense_Mutation]  [p.R248Q, p.R158H]   \n",
       "S005                             [Wildtype_Tumor]       [No_mutation]   \n",
       "...                                           ...                 ...   \n",
       "S111                            [Wildtype_Normal]       [No_mutation]   \n",
       "S112                            [Wildtype_Normal]       [No_mutation]   \n",
       "S113                            [Wildtype_Normal]       [No_mutation]   \n",
       "S114                            [Wildtype_Normal]       [No_mutation]   \n",
       "S115                            [Wildtype_Normal]       [No_mutation]   \n",
       "\n",
       "Name      TP53_Mutation_Status Sample_Status  \n",
       "Sample_ID                                     \n",
       "S001                  Mutation         Tumor  \n",
       "S002                  Mutation         Tumor  \n",
       "S003                  Wildtype         Tumor  \n",
       "S004                  Mutation         Tumor  \n",
       "S005                  Wildtype         Tumor  \n",
       "...                        ...           ...  \n",
       "S111           Wildtype_Normal        Normal  \n",
       "S112           Wildtype_Normal        Normal  \n",
       "S113           Wildtype_Normal        Normal  \n",
       "S114           Wildtype_Normal        Normal  \n",
       "S115           Wildtype_Normal        Normal  \n",
       "\n",
       "[108 rows x 13 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Classify all mutations as 'mutation' and all wildtypes as 'wildtype'\n",
    "reduced_transcript_mutations = reduced_transcript_mutations.replace({'Wildtype_Tumor':'Wildtype','Single_mutation':'Mutation','Multiple_mutation':'Mutation'})\n",
    "reduced_transcript_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for ind, row in reduced_transcript_mutations.iterrows():\n",
    "    if row['RB1_Mutation_Status'] == 'Mutation' and row[\"TP53_Mutation_Status\"] == 'Mutation':\n",
    "        reduced_transcript_mutations.loc[ind, 'Mutation_Exclusivity'] = 'Both'\n",
    "        \n",
    "    elif row['RB1_Mutation_Status'] == 'Mutation' and row[\"TP53_Mutation_Status\"] != 'Mutation':\n",
    "        reduced_transcript_mutations.loc[ind, 'Mutation_Exclusivity'] = 'TP53_Only'\n",
    "        \n",
    "    elif row['RB1_Mutation_Status'] != 'Mutation' and row[\"TP53_Mutation_Status\"] == 'Mutation':\n",
    "        reduced_transcript_mutations.loc[ind, 'Mutation_Exclusivity'] = \"RB1_Only\"\n",
    "        \n",
    "    else:\n",
    "        reduced_transcript_mutations.loc[ind, 'Mutation_Exclusivity'] = 'Neither'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>APAF1_transcriptomics</th>\n",
       "      <th>ARF1_transcriptomics</th>\n",
       "      <th>BID_transcriptomics</th>\n",
       "      <th>CASP3_transcriptomics</th>\n",
       "      <th>CASP7_transcriptomics</th>\n",
       "      <th>CASP9_transcriptomics</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>TP53_Mutation</th>\n",
       "      <th>TP53_Location</th>\n",
       "      <th>TP53_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "      <th>Mutation_Exclusivity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>121584.924414</td>\n",
       "      <td>8.755904e+05</td>\n",
       "      <td>172824.575576</td>\n",
       "      <td>282155.799107</td>\n",
       "      <td>94757.856468</td>\n",
       "      <td>96998.516011</td>\n",
       "      <td>[Splice_Site]</td>\n",
       "      <td>[p.X654_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G266R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>127543.335077</td>\n",
       "      <td>1.143540e+06</td>\n",
       "      <td>129224.679935</td>\n",
       "      <td>338958.989116</td>\n",
       "      <td>152333.359473</td>\n",
       "      <td>74437.051565</td>\n",
       "      <td>[Frame_Shift_Del]</td>\n",
       "      <td>[p.F117Lfs*8]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.K132R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>159717.892717</td>\n",
       "      <td>1.105102e+06</td>\n",
       "      <td>201906.077059</td>\n",
       "      <td>397178.747372</td>\n",
       "      <td>122607.182408</td>\n",
       "      <td>63949.687681</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>108777.078394</td>\n",
       "      <td>9.163683e+05</td>\n",
       "      <td>49391.078560</td>\n",
       "      <td>155861.917352</td>\n",
       "      <td>57094.934837</td>\n",
       "      <td>62775.668485</td>\n",
       "      <td>[Splice_Region, Splice_Site]</td>\n",
       "      <td>[p.D349=, p.X605_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation, Missense_Mutation]</td>\n",
       "      <td>[p.R248Q, p.R158H]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>132169.668704</td>\n",
       "      <td>1.052542e+06</td>\n",
       "      <td>81962.838430</td>\n",
       "      <td>272108.246951</td>\n",
       "      <td>84073.177294</td>\n",
       "      <td>57767.798647</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S101</td>\n",
       "      <td>103953.919377</td>\n",
       "      <td>1.002822e+06</td>\n",
       "      <td>122409.939821</td>\n",
       "      <td>555823.390118</td>\n",
       "      <td>89316.707465</td>\n",
       "      <td>69803.234116</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S102</td>\n",
       "      <td>99337.423473</td>\n",
       "      <td>1.335898e+06</td>\n",
       "      <td>160571.493943</td>\n",
       "      <td>304616.038824</td>\n",
       "      <td>79108.511221</td>\n",
       "      <td>54400.821984</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S104</td>\n",
       "      <td>110694.913426</td>\n",
       "      <td>1.036239e+06</td>\n",
       "      <td>70921.123553</td>\n",
       "      <td>347881.513801</td>\n",
       "      <td>125266.990346</td>\n",
       "      <td>49014.662793</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S105</td>\n",
       "      <td>265295.344171</td>\n",
       "      <td>1.176823e+06</td>\n",
       "      <td>116049.984065</td>\n",
       "      <td>396746.858407</td>\n",
       "      <td>173928.472208</td>\n",
       "      <td>79499.929809</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[In_Frame_Ins, Missense_Mutation]</td>\n",
       "      <td>[p.A161dup, p.R156G]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>RB1_Only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S106</td>\n",
       "      <td>76944.148015</td>\n",
       "      <td>1.323550e+06</td>\n",
       "      <td>59538.975702</td>\n",
       "      <td>97508.467562</td>\n",
       "      <td>72145.442496</td>\n",
       "      <td>44101.369384</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       APAF1_transcriptomics  ARF1_transcriptomics  BID_transcriptomics  \\\n",
       "Sample_ID                                                                     \n",
       "S001               121584.924414          8.755904e+05        172824.575576   \n",
       "S002               127543.335077          1.143540e+06        129224.679935   \n",
       "S003               159717.892717          1.105102e+06        201906.077059   \n",
       "S004               108777.078394          9.163683e+05         49391.078560   \n",
       "S005               132169.668704          1.052542e+06         81962.838430   \n",
       "...                          ...                   ...                  ...   \n",
       "S101               103953.919377          1.002822e+06        122409.939821   \n",
       "S102                99337.423473          1.335898e+06        160571.493943   \n",
       "S104               110694.913426          1.036239e+06         70921.123553   \n",
       "S105               265295.344171          1.176823e+06        116049.984065   \n",
       "S106                76944.148015          1.323550e+06         59538.975702   \n",
       "\n",
       "Name       CASP3_transcriptomics  CASP7_transcriptomics  \\\n",
       "Sample_ID                                                 \n",
       "S001               282155.799107           94757.856468   \n",
       "S002               338958.989116          152333.359473   \n",
       "S003               397178.747372          122607.182408   \n",
       "S004               155861.917352           57094.934837   \n",
       "S005               272108.246951           84073.177294   \n",
       "...                          ...                    ...   \n",
       "S101               555823.390118           89316.707465   \n",
       "S102               304616.038824           79108.511221   \n",
       "S104               347881.513801          125266.990346   \n",
       "S105               396746.858407          173928.472208   \n",
       "S106                97508.467562           72145.442496   \n",
       "\n",
       "Name       CASP9_transcriptomics                  RB1_Mutation  \\\n",
       "Sample_ID                                                        \n",
       "S001                96998.516011                 [Splice_Site]   \n",
       "S002                74437.051565             [Frame_Shift_Del]   \n",
       "S003                63949.687681              [Wildtype_Tumor]   \n",
       "S004                62775.668485  [Splice_Region, Splice_Site]   \n",
       "S005                57767.798647              [Wildtype_Tumor]   \n",
       "...                          ...                           ...   \n",
       "S101                69803.234116              [Wildtype_Tumor]   \n",
       "S102                54400.821984              [Wildtype_Tumor]   \n",
       "S104                49014.662793              [Wildtype_Tumor]   \n",
       "S105                79499.929809              [Wildtype_Tumor]   \n",
       "S106                44101.369384             [Wildtype_Normal]   \n",
       "\n",
       "Name                   RB1_Location RB1_Mutation_Status  \\\n",
       "Sample_ID                                                 \n",
       "S001                [p.X654_splice]            Mutation   \n",
       "S002                  [p.F117Lfs*8]            Mutation   \n",
       "S003                  [No_mutation]            Wildtype   \n",
       "S004       [p.D349=, p.X605_splice]            Mutation   \n",
       "S005                  [No_mutation]            Wildtype   \n",
       "...                             ...                 ...   \n",
       "S101                  [No_mutation]            Wildtype   \n",
       "S102                  [No_mutation]            Wildtype   \n",
       "S104                  [No_mutation]            Wildtype   \n",
       "S105                  [No_mutation]            Wildtype   \n",
       "S106                  [No_mutation]     Wildtype_Normal   \n",
       "\n",
       "Name                                TP53_Mutation         TP53_Location  \\\n",
       "Sample_ID                                                                 \n",
       "S001                          [Missense_Mutation]             [p.G266R]   \n",
       "S002                          [Missense_Mutation]             [p.K132R]   \n",
       "S003                             [Wildtype_Tumor]         [No_mutation]   \n",
       "S004       [Missense_Mutation, Missense_Mutation]    [p.R248Q, p.R158H]   \n",
       "S005                             [Wildtype_Tumor]         [No_mutation]   \n",
       "...                                           ...                   ...   \n",
       "S101                             [Wildtype_Tumor]         [No_mutation]   \n",
       "S102                             [Wildtype_Tumor]         [No_mutation]   \n",
       "S104                             [Wildtype_Tumor]         [No_mutation]   \n",
       "S105            [In_Frame_Ins, Missense_Mutation]  [p.A161dup, p.R156G]   \n",
       "S106                            [Wildtype_Normal]         [No_mutation]   \n",
       "\n",
       "Name      TP53_Mutation_Status Sample_Status Mutation_Exclusivity  \n",
       "Sample_ID                                                          \n",
       "S001                  Mutation         Tumor                 Both  \n",
       "S002                  Mutation         Tumor                 Both  \n",
       "S003                  Wildtype         Tumor              Neither  \n",
       "S004                  Mutation         Tumor                 Both  \n",
       "S005                  Wildtype         Tumor              Neither  \n",
       "...                        ...           ...                  ...  \n",
       "S101                  Wildtype         Tumor              Neither  \n",
       "S102                  Wildtype         Tumor              Neither  \n",
       "S104                  Wildtype         Tumor              Neither  \n",
       "S105                  Mutation         Tumor             RB1_Only  \n",
       "S106           Wildtype_Normal        Normal              Neither  \n",
       "\n",
       "[100 rows x 14 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduced_transcript_mutations.head(100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#num_freq_mut_genes = len(freq_mut) \n",
    "num_freq_mut_genes = 20\n",
    "prot_col_list = [\"APAF1\"+'_transcriptomics', \"ARF1_transcriptomics\",\"BID_transcriptomics\",\"CASP3_transcriptomics\",\"CASP7_transcriptomics\",\"CASP9_transcriptomics\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparison for Both and Wildtype\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Comparison</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>Significant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Both/Wildtype</td>\n",
       "      <td>CASP7_transcriptomics</td>\n",
       "      <td>0.040096</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>CASP3_transcriptomics</td>\n",
       "      <td>0.380327</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>CASP9_transcriptomics</td>\n",
       "      <td>0.389643</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>BID_transcriptomics</td>\n",
       "      <td>0.484439</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>APAF1_transcriptomics</td>\n",
       "      <td>0.546498</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ARF1_transcriptomics</td>\n",
       "      <td>0.937128</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          Comparison   P_Value  Significant\n",
       "Both/Wildtype  CASP7_transcriptomics  0.040096        False\n",
       "1              CASP3_transcriptomics  0.380327        False\n",
       "2              CASP9_transcriptomics  0.389643        False\n",
       "3                BID_transcriptomics  0.484439        False\n",
       "4              APAF1_transcriptomics  0.546498        False\n",
       "5               ARF1_transcriptomics  0.937128        False"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# compare wildtype (neither) and Both\n",
    "Both_wt = reduced_transcript_mutations.loc[(reduced_transcript_mutations['Mutation_Exclusivity'] != \"TP53_Only\")]\n",
    "Both_wt = Both_wt.loc[(Both_wt['Mutation_Exclusivity'] != \"RB1_Only\")]\n",
    "print(\"Doing t-test comparison for Both and Wildtype\")\n",
    "RB1_results_both = wrap_ttest_return_all(Both_wt, 'Mutation_Exclusivity',prot_col_list, num_freq_mut_genes)\n",
    "RB1_results_both = RB1_results_both.rename(index = {0: \"Both/Wildtype\"})\n",
    "RB1_results_both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Wrong number of items passed 6, placement implies 108",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-0a89845923cb>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfont_scale\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1.3\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m cis_boxplot = sns.boxplot(data = reduced_transcript_mutations, x = 'Mutation_Exclusivity',\n\u001b[0;32m----> 5\u001b[0;31m                           y = prot_col_list, order = ['Wildtype', 'Both', 'RB1_Only', \"TP53_Only\"], showfliers = False)  \n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0mcis_boxplot\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_title\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgene\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" Effect on E2F1 target genes\"\u001b[0m \u001b[0;34m+\u001b[0m\u001b[0;34m\" Proteomics in Glioblastoma Tumors\\n\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m cis_boxplot = sns.stripplot(data= reduced_transcript_mutations, x = 'Mutation_Exclusivity',\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/seaborn/categorical.py\u001b[0m in \u001b[0;36mboxplot\u001b[0;34m(x, y, hue, data, order, hue_order, orient, color, palette, saturation, width, dodge, fliersize, linewidth, whis, notch, ax, **kwargs)\u001b[0m\n\u001b[1;32m   2229\u001b[0m     plotter = _BoxPlotter(x, y, hue, data, order, hue_order,\n\u001b[1;32m   2230\u001b[0m                           \u001b[0morient\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpalette\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msaturation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2231\u001b[0;31m                           width, dodge, fliersize, linewidth)\n\u001b[0m\u001b[1;32m   2232\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2233\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0max\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/seaborn/categorical.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, x, y, hue, data, order, hue_order, orient, color, palette, saturation, width, dodge, fliersize, linewidth)\u001b[0m\n\u001b[1;32m    444\u001b[0m                  width, dodge, fliersize, linewidth):\n\u001b[1;32m    445\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 446\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mestablish_variables\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morient\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhue_order\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    447\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mestablish_colors\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpalette\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msaturation\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    448\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/seaborn/categorical.py\u001b[0m in \u001b[0;36mestablish_variables\u001b[0;34m(self, x, y, hue, data, orient, order, hue_order, units)\u001b[0m\n\u001b[1;32m    205\u001b[0m                 \u001b[0;31m# Group the numeric data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    206\u001b[0m                 plot_data, value_label = self._group_longform(vals, groups,\n\u001b[0;32m--> 207\u001b[0;31m                                                               group_names)\n\u001b[0m\u001b[1;32m    208\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    209\u001b[0m                 \u001b[0;31m# Now handle the hue levels for nested ordering\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/seaborn/categorical.py\u001b[0m in \u001b[0;36m_group_longform\u001b[0;34m(self, vals, grouper, order)\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;31m# Ensure that the groupby will work\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    245\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvals\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 246\u001b[0;31m             \u001b[0mvals\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvals\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    247\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    248\u001b[0m         \u001b[0;31m# Group the val data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, dtype, name, copy, fastpath)\u001b[0m\n\u001b[1;32m    311\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msanitize_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mraise_cast_failure\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    312\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 313\u001b[0;31m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSingleBlockManager\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfastpath\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    314\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    315\u001b[0m         \u001b[0mgeneric\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mNDFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfastpath\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, block, axis, do_integrity_check, fastpath)\u001b[0m\n\u001b[1;32m   1514\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1515\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mblock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mBlock\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1516\u001b[0;31m             \u001b[0mblock\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmake_block\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mblock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mplacement\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mslice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mndim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1517\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1518\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mblocks\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mblock\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/internals/blocks.py\u001b[0m in \u001b[0;36mmake_block\u001b[0;34m(values, placement, klass, ndim, dtype, fastpath)\u001b[0m\n\u001b[1;32m   3265\u001b[0m         \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mDatetimeArray\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_simple_new\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3266\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3267\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mklass\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mndim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mndim\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mplacement\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mplacement\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3268\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3269\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/internals/blocks.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, values, placement, ndim)\u001b[0m\n\u001b[1;32m   2773\u001b[0m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mobject\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2774\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2775\u001b[0;31m         \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mndim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mndim\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mplacement\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mplacement\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2776\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2777\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mproperty\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/internals/blocks.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, values, placement, ndim)\u001b[0m\n\u001b[1;32m    126\u001b[0m             raise ValueError(\n\u001b[1;32m    127\u001b[0m                 \u001b[0;34m\"Wrong number of items passed {val}, placement implies \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 128\u001b[0;31m                 \u001b[0;34m\"{mgr}\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mval\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmgr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmgr_locs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    129\u001b[0m             )\n\u001b[1;32m    130\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Wrong number of items passed 6, placement implies 108"
     ]
    }
   ],
   "source": [
    "gene = \"RB1\"\n",
    "plt.rcParams['figure.figsize']=(8,5)\n",
    "sns.set(font_scale = 1.3)\n",
    "cis_boxplot = sns.boxplot(data = reduced_transcript_mutations, x = 'Mutation_Exclusivity',\n",
    "                          y = prot_col_list, order = ['Wildtype', 'Both', 'RB1_Only', \"TP53_Only\"], showfliers = False)  \n",
    "cis_boxplot.set_title(gene + \" Effect on E2F1 target genes\" +\" Proteomics in Glioblastoma Tumors\\n\")\n",
    "cis_boxplot = sns.stripplot(data= reduced_transcript_mutations, x = 'Mutation_Exclusivity',\n",
    "                            y = prot_col_list,jitter = True, color = \".3\", order = ['Wildtype', \"Both\",'RB1_Only', \"TP53_Only\"])\n",
    "cis_boxplot.set(xlabel = \"\\n\"+gene + \" Mutation Status in Tumors\", ylabel = 'transcriptomics')\n",
    "cis_boxplot.set_xticklabels(cis_boxplot.get_xticklabels())\n",
    "\n",
    "# Missense\n",
    "#miss_pval = prot_results_miss_trunc.loc[prot_results_miss_trunc.index == \"Missense/Wildtype\"]\n",
    "#miss_pval_symbol = get_pval_symbol(miss_pval)\n",
    "#format_pval_annotation(miss_pval_symbol, x1 = 0, x2 = 1, line_start = 1.65, line_height = 0.15)\n",
    "\n",
    "# Truncation \n",
    "#trunc_pval = prot_results_miss_trunc.loc[prot_results_miss_trunc.index == \"Truncation/Wildtype\"]\n",
    "#trunc_pval_symbol = get_pval_symbol(trunc_pval)\n",
    "#format_pval_annotation(trunc_pval_symbol, x1 = 0, x2 = 2, line_start = 1.95, line_height = 0.15)\n",
    "\n",
    "plt.show()\n",
    "plt.clf()\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Proteomics \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: In joining the somatic_mutation table, no mutations were found for the following samples, so they were filled with Wildtype_Tumor or Wildtype_Normal: 99 samples for the RB1 gene, 77 samples for the TP53 gene (/Users/Lindsey/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py, line 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ARF1_proteomics</th>\n",
       "      <th>CASP3_proteomics</th>\n",
       "      <th>CASP7_proteomics</th>\n",
       "      <th>CASP7_proteomics</th>\n",
       "      <th>CASP9_proteomics</th>\n",
       "      <th>BID_proteomics</th>\n",
       "      <th>APAF1_proteomics</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>TP53_Mutation</th>\n",
       "      <th>TP53_Location</th>\n",
       "      <th>TP53_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>-1.079632</td>\n",
       "      <td>-0.103299</td>\n",
       "      <td>-0.478298</td>\n",
       "      <td>-0.478298</td>\n",
       "      <td>0.104011</td>\n",
       "      <td>0.121788</td>\n",
       "      <td>-0.082649</td>\n",
       "      <td>[Splice_Site]</td>\n",
       "      <td>[p.X654_splice]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G266R]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.068945</td>\n",
       "      <td>-0.051566</td>\n",
       "      <td>-0.051566</td>\n",
       "      <td>0.273644</td>\n",
       "      <td>0.156544</td>\n",
       "      <td>0.152627</td>\n",
       "      <td>[Frame_Shift_Del]</td>\n",
       "      <td>[p.F117Lfs*8]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.K132R]</td>\n",
       "      <td>Single_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.359319</td>\n",
       "      <td>-0.246385</td>\n",
       "      <td>-0.246385</td>\n",
       "      <td>0.113761</td>\n",
       "      <td>0.188308</td>\n",
       "      <td>-0.012196</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>-0.080557</td>\n",
       "      <td>-0.912002</td>\n",
       "      <td>-0.940367</td>\n",
       "      <td>-0.940367</td>\n",
       "      <td>-0.237273</td>\n",
       "      <td>-0.239060</td>\n",
       "      <td>-0.512124</td>\n",
       "      <td>[Splice_Region, Splice_Site]</td>\n",
       "      <td>[p.D349=, p.X605_splice]</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>[Missense_Mutation, Missense_Mutation]</td>\n",
       "      <td>[p.R248Q, p.R158H]</td>\n",
       "      <td>Multiple_mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-1.079538</td>\n",
       "      <td>0.185203</td>\n",
       "      <td>-0.215475</td>\n",
       "      <td>-0.215475</td>\n",
       "      <td>-0.284076</td>\n",
       "      <td>-0.345157</td>\n",
       "      <td>-0.295338</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S111</td>\n",
       "      <td>0.563308</td>\n",
       "      <td>-2.175506</td>\n",
       "      <td>-0.614856</td>\n",
       "      <td>-0.614856</td>\n",
       "      <td>-0.069430</td>\n",
       "      <td>-0.403765</td>\n",
       "      <td>-0.738951</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.586803</td>\n",
       "      <td>-1.149258</td>\n",
       "      <td>-1.149258</td>\n",
       "      <td>-0.438419</td>\n",
       "      <td>-0.419027</td>\n",
       "      <td>-0.747015</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S113</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.055661</td>\n",
       "      <td>-1.059149</td>\n",
       "      <td>-1.059149</td>\n",
       "      <td>-0.348722</td>\n",
       "      <td>-0.398427</td>\n",
       "      <td>-1.012836</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S114</td>\n",
       "      <td>0.046421</td>\n",
       "      <td>-1.974150</td>\n",
       "      <td>-1.338703</td>\n",
       "      <td>-1.338703</td>\n",
       "      <td>-0.620998</td>\n",
       "      <td>-0.775101</td>\n",
       "      <td>-0.985775</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.575213</td>\n",
       "      <td>-1.299305</td>\n",
       "      <td>-1.299305</td>\n",
       "      <td>-0.616942</td>\n",
       "      <td>-0.521992</td>\n",
       "      <td>-0.729908</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       ARF1_proteomics  CASP3_proteomics  CASP7_proteomics  \\\n",
       "Sample_ID                                                        \n",
       "S001             -1.079632         -0.103299         -0.478298   \n",
       "S002                   NaN         -0.068945         -0.051566   \n",
       "S003                   NaN         -0.359319         -0.246385   \n",
       "S004             -0.080557         -0.912002         -0.940367   \n",
       "S005             -1.079538          0.185203         -0.215475   \n",
       "...                    ...               ...               ...   \n",
       "S111              0.563308         -2.175506         -0.614856   \n",
       "S112                   NaN         -1.586803         -1.149258   \n",
       "S113                   NaN         -2.055661         -1.059149   \n",
       "S114              0.046421         -1.974150         -1.338703   \n",
       "S115                   NaN         -1.575213         -1.299305   \n",
       "\n",
       "Name       CASP7_proteomics  CASP9_proteomics  BID_proteomics  \\\n",
       "Sample_ID                                                       \n",
       "S001              -0.478298          0.104011        0.121788   \n",
       "S002              -0.051566          0.273644        0.156544   \n",
       "S003              -0.246385          0.113761        0.188308   \n",
       "S004              -0.940367         -0.237273       -0.239060   \n",
       "S005              -0.215475         -0.284076       -0.345157   \n",
       "...                     ...               ...             ...   \n",
       "S111              -0.614856         -0.069430       -0.403765   \n",
       "S112              -1.149258         -0.438419       -0.419027   \n",
       "S113              -1.059149         -0.348722       -0.398427   \n",
       "S114              -1.338703         -0.620998       -0.775101   \n",
       "S115              -1.299305         -0.616942       -0.521992   \n",
       "\n",
       "Name       APAF1_proteomics                  RB1_Mutation  \\\n",
       "Sample_ID                                                   \n",
       "S001              -0.082649                 [Splice_Site]   \n",
       "S002               0.152627             [Frame_Shift_Del]   \n",
       "S003              -0.012196              [Wildtype_Tumor]   \n",
       "S004              -0.512124  [Splice_Region, Splice_Site]   \n",
       "S005              -0.295338              [Wildtype_Tumor]   \n",
       "...                     ...                           ...   \n",
       "S111              -0.738951             [Wildtype_Normal]   \n",
       "S112              -0.747015             [Wildtype_Normal]   \n",
       "S113              -1.012836             [Wildtype_Normal]   \n",
       "S114              -0.985775             [Wildtype_Normal]   \n",
       "S115              -0.729908             [Wildtype_Normal]   \n",
       "\n",
       "Name                   RB1_Location RB1_Mutation_Status  \\\n",
       "Sample_ID                                                 \n",
       "S001                [p.X654_splice]     Single_mutation   \n",
       "S002                  [p.F117Lfs*8]     Single_mutation   \n",
       "S003                  [No_mutation]      Wildtype_Tumor   \n",
       "S004       [p.D349=, p.X605_splice]   Multiple_mutation   \n",
       "S005                  [No_mutation]      Wildtype_Tumor   \n",
       "...                             ...                 ...   \n",
       "S111                  [No_mutation]     Wildtype_Normal   \n",
       "S112                  [No_mutation]     Wildtype_Normal   \n",
       "S113                  [No_mutation]     Wildtype_Normal   \n",
       "S114                  [No_mutation]     Wildtype_Normal   \n",
       "S115                  [No_mutation]     Wildtype_Normal   \n",
       "\n",
       "Name                                TP53_Mutation       TP53_Location  \\\n",
       "Sample_ID                                                               \n",
       "S001                          [Missense_Mutation]           [p.G266R]   \n",
       "S002                          [Missense_Mutation]           [p.K132R]   \n",
       "S003                             [Wildtype_Tumor]       [No_mutation]   \n",
       "S004       [Missense_Mutation, Missense_Mutation]  [p.R248Q, p.R158H]   \n",
       "S005                             [Wildtype_Tumor]       [No_mutation]   \n",
       "...                                           ...                 ...   \n",
       "S111                            [Wildtype_Normal]       [No_mutation]   \n",
       "S112                            [Wildtype_Normal]       [No_mutation]   \n",
       "S113                            [Wildtype_Normal]       [No_mutation]   \n",
       "S114                            [Wildtype_Normal]       [No_mutation]   \n",
       "S115                            [Wildtype_Normal]       [No_mutation]   \n",
       "\n",
       "Name      TP53_Mutation_Status Sample_Status  \n",
       "Sample_ID                                     \n",
       "S001           Single_mutation         Tumor  \n",
       "S002           Single_mutation         Tumor  \n",
       "S003            Wildtype_Tumor         Tumor  \n",
       "S004         Multiple_mutation         Tumor  \n",
       "S005            Wildtype_Tumor         Tumor  \n",
       "...                        ...           ...  \n",
       "S111           Wildtype_Normal        Normal  \n",
       "S112           Wildtype_Normal        Normal  \n",
       "S113           Wildtype_Normal        Normal  \n",
       "S114           Wildtype_Normal        Normal  \n",
       "S115           Wildtype_Normal        Normal  \n",
       "\n",
       "[109 rows x 14 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = [\"RB1\",\"TP53\"]\n",
    "#num_freq_mut_genes = len(freq_mut)\n",
    "num_freq_mut_genes = 20\n",
    "# Step 1 - Create dataframe in order to do comparisons with wrap_ttest - drop \n",
    "proteomic_mutations = brain.join_omics_to_mutations(\n",
    "    mutations_genes = gene, omics_df_name = 'proteomics', omics_genes = [\"ARF1\",\"CASP3\",\"CASP7\",\"CASP7\",\"CASP9\",\"BID\",\"APAF1\"])\n",
    "\n",
    "proteomic_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ARF1_proteomics</th>\n",
       "      <th>CASP3_proteomics</th>\n",
       "      <th>CASP7_proteomics</th>\n",
       "      <th>CASP7_proteomics</th>\n",
       "      <th>CASP9_proteomics</th>\n",
       "      <th>BID_proteomics</th>\n",
       "      <th>APAF1_proteomics</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>TP53_Mutation</th>\n",
       "      <th>TP53_Location</th>\n",
       "      <th>TP53_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>-1.079632</td>\n",
       "      <td>-0.103299</td>\n",
       "      <td>-0.478298</td>\n",
       "      <td>-0.478298</td>\n",
       "      <td>0.104011</td>\n",
       "      <td>0.121788</td>\n",
       "      <td>-0.082649</td>\n",
       "      <td>[Splice_Site]</td>\n",
       "      <td>[p.X654_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G266R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.068945</td>\n",
       "      <td>-0.051566</td>\n",
       "      <td>-0.051566</td>\n",
       "      <td>0.273644</td>\n",
       "      <td>0.156544</td>\n",
       "      <td>0.152627</td>\n",
       "      <td>[Frame_Shift_Del]</td>\n",
       "      <td>[p.F117Lfs*8]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.K132R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.359319</td>\n",
       "      <td>-0.246385</td>\n",
       "      <td>-0.246385</td>\n",
       "      <td>0.113761</td>\n",
       "      <td>0.188308</td>\n",
       "      <td>-0.012196</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>-0.080557</td>\n",
       "      <td>-0.912002</td>\n",
       "      <td>-0.940367</td>\n",
       "      <td>-0.940367</td>\n",
       "      <td>-0.237273</td>\n",
       "      <td>-0.239060</td>\n",
       "      <td>-0.512124</td>\n",
       "      <td>[Splice_Region, Splice_Site]</td>\n",
       "      <td>[p.D349=, p.X605_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation, Missense_Mutation]</td>\n",
       "      <td>[p.R248Q, p.R158H]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-1.079538</td>\n",
       "      <td>0.185203</td>\n",
       "      <td>-0.215475</td>\n",
       "      <td>-0.215475</td>\n",
       "      <td>-0.284076</td>\n",
       "      <td>-0.345157</td>\n",
       "      <td>-0.295338</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S111</td>\n",
       "      <td>0.563308</td>\n",
       "      <td>-2.175506</td>\n",
       "      <td>-0.614856</td>\n",
       "      <td>-0.614856</td>\n",
       "      <td>-0.069430</td>\n",
       "      <td>-0.403765</td>\n",
       "      <td>-0.738951</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.586803</td>\n",
       "      <td>-1.149258</td>\n",
       "      <td>-1.149258</td>\n",
       "      <td>-0.438419</td>\n",
       "      <td>-0.419027</td>\n",
       "      <td>-0.747015</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S113</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.055661</td>\n",
       "      <td>-1.059149</td>\n",
       "      <td>-1.059149</td>\n",
       "      <td>-0.348722</td>\n",
       "      <td>-0.398427</td>\n",
       "      <td>-1.012836</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S114</td>\n",
       "      <td>0.046421</td>\n",
       "      <td>-1.974150</td>\n",
       "      <td>-1.338703</td>\n",
       "      <td>-1.338703</td>\n",
       "      <td>-0.620998</td>\n",
       "      <td>-0.775101</td>\n",
       "      <td>-0.985775</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.575213</td>\n",
       "      <td>-1.299305</td>\n",
       "      <td>-1.299305</td>\n",
       "      <td>-0.616942</td>\n",
       "      <td>-0.521992</td>\n",
       "      <td>-0.729908</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       ARF1_proteomics  CASP3_proteomics  CASP7_proteomics  \\\n",
       "Sample_ID                                                        \n",
       "S001             -1.079632         -0.103299         -0.478298   \n",
       "S002                   NaN         -0.068945         -0.051566   \n",
       "S003                   NaN         -0.359319         -0.246385   \n",
       "S004             -0.080557         -0.912002         -0.940367   \n",
       "S005             -1.079538          0.185203         -0.215475   \n",
       "...                    ...               ...               ...   \n",
       "S111              0.563308         -2.175506         -0.614856   \n",
       "S112                   NaN         -1.586803         -1.149258   \n",
       "S113                   NaN         -2.055661         -1.059149   \n",
       "S114              0.046421         -1.974150         -1.338703   \n",
       "S115                   NaN         -1.575213         -1.299305   \n",
       "\n",
       "Name       CASP7_proteomics  CASP9_proteomics  BID_proteomics  \\\n",
       "Sample_ID                                                       \n",
       "S001              -0.478298          0.104011        0.121788   \n",
       "S002              -0.051566          0.273644        0.156544   \n",
       "S003              -0.246385          0.113761        0.188308   \n",
       "S004              -0.940367         -0.237273       -0.239060   \n",
       "S005              -0.215475         -0.284076       -0.345157   \n",
       "...                     ...               ...             ...   \n",
       "S111              -0.614856         -0.069430       -0.403765   \n",
       "S112              -1.149258         -0.438419       -0.419027   \n",
       "S113              -1.059149         -0.348722       -0.398427   \n",
       "S114              -1.338703         -0.620998       -0.775101   \n",
       "S115              -1.299305         -0.616942       -0.521992   \n",
       "\n",
       "Name       APAF1_proteomics                  RB1_Mutation  \\\n",
       "Sample_ID                                                   \n",
       "S001              -0.082649                 [Splice_Site]   \n",
       "S002               0.152627             [Frame_Shift_Del]   \n",
       "S003              -0.012196              [Wildtype_Tumor]   \n",
       "S004              -0.512124  [Splice_Region, Splice_Site]   \n",
       "S005              -0.295338              [Wildtype_Tumor]   \n",
       "...                     ...                           ...   \n",
       "S111              -0.738951             [Wildtype_Normal]   \n",
       "S112              -0.747015             [Wildtype_Normal]   \n",
       "S113              -1.012836             [Wildtype_Normal]   \n",
       "S114              -0.985775             [Wildtype_Normal]   \n",
       "S115              -0.729908             [Wildtype_Normal]   \n",
       "\n",
       "Name                   RB1_Location RB1_Mutation_Status  \\\n",
       "Sample_ID                                                 \n",
       "S001                [p.X654_splice]            Mutation   \n",
       "S002                  [p.F117Lfs*8]            Mutation   \n",
       "S003                  [No_mutation]            Wildtype   \n",
       "S004       [p.D349=, p.X605_splice]            Mutation   \n",
       "S005                  [No_mutation]            Wildtype   \n",
       "...                             ...                 ...   \n",
       "S111                  [No_mutation]     Wildtype_Normal   \n",
       "S112                  [No_mutation]     Wildtype_Normal   \n",
       "S113                  [No_mutation]     Wildtype_Normal   \n",
       "S114                  [No_mutation]     Wildtype_Normal   \n",
       "S115                  [No_mutation]     Wildtype_Normal   \n",
       "\n",
       "Name                                TP53_Mutation       TP53_Location  \\\n",
       "Sample_ID                                                               \n",
       "S001                          [Missense_Mutation]           [p.G266R]   \n",
       "S002                          [Missense_Mutation]           [p.K132R]   \n",
       "S003                             [Wildtype_Tumor]       [No_mutation]   \n",
       "S004       [Missense_Mutation, Missense_Mutation]  [p.R248Q, p.R158H]   \n",
       "S005                             [Wildtype_Tumor]       [No_mutation]   \n",
       "...                                           ...                 ...   \n",
       "S111                            [Wildtype_Normal]       [No_mutation]   \n",
       "S112                            [Wildtype_Normal]       [No_mutation]   \n",
       "S113                            [Wildtype_Normal]       [No_mutation]   \n",
       "S114                            [Wildtype_Normal]       [No_mutation]   \n",
       "S115                            [Wildtype_Normal]       [No_mutation]   \n",
       "\n",
       "Name      TP53_Mutation_Status Sample_Status  \n",
       "Sample_ID                                     \n",
       "S001                  Mutation         Tumor  \n",
       "S002                  Mutation         Tumor  \n",
       "S003                  Wildtype         Tumor  \n",
       "S004                  Mutation         Tumor  \n",
       "S005                  Wildtype         Tumor  \n",
       "...                        ...           ...  \n",
       "S111           Wildtype_Normal        Normal  \n",
       "S112           Wildtype_Normal        Normal  \n",
       "S113           Wildtype_Normal        Normal  \n",
       "S114           Wildtype_Normal        Normal  \n",
       "S115           Wildtype_Normal        Normal  \n",
       "\n",
       "[109 rows x 14 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Classify all mutations as 'mutation' and all wildtypes as 'wildtype'\n",
    "proteomic_mutations = proteomic_mutations.replace({'Wildtype_Tumor':'Wildtype','Single_mutation':'Mutation','Multiple_mutation':'Mutation'})\n",
    "proteomic_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "for ind, row in proteomic_mutations.iterrows():\n",
    "    if row['RB1_Mutation_Status'] == 'Mutation' and row[\"TP53_Mutation_Status\"] == 'Mutation':\n",
    "        proteomic_mutations.loc[ind, 'Mutation_Exclusivity'] = 'Both'\n",
    "        \n",
    "    elif row['RB1_Mutation_Status'] == 'Mutation' and row[\"TP53_Mutation_Status\"] != 'Mutation':\n",
    "        proteomic_mutations.loc[ind, 'Mutation_Exclusivity'] = 'TP53_Only'\n",
    "        \n",
    "    elif row['RB1_Mutation_Status'] != 'Mutation' and row[\"TP53_Mutation_Status\"] == 'Mutation':\n",
    "        proteomic_mutations.loc[ind, 'Mutation_Exclusivity'] = \"RB1_Only\"\n",
    "        \n",
    "    else:\n",
    "        proteomic_mutations.loc[ind, 'Mutation_Exclusivity'] = 'Neither'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ARF1_proteomics</th>\n",
       "      <th>CASP3_proteomics</th>\n",
       "      <th>CASP7_proteomics</th>\n",
       "      <th>CASP7_proteomics</th>\n",
       "      <th>CASP9_proteomics</th>\n",
       "      <th>BID_proteomics</th>\n",
       "      <th>APAF1_proteomics</th>\n",
       "      <th>RB1_Mutation</th>\n",
       "      <th>RB1_Location</th>\n",
       "      <th>RB1_Mutation_Status</th>\n",
       "      <th>TP53_Mutation</th>\n",
       "      <th>TP53_Location</th>\n",
       "      <th>TP53_Mutation_Status</th>\n",
       "      <th>Sample_Status</th>\n",
       "      <th>Mutation_Exclusivity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S001</td>\n",
       "      <td>-1.079632</td>\n",
       "      <td>-0.103299</td>\n",
       "      <td>-0.478298</td>\n",
       "      <td>-0.478298</td>\n",
       "      <td>0.104011</td>\n",
       "      <td>0.121788</td>\n",
       "      <td>-0.082649</td>\n",
       "      <td>[Splice_Site]</td>\n",
       "      <td>[p.X654_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.G266R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.068945</td>\n",
       "      <td>-0.051566</td>\n",
       "      <td>-0.051566</td>\n",
       "      <td>0.273644</td>\n",
       "      <td>0.156544</td>\n",
       "      <td>0.152627</td>\n",
       "      <td>[Frame_Shift_Del]</td>\n",
       "      <td>[p.F117Lfs*8]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation]</td>\n",
       "      <td>[p.K132R]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.359319</td>\n",
       "      <td>-0.246385</td>\n",
       "      <td>-0.246385</td>\n",
       "      <td>0.113761</td>\n",
       "      <td>0.188308</td>\n",
       "      <td>-0.012196</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S004</td>\n",
       "      <td>-0.080557</td>\n",
       "      <td>-0.912002</td>\n",
       "      <td>-0.940367</td>\n",
       "      <td>-0.940367</td>\n",
       "      <td>-0.237273</td>\n",
       "      <td>-0.239060</td>\n",
       "      <td>-0.512124</td>\n",
       "      <td>[Splice_Region, Splice_Site]</td>\n",
       "      <td>[p.D349=, p.X605_splice]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>[Missense_Mutation, Missense_Mutation]</td>\n",
       "      <td>[p.R248Q, p.R158H]</td>\n",
       "      <td>Mutation</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Both</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S005</td>\n",
       "      <td>-1.079538</td>\n",
       "      <td>0.185203</td>\n",
       "      <td>-0.215475</td>\n",
       "      <td>-0.215475</td>\n",
       "      <td>-0.284076</td>\n",
       "      <td>-0.345157</td>\n",
       "      <td>-0.295338</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>[Wildtype_Tumor]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S111</td>\n",
       "      <td>0.563308</td>\n",
       "      <td>-2.175506</td>\n",
       "      <td>-0.614856</td>\n",
       "      <td>-0.614856</td>\n",
       "      <td>-0.069430</td>\n",
       "      <td>-0.403765</td>\n",
       "      <td>-0.738951</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.586803</td>\n",
       "      <td>-1.149258</td>\n",
       "      <td>-1.149258</td>\n",
       "      <td>-0.438419</td>\n",
       "      <td>-0.419027</td>\n",
       "      <td>-0.747015</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S113</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.055661</td>\n",
       "      <td>-1.059149</td>\n",
       "      <td>-1.059149</td>\n",
       "      <td>-0.348722</td>\n",
       "      <td>-0.398427</td>\n",
       "      <td>-1.012836</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S114</td>\n",
       "      <td>0.046421</td>\n",
       "      <td>-1.974150</td>\n",
       "      <td>-1.338703</td>\n",
       "      <td>-1.338703</td>\n",
       "      <td>-0.620998</td>\n",
       "      <td>-0.775101</td>\n",
       "      <td>-0.985775</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.575213</td>\n",
       "      <td>-1.299305</td>\n",
       "      <td>-1.299305</td>\n",
       "      <td>-0.616942</td>\n",
       "      <td>-0.521992</td>\n",
       "      <td>-0.729908</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>[Wildtype_Normal]</td>\n",
       "      <td>[No_mutation]</td>\n",
       "      <td>Wildtype_Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Neither</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name       ARF1_proteomics  CASP3_proteomics  CASP7_proteomics  \\\n",
       "Sample_ID                                                        \n",
       "S001             -1.079632         -0.103299         -0.478298   \n",
       "S002                   NaN         -0.068945         -0.051566   \n",
       "S003                   NaN         -0.359319         -0.246385   \n",
       "S004             -0.080557         -0.912002         -0.940367   \n",
       "S005             -1.079538          0.185203         -0.215475   \n",
       "...                    ...               ...               ...   \n",
       "S111              0.563308         -2.175506         -0.614856   \n",
       "S112                   NaN         -1.586803         -1.149258   \n",
       "S113                   NaN         -2.055661         -1.059149   \n",
       "S114              0.046421         -1.974150         -1.338703   \n",
       "S115                   NaN         -1.575213         -1.299305   \n",
       "\n",
       "Name       CASP7_proteomics  CASP9_proteomics  BID_proteomics  \\\n",
       "Sample_ID                                                       \n",
       "S001              -0.478298          0.104011        0.121788   \n",
       "S002              -0.051566          0.273644        0.156544   \n",
       "S003              -0.246385          0.113761        0.188308   \n",
       "S004              -0.940367         -0.237273       -0.239060   \n",
       "S005              -0.215475         -0.284076       -0.345157   \n",
       "...                     ...               ...             ...   \n",
       "S111              -0.614856         -0.069430       -0.403765   \n",
       "S112              -1.149258         -0.438419       -0.419027   \n",
       "S113              -1.059149         -0.348722       -0.398427   \n",
       "S114              -1.338703         -0.620998       -0.775101   \n",
       "S115              -1.299305         -0.616942       -0.521992   \n",
       "\n",
       "Name       APAF1_proteomics                  RB1_Mutation  \\\n",
       "Sample_ID                                                   \n",
       "S001              -0.082649                 [Splice_Site]   \n",
       "S002               0.152627             [Frame_Shift_Del]   \n",
       "S003              -0.012196              [Wildtype_Tumor]   \n",
       "S004              -0.512124  [Splice_Region, Splice_Site]   \n",
       "S005              -0.295338              [Wildtype_Tumor]   \n",
       "...                     ...                           ...   \n",
       "S111              -0.738951             [Wildtype_Normal]   \n",
       "S112              -0.747015             [Wildtype_Normal]   \n",
       "S113              -1.012836             [Wildtype_Normal]   \n",
       "S114              -0.985775             [Wildtype_Normal]   \n",
       "S115              -0.729908             [Wildtype_Normal]   \n",
       "\n",
       "Name                   RB1_Location RB1_Mutation_Status  \\\n",
       "Sample_ID                                                 \n",
       "S001                [p.X654_splice]            Mutation   \n",
       "S002                  [p.F117Lfs*8]            Mutation   \n",
       "S003                  [No_mutation]            Wildtype   \n",
       "S004       [p.D349=, p.X605_splice]            Mutation   \n",
       "S005                  [No_mutation]            Wildtype   \n",
       "...                             ...                 ...   \n",
       "S111                  [No_mutation]     Wildtype_Normal   \n",
       "S112                  [No_mutation]     Wildtype_Normal   \n",
       "S113                  [No_mutation]     Wildtype_Normal   \n",
       "S114                  [No_mutation]     Wildtype_Normal   \n",
       "S115                  [No_mutation]     Wildtype_Normal   \n",
       "\n",
       "Name                                TP53_Mutation       TP53_Location  \\\n",
       "Sample_ID                                                               \n",
       "S001                          [Missense_Mutation]           [p.G266R]   \n",
       "S002                          [Missense_Mutation]           [p.K132R]   \n",
       "S003                             [Wildtype_Tumor]       [No_mutation]   \n",
       "S004       [Missense_Mutation, Missense_Mutation]  [p.R248Q, p.R158H]   \n",
       "S005                             [Wildtype_Tumor]       [No_mutation]   \n",
       "...                                           ...                 ...   \n",
       "S111                            [Wildtype_Normal]       [No_mutation]   \n",
       "S112                            [Wildtype_Normal]       [No_mutation]   \n",
       "S113                            [Wildtype_Normal]       [No_mutation]   \n",
       "S114                            [Wildtype_Normal]       [No_mutation]   \n",
       "S115                            [Wildtype_Normal]       [No_mutation]   \n",
       "\n",
       "Name      TP53_Mutation_Status Sample_Status Mutation_Exclusivity  \n",
       "Sample_ID                                                          \n",
       "S001                  Mutation         Tumor                 Both  \n",
       "S002                  Mutation         Tumor                 Both  \n",
       "S003                  Wildtype         Tumor              Neither  \n",
       "S004                  Mutation         Tumor                 Both  \n",
       "S005                  Wildtype         Tumor              Neither  \n",
       "...                        ...           ...                  ...  \n",
       "S111           Wildtype_Normal        Normal              Neither  \n",
       "S112           Wildtype_Normal        Normal              Neither  \n",
       "S113           Wildtype_Normal        Normal              Neither  \n",
       "S114           Wildtype_Normal        Normal              Neither  \n",
       "S115           Wildtype_Normal        Normal              Neither  \n",
       "\n",
       "[109 rows x 15 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#num_freq_mut_genes = len(freq_mut) \n",
    "num_freq_mut_genes = 20\n",
    "prot_col_list = [\"APAF1\"+'_proteomics', \"ARF1_proteomics\",\"BID_proteomics\",\"CASP3_proteomics\",\"CASP7_proteomics\",\"CASP9_proteomics\"]\n",
    "proteomic_mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Doing t-test comparison for Both and Wildtype\n",
      "Incorrectly Formatted Dataframe!\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'rename'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-18-6010641fe989>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Doing t-test comparison for Both and Wildtype\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mRB1_results_both\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwrap_ttest_return_all\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mProt_Both_wt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Mutation_Exclusivity'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mprot_col_list\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnum_freq_mut_genes\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mRB1_results_both\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mRB1_results_both\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m\"Both/Wildtype\"\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0mBoth_wt\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'rename'"
     ]
    }
   ],
   "source": [
    "# compare wildtype (neither) and Both\n",
    "Prot_Both_wt = proteomic_mutations.loc[(proteomic_mutations['Mutation_Exclusivity'] != \"TP53_Only\")]\n",
    "Prot_Both_wt = Both_wt.loc[(Prot_Both_wt['Mutation_Exclusivity'] != \"RB1_Only\")]\n",
    "print(\"Doing t-test comparison for Both and Wildtype\")\n",
    "RB1_results_both = wrap_ttest_return_all(Prot_Both_wt, 'Mutation_Exclusivity',prot_col_list, num_freq_mut_genes)\n",
    "RB1_results_both = RB1_results_both.rename(index = {0: \"Both/Wildtype\"})\n",
    "Both_wt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
